diff --git a/Mage.Sets/src/mage/cards/e/EstridsInvocation.java b/Mage.Sets/src/mage/cards/e/EstridsInvocation.java
index edb60348..5942e603 100644
--- a/Mage.Sets/src/mage/cards/e/EstridsInvocation.java
+++ b/Mage.Sets/src/mage/cards/e/EstridsInvocation.java
@@ -40,6 +40,9 @@ public final class EstridsInvocation extends CardImpl {
                 + "If you do, return it to the battlefield "
                 + "under its owner's control.\""), true
         ));
+        this.addAbility(new BeginningOfUpkeepTriggeredAbility(
+                new EstridsInvocationEffect(), true
+        ));
     }
 
     private EstridsInvocation(final EstridsInvocation card) {
