diff --git a/lib/node_modules/@stdlib/stats/base/stdevwd/lib/ndarray.js b/lib/node_modules/@stdlib/stats/base/stdevwd/lib/ndarray.js
index f31809bdc91..922c515634e 100644
--- a/lib/node_modules/@stdlib/stats/base/stdevwd/lib/ndarray.js
+++ b/lib/node_modules/@stdlib/stats/base/stdevwd/lib/ndarray.js
@@ -20,8 +20,9 @@
 
 // MODULES //
 
-var variancewd = require( '@stdlib/stats/strided/variancewd' ).ndarray;
 var sqrt = require( '@stdlib/math/base/special/sqrt' );
+var arraylike2object = require( '@stdlib/array/base/arraylike2object' );
+var accessors = require( './accessors.js' );
 
 
 // MAIN //
@@ -29,6 +30,40 @@ var sqrt = require( '@stdlib/math/base/special/sqrt' );
 /**
 * Computes the standard deviation of a strided array using Welford's algorithm.
 *
+* ## Method
+*
+* -   This implementation uses Welford's algorithm for efficient computation, which can be derived as follows. Let
+*
+*     ```tex
+*     \begin{align*}
+*     S_n &= n \sigma_n^2 \\
+*         &= \sum_{i=1}^{n} (x_i - \mu_n)^2 \\
+*         &= \biggl(\sum_{i=1}^{n} x_i^2 \biggr) - n\mu_n^2
+*     \end{align*}
+*     ```
+*
+*     Accordingly,
+*
+*     ```tex
+*     \begin{align*}
+*     S_n - S_{n-1} &= \sum_{i=1}^{n} x_i^2 - n\mu_n^2 - \sum_{i=1}^{n-1} x_i^2 + (n-1)\mu_{n-1}^2 \\
+*                   &= x_n^2 - n\mu_n^2 + (n-1)\mu_{n-1}^2 \\
+*                   &= x_n^2 - \mu_{n-1}^2 + n(\mu_{n-1}^2 - \mu_n^2) \\
+*                   &= x_n^2 - \mu_{n-1}^2 + n(\mu_{n-1} - \mu_n)(\mu_{n-1} + \mu_n) \\
+*                   &= x_n^2 - \mu_{n-1}^2 + (\mu_{n-1} - x_n)(\mu_{n-1} + \mu_n) \\
+*                   &= x_n^2 - \mu_{n-1}^2 + \mu_{n-1}^2 - x_n\mu_n - x_n\mu_{n-1} + \mu_n\mu_{n-1} \\
+*                   &= x_n^2 - x_n\mu_n - x_n\mu_{n-1} + \mu_n\mu_{n-1} \\
+*                   &= (x_n - \mu_{n-1})(x_n - \mu_n) \\
+*                   &= S_{n-1} + (x_n - \mu_{n-1})(x_n - \mu_n)
+*     \end{align*}
+*     ```
+*
+*     where we use the identity
+*
+*     ```tex
+*     x_n - \mu_{n-1} = n (\mu_n - \mu_{n-1})
+*     ```
+*
 * ## References
 *
 * -   Welford, B. P. 1962. "Note on a Method for Calculating Corrected Sums of Squares and Products." _Technometrics_ 4 (3). Taylor & Francis: 419–20. doi:[10.1080/00401706.1962.10490022](https://doi.org/10.1080/00401706.1962.10490022).
@@ -51,7 +86,37 @@ var sqrt = require( '@stdlib/math/base/special/sqrt' );
 * // returns 2.5
 */
 function stdevwd( N, correction, x, stride, offset ) {
-	return sqrt( variancewd( N, correction, x, stride, offset ) );
+	var delta;
+	var mu;
+	var M2;
+	var ix;
+	var o;
+	var v;
+	var n;
+	var i;
+
+	n = N - correction;
+	if ( N <= 0 || n <= 0.0 ) {
+		return NaN;
+	}
+	if ( N === 1 || stride === 0 ) {
+		return 0.0;
+	}
+	o = arraylike2object( x );
+	if ( o.accessorProtocol ) {
+		return accessors( N, correction, o, stride, offset );
+	}
+	ix = offset;
+	M2 = 0.0;
+	mu = 0.0;
+	for ( i = 0; i < N; i++ ) {
+		v = x[ ix ];
+		delta = v - mu;
+		mu += delta / (i+1);
+		M2 += delta * ( v - mu );
+		ix += stride;
+	}
+	return sqrt( M2 / n );
 }
 
 
diff --git a/lib/node_modules/@stdlib/stats/base/stdevwd/lib/stdevwd.js b/lib/node_modules/@stdlib/stats/base/stdevwd/lib/stdevwd.js
index 326eaf7f016..d988945d289 100644
--- a/lib/node_modules/@stdlib/stats/base/stdevwd/lib/stdevwd.js
+++ b/lib/node_modules/@stdlib/stats/base/stdevwd/lib/stdevwd.js
@@ -20,8 +20,8 @@
 
 // MODULES //
 
-var variancewd = require( '@stdlib/stats/strided/variancewd' );
-var sqrt = require( '@stdlib/math/base/special/sqrt' );
+var stride2offset = require( '@stdlib/strided/base/stride2offset' );
+var ndarray = require( './ndarray.js' );
 
 
 // MAIN //
@@ -47,7 +47,7 @@ var sqrt = require( '@stdlib/math/base/special/sqrt' );
 * // returns ~2.0817
 */
 function stdevwd( N, correction, x, stride ) {
-	return sqrt( variancewd( N, correction, x, stride ) );
+	return ndarray( N, correction, x, stride, stride2offset( N, stride ) );
 }
 
 
diff --git a/lib/node_modules/@stdlib/stats/base/stdevwd/test/test.ndarray.js b/lib/node_modules/@stdlib/stats/base/stdevwd/test/test.ndarray.js
index 0b5022acab7..acec59c9bc9 100644
--- a/lib/node_modules/@stdlib/stats/base/stdevwd/test/test.ndarray.js
+++ b/lib/node_modules/@stdlib/stats/base/stdevwd/test/test.ndarray.js
@@ -24,6 +24,7 @@ var tape = require( 'tape' );
 var floor = require( '@stdlib/math/base/special/floor' );
 var sqrt = require( '@stdlib/math/base/special/sqrt' );
 var isnan = require( '@stdlib/math/base/assert/is-nan' );
+var toAccessorArray = require( '@stdlib/array/base/to-accessor-array' );
 var stdevwd = require( './../lib/ndarray.js' );
 
 
@@ -200,3 +201,110 @@ tape( 'the function supports an `offset` parameter', function test( t ) {
 
 	t.end();
 });
+
+tape( 'the function calculates the population standard deviation of a strided array (accessors)', function test( t ) {
+	var x;
+	var v;
+
+	x = toAccessorArray( [ 1.0, -2.0, -4.0, 5.0, 0.0, 3.0 ] );
+	v = stdevwd( x.length, 0, x, 1, 0 );
+	t.strictEqual( v, sqrt( 53.5/x.length ), 'returns expected value' );
+
+	x = toAccessorArray( [ -4.0, -4.0 ] );
+	v = stdevwd( x.length, 0, x, 1, 0 );
+	t.strictEqual( v, 0.0, 'returns expected value' );
+
+	x = toAccessorArray( [ NaN, 4.0 ] );
+	v = stdevwd( x.length, 0, x, 1, 0 );
+	t.strictEqual( isnan( v ), true, 'returns expected value' );
+
+	t.end();
+});
+
+tape( 'the function calculates the sample standard deviation of a strided array (accessors)', function test( t ) {
+	var x;
+	var v;
+
+	x = toAccessorArray( [ 1.0, -2.0, -4.0, 5.0, 0.0, 3.0 ] );
+	v = stdevwd( x.length, 1, x, 1, 0 );
+	t.strictEqual( v, sqrt( 53.5/(x.length-1) ), 'returns expected value' );
+
+	x = toAccessorArray( [ -4.0, -4.0 ] );
+	v = stdevwd( x.length, 1, x, 1, 0 );
+	t.strictEqual( v, 0.0, 'returns expected value' );
+
+	x = toAccessorArray( [ NaN, 4.0 ] );
+	v = stdevwd( x.length, 1, x, 1, 0 );
+	t.strictEqual( isnan( v ), true, 'returns expected value' );
+
+	t.end();
+});
+
+tape( 'the function supports a `stride` parameter (accessors)', function test( t ) {
+	var N;
+	var x;
+	var v;
+
+	x = toAccessorArray( [
+		1.0,  // 0
+		2.0,
+		2.0,  // 1
+		-7.0,
+		-2.0, // 2
+		3.0,
+		4.0,  // 3
+		2.0
+	] );
+
+	N = floor( x.length / 2 );
+	v = stdevwd( N, 1, x, 2, 0 );
+
+	t.strictEqual( v, 2.5, 'returns expected value' );
+	t.end();
+});
+
+tape( 'the function supports a negative `stride` parameter (accessors)', function test( t ) {
+	var N;
+	var x;
+	var v;
+
+	x = toAccessorArray( [
+		1.0,  // 3
+		2.0,
+		2.0,  // 2
+		-7.0,
+		-2.0, // 1
+		3.0,
+		4.0,  // 0
+		2.0
+	] );
+
+	N = floor( x.length / 2 );
+	v = stdevwd( N, 1, x, -2, 6 );
+
+	t.strictEqual( v, 2.5, 'returns expected value' );
+	t.end();
+});
+
+tape( 'the function supports an `offset` parameter (accessors)', function test( t ) {
+	var N;
+	var x;
+	var v;
+
+	x = toAccessorArray( [
+		2.0,
+		1.0,  // 0
+		2.0,
+		-2.0, // 1
+		-2.0,
+		2.0,  // 2
+		3.0,
+		4.0   // 3
+	] );
+	N = floor( x.length / 2 );
+
+	v = stdevwd( N, 1, x, 2, 1 );
+	t.strictEqual( v, 2.5, 'returns expected value' );
+
+	t.end();
+});
diff --git a/lib/node_modules/@stdlib/stats/base/stdevwd/test/test.stdevwd.js b/lib/node_modules/@stdlib/stats/base/stdevwd/test/test.stdevwd.js
index dcf840ce403..b381d0b4ef1 100644
--- a/lib/node_modules/@stdlib/stats/base/stdevwd/test/test.stdevwd.js
+++ b/lib/node_modules/@stdlib/stats/base/stdevwd/test/test.stdevwd.js
@@ -25,6 +25,7 @@ var floor = require( '@stdlib/math/base/special/floor' );
 var sqrt = require( '@stdlib/math/base/special/sqrt' );
 var isnan = require( '@stdlib/math/base/assert/is-nan' );
 var Float64Array = require( '@stdlib/array/float64' );
+var toAccessorArray = require( '@stdlib/array/base/to-accessor-array' );
 var stdevwd = require( './../lib/stdevwd.js' );
 
 
@@ -205,3 +206,87 @@ tape( 'the function supports view offsets', function test( t ) {
 
 	t.end();
 });
+
+tape( 'the function calculates the population standard deviation of a strided array (accessors)', function test( t ) {
+	var x;
+	var v;
+
+	x = toAccessorArray( [ 1.0, -2.0, -4.0, 5.0, 0.0, 3.0 ] );
+	v = stdevwd( x.length, 0, x, 1 );
+	t.strictEqual( v, sqrt( 53.5/x.length ), 'returns expected value' );
+
+	x = toAccessorArray( [ -4.0, -4.0 ] );
+	v = stdevwd( x.length, 0, x, 1 );
+	t.strictEqual( v, 0.0, 'returns expected value' );
+
+	x = toAccessorArray( [ NaN, 4.0 ] );
+	v = stdevwd( x.length, 0, x, 1 );
+	t.strictEqual( isnan( v ), true, 'returns expected value' );
+
+	t.end();
+});
+
+tape( 'the function calculates the sample standard deviation of a strided array (accessors)', function test( t ) {
+	var x;
+	var v;
+
+	x = toAccessorArray( [ 1.0, -2.0, -4.0, 5.0, 0.0, 3.0 ] );
+	v = stdevwd( x.length, 1, x, 1 );
+	t.strictEqual( v, sqrt( 53.5/(x.length-1) ), 'returns expected value' );
+
+	x = toAccessorArray( [ -4.0, -4.0 ] );
+	v = stdevwd( x.length, 1, x, 1 );
+	t.strictEqual( v, 0.0, 'returns expected value' );
+
+	x = toAccessorArray( [ NaN, 4.0 ] );
+	v = stdevwd( x.length, 1, x, 1 );
+	t.strictEqual( isnan( v ), true, 'returns expected value' );
+
+	t.end();
+});
+
+tape( 'the function supports a `stride` parameter (accessors)', function test( t ) {
+	var N;
+	var x;
+	var v;
+
+	x = toAccessorArray( [
+		1.0,  // 0
+		2.0,
+		2.0,  // 1
+		-7.0,
+		-2.0, // 2
+		3.0,
+		4.0,  // 3
+		2.0
+	] );
+
+	N = floor( x.length / 2 );
+	v = stdevwd( N, 1, x, 2 );
+
+	t.strictEqual( v, 2.5, 'returns expected value' );
+	t.end();
+});
+
+tape( 'the function supports a negative `stride` parameter (accessors)', function test( t ) {
+	var N;
+	var x;
+	var v;
+
+	x = toAccessorArray( [
+		1.0,  // 3
+		2.0,
+		2.0,  // 2
+		-7.0,
+		-2.0, // 1
+		3.0,
+		4.0,  // 0
+		2.0
+	] );
+
+	N = floor( x.length / 2 );
+	v = stdevwd( N, 1, x, -2 );
+
+	t.strictEqual( v, 2.5, 'returns expected value' );
+	t.end();
+});
