diff --git a/Cargo.lock b/Cargo.lock
index 3a9e669..360f12a 100644
--- a/Cargo.lock
+++ b/Cargo.lock
@@ -672,9 +672,9 @@ dependencies = [
 
 [[package]]
 name = "deranged"
-version = "0.5.5"
+version = "0.3.11"
 source = "registry+https://github.com/rust-lang/crates.io-index"
-checksum = "ececcb659e7ba858fb4f10388c250a7252eb0a27373f1a72b8748afdd248e587"
+checksum = "b42b6fa04a440b495c8b04d0e71b707c585f83cb9cb28cf8cd0d976c315e31b4"
 dependencies = [
  "powerfmt",
 ]
@@ -1959,9 +1959,9 @@ dependencies = [
 
 [[package]]
 name = "num-conv"
-version = "0.2.0"
+version = "0.1.0"
 source = "registry+https://github.com/rust-lang/crates.io-index"
-checksum = "cf97ec579c3c42f953ef76dbf8d55ac91fb219dde70e49aa4a6b7d74e9919050"
+checksum = "51d515d32fb182ee37cda2ccdcb92950d6a3c2893aa280e540671c2cd0f3b1d9"
 
 [[package]]
 name = "num-integer"
@@ -3208,30 +3208,30 @@ dependencies = [
 
 [[package]]
 name = "time"
-version = "0.3.47"
+version = "0.3.36"
 source = "registry+https://github.com/rust-lang/crates.io-index"
-checksum = "743bd48c283afc0388f9b8827b976905fb217ad9e647fae3a379a9283c4def2c"
+checksum = "5dfd88e563464686c916c7e46e623e520ddc6d79fa6641390f2e3fa86e83e885"
 dependencies = [
  "deranged",
  "itoa",
  "num-conv",
  "powerfmt",
- "serde_core",
+ "serde",
  "time-core",
  "time-macros",
 ]
 
 [[package]]
 name = "time-core"
-version = "0.1.8"
+version = "0.1.2"
 source = "registry+https://github.com/rust-lang/crates.io-index"
-checksum = "7694e1cfe791f8d31026952abf09c69ca6f6fa4e1a1229e18988f06a04a12dca"
+checksum = "ef927ca75afb808a4d64dd374f00a2adf8d0fcff8e7b184af886c3c87ec4a3f3"
 
 [[package]]
 name = "time-macros"
-version = "0.2.27"
+version = "0.2.18"
 source = "registry+https://github.com/rust-lang/crates.io-index"
-checksum = "2e70e4c5a0e0a8a4823ad65dfe1a6930e4f4d756dcd9dd7939022b5e8c501215"
+checksum = "3f252a68540fde3a3877aeea552b832b40ab9a69e318efd078774a01ddee1ccf"
 dependencies = [
  "num-conv",
  "time-core",
diff --git a/compiler-core/src/erlang.rs b/compiler-core/src/erlang.rs
index fa1ecc2..8e81582 100644
--- a/compiler-core/src/erlang.rs
+++ b/compiler-core/src/erlang.rs
@@ -690,7 +690,7 @@ fn string_inner(value: &str) -> Document<'_> {
         .replace_all(value, |caps: &Captures<'_>| {
             let slashes = caps.get(1).map_or("", |m| m.as_str());
 
-            if slashes.len().is_multiple_of(2) {
+            if slashes.len() % 2 == 0 {
                 format!("{slashes}u")
             } else {
                 format!("{slashes}x")
diff --git a/compiler-core/src/type_/expression.rs b/compiler-core/src/type_/expression.rs
index 0f4d562..54eeb7d 100644
--- a/compiler-core/src/type_/expression.rs
+++ b/compiler-core/src/type_/expression.rs
@@ -921,7 +921,7 @@ impl<'a, 'b> ExprTyper<'a, 'b> {
         // !!!True  // we can remove all but one negation.
         // ```
         if negations > 1 {
-            let location = if negations.is_multiple_of(2) {
+            let location = if negations % 2 == 0 {
                 SrcSpan {
                     start: starting_location.start,
                     end: location.start + 1,
@@ -1006,7 +1006,7 @@ impl<'a, 'b> ExprTyper<'a, 'b> {
         // ---1  // we can remove all but one negation.
         // ```
         if negations > 1 {
-            let location = if negations.is_multiple_of(2) {
+            let location = if negations % 2 == 0 {
                 SrcSpan {
                     start: starting_location.start,
                     end: end + 1,
diff --git a/hexpm/src/lib.rs b/hexpm/src/lib.rs
index 84fbcf1..8cbe1f1 100644
--- a/hexpm/src/lib.rs
+++ b/hexpm/src/lib.rs
@@ -1141,7 +1141,7 @@ impl OAuthDeviceAuthorisation {
         &mut self,
         response: http::Response<Vec<u8>>,
     ) -> Result<PollStep, ApiError> {
-        if self.start_time.elapsed() > Duration::from_mins(10) {
+        if self.start_time.elapsed() > Duration::from_secs(10 * 60) {
             return Err(ApiError::OAuthTimeout);
         }
