diff --git a/sparse_strips/vello_common/src/coarse.rs b/sparse_strips/vello_common/src/coarse.rs
index ced1a4c7..21f95334 100644
--- a/sparse_strips/vello_common/src/coarse.rs
+++ b/sparse_strips/vello_common/src/coarse.rs
@@ -844,7 +844,9 @@ impl WideTile {
     pub fn pop_clip(&mut self) {
         if !self.is_zero_clip() {
             self.pop_buf();
-            self.n_clip -= 1;
+            if self.n_clip > 0 {
+                self.n_clip -= 1;
+            }
         }
     }
 
@@ -855,7 +857,9 @@ impl WideTile {
 
     /// Removes the most recently added zero-winding clip region.
     pub fn pop_zero_clip(&mut self) {
-        self.n_zero_clip -= 1;
+        if self.n_zero_clip > 0 {
+            self.n_zero_clip -= 1;
+        }
     }
 
     /// Checks if the current clip region is a zero-winding clip.
@@ -930,7 +934,9 @@ impl WideTile {
             self.cmds.push(Cmd::PopBuf);
         }
 
-        self.n_bufs -= 1;
+        if self.n_bufs > 0 {
+            self.n_bufs -= 1;
+        }
     }
 
     /// Apply an opacity to the whole buffer.
diff --git a/sparse_strips/vello_sparse_tests/tests/clip.rs b/sparse_strips/vello_sparse_tests/tests/clip.rs
index ad6fb6dc..fd93ca12 100644
--- a/sparse_strips/vello_sparse_tests/tests/clip.rs
+++ b/sparse_strips/vello_sparse_tests/tests/clip.rs
@@ -326,3 +326,4 @@ fn clip_completely_in_out_of_bounds_wide_tile(ctx: &mut impl Renderer) {
     ctx.push_clip_layer(&Rect::new(300.0, 8.0, 350.0, 48.0).to_path(0.1));
     ctx.pop_layer();
 }
+
