diff --git a/Zend/zend_compile.c b/Zend/zend_compile.c
index 0a7ab7da1bd..3c3caa4d4c1 100644
--- a/Zend/zend_compile.c
+++ b/Zend/zend_compile.c
@@ -340,6 +340,7 @@ void zend_oparray_context_begin(zend_oparray_context *prev_context, zend_op_arra
 	CG(context).current_brk_cont = -1;
 	CG(context).last_brk_cont = 0;
 	CG(context).has_assigned_to_http_response_header = false;
+	CG(context).is_function_parameter = false;
 	CG(context).brk_cont_array = NULL;
 	CG(context).labels = NULL;
 	CG(context).in_jmp_frameless_branch = false;
@@ -2886,7 +2887,7 @@ static zend_result zend_try_compile_cv(znode *result, zend_ast *ast, uint32_t ty
 		}
 
 		if (!CG(context).has_assigned_to_http_response_header && zend_string_equals_literal(name, "http_response_header")) {
-			if (type == BP_VAR_R) {
+			if (type == BP_VAR_R && !CG(context).is_function_parameter) {
 				zend_error(E_DEPRECATED,
 					"The predefined locally scoped $http_response_header variable is deprecated,"
 					" call http_get_last_response_headers() instead");
@@ -2934,6 +2935,7 @@ static zend_op *zend_compile_simple_var_no_cv(znode *result, zend_ast *ast, uint
 		// TODO: Have a test case for this?
 		if (name_node.op_type == IS_CONST
 			&& type == BP_VAR_R
+			&& !CG(context).is_function_parameter
 			&& zend_string_equals_literal(Z_STR(name_node.u.constant), "http_response_header")) {
 			zend_error(E_DEPRECATED,
 				"The predefined locally scoped $http_response_header variable is deprecated,"
@@ -7795,6 +7797,8 @@ static void zend_compile_params(zend_ast *ast, zend_ast *return_type_ast, uint32
 		var_node.op_type = IS_CV;
 		var_node.u.op.var = lookup_cv(name);
 
+		CG(context).is_function_parameter = true;
+
 		if (EX_VAR_TO_NUM(var_node.u.op.var) != i) {
 			zend_error_noreturn(E_COMPILE_ERROR, "Redefinition of parameter $%s",
 				ZSTR_VAL(name));
@@ -7986,6 +7990,8 @@ static void zend_compile_params(zend_ast *ast, zend_ast *return_type_ast, uint32
 				}
 			}
 		}
+
+		CG(context).is_function_parameter = false;
 	}
 
 	/* These are assigned at the end to avoid uninitialized memory in case of an error */
diff --git a/Zend/zend_compile.h b/Zend/zend_compile.h
index 6c444340a9e..a5ec2f4d98b 100644
--- a/Zend/zend_compile.h
+++ b/Zend/zend_compile.h
@@ -209,6 +209,7 @@ typedef struct _zend_oparray_context {
 	zend_property_hook_kind active_property_hook_kind;
 	bool       in_jmp_frameless_branch;
 	bool has_assigned_to_http_response_header;
+	bool is_function_parameter;
 } zend_oparray_context;
 
 /* Class, property and method flags                  class|meth.|prop.|const*/
