diff --git a/quickjs.c b/quickjs.c
index 8b9a3b5..1fc03a9 100644
--- a/quickjs.c
+++ b/quickjs.c
@@ -57143,6 +57143,8 @@ void JS_AddIntrinsicTypedArrays(JSContext *ctx)
     JS_SetPropertyFunctionList(ctx, typed_array_base_func,
                                js_typed_array_base_funcs,
                                countof(js_typed_array_base_funcs));
+    /* %TypedArray% is not constructible but must be a constructor for class heritage */
+    JS_SetConstructorBit(ctx, typed_array_base_func, true);
     JS_SetConstructor(ctx, typed_array_base_func, typed_array_base_proto);
 
     /* Used to squelch a -Wcast-function-type warning. */
