diff --git a/crates/oxc_minifier/src/peephole/remove_unused_expression.rs b/crates/oxc_minifier/src/peephole/remove_unused_expression.rs
index fd29723..6254365 100644
--- a/crates/oxc_minifier/src/peephole/remove_unused_expression.rs
+++ b/crates/oxc_minifier/src/peephole/remove_unused_expression.rs
@@ -637,7 +637,7 @@ impl<'a> PeepholeOptimizations {
         if symbol_value.exported {
             return false;
         }
-        if symbol_value.read_references_count > 0 {
+        if symbol_value.read_references_count > 0 || symbol_value.write_references_count > 0 {
             return false;
         }
         *e = assign_expr.right.take_in(ctx.ast);
diff --git a/crates/oxc_minifier/tests/peephole/remove_unused_expression.rs b/crates/oxc_minifier/tests/peephole/remove_unused_expression.rs
index 2af8c72..1e9f367 100644
--- a/crates/oxc_minifier/tests/peephole/remove_unused_expression.rs
+++ b/crates/oxc_minifier/tests/peephole/remove_unused_expression.rs
@@ -140,6 +140,7 @@ fn test_logical_expression() {
     test_same("v = x != null || y");
     test("a == null && (a = b)", "a ??= b");
     test("a != null || (a = b)", "a ??= b");
+    test("let a; a == null && (a = b)", "let a; a ??= b");
     test_same("v = a == null && (a = b)");
     test_same("v = a != null || (a = b)");
     test("void (x == null && y)", "x ?? y");
