diff --git a/src/Riok.Mapperly/Descriptors/MappingBodyBuilders/BuilderContext/MembersMappingBuilderContext.cs b/src/Riok.Mapperly/Descriptors/MappingBodyBuilders/BuilderContext/MembersMappingBuilderContext.cs
index de763f6..9bcc840 100644
--- a/src/Riok.Mapperly/Descriptors/MappingBodyBuilders/BuilderContext/MembersMappingBuilderContext.cs
+++ b/src/Riok.Mapperly/Descriptors/MappingBodyBuilders/BuilderContext/MembersMappingBuilderContext.cs
@@ -137,17 +137,18 @@ public abstract class MembersMappingBuilderContext<T>(MappingBuilderContext buil
         [NotNullWhen(true)] out SourceMemberPath? sourcePath
     )
     {
-        // try to match in additional source members
+        // match against source type members first, since they are the primary source of data
         if (
             BuilderContext.SymbolAccessor.TryFindMemberPath(
-                _state.AdditionalSourceMembers,
+                Mapping.SourceType,
                 pathCandidates,
+                _state.IgnoredSourceMemberNames,
                 ignoreCase,
                 out var sourceMemberPath
             )
         )
         {
-            sourcePath = new SourceMemberPath(sourceMemberPath, SourceMemberType.AdditionalMappingMethodParameter);
+            sourcePath = new SourceMemberPath(sourceMemberPath, SourceMemberType.Member);
             return true;
         }
 
@@ -158,18 +159,17 @@ public abstract class MembersMappingBuilderContext<T>(MappingBuilderContext buil
             return true;
         }
 
-        // match against source type members
+        // try to match in additional source members as fallback
         if (
             BuilderContext.SymbolAccessor.TryFindMemberPath(
-                Mapping.SourceType,
+                _state.AdditionalSourceMembers,
                 pathCandidates,
-                _state.IgnoredSourceMemberNames,
                 ignoreCase,
                 out sourceMemberPath
             )
         )
         {
-            sourcePath = new SourceMemberPath(sourceMemberPath, SourceMemberType.Member);
+            sourcePath = new SourceMemberPath(sourceMemberPath, SourceMemberType.AdditionalMappingMethodParameter);
             return true;
         }
 
diff --git a/test/Riok.Mapperly.Tests/Mapping/UserMethodAdditionalParametersTest.cs b/test/Riok.Mapperly.Tests/Mapping/UserMethodAdditionalParametersTest.cs
index 97ad932..0e6c86f 100644
--- a/test/Riok.Mapperly.Tests/Mapping/UserMethodAdditionalParametersTest.cs
+++ b/test/Riok.Mapperly.Tests/Mapping/UserMethodAdditionalParametersTest.cs
@@ -106,15 +106,15 @@ public class UserMethodAdditionalParametersTest
             .GenerateMapper(source, TestHelperOptions.AllowDiagnostics)
             .Should()
             .HaveDiagnostic(
-                DiagnosticDescriptors.SourceMemberNotMapped,
-                "The member Value on the mapping source type A is not mapped to any member on the mapping target type B"
+                DiagnosticDescriptors.AdditionalParameterNotMapped,
+                "The additional mapping method parameter value of the method Map is not mapped"
             )
             .HaveAssertedAllDiagnostics()
             .HaveMapMethodBody(
                 """
                 var target = new global::B()
                 {
-                    Value = value,
+                    Value = src.Value,
                 };
                 return target;
                 """
@@ -318,15 +318,15 @@ public class UserMethodAdditionalParametersTest
             .GenerateMapper(source, TestHelperOptions.AllowDiagnostics)
             .Should()
             .HaveDiagnostic(
-                DiagnosticDescriptors.SourceMemberNotMapped,
-                "The member value on the mapping source type A is not mapped to any member on the mapping target type B"
+                DiagnosticDescriptors.AdditionalParameterNotMapped,
+                "The additional mapping method parameter value of the method Map is not mapped"
             )
             .HaveAssertedAllDiagnostics()
             .HaveMapMethodBody(
                 """
                 var target = new global::B()
                 {
-                    value = value,
+                    value = src.value,
                 };
                 return target;
                 """
