diff --git a/name.abuchen.portfolio.tests/src/name/abuchen/portfolio/datatransfer/pdf/fintechgroupbank/FinTechGroupBankPDFExtractorTest.java b/name.abuchen.portfolio.tests/src/name/abuchen/portfolio/datatransfer/pdf/fintechgroupbank/FinTechGroupBankPDFExtractorTest.java
index 2b9aca6e4..8ea96496f 100644
--- a/name.abuchen.portfolio.tests/src/name/abuchen/portfolio/datatransfer/pdf/fintechgroupbank/FinTechGroupBankPDFExtractorTest.java
+++ b/name.abuchen.portfolio.tests/src/name/abuchen/portfolio/datatransfer/pdf/fintechgroupbank/FinTechGroupBankPDFExtractorTest.java
@@ -6634,6 +6634,7 @@ public class FinTechGroupBankPDFExtractorTest
                         hasSource("FlatExDegiroKontoauszug08.txt"), hasNote("Portokosten - Versand Kundenformular"))));
     }
 
+
     @Test
     public void testFlatExDeGiroSammelabrechnung01()
     {
diff --git a/name.abuchen.portfolio/src/name/abuchen/portfolio/datatransfer/pdf/FinTechGroupBankPDFExtractor.java b/name.abuchen.portfolio/src/name/abuchen/portfolio/datatransfer/pdf/FinTechGroupBankPDFExtractor.java
index 853b46d0c..f6ff6a798 100644
--- a/name.abuchen.portfolio/src/name/abuchen/portfolio/datatransfer/pdf/FinTechGroupBankPDFExtractor.java
+++ b/name.abuchen.portfolio/src/name/abuchen/portfolio/datatransfer/pdf/FinTechGroupBankPDFExtractor.java
@@ -1694,7 +1694,8 @@ public class FinTechGroupBankPDFExtractor extends AbstractPDFExtractor
                         + "|\\/REC\\/.*" //
                         + "|Pr.mie .*" //
                         + "|R\\-Transaktion" //
-                        + "|Gutschrift aus Kulanz) " //
+                        + "|Gutschrift aus Kulanz" //
+                        + "|Ausgleichszahlung.*)" //
                         + "[\\s]{1,}[\\-\\.,\\d]+[\\+|\\-].*$");
         type.addBlock(depositRemovalblock);
         depositRemovalblock.set(new Transaction<AccountTransaction>()
@@ -1717,8 +1718,9 @@ public class FinTechGroupBankPDFExtractor extends AbstractPDFExtractor
                                         + "|\\/REC\\/.*" //
                                         + "|Pr.mie .*" //
                                         + "|[\\w]+\\-[\\w]+\\-[\\w]+\\-[\\w]+\\-[\\w]+" //
-                                        + "|R\\-Transaktion"
-                                        + "|Gutschrift aus Kulanz))" //
+                                        + "|R\\-Transaktion" //
+                                        + "|Gutschrift aus Kulanz" //
+                                        + "|Ausgleichszahlung.*))" //
                                         + "[\\s]{1,}" //
                                         + "(?<amount>[\\-\\.,\\d]+)" //
                                         + "(?<type>[\\+|\\-])$") //
