diff --git a/Core/Relationships/ResolvedRelationshipsTree.cs b/Core/Relationships/ResolvedRelationshipsTree.cs
index 4ced90d..c266a2d 100644
--- a/Core/Relationships/ResolvedRelationshipsTree.cs
+++ b/Core/Relationships/ResolvedRelationshipsTree.cs
@@ -85,6 +85,7 @@ public IEnumerable<ResolvedRelationship[]> Unsatisfied()
                             // Skip other versions of mods being installed
                         }
                         else if (module.BadRelationships(installing)
+                                  .Where(r => r.Type == RelationshipType.Depends)
                                   .Select(r => relationshipCache.GetValueOrDefault(r.Descriptor))
                                   .OfType<ResolvedRelationship>()
                                   .Select(badRR => new ResolvedRelationship[] { resRel, badRR })
