diff --git a/packages/svelte/src/internal/client/dom/elements/attributes.js b/packages/svelte/src/internal/client/dom/elements/attributes.js
index 2fa5d4541..c63d49064 100644
--- a/packages/svelte/src/internal/client/dom/elements/attributes.js
+++ b/packages/svelte/src/internal/client/dom/elements/attributes.js
@@ -241,7 +241,8 @@ export function set_custom_element_data(node, prop, value) {
 			(setters_cache.has(node.nodeName) ||
 			// customElements may not be available in browser extension contexts
 			!customElements ||
-			customElements.get(node.tagName.toLowerCase())
+			customElements.get(node.tagName.toLowerCase()) ||
+			customElements.get(node.getAttribute('is') || '')
 				? get_setters(node).includes(prop)
 				: value && typeof value === 'object')
 		) {
@@ -535,7 +536,9 @@ function get_attributes(element) {
 	return /** @type {Record<string | symbol, unknown>} **/ (
 		// @ts-expect-error
 		element.__attributes ??= {
-			[IS_CUSTOM_ELEMENT]: element.nodeName.includes('-'),
+			// A custom element can be either an autonomous custom element (<my-widget>)
+			// or a customized built-in element (<button is="my-button">).
+			[IS_CUSTOM_ELEMENT]: element.nodeName.includes('-') || element.hasAttribute('is'),
 			[IS_HTML]: element.namespaceURI === NAMESPACE_HTML
 		}
 	);
