diff --git a/naga/src/front/wgsl/parse/mod.rs b/naga/src/front/wgsl/parse/mod.rs
index 45c5514..e43450a 100644
--- a/naga/src/front/wgsl/parse/mod.rs
+++ b/naga/src/front/wgsl/parse/mod.rs
@@ -2032,6 +2032,7 @@ impl Parser {
         }
 
         let attrib_span = self.pop_rule_span(lexer);
+        let must_use = must_use.value;
         match (bind_group.value, bind_index.value) {
             (Some(group), Some(index)) => {
                 binding = Some(ast::ResourceBinding {
@@ -2063,6 +2064,9 @@ impl Parser {
             }
             (Token::Word("struct"), _) => {
                 ensure_no_diag_attrs("`struct`s".into(), diagnostic_filters)?;
+                if let Some(span) = must_use {
+                    return Err(Box::new(Error::UnknownAttribute(span)));
+                }
 
                 let name = lexer.next_ident()?;
 
@@ -2076,6 +2080,9 @@ impl Parser {
             }
             (Token::Word("alias"), _) => {
                 ensure_no_diag_attrs("`alias`es".into(), diagnostic_filters)?;
+                if let Some(span) = must_use {
+                    return Err(Box::new(Error::UnknownAttribute(span)));
+                }
 
                 let name = lexer.next_ident()?;
 
@@ -2086,6 +2093,9 @@ impl Parser {
             }
             (Token::Word("const"), _) => {
                 ensure_no_diag_attrs("`const`s".into(), diagnostic_filters)?;
+                if let Some(span) = must_use {
+                    return Err(Box::new(Error::UnknownAttribute(span)));
+                }
 
                 let (name, ty) = self.optionally_typed_ident(lexer, &mut ctx)?;
 
@@ -2102,6 +2112,9 @@ impl Parser {
             }
             (Token::Word("override"), _) => {
                 ensure_no_diag_attrs("`override`s".into(), diagnostic_filters)?;
+                if let Some(span) = must_use {
+                    return Err(Box::new(Error::UnknownAttribute(span)));
+                }
 
                 let (name, ty) = self.optionally_typed_ident(lexer, &mut ctx)?;
 
@@ -2122,6 +2135,9 @@ impl Parser {
             }
             (Token::Word("var"), _) => {
                 ensure_no_diag_attrs("`var`s".into(), diagnostic_filters)?;
+                if let Some(span) = must_use {
+                    return Err(Box::new(Error::UnknownAttribute(span)));
+                }
 
                 let mut var = self.variable_decl(lexer, &mut ctx)?;
                 var.binding = binding.take();
@@ -2139,7 +2155,7 @@ impl Parser {
                 let function = self.function_decl(
                     lexer,
                     diagnostic_filter_leaf,
-                    must_use.value,
+                    must_use,
                     out,
                     &mut dependencies,
                 )?;
@@ -2179,6 +2195,9 @@ impl Parser {
             }
             (Token::Word("const_assert"), _) => {
                 ensure_no_diag_attrs("`const_assert`s".into(), diagnostic_filters)?;
+                if let Some(span) = must_use {
+                    return Err(Box::new(Error::UnknownAttribute(span)));
+                }
 
                 // parentheses are optional
                 let paren = lexer.next_if(Token::Paren('('));
