diff --git a/naga/src/front/wgsl/parse/conv.rs b/naga/src/front/wgsl/parse/conv.rs
index 6270d4b..9fb003d 100644
--- a/naga/src/front/wgsl/parse/conv.rs
+++ b/naga/src/front/wgsl/parse/conv.rs
@@ -583,6 +583,9 @@ pub fn map_predeclared_type(
         PredeclaredType::TypeGenerator(TypeGenerator::CooperativeMatrix { .. }) => {
             Some(&[ImplementedEnableExtension::WgpuCooperativeMatrix])
         }
+        PredeclaredType::TypeGenerator(TypeGenerator::BindingArray) => {
+            Some(&[ImplementedEnableExtension::WgpuBindingArray])
+        }
         _ => None,
     };
     if let Some(extensions_needed) = extensions_needed {
diff --git a/naga/src/front/wgsl/parse/directive/enable_extension.rs b/naga/src/front/wgsl/parse/directive/enable_extension.rs
index a6d7285..8b5bfc8 100644
--- a/naga/src/front/wgsl/parse/directive/enable_extension.rs
+++ b/naga/src/front/wgsl/parse/directive/enable_extension.rs
@@ -22,6 +22,7 @@ pub(crate) struct EnableExtensions {
     draw_index: bool,
     primitive_index: bool,
     per_vertex: bool,
+    wgpu_binding_array: bool,
 }
 
 impl EnableExtensions {
@@ -38,6 +39,7 @@ impl EnableExtensions {
             draw_index: false,
             primitive_index: false,
             per_vertex: false,
+            wgpu_binding_array: false,
         }
     }
 
@@ -59,6 +61,7 @@ impl EnableExtensions {
             ImplementedEnableExtension::DrawIndex => &mut self.draw_index,
             ImplementedEnableExtension::PrimitiveIndex => &mut self.primitive_index,
             ImplementedEnableExtension::PerVertex => &mut self.per_vertex,
+            ImplementedEnableExtension::WgpuBindingArray => &mut self.wgpu_binding_array,
         };
         *field = true;
     }
@@ -79,6 +82,7 @@ impl EnableExtensions {
             ImplementedEnableExtension::DrawIndex => self.draw_index,
             ImplementedEnableExtension::PrimitiveIndex => self.primitive_index,
             ImplementedEnableExtension::PerVertex => self.per_vertex,
+            ImplementedEnableExtension::WgpuBindingArray => self.wgpu_binding_array,
         }
     }
 
@@ -129,6 +133,7 @@ impl EnableExtension {
     const RAY_TRACING_PIPELINE: &'static str = "wgpu_ray_tracing_pipeline";
     const COOPERATIVE_MATRIX: &'static str = "wgpu_cooperative_matrix";
     const SUBGROUPS: &'static str = "subgroups";
+    const BINDING_ARRAY: &'static str = "wgpu_binding_array";
     const PRIMITIVE_INDEX: &'static str = "primitive_index";
     const DRAW_INDEX: &'static str = "draw_index";
     const PER_VERTEX: &'static str = "wgpu_per_vertex";
@@ -156,6 +161,7 @@ impl EnableExtension {
             Self::DRAW_INDEX => Self::Implemented(ImplementedEnableExtension::DrawIndex),
             Self::PRIMITIVE_INDEX => Self::Implemented(ImplementedEnableExtension::PrimitiveIndex),
             Self::PER_VERTEX => Self::Implemented(ImplementedEnableExtension::PerVertex),
+            Self::BINDING_ARRAY => Self::Implemented(ImplementedEnableExtension::WgpuBindingArray),
             _ => return Err(Box::new(Error::UnknownEnableExtension(span, word))),
         })
     }
@@ -177,6 +183,7 @@ impl EnableExtension {
                 ImplementedEnableExtension::PrimitiveIndex => Self::PRIMITIVE_INDEX,
                 ImplementedEnableExtension::WgpuRayTracingPipeline => Self::RAY_TRACING_PIPELINE,
                 ImplementedEnableExtension::PerVertex => Self::PER_VERTEX,
+                ImplementedEnableExtension::WgpuBindingArray => Self::BINDING_ARRAY,
             },
             Self::Unimplemented(kind) => match kind {
                 UnimplementedEnableExtension::Subgroups => Self::SUBGROUPS,
@@ -227,6 +234,8 @@ pub enum ImplementedEnableExtension {
     PrimitiveIndex,
     /// Enables the `wgpu_per_vertex` extension, allows using `@interpolate(per_vertex)` attribute in WGSL, native only.
     PerVertex,
+    /// Enables the `wgpu_binding_array` extension, native only.
+    WgpuBindingArray,
 }
 
 impl ImplementedEnableExtension {
@@ -243,6 +252,7 @@ impl ImplementedEnableExtension {
         Self::DrawIndex,
         Self::PrimitiveIndex,
         Self::PerVertex,
+        Self::WgpuBindingArray,
     ];
 
     /// Returns slice of all variants of [`ImplementedEnableExtension`].
@@ -265,6 +275,7 @@ impl ImplementedEnableExtension {
             Self::DrawIndex => C::DRAW_INDEX,
             Self::PrimitiveIndex => C::PRIMITIVE_INDEX,
             Self::PerVertex => C::PER_VERTEX,
+            Self::WgpuBindingArray => C::TEXTURE_AND_SAMPLER_BINDING_ARRAY,
         }
     }
 }
diff --git a/naga/src/valid/mod.rs b/naga/src/valid/mod.rs
index cbc10ce..4f26332 100644
--- a/naga/src/valid/mod.rs
+++ b/naga/src/valid/mod.rs
@@ -238,6 +238,7 @@ impl Capabilities {
             Self::COOPERATIVE_MATRIX => Some(Ext::WgpuCooperativeMatrix),
             Self::RAY_TRACING_PIPELINE => Some(Ext::WgpuRayTracingPipeline),
             Self::PER_VERTEX => Some(Ext::PerVertex),
+            Self::TEXTURE_AND_SAMPLER_BINDING_ARRAY => Some(Ext::WgpuBindingArray),
             _ => None,
         }
     }
diff --git a/naga/tests/in/wgsl/binding-arrays.wgsl b/naga/tests/in/wgsl/binding-arrays.wgsl
index d1bf121..0f7e955 100644
--- a/naga/tests/in/wgsl/binding-arrays.wgsl
+++ b/naga/tests/in/wgsl/binding-arrays.wgsl
@@ -1,3 +1,5 @@
+enable wgpu_binding_array;
+
 struct UniformIndex {
     index: u32
 };
diff --git a/naga/tests/in/wgsl/binding-buffer-arrays.wgsl b/naga/tests/in/wgsl/binding-buffer-arrays.wgsl
index fb25623..7a8748c 100644
--- a/naga/tests/in/wgsl/binding-buffer-arrays.wgsl
+++ b/naga/tests/in/wgsl/binding-buffer-arrays.wgsl
@@ -1,3 +1,5 @@
+enable wgpu_binding_array;
+
 struct UniformIndex {
     index: u32
 }
diff --git a/naga/tests/naga/wgsl_errors.rs b/naga/tests/naga/wgsl_errors.rs
index 97bed72..7adb1df 100644
--- a/naga/tests/naga/wgsl_errors.rs
+++ b/naga/tests/naga/wgsl_errors.rs
@@ -5013,6 +5013,31 @@ fn fs_main(@location(0) @interpolate(per_vertex) v: array<f32, 3>) -> @location(
     );
 }
 
+#[test]
+fn binding_array_enable_extension() {
+    check_extension_validation!(
+        Capabilities::TEXTURE_AND_SAMPLER_BINDING_ARRAY,
+        r#"@group(0) @binding(0)
+var texture_arr: binding_array<texture_2d<f32>>;
+        "#,
+        r#"error: the `wgpu_binding_array` enable extension is not enabled
+  ┌─ wgsl:2:18
+  │
+2 │ var texture_arr: binding_array<texture_2d<f32>>;
+  │                  ^^^^^^^^^^^^^ the `wgpu_binding_array` "Enable Extension" is needed for this functionality, but it is not currently enabled.
+  │
+  = note: You can enable this extension by adding `enable wgpu_binding_array;` at the top of the shader, before any other items.
+
+"#,
+        Err(naga::valid::ValidationError::GlobalVariable {
+            source: naga::valid::GlobalVariableError::UnsupportedCapability(
+                Capabilities::TEXTURE_AND_SAMPLER_BINDING_ARRAY
+            ),
+            ..
+        })
+    );
+}
+
 /// Checks that every ray tracing pipeline binding in naga is invalid in other stages.
 #[test]
 fn check_ray_tracing_pipeline_bindings() {
