diff --git a/torchtune/models/gemma/_tokenizer.py b/torchtune/models/gemma/_tokenizer.py
index d4655c5f..d38afe26 100644
--- a/torchtune/models/gemma/_tokenizer.py
+++ b/torchtune/models/gemma/_tokenizer.py
@@ -99,6 +99,8 @@ class GemmaTokenizer(ModelTokenizer, Transform):
     def tokenize_messages(
         self,
         messages: List[Message],
+        *,
+        add_eos: bool = True,
     ) -> Tuple[List[int], List[bool]]:
         r"""Tokenize a list of messages one at a time then concatenate them,
         returning a list of tokens and a list of masks.
@@ -125,6 +127,7 @@ class GemmaTokenizer(ModelTokenizer, Transform):
         Args:
             messages (List[Message]): A list of messages, each containing role, content,
                 and masked attributes.
+            add_eos (bool): Whether to add the EOS token at the end of assistant messages. Default True.
 
         Returns:
             Tuple[List[int], List[bool]]: The tokenized messages
@@ -139,6 +142,7 @@ class GemmaTokenizer(ModelTokenizer, Transform):
             messages=templated_messages,
             bos_id=self.bos_id,
             eos_id=self.eos_id,
+            add_eos=add_eos,
         )
 
     def __call__(
diff --git a/torchtune/models/mistral/_tokenizer.py b/torchtune/models/mistral/_tokenizer.py
index b541c993..aadd9913 100644
--- a/torchtune/models/mistral/_tokenizer.py
+++ b/torchtune/models/mistral/_tokenizer.py
@@ -122,7 +122,7 @@ class MistralTokenizer(ModelTokenizer, Transform):
         return self._spm_model.decode(token_ids)
 
     def tokenize_messages(
-        self, messages: List[Message]
+        self, messages: List[Message], *, add_eos: bool = True
     ) -> Tuple[List[int], List[bool]]:
         r"""Tokenize a list of messages one at a time then concatenate them,
         returning a list of tokens and a list of masks.
@@ -154,6 +154,7 @@ class MistralTokenizer(ModelTokenizer, Transform):
         Args:
             messages (List[Message]): A list of messages, each containing role, content,
                 and masked attributes.
+            add_eos (bool): Whether to add the EOS token at the end of assistant messages. Default True.
 
         Returns:
             Tuple[List[int], List[bool]]: The tokenized messages
@@ -168,6 +169,7 @@ class MistralTokenizer(ModelTokenizer, Transform):
             messages=templated_messages,
             bos_id=self.bos_id,
             eos_id=self.eos_id,
+            add_eos=add_eos,
         )
 
     def __call__(
diff --git a/torchtune/modules/tokenizers/_utils.py b/torchtune/modules/tokenizers/_utils.py
index b580eda1..0fe67485 100644
--- a/torchtune/modules/tokenizers/_utils.py
+++ b/torchtune/modules/tokenizers/_utils.py
@@ -78,6 +78,7 @@ def tokenize_messages_no_special_tokens(
     *,
     bos_id: Optional[int] = None,
     eos_id: Optional[int] = None,
+    add_eos: bool = True,
 ) -> Tuple[List[int], List[bool]]:
     r"""Tokenize a list of messages one at a time then concatenate them,
     returning a list of tokens and a list of masks. Does not add any special
@@ -111,6 +112,7 @@ def tokenize_messages_no_special_tokens(
         bos_id (Optional[int]): Beginning-of-sequence token id. If None, no BOS token will
             be added. Default None.
         eos_id (Optional[int]): End-of-sequence token id. If None, no EOS token will be added. Default None.
+        add_eos (bool): Whether to add the EOS token at the end of assistant messages. Default True.
 
     Returns:
         Tuple[List[int], List[bool]]: The tokenized messages.
@@ -156,9 +158,9 @@ def tokenize_messages_no_special_tokens(
         tokenized_messages.extend(tokens)
         mask.extend([message.masked] * len(tokens))
 
-        # If assistant message, append EOS at end
+        # If assistant message, append EOS at end if add_eos is True
         if end_of_turn:
-            if eos_id is not None:
+            if eos_id is not None and add_eos:
                 tokenized_messages.append(eos_id)
                 mask.append(message.masked)
             end_of_turn = False
