diff --git a/packages/core/src/utils/EntityComparator.ts b/packages/core/src/utils/EntityComparator.ts
index 7b81203..c9025b8 100644
--- a/packages/core/src/utils/EntityComparator.ts
+++ b/packages/core/src/utils/EntityComparator.ts
@@ -634,21 +634,30 @@ export class EntityComparator {
     if (object) {
       ret += `${padding}if (${nullCond}) ret${dataKey} = null;\n`;
     } else {
-      ret += `${padding}if (${nullCond}) {\n`;
-      ret +=
-        meta.props
+      const getNullAssignments = (embeddedProp: EntityProperty<T>, embeddedDataKey: string): string[] => {
+        return meta.props
           .filter(
             p =>
-              p.embedded?.[0] === prop.name &&
+              p.embedded?.[0] === embeddedProp.name &&
               // object for JSON embeddable
               (p.object || p.persist !== false),
           )
-          .map(childProp => {
+          .flatMap(childProp => {
             const childDataKey =
-              meta.embeddable || prop.object ? dataKey + this.wrap(childProp.embedded![1]) : this.wrap(childProp.name);
+              meta.embeddable || embeddedProp.object
+                ? embeddedDataKey + this.wrap(childProp.embedded![1])
+                : this.wrap(childProp.name);
+
+            if (childProp.kind === ReferenceKind.EMBEDDED && !childProp.object && !childProp.array) {
+              return getNullAssignments(childProp, childDataKey);
+            }
+
             return `${padding}  ret${childDataKey} = null;`;
-          })
-          .join('\n') + `\n`;
+          });
+      };
+
+      ret += `${padding}if (${nullCond}) {\n`;
+      ret += getNullAssignments(prop, dataKey).join('\n') + `\n`;
       ret += `${padding}}\n`;
     }
 
