diff --git a/eng/Versions.props b/eng/Versions.props
index d9d60dcf6d..1cfa529f66 100644
--- a/eng/Versions.props
+++ b/eng/Versions.props
@@ -11,7 +11,7 @@
     -->
     <StabilizePackageVersion Condition="'$(StabilizePackageVersion)' == ''">false</StabilizePackageVersion>
     <DotNetFinalVersionKind Condition="'$(StabilizePackageVersion)' == 'true'">release</DotNetFinalVersionKind>
-    <DefaultNetCoreTargetFramework>net10.0</DefaultNetCoreTargetFramework>
+    <DefaultNetCoreTargetFramework>net9.0</DefaultNetCoreTargetFramework>
   </PropertyGroup>
   <PropertyGroup Label="Arcade settings">
     <UsingToolXliff>False</UsingToolXliff>
diff --git a/global.json b/global.json
index 50294104c6..65198d70f8 100644
--- a/global.json
+++ b/global.json
@@ -1,6 +1,6 @@
 {
   "sdk": {
-    "version": "10.0.100-rc.1.25411.109",
+    "version": "9.0.306",
     "allowPrerelease": true,
     "rollForward": "latestMajor",
     "paths": [
diff --git a/src/EFCore.Design/Migrations/Design/CSharpSnapshotGenerator.cs b/src/EFCore.Design/Migrations/Design/CSharpSnapshotGenerator.cs
index 9938ce572e..feb6b64d1e 100644
--- a/src/EFCore.Design/Migrations/Design/CSharpSnapshotGenerator.cs
+++ b/src/EFCore.Design/Migrations/Design/CSharpSnapshotGenerator.cs
@@ -569,7 +569,7 @@ public virtual void Generate(string modelBuilderName, IModel model, IndentedStri
             .AppendLine()
             .Append(builderName)
             .Append(complexProperty.IsCollection ? ".ComplexCollection(" : ".ComplexProperty(")
-            .Append($"typeof({Code.Reference(Model.DefaultPropertyBagType)}), ")
+            .Append($"typeof({Code.Reference(complexProperty.ClrType)}), ")
             .Append($"{Code.Literal(complexProperty.Name)}, {Code.Literal(complexType.Name)}, ")
             .Append(complexTypeBuilderName)
             .AppendLine(" =>");
