diff --git a/pkg/apis/kops/containerdconfig.go b/pkg/apis/kops/containerdconfig.go
index 9efe761517..85d9996afd 100644
--- a/pkg/apis/kops/containerdconfig.go
+++ b/pkg/apis/kops/containerdconfig.go
@@ -74,6 +74,9 @@ type NvidiaGPUConfig struct {
 	Enabled *bool `json:"enabled,omitempty"`
 	// DCGMExporterConfig configures the DCGM exporter
 	DCGMExporter *DCGMExporterConfig `json:"dcgmExporter,omitempty"`
+	// DevicePluginVersion is the version tag for the NVIDIA k8s-device-plugin image.
+	// Default is "v0.12.2".
+	DevicePluginVersion string `json:"devicePluginVersion,omitempty"`
 }
 
 // DCGMExporterConfig configures the DCGM exporter.
diff --git a/pkg/apis/kops/v1alpha2/containerdconfig.go b/pkg/apis/kops/v1alpha2/containerdconfig.go
index e4430b7908..3152d0ffae 100644
--- a/pkg/apis/kops/v1alpha2/containerdconfig.go
+++ b/pkg/apis/kops/v1alpha2/containerdconfig.go
@@ -71,6 +71,9 @@ type NvidiaGPUConfig struct {
 	Enabled *bool `json:"enabled,omitempty"`
 	// DCGMExporterConfig configures the DCGM exporter
 	DCGMExporter *DCGMExporterConfig `json:"dcgmExporter,omitempty"`
+	// DevicePluginVersion is the version tag for the NVIDIA k8s-device-plugin image.
+	// Default is "v0.12.2".
+	DevicePluginVersion string `json:"devicePluginVersion,omitempty"`
 }
 
 // DCGMExporterConfig configures the DCGM exporter.
diff --git a/pkg/apis/kops/v1alpha3/containerdconfig.go b/pkg/apis/kops/v1alpha3/containerdconfig.go
index be69f3e9a7..8522d3e84b 100644
--- a/pkg/apis/kops/v1alpha3/containerdconfig.go
+++ b/pkg/apis/kops/v1alpha3/containerdconfig.go
@@ -71,6 +71,9 @@ type NvidiaGPUConfig struct {
 	Enabled *bool `json:"enabled,omitempty"`
 	// DCGMExporterConfig configures the DCGM exporter
 	DCGMExporter *DCGMExporterConfig `json:"dcgmExporter,omitempty"`
+	// DevicePluginVersion is the version tag for the NVIDIA k8s-device-plugin image.
+	// Default is "v0.12.2".
+	DevicePluginVersion string `json:"devicePluginVersion,omitempty"`
 }
 
 // DCGMExporterConfig configures the DCGM exporter.
diff --git a/upup/models/cloudup/resources/addons/nvidia.addons.k8s.io/k8s-1.16.yaml.template b/upup/models/cloudup/resources/addons/nvidia.addons.k8s.io/k8s-1.16.yaml.template
index 1aff485afb..780a6a054b 100644
--- a/upup/models/cloudup/resources/addons/nvidia.addons.k8s.io/k8s-1.16.yaml.template
+++ b/upup/models/cloudup/resources/addons/nvidia.addons.k8s.io/k8s-1.16.yaml.template
@@ -17,7 +17,7 @@ spec:
         name: nvidia-device-plugin-ds
     spec:
       containers:
-      - image: nvcr.io/nvidia/k8s-device-plugin:v0.12.2
+      - image: nvcr.io/nvidia/k8s-device-plugin:{{ if .Containerd.NvidiaGPU.DevicePluginVersion }}{{ .Containerd.NvidiaGPU.DevicePluginVersion }}{{ else }}v0.12.2{{ end }}
         name: nvidia-device-plugin-ctr
         args: ["--fail-on-init-error=false"]
         securityContext:
