diff --git a/harper-core/src/linting/lint_group.rs b/harper-core/src/linting/lint_group.rs
index 1a7c3ad..e48fba6 100644
--- a/harper-core/src/linting/lint_group.rs
+++ b/harper-core/src/linting/lint_group.rs
@@ -78,6 +78,7 @@ use super::few_units_of_time_ago::FewUnitsOfTimeAgo;
 use super::filler_words::FillerWords;
 use super::find_fine::FindFine;
 use super::first_aid_kit::FirstAidKit;
+use super::flaunt_flout::FlauntFlout;
 use super::flesh_out_vs_full_fledged::FleshOutVsFullFledged;
 use super::for_noun::ForNoun;
 use super::free_predicate::FreePredicate;
@@ -664,6 +665,7 @@ impl LintGroup {
         insert_expr_rule!(FillerWords, true);
         insert_struct_rule!(FindFine, true);
         insert_expr_rule!(FirstAidKit, true);
+        insert_expr_rule!(FlauntFlout, true);
         insert_expr_rule!(FleshOutVsFullFledged, true);
         insert_expr_rule!(ForNoun, true);
         insert_expr_rule!(FreePredicate, true);
diff --git a/harper-core/src/linting/mod.rs b/harper-core/src/linting/mod.rs
index 0b7af3e..7ba70a9 100644
--- a/harper-core/src/linting/mod.rs
+++ b/harper-core/src/linting/mod.rs
@@ -73,6 +73,7 @@ mod few_units_of_time_ago;
 mod filler_words;
 mod find_fine;
 mod first_aid_kit;
+mod flaunt_flout;
 mod flesh_out_vs_full_fledged;
 mod for_noun;
 mod free_predicate;
