diff --git a/src/MudBlazor/Services/Dialog/DialogService.cs b/src/MudBlazor/Services/Dialog/DialogService.cs
index 4671505..97d3f24 100644
--- a/src/MudBlazor/Services/Dialog/DialogService.cs
+++ b/src/MudBlazor/Services/Dialog/DialogService.cs
@@ -80,6 +80,12 @@ namespace MudBlazor
             return ShowAsync<T>(title, DialogParameters.Default, options);
         }
 
+        /// <inheritdoc />
+        public Task<IDialogReference> ShowAsync<[DynamicallyAccessedMembers(DynamicallyAccessedMemberTypes.All)] T>(DialogParameters<T> parameters) where T : IComponent
+        {
+            return ShowAsync<T>(string.Empty, parameters, DialogOptions.Default);
+        }
+
         /// <inheritdoc />
         public Task<IDialogReference> ShowAsync<[DynamicallyAccessedMembers(DynamicallyAccessedMemberTypes.All)] T>(string? title, DialogParameters parameters) where T : IComponent
         {
diff --git a/src/MudBlazor/Services/Dialog/IDialogService.cs b/src/MudBlazor/Services/Dialog/IDialogService.cs
index 2475b3a..6debe6e 100644
--- a/src/MudBlazor/Services/Dialog/IDialogService.cs
+++ b/src/MudBlazor/Services/Dialog/IDialogService.cs
@@ -52,6 +52,14 @@ namespace MudBlazor
         /// <returns>A reference to the dialog.</returns>
         Task<IDialogReference> ShowAsync<[DynamicallyAccessedMembers(DynamicallyAccessedMemberTypes.All)] TComponent>(string? title, DialogOptions options) where TComponent : IComponent;
 
+        /// <summary>
+        /// Displays a dialog with parameters.
+        /// </summary>
+        /// <typeparam name="TComponent">The dialog to display.</typeparam>
+        /// <param name="parameters">The custom parameters to set within the dialog.</param>
+        /// <returns>A reference to the dialog.</returns>
+        Task<IDialogReference> ShowAsync<[DynamicallyAccessedMembers(DynamicallyAccessedMemberTypes.All)] TComponent>(DialogParameters<TComponent> parameters) where TComponent : IComponent;
+
         /// <summary>
         /// Displays a dialog with a custom title and parameters.
         /// </summary>
