diff --git a/src/cfnlint/rules/resources/lmbd/SnapStartSupported.py b/src/cfnlint/rules/resources/lmbd/SnapStartSupported.py
index b55b662a4..1c8acda83 100644
--- a/src/cfnlint/rules/resources/lmbd/SnapStartSupported.py
+++ b/src/cfnlint/rules/resources/lmbd/SnapStartSupported.py
@@ -18,7 +18,7 @@ class SnapStartSupported(CfnLintKeyword):
     id = "E2530"
     shortdesc = "SnapStart supports the configured runtime"
     description = (
-        "To properly leverage SnapStart, you must have a runtime of Java11 or greater"
+        "To properly leverage SnapStart, you must have a supported runtime (Java11 or greater, Python 3.12, or .NET)"
     )
     source_url = "https://docs.aws.amazon.com/lambda/latest/dg/snapstart.html"
     tags = ["resources", "lambda"]
@@ -97,7 +97,8 @@ class SnapStartSupported(CfnLintKeyword):
             if (
                 runtime
                 and (not runtime.startswith("java"))
-                and runtime not in ["java8.al2", "java8"]
+                and (not runtime.startswith("dotnet"))
+                and runtime not in ["java8.al2", "java8", "python3.12"]
             ):
                 yield ValidationError(
                     f"{runtime!r} is not supported for 'SnapStart' enabled functions",
