diff --git a/config/src/main/java/com/alibaba/nacos/config/server/controller/v3/ConfigControllerV3.java b/config/src/main/java/com/alibaba/nacos/config/server/controller/v3/ConfigControllerV3.java
index 36f19d5..bfba96d 100644
--- a/config/src/main/java/com/alibaba/nacos/config/server/controller/v3/ConfigControllerV3.java
+++ b/config/src/main/java/com/alibaba/nacos/config/server/controller/v3/ConfigControllerV3.java
@@ -498,9 +498,14 @@ public class ConfigControllerV3 {
     private Result<Map<String, Object>> parseImportDataV2(String srcUser, ZipUtils.UnZipResult unziped,
             List<ConfigAllInfo> configInfoList, List<Map<String, String>> unrecognizedList, String namespaceId) {
         ZipUtils.ZipItem metaDataItem = unziped.getMetaDataItem();
-        String metaData = metaDataItem.getItemData();
         Map<String, Object> failedData = new HashMap<>(4);
-        
+
+        if (metaDataItem == null || StringUtils.isBlank(metaDataItem.getItemData())) {
+            failedData.put("succCount", 0);
+            return Result.failure(ErrorCode.METADATA_ILLEGAL, failedData);
+        }
+        String metaData = metaDataItem.getItemData();
+
         ConfigMetadata configMetadata = YamlParserUtil.loadObject(metaData, ConfigMetadata.class);
         if (configMetadata == null || CollectionUtils.isEmpty(configMetadata.getMetadata())) {
             failedData.put("succCount", 0);
@@ -515,12 +520,12 @@ public class ConfigControllerV3 {
                 return Result.failure(ErrorCode.METADATA_ILLEGAL, failedData);
             }
         }
-        
+
         List<ZipUtils.ZipItem> zipItemList = unziped.getZipItemList();
         Set<String> metaDataKeys = configExportItems.stream()
                 .map(metaItem -> GroupKey.getKey(metaItem.getDataId(), metaItem.getGroup()))
                 .collect(Collectors.toSet());
-        
+
         Map<String, String> configContentMap = new HashMap<>(zipItemList.size());
         int itemNameLength = 2;
         zipItemList.forEach(item -> {
@@ -532,7 +537,7 @@ public class ConfigControllerV3 {
                 unrecognizedList.add(unrecognizedItem);
                 return;
             }
-            
+
             String group = groupAdnDataId[0];
             String dataId = groupAdnDataId[1];
             String key = GroupKey.getKey(dataId, group);
@@ -546,7 +551,7 @@ public class ConfigControllerV3 {
             String itemData = item.getItemData();
             configContentMap.put(key, itemData);
         });
-        
+
         for (ConfigMetadata.ConfigExportItem configExportItem : configExportItems) {
             String dataId = configExportItem.getDataId();
             String group = configExportItem.getGroup();
@@ -561,7 +566,7 @@ public class ConfigControllerV3 {
             // encrypted
             Pair<String, String> pair = EncryptionHandler.encryptHandler(dataId, content);
             content = pair.getSecond();
-            
+
             ConfigAllInfo ci = new ConfigAllInfo();
             ci.setGroup(group);
             ci.setDataId(dataId);
@@ -574,7 +579,7 @@ public class ConfigControllerV3 {
             ci.setCreateUser(srcUser);
             configInfoList.add(ci);
         }
-        
+
         return null;
     }
