diff --git a/plugins/inputs/turbostat/turbostat.go b/plugins/inputs/turbostat/turbostat.go
index 48671e6..a218f51 100644
--- a/plugins/inputs/turbostat/turbostat.go
+++ b/plugins/inputs/turbostat/turbostat.go
@@ -69,9 +69,11 @@ type Turbostat struct {
 }
 
 type column struct {
-	name   string
-	isTag  bool
-	isTime bool
+	name     string
+	header   string
+	isTag    bool
+	isTime   bool
+	isHeader bool
 }
 
 func (*Turbostat) SampleConfig() string {
@@ -128,11 +130,7 @@ func (t *Turbostat) Start(acc telegraf.Accumulator) error {
 			columns = append(columns, createColumn(s))
 		}
 		for scanner.Scan() {
-			line := scanner.Text()
-			if line == header {
-				continue
-			}
-			values := strings.Fields(line)
+			values := strings.Fields(scanner.Text())
 			acc.AddError(processValues(acc, columns, values))
 		}
 		acc.AddError(scanner.Err())
@@ -165,6 +163,9 @@ func processValues(acc telegraf.Accumulator, columns []column, values []string)
 	timestamp := time.Now()
 	for i, value := range values {
 		column := columns[i]
+		if column.isHeader && value == column.header {
+			return nil
+		}
 		switch {
 		case column.isTime:
 			f, err := strconv.ParseFloat(value, 64)
@@ -191,6 +192,7 @@ func processValues(acc telegraf.Accumulator, columns []column, values []string)
 
 // Creates a column struct from a Turbostat column name.
 func createColumn(s string) column {
+	header := s
 	// Add underscores around words known not to have delimiters.
 	s = replacer.Replace(s)
 	// Split the column name into lower case tokens.
@@ -204,14 +206,14 @@ func createColumn(s string) column {
 
 	// Create and return the column.
 	name := strings.Join(tokens, "_")
+	col := column{name: name, header: header, isHeader: true}
 	switch name {
 	case "time_of_day_seconds":
-		return column{name: name, isTime: true}
+		col.isTime = true
 	case "package", "node", "die", "core", "cpu", "apic", "x2apic":
-		return column{name: name, isTag: true}
-	default:
-		return column{name: name}
+		col.isTag = true
 	}
+	return col
 }
 
 func init() {
diff --git a/plugins/inputs/turbostat/turbostat_test.go b/plugins/inputs/turbostat/turbostat_test.go
index a10f56e..955cc29 100644
--- a/plugins/inputs/turbostat/turbostat_test.go
+++ b/plugins/inputs/turbostat/turbostat_test.go
@@ -25,15 +25,15 @@ func TestCreateColumn(t *testing.T) {
 		in  string
 		out column
 	}{
-		{in: "Core", out: column{name: "core", isTag: true}},
-		{in: "Busy%", out: column{name: "busy_percent"}},
-		{in: "Bzy_MHz", out: column{name: "busy_frequency_mhz"}},
-		{in: "C3-", out: column{name: "c3_minus"}},
-		{in: "CoreThr", out: column{name: "core_throttle"}},
-		{in: "Cor_J", out: column{name: "core_energy_joule"}},
-		{in: "CorWatt", out: column{name: "core_power_watt"}},
-		{in: "UMHz1.0", out: column{name: "uncore_frequency_mhz_1_0"}},
-		{in: "Time_Of_Day_Seconds", out: column{name: "time_of_day_seconds", isTime: true}},
+		{in: "Core", out: column{name: "core", header: "Core", isTag: true, isHeader: true}},
+		{in: "Busy%", out: column{name: "busy_percent", header: "Busy%", isHeader: true}},
+		{in: "Bzy_MHz", out: column{name: "busy_frequency_mhz", header: "Bzy_MHz", isHeader: true}},
+		{in: "C3-", out: column{name: "c3_minus", header: "C3-", isHeader: true}},
+		{in: "CoreThr", out: column{name: "core_throttle", header: "CoreThr", isHeader: true}},
+		{in: "Cor_J", out: column{name: "core_energy_joule", header: "Cor_J", isHeader: true}},
+		{in: "CorWatt", out: column{name: "core_power_watt", header: "CorWatt", isHeader: true}},
+		{in: "UMHz1.0", out: column{name: "uncore_frequency_mhz_1_0", header: "UMHz1.0", isHeader: true}},
+		{in: "Time_Of_Day_Seconds", out: column{name: "time_of_day_seconds", header: "Time_Of_Day_Seconds", isTime: true, isHeader: true}},
 	}
 	for _, tt := range tests {
 		t.Run(tt.in, func(t *testing.T) {
