diff --git a/grammar/rainerscript.c b/grammar/rainerscript.c
index 82638305f..4cb7617d8 100644
--- a/grammar/rainerscript.c
+++ b/grammar/rainerscript.c
@@ -1990,7 +1990,7 @@ static void ATTR_NONNULL() doFunct_RandomGen(struct cnffunc *__restrict__ const
     }
     x = labs(randomNumber());
     if (max > MAX_RANDOM_NUMBER) {
-        DBGPRINTF(
+        LogMsg(0, RS_RET_VAL_OUT_OF_RANGE, LOG_WARNING,
             "rainerscript: desired random-number range [0 - %lld] "
             "is wider than supported limit of [0 - %d)\n",
             max, MAX_RANDOM_NUMBER);
diff --git a/tests/check_relpEngineVersion b/tests/check_relpEngineVersion
new file mode 100755
index 000000000..deb4b431f
Binary files /dev/null and b/tests/check_relpEngineVersion differ
diff --git a/tests/testbench_test_failed_rsyslog b/tests/testbench_test_failed_rsyslog
new file mode 100644
index 000000000..cc951e233
--- /dev/null
+++ b/tests/testbench_test_failed_rsyslog
@@ -0,0 +1 @@
+./tests/omfile-read-only.sh
