diff --git a/ext/reflection/php_reflection.c b/ext/reflection/php_reflection.c
index 3a37304dc2d..27b2ec4e5d7 100644
--- a/ext/reflection/php_reflection.c
+++ b/ext/reflection/php_reflection.c
@@ -1043,6 +1043,9 @@ static void _property_string(smart_str *str, zend_property_info *prop, const cha
 			smart_str_appendc(str, ' ');
 			zend_string_release(type_str);
 		}
+		if (prop->hooks) {
+			smart_str_appends(str, "<hooks> ");
+		}
 		if (!prop_name) {
 			const char *class_name;
 			zend_unmangle_property_name(prop->name, &class_name, &prop_name);
