diff --git a/weasyprint/formatting_structure/boxes.py b/weasyprint/formatting_structure/boxes.py
index 548e6be3..18d35bd4 100644
--- a/weasyprint/formatting_structure/boxes.py
+++ b/weasyprint/formatting_structure/boxes.py
@@ -308,7 +308,10 @@ class Box:
             isinstance(self, ReplacedBox) or
             self.style['overflow'] in ('auto', 'scroll') or
             (self.style['overflow'] == 'hidden' and
-             self.style['height'] != 'auto'))
+             self.style['height'] != 'auto' and
+             not ('flex' in self.style['display'] or
+                  self.style.get('flex') or
+                  self.parent and self.parent.style['display'].startswith('flex'))))
 
     def establishes_formatting_context(self):
         """Return whether this box establishes a block formatting context."""
