diff --git a/packages/autofmt/src/writer.rs b/packages/autofmt/src/writer.rs
index d9dfbd3..f5bb28c 100644
--- a/packages/autofmt/src/writer.rs
+++ b/packages/autofmt/src/writer.rs
@@ -426,7 +426,10 @@ impl<'a> Writer<'a> {
         {
             let comments = self.accumulate_full_line_comments(brace.span.span().end());
             if !comments.is_empty() {
+                self.clear_current_indent();
+                self.out.indent_level += 1;
                 self.apply_line_comments(comments)?;
+                self.out.indent_level -= 1;
                 self.out.tab()?;
             }
         }
@@ -706,6 +709,11 @@ impl<'a> Writer<'a> {
         Ok(())
     }
 
+    fn clear_current_indent(&mut self) {
+        let trim_to = self.out.buf.trim_end_matches([' ', '\t']).len();
+        self.out.buf.truncate(trim_to);
+    }
+
     fn write_comments(&mut self, loc: LineColumn) -> Result {
         let comments = self.accumulate_full_line_comments(loc);
         self.apply_line_comments(comments)?;
diff --git a/packages/cli-harnesses/harness-fullstack-with-optional-tokio/Cargo.toml b/packages/cli-harnesses/harness-fullstack-with-optional-tokio/Cargo.toml
index 5b0d69a..e9ae65c 100644
--- a/packages/cli-harnesses/harness-fullstack-with-optional-tokio/Cargo.toml
+++ b/packages/cli-harnesses/harness-fullstack-with-optional-tokio/Cargo.toml
@@ -7,7 +7,7 @@ publish = false
 
 [dependencies]
 dioxus = { workspace = true, features = ["fullstack"] }
-serde = { workspace = true }
+serde = "1.0.219"
 tokio = { workspace = true, features = ["full"], optional = true }
 
 
@@ -16,4 +16,4 @@ default = []
 server = ["dioxus/server", "dep:tokio"]
 web = ["dioxus/web"]
 
-    
+    
\ No newline at end of file
