diff --git a/src/MudBlazor/Services/Dialog/DialogService.cs b/src/MudBlazor/Services/Dialog/DialogService.cs
index 4671505..88ca4e9 100644
--- a/src/MudBlazor/Services/Dialog/DialogService.cs
+++ b/src/MudBlazor/Services/Dialog/DialogService.cs
@@ -68,6 +68,12 @@ namespace MudBlazor
             return ShowAsync<T>(string.Empty, DialogParameters.Default, DialogOptions.Default);
         }
 
+        /// <inheritdoc />
+        public Task<IDialogReference> ShowAsync<[DynamicallyAccessedMembers(DynamicallyAccessedMemberTypes.All)] T>(DialogParameters<T> parameters) where T : IComponent
+        {
+            return ShowAsync<T>(string.Empty, parameters, DialogOptions.Default);
+        }
+
         /// <inheritdoc />
         public Task<IDialogReference> ShowAsync<[DynamicallyAccessedMembers(DynamicallyAccessedMemberTypes.All)] T>(string? title) where T : IComponent
         {
diff --git a/src/MudBlazor/Services/Dialog/IDialogService.cs b/src/MudBlazor/Services/Dialog/IDialogService.cs
index 2475b3a..4092208 100644
--- a/src/MudBlazor/Services/Dialog/IDialogService.cs
+++ b/src/MudBlazor/Services/Dialog/IDialogService.cs
@@ -35,6 +35,17 @@ namespace MudBlazor
         /// <returns>A reference to the dialog.</returns>
         Task<IDialogReference> ShowAsync<[DynamicallyAccessedMembers(DynamicallyAccessedMemberTypes.All)] TComponent>() where TComponent : IComponent;
 
+        /// <summary>
+        /// Displays a dialog with parameters.
+        /// </summary>
+        /// <typeparam name="TComponent">The dialog to display.</typeparam>
+        /// <param name="parameters">The custom parameters to set within the dialog.</param>
+        /// <returns>A reference to the dialog.</returns>
+        Task<IDialogReference> ShowAsync<[DynamicallyAccessedMembers(DynamicallyAccessedMemberTypes.All)] TComponent>(DialogParameters<TComponent> parameters) where TComponent : IComponent
+        {
+            return ShowAsync<TComponent>(string.Empty, parameters);
+        }
+
         /// <summary>
         /// Displays a dialog with a custom title.
         /// </summary>
