diff --git a/src/Spectre.Console/Prompts/TextPrompt.cs b/src/Spectre.Console/Prompts/TextPrompt.cs
index 5418ef8..b87c910 100644
--- a/src/Spectre.Console/Prompts/TextPrompt.cs
+++ b/src/Spectre.Console/Prompts/TextPrompt.cs
@@ -109,7 +109,7 @@ public sealed class TextPrompt<T> : IPrompt<T>, IHasCulture
     public TextPrompt(string prompt, StringComparer? comparer = null)
     {
         _prompt = prompt ?? throw new System.ArgumentNullException(nameof(prompt));
-        _comparer = comparer;
+        _comparer = comparer ?? StringComparer.CurrentCultureIgnoreCase;
     }
 
     /// <summary>
