diff --git a/compiler-core/src/erlang.rs b/compiler-core/src/erlang.rs
index fa1ecc2..8e81582 100644
--- a/compiler-core/src/erlang.rs
+++ b/compiler-core/src/erlang.rs
@@ -690,7 +690,7 @@ fn string_inner(value: &str) -> Document<'_> {
         .replace_all(value, |caps: &Captures<'_>| {
             let slashes = caps.get(1).map_or("", |m| m.as_str());
 
-            if slashes.len().is_multiple_of(2) {
+            if slashes.len() % 2 == 0 {
                 format!("{slashes}u")
             } else {
                 format!("{slashes}x")
diff --git a/compiler-core/src/format.rs b/compiler-core/src/format.rs
index ee840a1..cfcfbc9 100644
--- a/compiler-core/src/format.rs
+++ b/compiler-core/src/format.rs
@@ -1386,13 +1386,27 @@ impl<'comments> Formatter<'comments> {
             };
         }
 
-        self.append_inlinable_wrapped_arguments(
-            "#".to_doc(),
-            elements,
-            location,
-            |e| e,
-            |self_, e| self_.comma_separated_item(e, elements.len()),
-        )
+        if elements.len() == 2
+            && matches!(elements.last(), Some(last) if is_breakable_argument(last, elements.len()))
+            && !matches!(elements.last(), Some(UntypedExpr::Tuple { .. }))
+        {
+            self.append_inlinable_wrapped_arguments(
+                "#".to_doc(),
+                elements,
+                location,
+                |e| e,
+                |self_, e| self_.comma_separated_item(e, elements.len()),
+            )
+        } else {
+            let elements = elements
+                .iter()
+                .map(|element| self.comma_separated_item(element, elements.len()))
+                .collect_vec();
+
+            "#".to_doc()
+                .append(self.wrap_arguments(elements, location.end))
+                .group()
+        }
     }
 
     // Appends to the given docs a comma-separated list of documents wrapped by
diff --git a/compiler-core/src/type_/expression.rs b/compiler-core/src/type_/expression.rs
index 0f4d562..54eeb7d 100644
--- a/compiler-core/src/type_/expression.rs
+++ b/compiler-core/src/type_/expression.rs
@@ -921,7 +921,7 @@ impl<'a, 'b> ExprTyper<'a, 'b> {
         // !!!True  // we can remove all but one negation.
         // ```
         if negations > 1 {
-            let location = if negations.is_multiple_of(2) {
+            let location = if negations % 2 == 0 {
                 SrcSpan {
                     start: starting_location.start,
                     end: location.start + 1,
@@ -1006,7 +1006,7 @@ impl<'a, 'b> ExprTyper<'a, 'b> {
         // ---1  // we can remove all but one negation.
         // ```
         if negations > 1 {
-            let location = if negations.is_multiple_of(2) {
+            let location = if negations % 2 == 0 {
                 SrcSpan {
                     start: starting_location.start,
                     end: end + 1,
diff --git a/hexpm/src/lib.rs b/hexpm/src/lib.rs
index 84fbcf1..8cbe1f1 100644
--- a/hexpm/src/lib.rs
+++ b/hexpm/src/lib.rs
@@ -1141,7 +1141,7 @@ impl OAuthDeviceAuthorisation {
         &mut self,
         response: http::Response<Vec<u8>>,
     ) -> Result<PollStep, ApiError> {
-        if self.start_time.elapsed() > Duration::from_mins(10) {
+        if self.start_time.elapsed() > Duration::from_secs(10 * 60) {
             return Err(ApiError::OAuthTimeout);
         }
