diff --git a/compat/src/Children.js b/compat/src/Children.js
index 0295d936..9914ae13 100644
--- a/compat/src/Children.js
+++ b/compat/src/Children.js
@@ -1,8 +1,11 @@
 import { toChildArray } from 'preact';
 
-const mapFn = (children, fn) => {
+const mapFn = (children, fn, ctx) => {
 	if (children == null) return null;
-	return toChildArray(toChildArray(children).map(fn));
+	children = toChildArray(children);
+	return toChildArray(
+		ctx == null ? children.map(fn) : children.map(fn, ctx)
+	);
 };
 
 // This API is completely unnecessary for Preact, so it's basically passthrough.
diff --git a/compat/src/index.d.ts b/compat/src/index.d.ts
index ce437a10..ff613e06 100644
--- a/compat/src/index.d.ts
+++ b/compat/src/index.d.ts
@@ -407,11 +407,13 @@ declare namespace React {
 	export const Children: {
 		map<T extends preact1.ComponentChild, R>(
 			children: T | T[],
-			fn: (child: T, i: number) => R
+			fn: (child: T, i: number) => R,
+			ctx?: any
 		): R[];
 		forEach<T extends preact1.ComponentChild>(
 			children: T | T[],
-			fn: (child: T, i: number) => void
+			fn: (child: T, i: number) => void,
+			ctx?: any
 		): void;
 		count: (children: preact1.ComponentChildren) => number;
 		only: (children: preact1.ComponentChildren) => preact1.ComponentChild;
