diff --git a/hugofs/component_fs.go b/hugofs/component_fs.go
index ec7f06a07..89f848782 100644
--- a/hugofs/component_fs.go
+++ b/hugofs/component_fs.go
@@ -320,19 +320,22 @@ func (fs *componentFs) applyMeta(fi FileNameIsDir, name string) (FileMetaInfo, b
 
 	meta.PathInfo = pi
 	if !fim.IsDir() {
-		if fileLang := meta.PathInfo.Lang(); fileLang != "" {
-			if idx, ok := fs.opts.Cfg.PathParser().LanguageIndex[fileLang]; ok {
-				// A valid lang set in filename.
-				// Give priority to myfile.sv.txt inside the sv filesystem.
-				meta.Weight++
-				meta.SitesMatrix = meta.SitesMatrix.WithLanguageIndices(idx)
-				if idx > 0 {
-					// Not the default language, add some weight.
-					meta.SitesMatrix = sitesmatrix.NewWeightedVectorStore(meta.SitesMatrix, 10)
-				}
-
+		pathMatrix := fs.opts.Cfg.PathParser().SitesMatrixFromPath(pi)
+		if len(pi.Langs()) > 0 || pi.Lang() != "" {
+			// Give priority to myfile.sv.txt inside the sv filesystem.
+			meta.Weight++
+			meta.SitesMatrix = meta.SitesMatrix.WithDimensionIndices(sitesmatrix.Language, pathMatrix.KeysSortedDimension(sitesmatrix.Language)...)
+			if idx := fs.opts.Cfg.PathParser().LanguageIndex[pi.Lang()]; idx > 0 {
+				// Not the default language, add some weight.
+				meta.SitesMatrix = sitesmatrix.NewWeightedVectorStore(meta.SitesMatrix, 10)
 			}
 		}
+		if len(pi.Versions()) > 0 {
+			meta.SitesMatrix = meta.SitesMatrix.WithDimensionIndices(sitesmatrix.Version, pathMatrix.KeysSortedDimension(sitesmatrix.Version)...)
+		}
+		if len(pi.Roles()) > 0 {
+			meta.SitesMatrix = meta.SitesMatrix.WithDimensionIndices(sitesmatrix.Role, pathMatrix.KeysSortedDimension(sitesmatrix.Role)...)
+		}
 		switch meta.Component {
 		case files.ComponentFolderLayouts:
 			// Eg. index.fr.html when French isn't defined,
diff --git a/hugolib/sitesmatrix/dimensions.go b/hugolib/sitesmatrix/dimensions.go
index d7a96a038..030a4bdb0 100644
--- a/hugolib/sitesmatrix/dimensions.go
+++ b/hugolib/sitesmatrix/dimensions.go
@@ -220,6 +220,9 @@ type VectorStore interface {
 	HasRole(role int) bool
 	MustHash() uint64
 
+	WithDimensionIndices(d int, idxs ...int) VectorStore
+	KeysSortedDimension(d int) []int
+
 	// Used in tests.
 	KeysSorted() ([]int, []int, []int)
 	Vectors() []Vector
diff --git a/hugolib/sitesmatrix/vectorstores.go b/hugolib/sitesmatrix/vectorstores.go
index 174dd6ea9..b64916800 100644
--- a/hugolib/sitesmatrix/vectorstores.go
+++ b/hugolib/sitesmatrix/vectorstores.go
@@ -17,7 +17,6 @@ import (
 	"cmp"
 	"fmt"
 	"iter"
-	"maps"
 	"slices"
 	"sort"
 	"sync"
@@ -72,20 +71,16 @@ func (s *vectorStoreMap) setVector(vec Vector) {
 }
 
 func (s *vectorStoreMap) KeysSorted() ([]int, []int, []int) {
-	var k0, k1, k2 []int
+	return s.KeysSortedDimension(Language), s.KeysSortedDimension(Version), s.KeysSortedDimension(Role)
+}
+
+func (s *vectorStoreMap) KeysSortedDimension(d int) []int {
+	var keys []int
 	for v := range s.sets {
-		k0 = append(k0, v.Language())
-		k1 = append(k1, v.Version())
-		k2 = append(k2, v.Role())
+		keys = append(keys, v[d])
 	}
-	sort.Ints(k0)
-	sort.Ints(k1)
-	sort.Ints(k2)
-	k0 = slices.Compact(k0)
-	k1 = slices.Compact(k1)
-	k2 = slices.Compact(k2)
-
-	return k0, k1, k2
+	sort.Ints(keys)
+	return slices.Compact(keys)
 }
 
 func (s *vectorStoreMap) Hash() (uint64, error) {
@@ -210,13 +205,17 @@ func (s *vectorStoreMap) Vectors() []Vector {
 }
 
 func (s *vectorStoreMap) WithLanguageIndices(i int) VectorStore {
-	c := s.clone()
+	return s.WithDimensionIndices(Language, i)
+}
 
-	for v := range c.sets {
-		v[Language] = i
-		c.sets[v] = struct{}{}
+func (s *vectorStoreMap) WithDimensionIndices(d int, idxs ...int) VectorStore {
+	c := newVectorStoreMap(len(s.sets) * len(idxs))
+	for v := range s.sets {
+		for _, idx := range idxs {
+			v[d] = idx
+			c.setVector(v)
+		}
 	}
-
 	return c
 }
 
@@ -247,13 +246,6 @@ func (s *vectorStoreMap) HasRole(i int) bool {
 	return false
 }
 
-func (s *vectorStoreMap) clone() *vectorStoreMap {
-	c := *s
-	c.h = &hashOnce{}
-	c.sets = maps.Clone(s.sets)
-	return &c
-}
-
 func NewIntSetsBuilder(cfg *ConfiguredDimensions) *IntSetsBuilder {
 	if cfg == nil {
 		panic("cfg is required")
@@ -498,13 +490,23 @@ func (s *IntSets) Vectors() []Vector {
 }
 
 func (s *IntSets) KeysSorted() ([]int, []int, []int) {
+	return s.KeysSortedDimension(Language), s.KeysSortedDimension(Version), s.KeysSortedDimension(Role)
+}
+
+func (s *IntSets) KeysSortedDimension(d int) []int {
 	if s == nil {
-		return nil, nil, nil
+		return nil
+	}
+	switch d {
+	case Language:
+		return s.languages.KeysSorted()
+	case Version:
+		return s.versions.KeysSorted()
+	case Role:
+		return s.roles.KeysSorted()
+	default:
+		panic("invalid dimension")
 	}
-	languages := s.languages.KeysSorted()
-	versions := s.versions.KeysSorted()
-	roles := s.roles.KeysSorted()
-	return languages, versions, roles
 }
 
 func (s *IntSets) HasLanguage(lang int) bool {
@@ -591,8 +593,22 @@ func (s IntSets) shallowClone() *IntSets {
 
 // WithLanguageIndices replaces the current language set with a single language index.
 func (s *IntSets) WithLanguageIndices(i int) VectorStore {
+	return s.WithDimensionIndices(Language, i)
+}
+
+func (s *IntSets) WithDimensionIndices(d int, idxs ...int) VectorStore {
 	c := s.shallowClone()
-	c.languages = hmaps.NewOrderedIntSet(i)
+	set := hmaps.NewOrderedIntSet(idxs...)
+	switch d {
+	case Language:
+		c.languages = set
+	case Version:
+		c.versions = set
+	case Role:
+		c.roles = set
+	default:
+		panic("invalid dimension")
+	}
 	return c.init()
 }
