diff --git a/docs/reference/platforms/linux/flatpak.rst b/docs/reference/platforms/linux/flatpak.rst
index eab3c9f4..f23fb7ed 100644
--- a/docs/reference/platforms/linux/flatpak.rst
+++ b/docs/reference/platforms/linux/flatpak.rst
@@ -96,7 +96,6 @@ Briefcase adds the following finish arguments by default:
 * ``filesystem=xdg-config``
 * ``filesystem=xdg-data``
 * ``filesystem=xdg-documents``
-* ``socket=session-bus``
 
 These can be disabled by explicitly setting their value to ``False``; for example, to
 disable audio access, you would specify::
diff --git a/src/briefcase/platforms/linux/flatpak.py b/src/briefcase/platforms/linux/flatpak.py
index 563248b0..127e10f7 100644
--- a/src/briefcase/platforms/linux/flatpak.py
+++ b/src/briefcase/platforms/linux/flatpak.py
@@ -139,8 +139,8 @@ class LinuxFlatpakCreateCommand(LinuxFlatpakMixin, CreateCommand):
             "filesystem=xdg-config": True,
             "filesystem=xdg-data": True,
             "filesystem=xdg-documents": True,
-            # DBus access
-            "socket=session-bus": True,
+            # DBus access is not included by default as it breaks sandboxing
+            # and presents a security risk according to Flatpak documentation
         }
 
         finish_args.update(getattr(app, "finish_arg", {}))
diff --git a/tests/platforms/linux/flatpak/test_create.py b/tests/platforms/linux/flatpak/test_create.py
index b2d748df..693937e7 100644
--- a/tests/platforms/linux/flatpak/test_create.py
+++ b/tests/platforms/linux/flatpak/test_create.py
@@ -54,7 +54,6 @@ DEFAULT_FINISH_ARGS = {
     "filesystem=xdg-config": True,
     "filesystem=xdg-data": True,
     "filesystem=xdg-documents": True,
-    "socket=session-bus": True,
 }
 
 
@@ -221,7 +220,6 @@ DEFAULT_FINISH_ARGS = {
                     "filesystem=xdg-config": True,
                     "filesystem=xdg-data": True,
                     "filesystem=xdg-documents": True,
-                    "socket=session-bus": True,
                     "allow=bluetooth": True,
                 },
             },
