diff --git a/packages/@stylexjs/babel-plugin/src/visitors/parse-stylex-create-arg.js b/packages/@stylexjs/babel-plugin/src/visitors/parse-stylex-create-arg.js
index 701d9112..41848269 100644
--- a/packages/@stylexjs/babel-plugin/src/visitors/parse-stylex-create-arg.js
+++ b/packages/@stylexjs/babel-plugin/src/visitors/parse-stylex-create-arg.js
@@ -149,10 +149,7 @@ function evaluatePartialObjectRecursively(
       if (!keyResult.confident) {
         return { confident: false, deopt: keyResult.deopt, value: null };
       }
-      let key = keyResult.value;
-      if (key.startsWith('var(') && key.endsWith(')')) {
-        key = key.slice(4, -1);
-      }
+      const key = keyResult.value;
 
       const valuePath: NodePath<t.Expression | t.PatternLike> =
         prop.get('value');
@@ -234,7 +231,7 @@ function evaluatePartialObjectRecursively(
                   t.identifier('undefined'),
                 );
           inlineStyles[varName] = {
-            path: [...keyPath, key],
+            path: fullKeyPath,
             originalExpression: expression,
             expression: inlineStyleExpression,
           };
