diff --git a/src/cluster.c b/src/cluster.c
index dd83a3324..3201b540c 100644
--- a/src/cluster.c
+++ b/src/cluster.c
@@ -1594,6 +1594,8 @@ void resetClusterStats(void) {
     if (!server.cluster_enabled) return;
 
     clusterSlotStatResetAll();
+    memset(server.cluster->stats_bus_messages_sent, 0, sizeof(server.cluster->stats_bus_messages_sent));
+    memset(server.cluster->stats_bus_messages_received, 0, sizeof(server.cluster->stats_bus_messages_received));
 }
 
 
