diff --git a/src/MudBlazor/Components/Select/MudSelect.razor.cs b/src/MudBlazor/Components/Select/MudSelect.razor.cs
index f84998b..05a407a 100644
--- a/src/MudBlazor/Components/Select/MudSelect.razor.cs
+++ b/src/MudBlazor/Components/Select/MudSelect.razor.cs
@@ -703,17 +703,6 @@ namespace MudBlazor
             UpdateIcon();
             StateHasChanged();
 
-            //Scroll the active item on each opening
-            if (_activeItemId != null)
-            {
-                var index = Items.FindIndex(x => x.ItemId == _activeItemId);
-                if (index > 0)
-                {
-                    var item = Items[index];
-                    await ScrollToItemAsync(item);
-                }
-            }
-
             //disable escape propagation: if selectmenu is open, only the select popover should close and underlying components should not handle escape key
             await KeyInterceptorService.UpdateKeyAsync(ElementId, new("Escape", stopDown: "key+none"));
         }
@@ -1453,6 +1442,17 @@ namespace MudBlazor
                     {
                         await HighlightItemForValueAsync(ReadValue);
                     }
+
+                    // Scroll to the highlighted item (now that items are registered)
+                    if (_activeItemId != null)
+                    {
+                        var index = Items.FindIndex(x => x.ItemId == _activeItemId);
+                        if (index > 0)
+                        {
+                            var item = Items[index];
+                            await ScrollToItemAsync(item);
+                        }
+                    }
                 });
             }
         }
