diff --git a/execution/rerun/rerun.go b/execution/rerun/rerun.go
index 9ac1725..f3d47db 100644
--- a/execution/rerun/rerun.go
+++ b/execution/rerun/rerun.go
@@ -70,6 +70,16 @@ func (m *failedMetadata) addFailedItem(itemName string, item string) {
 	m.failedItemsMap[itemName][item] = true
 }
 
+func (m *failedMetadata) deleteFailedItem(itemName string, item string) {
+	if _, ok := m.failedItemsMap[itemName]; !ok {
+		return
+	}
+	delete(m.failedItemsMap[itemName], item)
+	if len(m.failedItemsMap[itemName]) == 0 {
+		delete(m.failedItemsMap, itemName)
+	}
+}
+
 // ListenFailedScenarios listens to execution events and writes the failed scenarios to JSON file
 func ListenFailedScenarios(wg *sync.WaitGroup, specDirs []string) {
 	ch := make(chan event.ExecutionEvent)
@@ -97,11 +107,14 @@ func ListenFailedScenarios(wg *sync.WaitGroup, specDirs []string) {
 }
 
 func prepareScenarioFailedMetadata(res *result.ScenarioResult, sce *gauge.Scenario, executionInfo *gauge_messages.ExecutionInfo) {
+	specPath := executionInfo.GetCurrentSpec().GetFileName()
+	failedScenario := util.RelPathToProjectRoot(specPath)
+	failedItem := fmt.Sprintf("%s:%v", failedScenario, sce.Span.Start)
 	if res.GetFailed() {
-		specPath := executionInfo.GetCurrentSpec().GetFileName()
-		failedScenario := util.RelPathToProjectRoot(specPath)
-		failedMeta.addFailedItem(specPath, fmt.Sprintf("%s:%v", failedScenario, sce.Span.Start))
+		failedMeta.addFailedItem(specPath, failedItem)
+		return
 	}
+	failedMeta.deleteFailedItem(specPath, failedItem)
 }
 
 func addSpecFailedMetadata(res result.Result, args []string) {
