diff --git a/slatedb/src/bytes_range.rs b/slatedb/src/bytes_range.rs
index f05936b..71893bb 100644
--- a/slatedb/src/bytes_range.rs
+++ b/slatedb/src/bytes_range.rs
@@ -43,23 +43,8 @@ impl RangeBounds<Bytes> for BytesRange {
     }
 }
 
-fn is_bound_non_empty(bound: &Bound<Bytes>) -> bool {
-    match bound {
-        Included(b) | Excluded(b) => !b.is_empty(),
-        Unbounded => true,
-    }
-}
-
 impl BytesRange {
     pub(crate) fn new(start_bound: Bound<Bytes>, end_bound: Bound<Bytes>) -> Self {
-        assert!(
-            is_bound_non_empty(&start_bound),
-            "Start bound must be non-empty"
-        );
-        assert!(
-            is_bound_non_empty(&end_bound),
-            "End bound must be non-empty"
-        );
         let inner = ComparableRange::new(start_bound, end_bound);
         assert!(inner.non_empty(), "Range must be non-empty");
         Self { inner }
