diff --git a/Zend/zend_compile.c b/Zend/zend_compile.c
index db611be49ab..3583fb0e659 100644
--- a/Zend/zend_compile.c
+++ b/Zend/zend_compile.c
@@ -2726,7 +2726,8 @@ static inline bool zend_is_call(zend_ast *ast) /* {{{ */
 	return ast->kind == ZEND_AST_CALL
 		|| ast->kind == ZEND_AST_METHOD_CALL
 		|| ast->kind == ZEND_AST_NULLSAFE_METHOD_CALL
-		|| ast->kind == ZEND_AST_STATIC_CALL;
+		|| ast->kind == ZEND_AST_STATIC_CALL
+		|| ast->kind == ZEND_AST_PIPE;
 }
 /* }}} */
 
@@ -6459,7 +6460,7 @@ static bool zend_is_pipe_optimizable_callable_name(zend_ast *ast)
 	return true;
 }
 
-static void zend_compile_pipe(znode *result, zend_ast *ast)
+static void zend_compile_pipe(znode *result, zend_ast *ast, uint32_t type)
 {
 	zend_ast *operand_ast = ast->child[0];
 	zend_ast *callable_ast = ast->child[1];
@@ -6508,7 +6509,7 @@ static void zend_compile_pipe(znode *result, zend_ast *ast)
 			callable_ast, arg_list_ast);
 	}
 
-	zend_compile_expr(result, fcall_ast);
+	zend_compile_var(result, fcall_ast, type, 0);
 	CG(zend_lineno) = fcall_ast->lineno;
 	zend_do_extended_stmt(result);
 }
@@ -11873,7 +11874,7 @@ static void zend_compile_expr_inner(znode *result, zend_ast *ast) /* {{{ */
 			zend_compile_match(result, ast);
 			return;
 		case ZEND_AST_PIPE:
-			zend_compile_pipe(result, ast);
+			zend_compile_pipe(result, ast, BP_VAR_R);
 			return;
 		default:
 			ZEND_ASSERT(0 /* not supported */);
@@ -11929,6 +11930,9 @@ static zend_op *zend_compile_var_inner(znode *result, zend_ast *ast, uint32_t ty
 		case ZEND_AST_STATIC_CALL:
 			zend_compile_static_call(result, ast, type);
 			return NULL;
+		case ZEND_AST_PIPE:
+			zend_compile_pipe(result, ast, type);
+			return NULL;
 		case ZEND_AST_ZNODE:
 			*result = *zend_ast_get_znode(ast);
 			return NULL;
