diff --git a/common/socket/src/main/java/io/helidon/common/socket/NioSocket.java b/common/socket/src/main/java/io/helidon/common/socket/NioSocket.java
index c3ff018..c6c90ba 100644
--- a/common/socket/src/main/java/io/helidon/common/socket/NioSocket.java
+++ b/common/socket/src/main/java/io/helidon/common/socket/NioSocket.java
@@ -144,7 +144,9 @@ public sealed class NioSocket implements HelidonSocket permits TlsNioSocket {
                 writeBuffer.clear();
                 buffer.writeTo(writeBuffer, buffer.available());
                 writeBuffer.flip();
-                delegate.write(writeBuffer);
+                while (writeBuffer.hasRemaining()) {
+                    delegate.write(writeBuffer);
+                }
             }
         } catch (IOException e) {
             throw new UncheckedIOException(e);
