diff --git a/src/instructlab/rag/taxonomy_utils.py b/src/instructlab/rag/taxonomy_utils.py
index 33ae9a0a..acfb769a 100644
--- a/src/instructlab/rag/taxonomy_utils.py
+++ b/src/instructlab/rag/taxonomy_utils.py
@@ -26,7 +26,10 @@ def lookup_knowledge_files(taxonomy_path, taxonomy_base, temp_dir) -> list[Path]
     )
     knowledge_files: list[Path] = []
     for leaf_node in leaf_nodes.values():
-        knowledge_files.extend(leaf_node[0]["filepaths"])
+        if leaf_node and isinstance(leaf_node, list) and len(leaf_node) > 0:
+            # Check if 'filepaths' key exists in the leaf node
+            if "filepaths" in leaf_node[0]:
+                knowledge_files.extend(leaf_node[0]["filepaths"])
 
     grouped_knowledge_files = []
     for knowledge_file in knowledge_files:
@@ -34,7 +37,7 @@ def lookup_knowledge_files(taxonomy_path, taxonomy_base, temp_dir) -> list[Path]
             knowledge_file.rename(Path(temp_dir) / knowledge_file.name)
         )
 
-    return knowledge_files
+    return grouped_knowledge_files
 
 
 def lookup_processed_documents_folder(output_dir: str) -> Optional[Path]:
