diff --git a/global.json b/global.json
index e5eb8f6..2788290 100644
--- a/global.json
+++ b/global.json
@@ -1,6 +1,6 @@
 {
   "sdk": {
-    "version": "10.0.100",
+    "version": "10.0.203",
     "allowPrerelease": false,
     "rollForward": "latestPatch"
   }
diff --git a/src/Core/RevEng.Core.100/RevEng.Core.100.csproj b/src/Core/RevEng.Core.100/RevEng.Core.100.csproj
index 0f8b01a..f1cda81 100644
--- a/src/Core/RevEng.Core.100/RevEng.Core.100.csproj
+++ b/src/Core/RevEng.Core.100/RevEng.Core.100.csproj
@@ -29,7 +29,7 @@
     <PackageReference Include="Bricelam.EntityFrameworkCore.Pluralizer" Version="1.0.0" />
     <!--<PackageReference Include="EFCore.Snowflake" Version="9.0.4" />-->
     <PackageReference Include="EntityFrameworkCore.Scaffolding.Handlebars" Version="10.0.0-beta1" />
-    <!--<PackageReference Include="EntityFrameworkCore.Sqlite.NodaTime" Version="9.1.0" />-->
+    <PackageReference Include="EntityFrameworkCore.Sqlite.NodaTime" Version="10.0.0" />
     <PackageReference Include="FirebirdSql.EntityFrameworkCore.Firebird" Version="13.0.0" />
     <PackageReference Include="Microsoft.Data.SqlClient" Version="6.1.4" />
     <PackageReference Include="Microsoft.EntityFrameworkCore.Design" Version="10.0.6" />
diff --git a/src/Core/RevEng.Core.80/ServiceProviderBuilder.cs b/src/Core/RevEng.Core.80/ServiceProviderBuilder.cs
index 397ac1a..67bd4cf 100644
--- a/src/Core/RevEng.Core.80/ServiceProviderBuilder.cs
+++ b/src/Core/RevEng.Core.80/ServiceProviderBuilder.cs
@@ -171,13 +171,12 @@ namespace RevEng.Core
                     var sqliteProvider = new SqliteDesignTimeServices();
                     sqliteProvider.ConfigureDesignTimeServices(serviceCollection);
 
-#if !CORE100
                     if (options.UseNodaTime)
                     {
                         var nodaTime = new SqliteNodaTimeDesignTimeServices();
                         nodaTime.ConfigureDesignTimeServices(serviceCollection);
                     }
-#endif
+
                     break;
 
                 default:
diff --git a/src/GUI/RevEng.Shared/Providers.cs b/src/GUI/RevEng.Shared/Providers.cs
index 1fae0ce..4d5108e 100644
--- a/src/GUI/RevEng.Shared/Providers.cs
+++ b/src/GUI/RevEng.Shared/Providers.cs
@@ -262,6 +262,10 @@ namespace RevEng.Common
                             pkgVersion = "9.1.0";
                             break;
 
+                        case CodeGenerationMode.EFCore10:
+                            pkgVersion = "10.0.0";
+                            break;
+
                         default: throw new NotImplementedException();
                     }
