diff --git a/smithy-model/src/main/java/software/amazon/smithy/model/shapes/SmithyIdlModelSerializer.java b/smithy-model/src/main/java/software/amazon/smithy/model/shapes/SmithyIdlModelSerializer.java
index 6773c71..d80e648 100644
--- a/smithy-model/src/main/java/software/amazon/smithy/model/shapes/SmithyIdlModelSerializer.java
+++ b/smithy-model/src/main/java/software/amazon/smithy/model/shapes/SmithyIdlModelSerializer.java
@@ -943,15 +943,15 @@ public final class SmithyIdlModelSerializer {
          * Serialize a node into the Smithy IDL format.
          *
          * <p>This uses the given shape to influence serialization. For example, a string shape marked with the idRef
-         * trait will be serialized as a shape id rather than a string.
+         * trait will be serialized as a shape id rather than a string when the referenced shape is present.
          *
          * @param node The node to serialize.
          * @param shape The shape of the node.
          */
         private void serialize(Node node, Shape shape) {
-            // ShapeIds are represented differently than strings, so if a shape looks like it's
-            // representing a shapeId we need to serialize it without quotes.
-            if (isShapeId(shape)) {
+            // ShapeIds are represented differently than strings, so if a node is an idRef that targets a shape in the
+            // model, serialize it without quotes.
+            if (isShapeId(node, shape)) {
                 serializeShapeId(node.expectStringNode());
                 return;
             }
@@ -975,11 +975,15 @@ public final class SmithyIdlModelSerializer {
             }
         }
 
-        private boolean isShapeId(Shape shape) {
-            if (shape == null) {
+        private boolean isShapeId(Node node, Shape shape) {
+            if (shape == null || !node.isStringNode()) {
                 return false;
             }
-            return shape.getMemberTrait(model, IdRefTrait.class).isPresent();
+
+            Optional<ShapeId> shapeId = node.expectStringNode().asShapeId();
+            return shapeId.isPresent()
+                    && model.getShape(shapeId.get()).isPresent()
+                    && shape.getMemberTrait(model, IdRefTrait.class).isPresent();
         }
 
         private void serializeString(StringNode node) {
