diff --git a/librz/bin/format/elf/elf_relocs_conversion.c b/librz/bin/format/elf/elf_relocs_conversion.c
index 2daef857e6..523c80d3a6 100644
--- a/librz/bin/format/elf/elf_relocs_conversion.c
+++ b/librz/bin/format/elf/elf_relocs_conversion.c
@@ -299,8 +299,8 @@ static RzBinReloc *reloc_convert_arm(ELFOBJ *bin, RzBinElfReloc *rel, ut64 GOT)
 	case R_ARM_ABS16: ADD(16, 0, "R_ARM_ABS16", RZ_RELOC_BASE_SYMBOL);
 	case R_ARM_ABS8: ADD(8, 0, "R_ARM_ABS8", RZ_RELOC_BASE_SYMBOL);
 	case R_ARM_SBREL32: ADD_T(32, T, B, "R_ARM_SBREL32", RZ_RELOC_BASE_SYMBOL);
-	case R_ARM_GLOB_DAT: ADD_T(32, T, 0, "R_ARM_GLOB_DAT", RZ_RELOC_BASE_SYMBOL);
-	case R_ARM_JUMP_SLOT: ADD_T(32, T, 0, "R_ARM_JUMP_SLOT", RZ_RELOC_BASE_SYMBOL);
+	case R_ARM_GLOB_DAT: ADD(32, 0, "R_ARM_GLOB_DAT", RZ_RELOC_BASE_SYMBOL);
+	case R_ARM_JUMP_SLOT: ADD(32, 0, "R_ARM_JUMP_SLOT", RZ_RELOC_BASE_SYMBOL);
 	case R_ARM_COPY: ADD(32, 0, "R_ARM_COPY", RZ_RELOC_BASE_UNKNOWN); // copy symbol at runtime
 	case R_ARM_RELATIVE: ADD(32, 0, "R_ARM_RELATIVE", RZ_RELOC_BASE_BASE);
 	case R_ARM_GOTOFF32: ADD_T(32, T, -GOT, "R_ARM_GOTOFF32", RZ_RELOC_BASE_SYMBOL);
