diff --git a/bookmarks/models.py b/bookmarks/models.py
index 38d8c30..c11ed5e 100644
--- a/bookmarks/models.py
+++ b/bookmarks/models.py
@@ -138,6 +138,18 @@ def bookmark_asset_deleted(sender, instance, **kwargs):
                 logger.error(f"Failed to delete asset file: {filepath}", exc_info=error)
 
 
+@receiver(post_delete, sender=Bookmark)
+def bookmark_deleted(sender, instance, **kwargs):
+    if instance.preview_image_file:
+        filepath = os.path.join(settings.LD_PREVIEW_FOLDER, instance.preview_image_file)
+        if os.path.isfile(filepath):
+            try:
+                os.remove(filepath)
+                logger.debug(f"Deleted preview image file: {filepath}")
+            except Exception as error:
+                logger.error(f"Failed to delete preview image file: {filepath}", exc_info=error)
+
+
 class BookmarkForm(forms.ModelForm):
     # Use URLField for URL
     url = forms.CharField(validators=[BookmarkURLValidator()])
