diff --git a/src/MudBlazor/Components/Collapse/MudCollapse.razor b/src/MudBlazor/Components/Collapse/MudCollapse.razor
index e9ec3de..63cb738 100644
--- a/src/MudBlazor/Components/Collapse/MudCollapse.razor
+++ b/src/MudBlazor/Components/Collapse/MudCollapse.razor
@@ -4,7 +4,7 @@
 <div @ontransitionend="this.AsNonRenderingEventHandler(AnimationEndAsync)"
      @attributes="UserAttributes" class="@Classname" style="@Stylename">
     <div class="mud-collapse-wrapper">
-        <div class="mud-collapse-wrapper-inner">
+        <div class="mud-collapse-wrapper-inner" @ontransitionend:stopPropagation="true">
             @ChildContent
         </div>
     </div>
diff --git a/src/MudBlazor/Components/Collapse/MudCollapse.razor.cs b/src/MudBlazor/Components/Collapse/MudCollapse.razor.cs
index 612a688..0ac1167 100644
--- a/src/MudBlazor/Components/Collapse/MudCollapse.razor.cs
+++ b/src/MudBlazor/Components/Collapse/MudCollapse.razor.cs
@@ -107,6 +107,10 @@ namespace MudBlazor
                 _state = CollapseState.Exited;
                 StateHasChanged();
             }
+            else
+            {
+                return Task.CompletedTask;
+            }
 
             return OnAnimationEnd.InvokeAsync(_expandedState.Value);
         }
