diff --git a/packages/svelte/src/internal/client/dom/elements/attributes.js b/packages/svelte/src/internal/client/dom/elements/attributes.js
index 5db685cf3..8f626d331 100644
--- a/packages/svelte/src/internal/client/dom/elements/attributes.js
+++ b/packages/svelte/src/internal/client/dom/elements/attributes.js
@@ -514,9 +514,11 @@ export function attribute_effect(
 
 	if (is_select) {
 		var select = /** @type {HTMLSelectElement} */ (element);
+		var mounting = true;
 
 		effect(() => {
-			select_option(select, /** @type {Record<string | symbol, any>} */ (prev).value);
+			select_option(select, /** @type {Record<string | symbol, any>} */ (prev).value, mounting);
+			mounting = false;
 			init_select(select);
 		});
 	}
