diff --git a/src/cfnlint/rules/functions/GetAtt.py b/src/cfnlint/rules/functions/GetAtt.py
index 5cb39ac72..81e68bff6 100644
--- a/src/cfnlint/rules/functions/GetAtt.py
+++ b/src/cfnlint/rules/functions/GetAtt.py
@@ -34,7 +34,7 @@ class GetAtt(BaseFn):
     def schema(self, validator, instance) -> dict[str, Any]:
         resource_functions = []
         if validator.context.transforms.has_language_extensions_transform():
-            resource_functions = ["Ref"]
+            resource_functions = ["Ref", "Fn::Sub"]
 
         return {
             "type": ["string", "array"],
