diff --git a/Core/Relationships/RelationshipResolver.cs b/Core/Relationships/RelationshipResolver.cs
index 0c1314a..046b859 100644
--- a/Core/Relationships/RelationshipResolver.cs
+++ b/Core/Relationships/RelationshipResolver.cs
@@ -281,7 +281,10 @@ private void AddModulesToInstall(CkanModule[] modules)
                 IReadOnlyList<CkanModule>? candidates = null;
                 try
                 {
-                    candidates = resolved.Candidates(descriptor, modlist.Values,
+                    candidates = resolved.Candidates(descriptor,
+                                                     soft_resolve && options.proceed_with_inconsistencies
+                                                         ? Array.Empty<CkanModule>()
+                                                         : modlist.Values,
                                                      registry, game);
                     log.DebugFormat("Got {0} candidates for {1}",
                                     candidates.Count, descriptor);
