diff --git a/conan/tools/system/package_manager.py b/conan/tools/system/package_manager.py
index 98a5dda3e..50ad83778 100644
--- a/conan/tools/system/package_manager.py
+++ b/conan/tools/system/package_manager.py
@@ -320,8 +320,11 @@ class Chocolatey(_SystemPackageManagerTool):
     tool_name = "choco"
     install_command = "{tool} install --yes {packages}"
     update_command = "{tool} outdated"
+    # Look for the package name in the output of the choco search command
+    # The --exact flag in choco search ensures we're looking for the exact package name
+    # We use findstr to check if the package name appears in the output
     check_command = '{tool} search --local-only --exact {package} | ' \
-                    'findstr /c:"1 packages installed."'
+                    'findstr /i /c:"{package}"'
 
 
 class PacMan(_SystemPackageManagerTool):
