diff --git a/src/MudBlazor/Services/Dialog/DialogService.cs b/src/MudBlazor/Services/Dialog/DialogService.cs
index 4671505..a910f4b 100644
--- a/src/MudBlazor/Services/Dialog/DialogService.cs
+++ b/src/MudBlazor/Services/Dialog/DialogService.cs
@@ -80,6 +80,18 @@ namespace MudBlazor
             return ShowAsync<T>(title, DialogParameters.Default, options);
         }
 
+        /// <inheritdoc />
+        public Task<IDialogReference> ShowAsync<[DynamicallyAccessedMembers(DynamicallyAccessedMemberTypes.All)] T>(DialogParameters<T> parameters) where T : IComponent
+        {
+            return ShowAsync<T>(string.Empty, parameters, DialogOptions.Default);
+        }
+
+        /// <inheritdoc />
+        public Task<IDialogReference> ShowAsync<[DynamicallyAccessedMembers(DynamicallyAccessedMemberTypes.All)] T>(DialogParameters<T> parameters, DialogOptions? options) where T : IComponent
+        {
+            return ShowAsync<T>(string.Empty, parameters, options ?? DialogOptions.Default);
+        }
+
         /// <inheritdoc />
         public Task<IDialogReference> ShowAsync<[DynamicallyAccessedMembers(DynamicallyAccessedMemberTypes.All)] T>(string? title, DialogParameters parameters) where T : IComponent
         {
@@ -111,6 +123,18 @@ namespace MudBlazor
             return ShowAsync(component, title, DialogParameters.Default, options);
         }
 
+        /// <inheritdoc />
+        public Task<IDialogReference> ShowAsync([DynamicallyAccessedMembers(DynamicallyAccessedMemberTypes.All)] Type component, DialogParameters parameters)
+        {
+            return ShowAsync(component, string.Empty, parameters, DialogOptions.Default);
+        }
+
+        /// <inheritdoc />
+        public Task<IDialogReference> ShowAsync([DynamicallyAccessedMembers(DynamicallyAccessedMemberTypes.All)] Type component, DialogParameters parameters, DialogOptions options)
+        {
+            return ShowAsync(component, string.Empty, parameters, options);
+        }
+
         /// <inheritdoc />
         public Task<IDialogReference> ShowAsync([DynamicallyAccessedMembers(DynamicallyAccessedMemberTypes.All)] Type component, string? title, DialogParameters parameters)
         {
diff --git a/src/MudBlazor/Services/Dialog/IDialogService.cs b/src/MudBlazor/Services/Dialog/IDialogService.cs
index 2475b3a..231b304 100644
--- a/src/MudBlazor/Services/Dialog/IDialogService.cs
+++ b/src/MudBlazor/Services/Dialog/IDialogService.cs
@@ -52,6 +52,23 @@ namespace MudBlazor
         /// <returns>A reference to the dialog.</returns>
         Task<IDialogReference> ShowAsync<[DynamicallyAccessedMembers(DynamicallyAccessedMemberTypes.All)] TComponent>(string? title, DialogOptions options) where TComponent : IComponent;
 
+        /// <summary>
+        /// Displays a dialog with parameters.
+        /// </summary>
+        /// <typeparam name="TComponent">The dialog to display.</typeparam>
+        /// <param name="parameters">The custom parameters to set within the dialog.</param>
+        /// <returns>A reference to the dialog.</returns>
+        Task<IDialogReference> ShowAsync<[DynamicallyAccessedMembers(DynamicallyAccessedMemberTypes.All)] TComponent>(DialogParameters<TComponent> parameters) where TComponent : IComponent;
+
+        /// <summary>
+        /// Displays a dialog with parameters and options.
+        /// </summary>
+        /// <typeparam name="TComponent">The dialog to display.</typeparam>
+        /// <param name="parameters">The custom parameters to set within the dialog.</param>
+        /// <param name="options">The custom display options for the dialog.</param>
+        /// <returns>A reference to the dialog.</returns>
+        Task<IDialogReference> ShowAsync<[DynamicallyAccessedMembers(DynamicallyAccessedMemberTypes.All)] TComponent>(DialogParameters<TComponent> parameters, DialogOptions? options) where TComponent : IComponent;
+
         /// <summary>
         /// Displays a dialog with a custom title and parameters.
         /// </summary>
@@ -95,6 +112,23 @@ namespace MudBlazor
         /// <returns>A reference to the dialog.</returns>
         Task<IDialogReference> ShowAsync([DynamicallyAccessedMembers(DynamicallyAccessedMemberTypes.All)] Type component, string? title, DialogOptions options);
 
+        /// <summary>
+        /// Displays a dialog with parameters.
+        /// </summary>
+        /// <param name="component">The dialog to display.</param>
+        /// <param name="parameters">The custom parameters to set within the dialog.</param>
+        /// <returns>A reference to the dialog.</returns>
+        Task<IDialogReference> ShowAsync([DynamicallyAccessedMembers(DynamicallyAccessedMemberTypes.All)] Type component, DialogParameters parameters);
+
+        /// <summary>
+        /// Displays a dialog with parameters and options.
+        /// </summary>
+        /// <param name="component">The dialog to display.</param>
+        /// <param name="parameters">The custom parameters to set within the dialog.</param>
+        /// <param name="options">The custom display options for the dialog.</param>
+        /// <returns>A reference to the dialog.</returns>
+        Task<IDialogReference> ShowAsync([DynamicallyAccessedMembers(DynamicallyAccessedMemberTypes.All)] Type component, DialogParameters parameters, DialogOptions options);
+
         /// <summary>
         /// Displays a dialog with a custom title and parameters.
         /// </summary>
