diff --git a/internal/tui/components/detailedit/controller.go b/internal/tui/components/detailedit/controller.go
index d86fec5..8100819 100644
--- a/internal/tui/components/detailedit/controller.go
+++ b/internal/tui/components/detailedit/controller.go
@@ -241,6 +241,13 @@ func (c Controller) Update(msg tea.Msg) (Controller, tea.Cmd, *Submit, bool) {
 			return c, nil, nil, false
 		}
 
+	case tea.PasteMsg:
+		if !c.Active() {
+			return c, nil, nil, false
+		}
+		c.inputBox, taCmd = c.inputBox.Update(msg)
+		return c, taCmd, nil, true
+
 	case tea.KeyMsg:
 		if !c.Active() {
 			return c, nil, nil, false
