diff --git a/internal/api/ui/login/invite_user_handler.go b/internal/api/ui/login/invite_user_handler.go
index 18ba50248..3cde4351a 100644
--- a/internal/api/ui/login/invite_user_handler.go
+++ b/internal/api/ui/login/invite_user_handler.go
@@ -109,12 +109,14 @@ func (l *Login) checkUserInviteCode(w http.ResponseWriter, r *http.Request, auth
 }
 
 func (l *Login) resendUserInvite(w http.ResponseWriter, r *http.Request, authReq *domain.AuthRequest, userID, orgID, loginName string) {
-	var userOrgID, authRequestID string
+	var authRequestID string
 	if authReq != nil {
-		userOrgID = authReq.UserOrgID
 		authRequestID = authReq.ID
 	}
-	_, err := l.command.ResendInviteCode(setUserContext(r.Context(), userID, userOrgID), userID, userOrgID, authRequestID)
+	// The resend action must use the orgID from the invite link / form.
+	// The auth request can refer to a membership context (e.g. from an app) which might not exist for newly invited users.
+	// Using that orgID can lead to permission checks failing with `membership not found`.
+	_, err := l.command.ResendInviteCode(setUserContext(r.Context(), userID, orgID), userID, orgID, authRequestID)
 	l.renderInviteUser(w, r, authReq, userID, orgID, loginName, "", err)
 }
