diff --git a/checkov/terraform/checks/resource/gcp/GoogleSubnetworkPrivateGoogleEnabled.py b/checkov/terraform/checks/resource/gcp/GoogleSubnetworkPrivateGoogleEnabled.py
index 790d536b6..5a8da6815 100644
--- a/checkov/terraform/checks/resource/gcp/GoogleSubnetworkPrivateGoogleEnabled.py
+++ b/checkov/terraform/checks/resource/gcp/GoogleSubnetworkPrivateGoogleEnabled.py
@@ -5,6 +5,9 @@ from typing import Any
 from checkov.common.models.enums import CheckCategories, CheckResult
 from checkov.terraform.checks.resource.base_resource_value_check import BaseResourceValueCheck
 
+# flow logs and private_ip_google_access can't be enabled for subnetworks with the following purpose set
+PURPOSE_EXCEPTIONS = ["INTERNAL_HTTPS_LOAD_BALANCER", "REGIONAL_MANAGED_PROXY", "GLOBAL_MANAGED_PROXY"]
+
 
 class GoogleSubnetworkPrivateGoogleEnabled(BaseResourceValueCheck):
     def __init__(self) -> None:
@@ -16,7 +19,7 @@ class GoogleSubnetworkPrivateGoogleEnabled(BaseResourceValueCheck):
 
     def scan_resource_conf(self, conf: dict[str, list[Any]]) -> CheckResult:
         purpose = conf.get("purpose")
-        if purpose and isinstance(purpose, list) and purpose[0] == "INTERNAL_HTTPS_LOAD_BALANCER":
+        if purpose and isinstance(purpose, list) and purpose[0] in PURPOSE_EXCEPTIONS:
             return CheckResult.UNKNOWN
 
         return super().scan_resource_conf(conf)
