diff --git a/src/recipe.rs b/src/recipe.rs
index 6b5727b6..bf70dcdc 100644
--- a/src/recipe.rs
+++ b/src/recipe.rs
@@ -193,7 +193,8 @@ impl<'src, D> Recipe<'src, D> {
       }
     }
 
-    let evaluator = Evaluator::new(context, is_dependency, scope);
+    let mut evaluator = Evaluator::new(context, is_dependency, scope);
+    evaluator.assignments = Some(&context.module.assignments);
 
     if self.is_script() {
       self.run_script(context, scope, positional, evaluator)
