diff --git a/cli/src/commands/git/init.rs b/cli/src/commands/git/init.rs
index ae3672c38..32d83241b 100644
--- a/cli/src/commands/git/init.rs
+++ b/cli/src/commands/git/init.rs
@@ -240,14 +240,30 @@ fn init_git_refs(
     Ok(repo)
 }
 
-// Set repository level `trunk()` alias to the default branch for "origin".
+// Set repository level `trunk()` alias to the default branch for preferred
+// remote.
+//
+// If the Git repository has `upstream` remote (common when created via GitHub
+// CLI `gh repo fork`), prefer `upstream` over `origin`.
 pub fn maybe_set_repository_level_trunk_alias(
     ui: &Ui,
     workspace_command: &WorkspaceCommandHelper,
 ) -> Result<(), CommandError> {
     let git_repo = git::get_git_repo(workspace_command.repo().store())?;
+
+    // Prefer upstream if it exists.
+    let remote = if git_repo
+        .try_find_reference("refs/remotes/upstream/HEAD")
+        .map_err(internal_error)?
+        .is_some()
+    {
+        "upstream"
+    } else {
+        "origin"
+    };
+
     if let Some(reference) = git_repo
-        .try_find_reference("refs/remotes/origin/HEAD")
+        .try_find_reference(&format!("refs/remotes/{remote}/HEAD"))
         .map_err(internal_error)?
     {
         if let Some(reference_name) = reference.target().try_name() {
@@ -256,7 +272,7 @@ pub fn maybe_set_repository_level_trunk_alias(
                 .and_then(|name| parse_git_ref(name.as_ref()))
             {
                 // TODO: Can we assume the symbolic target points to the same remote?
-                let symbol = symbol.name.to_remote_symbol("origin".as_ref());
+                let symbol = symbol.name.to_remote_symbol(remote.as_ref());
                 write_repository_level_trunk_alias(ui, workspace_command.repo_path(), symbol)?;
             }
         };
diff --git a/cli/tests/test_git_init.rs b/cli/tests/test_git_init.rs
index 433a61a3f..300a383ee 100644
--- a/cli/tests/test_git_init.rs
+++ b/cli/tests/test_git_init.rs
@@ -200,6 +200,7 @@ fn test_git_init_external_import_trunk(bare: bool) {
         "refs/remotes/origin/trunk",
     );
 
+
     let output = test_env.run_jj_in(
         ".",
         [
