diff --git a/harper-core/src/linting/lint_group.rs b/harper-core/src/linting/lint_group.rs
index 1a7c3ad..6a083fb 100644
--- a/harper-core/src/linting/lint_group.rs
+++ b/harper-core/src/linting/lint_group.rs
@@ -226,6 +226,7 @@ use super::update_place_names::UpdatePlaceNames;
 use super::use_title_case::UseTitleCase;
 use super::verb_to_adjective::VerbToAdjective;
 use super::very_unique::VeryUnique;
+use super::verse_verb::VerseVerb;
 use super::vice_versa::ViceVersa;
 use super::vicious_loop::ViciousCircle;
 use super::vicious_loop::ViciousCircleOrCycle;
@@ -809,6 +810,7 @@ impl LintGroup {
         insert_struct_rule_with_dict!(UseTitleCase, true);
         insert_expr_rule!(VerbToAdjective, true);
         insert_expr_rule!(VeryUnique, true);
+        insert_expr_rule!(VerseVerb, true);
         insert_expr_rule!(ViceVersa, true);
         insert_expr_rule!(ViciousCircle, true);
         insert_expr_rule!(ViciousCircleOrCycle, false);
diff --git a/harper-core/src/linting/mod.rs b/harper-core/src/linting/mod.rs
index 0b7af3e..be09a01 100644
--- a/harper-core/src/linting/mod.rs
+++ b/harper-core/src/linting/mod.rs
@@ -237,6 +237,7 @@ mod update_place_names;
 mod use_title_case;
 mod verb_to_adjective;
 mod very_unique;
+mod verse_verb;
 mod vice_versa;
 mod vicious_loop;
 mod was_aloud;
