diff --git a/Mage.Sets/src/mage/cards/j/JeganthaTheWellspring.java b/Mage.Sets/src/mage/cards/j/JeganthaTheWellspring.java
index ca321fb4..b2b7ddb7 100644
--- a/Mage.Sets/src/mage/cards/j/JeganthaTheWellspring.java
+++ b/Mage.Sets/src/mage/cards/j/JeganthaTheWellspring.java
@@ -106,7 +106,12 @@ class JeganthaManaCondition extends ManaCondition {
 
     @Override
     public boolean apply(Game game, Ability source, UUID originalId, Cost costsToPay) {
-        // TODO find a better method.  this one forces the user to pay off the generic mana before continuing.
-        return source.getManaCostsToPay().getUnpaid().getMana().getGeneric() == 0;
+        Mana unpaid = source.getManaCostsToPay().getUnpaid().getMana();
+        return unpaid.getWhite() > 0
+                || unpaid.getBlue() > 0
+                || unpaid.getBlack() > 0
+                || unpaid.getRed() > 0
+                || unpaid.getGreen() > 0
+                || unpaid.getColorless() > 0;
     }
 }
