diff --git a/src/main/java/io/github/hectorvent/floci/services/lambda/LambdaController.java b/src/main/java/io/github/hectorvent/floci/services/lambda/LambdaController.java
index 134ce83..4de0764 100644
--- a/src/main/java/io/github/hectorvent/floci/services/lambda/LambdaController.java
+++ b/src/main/java/io/github/hectorvent/floci/services/lambda/LambdaController.java
@@ -473,6 +473,8 @@ public class LambdaController {
         node.put("State", fn.getState());
         if (fn.getStateReason() != null) node.put("StateReason", fn.getStateReason());
         if (fn.getStateReasonCode() != null) node.put("StateReasonCode", fn.getStateReasonCode());
+        node.put("LastUpdateStatus", "Successful");
+        node.put("LastUpdateStatusReason", "The function was successfully updated");
         node.put("CodeSize", fn.getCodeSizeBytes());
         node.put("PackageType", fn.getPackageType());
         if (fn.getImageUri() != null) node.put("ImageUri", fn.getImageUri());
