diff --git a/lib/node_modules/@stdlib/stats/base/nanstdevwd/test/test.ndarray.js b/lib/node_modules/@stdlib/stats/base/nanstdevwd/test/test.ndarray.js
index cd085c15467..4154816127b 100644
--- a/lib/node_modules/@stdlib/stats/base/nanstdevwd/test/test.ndarray.js
+++ b/lib/node_modules/@stdlib/stats/base/nanstdevwd/test/test.ndarray.js
@@ -316,3 +316,4 @@ tape( 'the function supports an `offset` parameter', function test( t ) {
 
 	t.end();
 });
+
