diff --git a/src/js/components/FileInput/FileInput.js b/src/js/components/FileInput/FileInput.js
index 98ae9e121..1c2e6dd18 100644
--- a/src/js/components/FileInput/FileInput.js
+++ b/src/js/components/FileInput/FileInput.js
@@ -353,6 +353,7 @@ const FileInput = forwardRef(
                           id: 'fileInput.browse',
                           messages,
                         })}
+                        disabled={disabled}
                         onClick={() => {
                           inputRef.current.click();
                           inputRef.current.focus();
@@ -365,6 +366,7 @@ const FileInput = forwardRef(
                         alignSelf="center"
                         ref={controlRef}
                         margin="small"
+                        disabled={disabled}
                         onClick={() => {
                           inputRef.current.click();
                           inputRef.current.focus();
@@ -427,6 +429,7 @@ const FileInput = forwardRef(
                         id: 'fileInput.browse',
                         messages,
                       })}
+                      disabled={disabled}
                       onClick={() => {
                         inputRef.current.click();
                         inputRef.current.focus();
@@ -439,6 +442,7 @@ const FileInput = forwardRef(
                       alignSelf="center"
                       ref={controlRef}
                       margin="small"
+                      disabled={disabled}
                       onClick={() => {
                         inputRef.current.click();
                         inputRef.current.focus();
@@ -524,6 +528,7 @@ const FileInput = forwardRef(
                             id: 'fileInput.browse',
                             messages,
                           })}
+                          disabled={disabled}
                           onClick={() => {
                             inputRef.current.click();
                             inputRef.current.focus();
@@ -535,6 +540,7 @@ const FileInput = forwardRef(
                           tabIndex={-1}
                           ref={controlRef}
                           margin="small"
+                          disabled={disabled}
                           onClick={() => {
                             inputRef.current.click();
                             inputRef.current.focus();
diff --git a/src/js/components/FileInput/__tests__/__snapshots__/FileInput-test.tsx.snap b/src/js/components/FileInput/__tests__/__snapshots__/FileInput-test.tsx.snap
index 3b722bf85..86e57380d 100644
--- a/src/js/components/FileInput/__tests__/__snapshots__/FileInput-test.tsx.snap
+++ b/src/js/components/FileInput/__tests__/__snapshots__/FileInput-test.tsx.snap
@@ -1085,10 +1085,9 @@ exports[`FileInput disabled 1`] = `
   cursor: pointer;
   align-self: center;
   margin: 12px;
-}
-
-.c6:hover {
-  text-decoration: underline;
+  opacity: 0.3;
+  cursor: default;
+  text-decoration: none;
 }
 
 .c5 {
@@ -1207,6 +1206,7 @@ exports[`FileInput disabled 1`] = `
       </span>
       <a
         class="c6"
+        disabled=""
         tabindex="-1"
       >
         browse
@@ -1334,10 +1334,9 @@ exports[`FileInput disabled with file selected 1`] = `
   text-decoration: none;
   cursor: pointer;
   margin: 12px;
-}
-
-.c13:hover {
-  text-decoration: underline;
+  opacity: 0.3;
+  cursor: default;
+  text-decoration: none;
 }
 
 .c2 {
@@ -1585,6 +1584,7 @@ exports[`FileInput disabled with file selected 1`] = `
               </button>
               <a
                 class="c13"
+                disabled=""
                 tabindex="-1"
               >
                 browse
