diff --git a/crates/package/src/lib.rs b/crates/package/src/lib.rs
index 7dfc9e0..c33f7bb 100644
--- a/crates/package/src/lib.rs
+++ b/crates/package/src/lib.rs
@@ -50,7 +50,16 @@
 //! ```no_run
 //! # use leo_ast::NetworkName;
 //! use leo_package::Package;
-//! let package = Package::from_directory("path/to/package", "/home/me/.aleo", false, false, Some(NetworkName::TestnetV0), Some("http://localhost:3030")).unwrap();
+//! let package = Package::from_directory(
+//!     "path/to/package",
+//!     "/home/me/.aleo",
+//!     false,
+//!     false,
+//!     Some(NetworkName::TestnetV0),
+//!     Some("http://localhost:3030"),
+//!     0,
+//! )
+//! .unwrap();
 //! ```
 //! This will read the manifest and keep their data in `package.manifest`.
 //! It will also process dependencies and store them in topological order in `package.compilation_units`. This processing
diff --git a/crates/passes/src/write_transforming/visitor.rs b/crates/passes/src/write_transforming/visitor.rs
index d3206fb..2b65d4e 100644
--- a/crates/passes/src/write_transforming/visitor.rs
+++ b/crates/passes/src/write_transforming/visitor.rs
@@ -80,8 +80,9 @@ impl<'a> WriteTransformingVisitor<'a> {
         if let Some(members) = self.array_members.get(&name.name).cloned() {
             for (i, member) in members.iter().cloned().enumerate() {
                 // Create a definition for each array index.
+                let index_type = if i <= u8::MAX as usize { IntegerType::U8 } else { IntegerType::U32 };
                 let index = Literal::integer(
-                    IntegerType::U8,
+                    index_type,
                     i.to_string(),
                     Default::default(),
                     self.state.node_builder.next_id(),
