diff --git a/src/main/java/org/json/XML.java b/src/main/java/org/json/XML.java
index 4bf4759..cc35da1 100644
--- a/src/main/java/org/json/XML.java
+++ b/src/main/java/org/json/XML.java
@@ -355,19 +355,23 @@ public class XML {
                                 && TYPE_ATTR.equals(string)) {
                             xmlXsiTypeConverter = config.getXsiTypeMap().get(token);
                         } else if (!nilAttributeFound) {
-                            Object obj = stringToValue((String) token);
-                            if (obj instanceof Boolean) {
-                                jsonObject.accumulate(string,
-                                        config.isKeepBooleanAsString()
-                                                ? ((String) token)
-                                                : obj);
-                            } else if (obj instanceof Number) {
-                                jsonObject.accumulate(string,
-                                        config.isKeepNumberAsString()
-                                                ? ((String) token)
-                                                : obj);
+                            if (config.keepStrings) {
+                                jsonObject.accumulate(string, (String) token);
                             } else {
-                                jsonObject.accumulate(string, stringToValue((String) token));
+                                Object obj = stringToValue((String) token);
+                                if (obj instanceof Boolean) {
+                                    jsonObject.accumulate(string,
+                                            config.isKeepBooleanAsString()
+                                                    ? ((String) token)
+                                                    : obj);
+                                } else if (obj instanceof Number) {
+                                    jsonObject.accumulate(string,
+                                            config.isKeepNumberAsString()
+                                                    ? ((String) token)
+                                                    : obj);
+                                } else {
+                                    jsonObject.accumulate(string, stringToValue((String) token));
+                                }
                             }
                         }
                         token = null;
@@ -417,19 +421,23 @@ public class XML {
                                     jsonObject.accumulate(config.getcDataTagName(),
                                             stringToValue(string, xmlXsiTypeConverter));
                                 } else {
-                                    Object obj = stringToValue((String) token);
-                                    if (obj instanceof Boolean) {
-                                        jsonObject.accumulate(config.getcDataTagName(),
-                                                config.isKeepBooleanAsString()
-                                                        ? ((String) token)
-                                                        : obj);
-                                    } else if (obj instanceof Number) {
-                                        jsonObject.accumulate(config.getcDataTagName(),
-                                                config.isKeepNumberAsString()
-                                                        ? ((String) token)
-                                                        : obj);
+                                    if (config.keepStrings) {
+                                        jsonObject.accumulate(config.getcDataTagName(), (String) token);
                                     } else {
-                                        jsonObject.accumulate(config.getcDataTagName(), stringToValue((String) token));
+                                        Object obj = stringToValue((String) token);
+                                        if (obj instanceof Boolean) {
+                                            jsonObject.accumulate(config.getcDataTagName(),
+                                                    config.isKeepBooleanAsString()
+                                                            ? ((String) token)
+                                                            : obj);
+                                        } else if (obj instanceof Number) {
+                                            jsonObject.accumulate(config.getcDataTagName(),
+                                                    config.isKeepNumberAsString()
+                                                            ? ((String) token)
+                                                            : obj);
+                                        } else {
+                                            jsonObject.accumulate(config.getcDataTagName(), stringToValue((String) token));
+                                        }
                                     }
                                 }
                             }
