diff --git a/src/Microsoft.ML.Tokenizers/Model/TiktokenTokenizer.cs b/src/Microsoft.ML.Tokenizers/Model/TiktokenTokenizer.cs
index 69cc3dfb9..af47daad3 100644
--- a/src/Microsoft.ML.Tokenizers/Model/TiktokenTokenizer.cs
+++ b/src/Microsoft.ML.Tokenizers/Model/TiktokenTokenizer.cs
@@ -1013,6 +1013,14 @@ namespace Microsoft.ML.Tokenizers
         private const string IMStart = "<|im_start|>";
         private const string IMEnd = "<|im_end|>";
         private const string IMSep = "<|im_sep|>";
+        private const string StartOfText = "<|startoftext|>";
+        private const string Return = "<|return|>";
+        private const string Constrain = "<|constrain|>";
+        private const string Channel = "<|channel|>";
+        private const string Start = "<|start|>";
+        private const string End = "<|end|>";
+        private const string Message = "<|message|>";
+        private const string Call = "<|call|>";
 
         private enum ModelEncoding
         {
@@ -1022,7 +1030,8 @@ namespace Microsoft.ML.Tokenizers
             P50kEdit,
             R50kBase,
             GPT2,
-            O200kBase
+            O200kBase,
+            O200kHarmony
         }
 
         private const string Phi4ModelName = "phi-4";
@@ -1032,6 +1041,9 @@ namespace Microsoft.ML.Tokenizers
                                                                 // chat
                                                                 ( "o1-", ModelEncoding.O200kBase ),       // e.g. o1-mini
                                                                 ( "o3-", ModelEncoding.O200kBase ),       // e.g. o3-mini
+                                                                ( "o4-mini-", ModelEncoding.O200kBase ),  // e.g. o4-mini
+                                                                ( "gpt-oss-", ModelEncoding.O200kHarmony), // e.g. gpt-oss-mini
+                                                                ( "gpt-4.5-", ModelEncoding.O200kBase),   // e.g., gpt-4.5-mini
                                                                 ( "gpt-4.1-", ModelEncoding.O200kBase),   // e.g., gpt-4.1-mini
                                                                 ( "gpt-4o-", ModelEncoding.O200kBase),    // e.g., gpt-4o-2024-05-13
                                                                 ( "gpt-4-", ModelEncoding.Cl100kBase),    // e.g., gpt-4-0314, etc., plus gpt-4-32k
@@ -1047,6 +1059,7 @@ namespace Microsoft.ML.Tokenizers
                                                                 { "o1", ModelEncoding.O200kBase },
                                                                 { "o3", ModelEncoding.O200kBase },
                                                                 { "o4-mini", ModelEncoding.O200kBase },
+                                                                { "gpt-4.5", ModelEncoding.O200kBase },
                                                                 { "gpt-4.1", ModelEncoding.O200kBase },
                                                                 { "gpt-4", ModelEncoding.Cl100kBase },
                                                                 { "gpt-3.5-turbo", ModelEncoding.Cl100kBase },
@@ -1147,6 +1160,19 @@ namespace Microsoft.ML.Tokenizers
                 case ModelEncoding.O200kBase:
                     return (new Dictionary<string, int> { { EndOfText, 199999 }, { EndOfPrompt, 200018 } }, O200kBaseRegex(), O200kBaseFile, Type.GetType(O200kBaseTypeName), O200kBasePackageName);
 
+                case ModelEncoding.O200kHarmony:
+                    return (new Dictionary<string, int> {
+                        { StartOfText, 199998 },
+                        { EndOfText, 199999 },
+                        { Return, 200002 },
+                        { Constrain, 200003 },
+                        { Channel, 200005 },
+                        { Start, 200006 },
+                        { End, 200007 },
+                        { Message, 200008 },
+                        { Call, 200012 }
+                    }, O200kBaseRegex(), O200kBaseFile, Type.GetType(O200kBaseTypeName), O200kBasePackageName);
+
                 case ModelEncoding.P50kBase:
                     return (new Dictionary<string, int> { { EndOfText, 50256 } }, P50kBaseRegex(), P50RanksFile, Type.GetType(P50kBaseTypeName), P50kBasePackageName);
 
@@ -1179,6 +1205,7 @@ namespace Microsoft.ML.Tokenizers
         internal const string P50kEditEncodingName = "p50k_edit";
         internal const string R50kBaseEncodingName = "r50k_base";
         internal const string O200kBaseEncodingName = "o200k_base";
+        internal const string O200kHarmonyEncodingName = "o200k_harmony";
 
         internal const string Cl100kBasePackageName = "Microsoft.ML.Tokenizers.Data.Cl100kBase";
         internal const string Gpt2PackageName = "Microsoft.ML.Tokenizers.Data.Gpt2";
@@ -1474,6 +1501,10 @@ namespace Microsoft.ML.Tokenizers
             {
                 modelEncoding = ModelEncoding.O200kBase;
             }
+            else if (encodingName.Equals(O200kHarmonyEncodingName, StringComparison.OrdinalIgnoreCase))
+            {
+                modelEncoding = ModelEncoding.O200kHarmony;
+            }
             else if (encodingName.Equals(P50kBaseEncodingName, StringComparison.OrdinalIgnoreCase))
             {
                 modelEncoding = ModelEncoding.P50kBase;
@@ -1488,7 +1519,7 @@ namespace Microsoft.ML.Tokenizers
             }
             else
             {
-                throw new ArgumentException($"The encoding name '{encodingName}' is not supported. The only supported encoding names are: {TiktokenTokenizer.Cl100kBaseEncodingName}, {TiktokenTokenizer.P50kBaseEncodingName}, {TiktokenTokenizer.P50kEditEncodingName}, and {TiktokenTokenizer.R50kBaseEncodingName}.", nameof(encodingName));
+                throw new ArgumentException($"The encoding name '{encodingName}' is not supported. The only supported encoding names are: {TiktokenTokenizer.Cl100kBaseEncodingName}, {TiktokenTokenizer.O200kBaseEncodingName}, {TiktokenTokenizer.O200kHarmonyEncodingName}, {TiktokenTokenizer.P50kBaseEncodingName}, {TiktokenTokenizer.P50kEditEncodingName}, and {TiktokenTokenizer.R50kBaseEncodingName}.", nameof(encodingName));
             }
 
             return CreateForModel(modelEncoding, modelName: null, extraSpecialTokens, normalizer);
diff --git a/test/Microsoft.ML.Tokenizers.Tests/TiktokenTests.cs b/test/Microsoft.ML.Tokenizers.Tests/TiktokenTests.cs
index 1ace27f1f..fc18978c8 100644
--- a/test/Microsoft.ML.Tokenizers.Tests/TiktokenTests.cs
+++ b/test/Microsoft.ML.Tokenizers.Tests/TiktokenTests.cs
@@ -402,6 +402,9 @@ namespace Microsoft.ML.Tokenizers.Tests
         [InlineData("o3-")]
         [InlineData("o3-mini")]
         [InlineData("o4-mini")]
+        [InlineData("o4-mini-2024")]
+        [InlineData("gpt-4.5")]
+        [InlineData("gpt-4.5-mini")]
         [InlineData("gpt-4.1")]
         [InlineData("gpt-4.1-mini")]
         [InlineData("gpt-4o")]
@@ -450,6 +453,7 @@ namespace Microsoft.ML.Tokenizers.Tests
         [InlineData("code-search-ada-code-001")]
         [InlineData("gpt2")]
         [InlineData("phi-4")]
+        [InlineData("gpt-oss-mini")]
         public void TestAllSupportedModelNames(string modelName)
         {
             Tokenizer tokenizer = TiktokenTokenizer.CreateForModel(modelName);
@@ -463,6 +467,7 @@ namespace Microsoft.ML.Tokenizers.Tests
         [InlineData("p50k_edit")]
         [InlineData("cl100k_base")]
         [InlineData("o200k_base")]
+        [InlineData("o200k_harmony")]
         public void TestAllSupportedEncodingNames(string encodingName)
         {
             Tokenizer tokenizer = TiktokenTokenizer.CreateForEncoding(encodingName);
@@ -476,6 +481,7 @@ namespace Microsoft.ML.Tokenizers.Tests
                 "p50k_edit" => "text-davinci-edit-001",
                 "cl100k_base" => "gpt-4",
                 "o200k_base" => "gpt-4o",
+                "o200k_harmony" => "gpt-oss-mini",
                 _ => throw new ArgumentException("Invalid encoding name"),
             };
 
@@ -502,6 +508,7 @@ namespace Microsoft.ML.Tokenizers.Tests
             Assert.Throws<ArgumentException>(() => TiktokenTokenizer.CreateForEncoding("p50k_edit_"));
             Assert.Throws<ArgumentException>(() => TiktokenTokenizer.CreateForEncoding("cl100k_base_"));
             Assert.Throws<ArgumentException>(() => TiktokenTokenizer.CreateForEncoding("o200k_base_"));
+            Assert.Throws<ArgumentException>(() => TiktokenTokenizer.CreateForEncoding("o200k_harmony_"));
         }
 
         [InlineData("gpt-4")]
