diff --git a/src/main/java/io/github/hectorvent/floci/core/common/AwsJsonController.java b/src/main/java/io/github/hectorvent/floci/core/common/AwsJsonController.java
index 88a11f8..4791666 100644
--- a/src/main/java/io/github/hectorvent/floci/core/common/AwsJsonController.java
+++ b/src/main/java/io/github/hectorvent/floci/core/common/AwsJsonController.java
@@ -70,6 +70,17 @@ public class AwsJsonController {
         this.cloudWatchMetricsJsonHandler = cloudWatchMetricsJsonHandler;
     }
 
+    @POST
+    @Path("{accountId:[0-9]+}/{queueName:.+}")
+    @Consumes("application/x-amz-json-1.0")
+    @Produces(MediaType.APPLICATION_JSON)
+    public Response handleSqsQueueJsonRequest(
+            @HeaderParam("X-Amz-Target") String target,
+            @Context HttpHeaders httpHeaders,
+            String body) {
+        return handleJsonRequest(target, httpHeaders, body);
+    }
+
     @POST
     @Consumes("application/x-amz-json-1.0")
     @Produces(MediaType.APPLICATION_JSON)
