diff --git a/langchain4j-ollama/src/main/java/dev/langchain4j/model/ollama/RunningOllamaModel.java b/langchain4j-ollama/src/main/java/dev/langchain4j/model/ollama/RunningOllamaModel.java
index 59b4c60..9d84052 100644
--- a/langchain4j-ollama/src/main/java/dev/langchain4j/model/ollama/RunningOllamaModel.java
+++ b/langchain4j-ollama/src/main/java/dev/langchain4j/model/ollama/RunningOllamaModel.java
@@ -23,18 +23,20 @@ public class RunningOllamaModel {
     @JsonDeserialize(using = OllamaDateDeserializer.class)
     private OffsetDateTime expiresAt;
     private Long sizeVram;
+    private Long contextLength;
 
     RunningOllamaModel() {
     }
 
-    RunningOllamaModel(String name, String model, Long size, OllamaModelDetails details, String digest, OffsetDateTime expiresAt, Long sizeVram) {
-        this.name = name;
-        this.model = model;
-        this.size = size;
-        this.details = details;
-        this.digest = digest;
-        this.expiresAt = expiresAt;
-        this.sizeVram = sizeVram;
+    private RunningOllamaModel(Builder builder) {
+        this.name = builder.name;
+        this.model = builder.model;
+        this.size = builder.size;
+        this.details = builder.details;
+        this.digest = builder.digest;
+        this.expiresAt = builder.expiresAt;
+        this.sizeVram = builder.sizeVram;
+        this.contextLength = builder.contextLength;
     }
 
     public String getName() {
@@ -93,6 +95,14 @@ public class RunningOllamaModel {
         this.sizeVram = sizeVram;
     }
 
+    public Long getContextLength() {
+        return contextLength;
+    }
+
+    public void setContextLength(Long contextLength) {
+        this.contextLength = contextLength;
+    }
+
     static Builder builder() {
         return new Builder();
     }
@@ -106,6 +116,7 @@ public class RunningOllamaModel {
         private OllamaModelDetails details;
         private OffsetDateTime expiresAt;
         private Long sizeVram;
+        private Long contextLength;
 
         Builder name(String name) {
             this.name = name;
@@ -142,8 +153,13 @@ public class RunningOllamaModel {
             return this;
         }
 
+        Builder contextLength(Long contextLength) {
+            this.contextLength = contextLength;
+            return this;
+        }
+
         RunningOllamaModel build() {
-            return new RunningOllamaModel(name, model, size, details, digest, expiresAt, sizeVram);
+            return new RunningOllamaModel(this);
         }
     }
 }
