diff --git a/modules/openapi-generator/src/main/java/org/openapitools/codegen/languages/KotlinSpringServerCodegen.java b/modules/openapi-generator/src/main/java/org/openapitools/codegen/languages/KotlinSpringServerCodegen.java
index 995c05ae1a1..e19cdbad7b6 100644
--- a/modules/openapi-generator/src/main/java/org/openapitools/codegen/languages/KotlinSpringServerCodegen.java
+++ b/modules/openapi-generator/src/main/java/org/openapitools/codegen/languages/KotlinSpringServerCodegen.java
@@ -651,6 +651,10 @@ public class KotlinSpringServerCodegen extends AbstractKotlinCodegen
                 supportingFiles.add(new SupportingFile("springBootApplication.mustache",
                         sanitizeDirectory(sourceFolder + File.separator + basePackage), "Application.kt"));
 
+                // Add enum converter configuration for non-default enumPropertyNaming
+                supportingFiles.add(new SupportingFile("converter.mustache",
+                        sanitizeDirectory(sourceFolder + File.separator + configPackage), "EnumConverterConfiguration.kt"));
+
                 if (useSwaggerUI && selectedDocumentationProviderRequiresSwaggerUiBootstrap()) {
                     supportingFiles.add(new SupportingFile("swagger-ui.mustache",
                             resourcesFolder + "/static", "swagger-ui.html"));
