diff --git a/src/libraries/System.IO.Compression/src/System/IO/Compression/ZipBlocks.Async.cs b/src/libraries/System.IO.Compression/src/System/IO/Compression/ZipBlocks.Async.cs
index 8305b5e42bf..f68f3ab6a5f 100644
--- a/src/libraries/System.IO.Compression/src/System/IO/Compression/ZipBlocks.Async.cs
+++ b/src/libraries/System.IO.Compression/src/System/IO/Compression/ZipBlocks.Async.cs
@@ -61,9 +61,11 @@ public static async Task<Zip64EndOfCentralDirectoryLocator> TryReadBlockAsync(St
 
         byte[] blockContents = new byte[TotalSize];
         int bytesRead = await stream.ReadAtLeastAsync(blockContents, blockContents.Length, throwOnEndOfStream: false, cancellationToken).ConfigureAwait(false);
-        bool zip64eocdLocatorProper = TryReadBlockCore(blockContents, bytesRead, out Zip64EndOfCentralDirectoryLocator? zip64EOCDLocator);
 
-        Debug.Assert(zip64eocdLocatorProper && zip64EOCDLocator != null); // we just found this using the signature finder, so it should be okay
+        if (!TryReadBlockCore(blockContents, bytesRead, out Zip64EndOfCentralDirectoryLocator? zip64EOCDLocator))
+        {
+            throw new InvalidDataException(SR.Zip64EOCDNotWhereExpected);
+        }
 
         return zip64EOCDLocator;
     }
diff --git a/src/libraries/System.IO.Compression/src/System/IO/Compression/ZipBlocks.cs b/src/libraries/System.IO.Compression/src/System/IO/Compression/ZipBlocks.cs
index 5e70cf29fc5..e6e906cd4a0 100644
--- a/src/libraries/System.IO.Compression/src/System/IO/Compression/ZipBlocks.cs
+++ b/src/libraries/System.IO.Compression/src/System/IO/Compression/ZipBlocks.cs
@@ -438,9 +438,10 @@ public static Zip64EndOfCentralDirectoryLocator TryReadBlock(Stream stream)
         {
             Span<byte> blockContents = stackalloc byte[TotalSize];
             int bytesRead = stream.ReadAtLeast(blockContents, blockContents.Length, throwOnEndOfStream: false);
-            bool zip64eocdLocatorProper = TryReadBlockCore(blockContents, bytesRead, out Zip64EndOfCentralDirectoryLocator? zip64EOCDLocator);
-
-            Debug.Assert(zip64eocdLocatorProper && zip64EOCDLocator != null); // we just found this using the signature finder, so it should be okay
+            if (!TryReadBlockCore(blockContents, bytesRead, out Zip64EndOfCentralDirectoryLocator? zip64EOCDLocator))
+            {
+                throw new InvalidDataException(SR.Zip64EOCDNotWhereExpected);
+            }
 
             return zip64EOCDLocator;
         }
