diff --git a/src/Riok.Mapperly/Descriptors/MappingBuilders/CtorMappingBuilder.cs b/src/Riok.Mapperly/Descriptors/MappingBuilders/CtorMappingBuilder.cs
index 8cbec2d..afd04e5 100644
--- a/src/Riok.Mapperly/Descriptors/MappingBuilders/CtorMappingBuilder.cs
+++ b/src/Riok.Mapperly/Descriptors/MappingBuilders/CtorMappingBuilder.cs
@@ -36,6 +36,9 @@ public static class CtorMappingBuilder
         if (ctx.Target is not INamedTypeSymbol namedTarget)
             return null;
 
+        if (HasMapperConstructor(namedTarget, ctx.SymbolAccessor))
+            return null;
+
         // resolve ctors which have the source as single argument
         var ctor = FindSingleArgCtor(namedTarget, ctx.Source, ctx.SymbolAccessor);
         if (ctor == null)
@@ -44,6 +47,9 @@ public static class CtorMappingBuilder
         return new CtorMapping(ctx.Source, ctx.Target, ctx.InstanceConstructors.BuildForConstructor(ctor));
     }
 
+    private static bool HasMapperConstructor(INamedTypeSymbol target, SymbolAccessor symbolAccessor) =>
+        target.InstanceConstructors.Any(x => symbolAccessor.HasAttribute<MapperConstructorAttribute>(x));
+
     private static IMethodSymbol? FindSingleArgCtor(INamedTypeSymbol target, ITypeSymbol sourceType, SymbolAccessor symbolAccessor) =>
         target
             .InstanceConstructors.Where(symbolAccessor.IsConstructorAccessible)
