diff --git a/hexpm/src/lib.rs b/hexpm/src/lib.rs
index 328aafa..26bbf02 100644
--- a/hexpm/src/lib.rs
+++ b/hexpm/src/lib.rs
@@ -1142,7 +1142,7 @@ impl OAuthDeviceAuthorisation {
         &mut self,
         response: http::Response<Vec<u8>>,
     ) -> Result<PollStep, ApiError> {
-        if self.start_time.elapsed() > Duration::from_mins(10) {
+        if self.start_time.elapsed() > Duration::from_secs(60 * 10) {
             return Err(ApiError::OAuthTimeout);
         }
 
diff --git a/language-server/src/code_action.rs b/language-server/src/code_action.rs
index 89819e4..5a442ec 100644
--- a/language-server/src/code_action.rs
+++ b/language-server/src/code_action.rs
@@ -1256,10 +1256,10 @@ pub fn code_action_add_missing_patterns(
             .map(|clause| clause.location.end)
             .unwrap_or(location.end - 1);
 
-        for pattern in missing {
-            let new_text = format!("\n{indent}  {pattern} -> todo");
-            edits.insert(insert_at, new_text);
-        }
+        let missing_patterns = missing
+            .iter()
+            .map(|pattern| format!("\n{indent}  {pattern} -> todo"))
+            .join("");
 
         // Add a newline + indent after the last pattern if there are no clauses
         //
@@ -1298,9 +1298,23 @@ pub fn code_action_add_missing_patterns(
                 }
             }
 
-            // Remove any blank spaces/lines between the start brace and end brace
-            edits.delete(SrcSpan::new(start_brace_location, insert_at));
-            edits.insert(insert_at, format!("\n{indent}"));
+            // Remove any blank spaces/lines between the start brace and end brace,
+            // but don't remove comments that may be there. Comments are not clauses,
+            // but they may contain code that the user wants to keep around for
+            // reference when filling in the missing patterns.
+            let empty_case_contents =
+                &module.code[start_brace_location as usize..insert_at as usize];
+            let empty_case_contains_comments = !empty_case_contents.trim().is_empty();
+            if empty_case_contains_comments {
+                edits.insert(insert_at, missing_patterns);
+            } else {
+                edits.replace(
+                    SrcSpan::new(start_brace_location, insert_at),
+                    format!("{missing_patterns}\n{indent}"),
+                );
+            }
+        } else {
+            edits.insert(insert_at, missing_patterns);
         }
 
         CodeActionBuilder::new("Add missing patterns")
