diff --git a/pkg/cidata/cidata.TEMPLATE.d/user-data b/pkg/cidata/cidata.TEMPLATE.d/user-data
index 3842f8d4..8330d53d 100644
--- a/pkg/cidata/cidata.TEMPLATE.d/user-data
+++ b/pkg/cidata/cidata.TEMPLATE.d/user-data
@@ -20,6 +20,8 @@ mounts:
     {{- range $m := $.Mounts}}
 - [{{$m.Tag}}, {{$m.MountPoint}}, {{$m.Type}}, "{{$m.Options}}", "0", "0"]
     {{- end }}
+  {{- else if not .RosettaEnabled }}
+[]
   {{- end }}
 {{- end }}
 
@@ -29,7 +31,7 @@ timezone: {{.TimeZone}}
 
 users:
   - name: "{{.User}}"
-    uid: "{{.UID}}"
+    uid: {{.UID}}
 {{- if .Comment }}
     gecos: {{ printf "%q" .Comment }}
 {{- end }}
@@ -37,7 +39,7 @@ users:
     shell: {{.Shell}}
     sudo: ALL=(ALL) NOPASSWD:ALL
     lock_passwd: true
-    ssh-authorized-keys:
+    ssh_authorized_keys:
     {{- range $val := .SSHPubKeys }}
       - {{ printf "%q" $val }}
     {{- end }}
