diff --git a/introspection.go b/introspection.go
index 51feb42..26810ee 100644
--- a/introspection.go
+++ b/introspection.go
@@ -59,7 +59,7 @@ func init() {
 
 	TypeKindEnumType = NewEnum(EnumConfig{
 		Name:        "__TypeKind",
-		Description: "An enum describing what kind of type a given `__Type` is",
+		Description: "An enum describing what kind of type a given `__Type` is.",
 		Values: EnumValueConfigMap{
 			"SCALAR": &EnumValueConfig{
 				Value:       TypeKindScalar,
@@ -146,7 +146,7 @@ func init() {
 			},
 			"OBJECT": &EnumValueConfig{
 				Value:       DirectiveLocationObject,
-				Description: "Location adjacent to a object definition.",
+				Description: "Location adjacent to an object type definition.",
 			},
 			"FIELD_DEFINITION": &EnumValueConfig{
 				Value:       DirectiveLocationFieldDefinition,
diff --git a/introspection_test.go b/introspection_test.go
index c0e62bf..d88c5a1 100644
--- a/introspection_test.go
+++ b/introspection_test.go
@@ -1422,7 +1422,7 @@ func TestIntrospection_ExposesDescriptionsOnEnums(t *testing.T) {
 		Data: map[string]interface{}{
 			"typeKindType": map[string]interface{}{
 				"name":        "__TypeKind",
-				"description": "An enum describing what kind of type a given `__Type` is",
+				"description": "An enum describing what kind of type a given `__Type` is.",
 				"enumValues": []interface{}{
 					map[string]interface{}{
 						"name":        "SCALAR",
