diff --git a/src/Quartz/Impl/Triggers/DailyTimeIntervalTriggerImpl.cs b/src/Quartz/Impl/Triggers/DailyTimeIntervalTriggerImpl.cs
index 7627b0a..1ec45e8 100644
--- a/src/Quartz/Impl/Triggers/DailyTimeIntervalTriggerImpl.cs
+++ b/src/Quartz/Impl/Triggers/DailyTimeIntervalTriggerImpl.cs
@@ -734,11 +734,11 @@ public override void SetPreviousFireTimeUtc(DateTimeOffset? previousFireTimeUtc)
 
         // f. Continue to calculate the fireTime by incremental unit of intervals.
         // recall that if fireTime was less that fireTimeStartDate, we didn't get this far
-        DateTimeOffset startOfDayUtc = TimeZoneUtil.ConvertTime(fireTimeStartDate, TimeZone);
-        long secondsAfterStart = (long) (fireTime.Value - startOfDayUtc).TotalSeconds;
+        DateTimeOffset startOfDayLocal = TimeZoneUtil.ConvertTime(fireTimeStartDate, TimeZone);
+        long secondsAfterStart = (long) (fireTime.Value.DateTime - startOfDayLocal.DateTime).TotalSeconds;
         long repeatLong = RepeatInterval;
 
-        DateTimeOffset sTime = fireTimeStartDate.ToUniversalTime();
+        DateTime sTime = fireTimeStartDate.DateTime;
         IntervalUnit repeatUnit = RepeatIntervalUnit;
         long jumpCount = 0;
         if (repeatUnit == IntervalUnit.Second)
@@ -750,7 +750,7 @@ public override void SetPreviousFireTimeUtc(DateTimeOffset? previousFireTimeUtc)
             }
 
             sTime = sTime.AddSeconds(RepeatInterval * (int) jumpCount);
-            fireTime = TimeZoneUtil.ConvertTime(sTime, TimeZone);
+            fireTime = new DateTimeOffset(sTime, TimeZoneUtil.GetUtcOffset(sTime, TimeZone));
         }
         else if (repeatUnit == IntervalUnit.Minute)
         {
@@ -760,7 +760,7 @@ public override void SetPreviousFireTimeUtc(DateTimeOffset? previousFireTimeUtc)
                 jumpCount++;
             }
             sTime = sTime.AddMinutes(RepeatInterval * (int) jumpCount);
-            fireTime = TimeZoneUtil.ConvertTime(sTime, TimeZone);
+            fireTime = new DateTimeOffset(sTime, TimeZoneUtil.GetUtcOffset(sTime, TimeZone));
         }
         else if (repeatUnit == IntervalUnit.Hour)
         {
@@ -770,7 +770,7 @@ public override void SetPreviousFireTimeUtc(DateTimeOffset? previousFireTimeUtc)
                 jumpCount++;
             }
             sTime = sTime.AddHours(RepeatInterval * (int) jumpCount);
-            fireTime = TimeZoneUtil.ConvertTime(sTime, TimeZone);
+            fireTime = new DateTimeOffset(sTime, TimeZoneUtil.GetUtcOffset(sTime, TimeZone));
         }
 
         // g. Check if we've exceeded the per-day repeat count.
