diff --git a/Zend/zend_compile.c b/Zend/zend_compile.c
index 5527f55a25c..d311a9f85ea 100644
--- a/Zend/zend_compile.c
+++ b/Zend/zend_compile.c
@@ -7492,6 +7492,18 @@ static void zend_compile_attributes(
 			zend_string *lcname = zend_string_tolower_ex(name, false);
 			zend_ast_list *args = el->child[1] ? zend_ast_get_list(el->child[1]) : NULL;
 
+			/* Attribute names must resolve to classes. */
+			if (el->child[0]->kind != ZEND_AST_ZVAL) {
+				zend_error_noreturn(E_COMPILE_ERROR,
+					"Attribute class \"%s\" must be a class", ZSTR_VAL(name));
+			}
+
+			if (CG(active_class_entry)
+					&& (CG(active_class_entry)->ce_flags & (ZEND_ACC_ENUM|ZEND_ACC_TRAIT|ZEND_ACC_INTERFACE))) {
+				zend_error_noreturn(E_COMPILE_ERROR,
+					"Attribute class \"%s\" must be a class", ZSTR_VAL(CG(active_class_entry)->name));
+			}
+
 			config = zend_internal_attribute_get(lcname);
 			zend_string_release(lcname);
