diff --git a/src/MudBlazor/Components/Select/MudSelect.razor.cs b/src/MudBlazor/Components/Select/MudSelect.razor.cs
index 81deab0..f2b2e51 100644
--- a/src/MudBlazor/Components/Select/MudSelect.razor.cs
+++ b/src/MudBlazor/Components/Select/MudSelect.razor.cs
@@ -783,7 +783,13 @@ namespace MudBlazor
         {
             // Apply comparer and refresh selected values
             _selectedValues = new HashSet<T?>(_selectedValues, arg.Value);
-            await UpdateSelectedValuesStateAsync(arg.Value);
+
+            if (MultiSelection)
+            {
+                await UpdateTextPropertyAsync(false);
+            }
+
+            await _context.NotifySelectionChangedAsync();
         }
 
         private async Task OnSelectedValuesChangedAsync(ParameterChangedEventArgs<IReadOnlyCollection<T?>?> arg)
