diff --git a/langchain4j-onnx-scoring/src/main/java/dev/langchain4j/model/scoring/onnx/AbstractInProcessScoringModel.java b/langchain4j-onnx-scoring/src/main/java/dev/langchain4j/model/scoring/onnx/AbstractInProcessScoringModel.java
index 2dec182..519494a 100644
--- a/langchain4j-onnx-scoring/src/main/java/dev/langchain4j/model/scoring/onnx/AbstractInProcessScoringModel.java
+++ b/langchain4j-onnx-scoring/src/main/java/dev/langchain4j/model/scoring/onnx/AbstractInProcessScoringModel.java
@@ -9,7 +9,7 @@ import dev.langchain4j.model.scoring.ScoringModel;
 import java.util.List;
 import java.util.stream.Collectors;
 
-abstract class AbstractInProcessScoringModel implements ScoringModel {
+abstract class AbstractInProcessScoringModel implements ScoringModel, AutoCloseable {
 
     public AbstractInProcessScoringModel() {
     }
@@ -29,4 +29,12 @@ abstract class AbstractInProcessScoringModel implements ScoringModel {
                 segments.stream().map(TextSegment::text).collect(Collectors.toList()));
         return Response.from(scoresAndTokenCount.scores, new TokenUsage(scoresAndTokenCount.tokenCount));
     }
+
+    /**
+     * Closes the underlying ONNX runtime session and releases native resources.
+     */
+    @Override
+    public void close() {
+        this.model().close();
+    }
 }
diff --git a/langchain4j-onnx-scoring/src/main/java/dev/langchain4j/model/scoring/onnx/OnnxScoringBertCrossEncoder.java b/langchain4j-onnx-scoring/src/main/java/dev/langchain4j/model/scoring/onnx/OnnxScoringBertCrossEncoder.java
index b1a74da..b5a5876 100644
--- a/langchain4j-onnx-scoring/src/main/java/dev/langchain4j/model/scoring/onnx/OnnxScoringBertCrossEncoder.java
+++ b/langchain4j-onnx-scoring/src/main/java/dev/langchain4j/model/scoring/onnx/OnnxScoringBertCrossEncoder.java
@@ -113,4 +113,12 @@ class OnnxScoringBertCrossEncoder {
         return 1 / (1 + Math.exp(-x));
     }
 
+    public void close() {
+        try {
+            this.session.close();
+        } catch (OrtException e) {
+            throw new RuntimeException(e);
+        }
+    }
+
 }
