diff --git a/src/Microsoft.ML.Tokenizers/Model/TiktokenTokenizer.cs b/src/Microsoft.ML.Tokenizers/Model/TiktokenTokenizer.cs
index 69cc3dfb9..f68ae3ae2 100644
--- a/src/Microsoft.ML.Tokenizers/Model/TiktokenTokenizer.cs
+++ b/src/Microsoft.ML.Tokenizers/Model/TiktokenTokenizer.cs
@@ -1030,6 +1030,7 @@ namespace Microsoft.ML.Tokenizers
         private static readonly (string Prefix, ModelEncoding Encoding)[] _modelPrefixToEncoding =
                                                             [
                                                                 // chat
+                                                                ( "gpt-oss-", ModelEncoding.O200kBase ),   // e.g. gpt-oss-1, gpt-oss-2 (harmony models)
                                                                 ( "o1-", ModelEncoding.O200kBase ),       // e.g. o1-mini
                                                                 ( "o3-", ModelEncoding.O200kBase ),       // e.g. o3-mini
                                                                 ( "gpt-4.1-", ModelEncoding.O200kBase),   // e.g., gpt-4.1-mini
@@ -1042,6 +1043,14 @@ namespace Microsoft.ML.Tokenizers
         private static readonly Dictionary<string, ModelEncoding> _modelToEncoding =
                                                             new Dictionary<string, ModelEncoding>(StringComparer.OrdinalIgnoreCase)
                                                             {
+                                                                // harmony OSS models
+                                                                { "gpt-oss-1", ModelEncoding.O200kBase },
+                                                                { "gpt-oss-2", ModelEncoding.O200kBase },
+                                                                { "gpt-oss-3", ModelEncoding.O200kBase },
+                                                                { "gpt-oss-mini", ModelEncoding.O200kBase },
+                                                                { "gpt-oss-micro", ModelEncoding.O200kBase },
+                                                                { "gpt-oss-nano", ModelEncoding.O200kBase },
+
                                                                 // chat
                                                                 { "gpt-4o", ModelEncoding.O200kBase },
                                                                 { "o1", ModelEncoding.O200kBase },
@@ -1145,7 +1154,18 @@ namespace Microsoft.ML.Tokenizers
                     return (new Dictionary<string, int> { { EndOfText, 50256 }, }, P50kBaseRegex(), GPT2File, Type.GetType(Gpt2TypeName), Gpt2PackageName);
 
                 case ModelEncoding.O200kBase:
-                    return (new Dictionary<string, int> { { EndOfText, 199999 }, { EndOfPrompt, 200018 } }, O200kBaseRegex(), O200kBaseFile, Type.GetType(O200kBaseTypeName), O200kBasePackageName);
+                    return (new Dictionary<string, int> {
+                        { "<|startoftext|>", 199998 },
+                        { EndOfText, 199999 },
+                        { "<|return|>", 200002 },
+                        { "<|constrain|>", 200003 },
+                        { "<|channel|>", 200005 },
+                        { "<|start|>", 200006 },
+                        { "<|end|>", 200007 },
+                        { "<|message|>", 200008 },
+                        { "<|call|>", 200012 },
+                        { EndOfPrompt, 200018 }
+                    }, O200kBaseRegex(), O200kBaseFile, Type.GetType(O200kBaseTypeName), O200kBasePackageName);
 
                 case ModelEncoding.P50kBase:
                     return (new Dictionary<string, int> { { EndOfText, 50256 } }, P50kBaseRegex(), P50RanksFile, Type.GetType(P50kBaseTypeName), P50kBasePackageName);
