diff --git a/naming/src/main/java/com/alibaba/nacos/naming/core/CatalogServiceV2Impl.java b/naming/src/main/java/com/alibaba/nacos/naming/core/CatalogServiceV2Impl.java
index a87dfa3..53be539 100644
--- a/naming/src/main/java/com/alibaba/nacos/naming/core/CatalogServiceV2Impl.java
+++ b/naming/src/main/java/com/alibaba/nacos/naming/core/CatalogServiceV2Impl.java
@@ -113,7 +113,7 @@ public class CatalogServiceV2Impl implements CatalogService {
             throw new NacosException(NacosException.NOT_FOUND,
                     String.format("service %s@@%s is not found!", groupName, serviceName));
         }
-        if (!serviceStorage.getClusters(service).contains(clusterName)) {
+        if (StringUtils.isNotBlank(clusterName) && !serviceStorage.getClusters(service).contains(clusterName)) {
             throw new NacosException(NacosException.NOT_FOUND, "cluster " + clusterName + " is not found!");
         }
         ServiceInfo serviceInfo = serviceStorage.getData(service);
diff --git a/naming/src/main/java/com/alibaba/nacos/naming/model/form/InstanceListForm.java b/naming/src/main/java/com/alibaba/nacos/naming/model/form/InstanceListForm.java
index 025677a..ac5d8db 100644
--- a/naming/src/main/java/com/alibaba/nacos/naming/model/form/InstanceListForm.java
+++ b/naming/src/main/java/com/alibaba/nacos/naming/model/form/InstanceListForm.java
@@ -18,10 +18,9 @@ package com.alibaba.nacos.naming.model.form;
 
 import com.alibaba.nacos.api.common.Constants;
 import com.alibaba.nacos.api.exception.api.NacosApiException;
+import com.alibaba.nacos.api.model.NacosForm;
 import com.alibaba.nacos.api.model.v2.ErrorCode;
 import com.alibaba.nacos.common.utils.StringUtils;
-import com.alibaba.nacos.api.model.NacosForm;
-import com.alibaba.nacos.naming.misc.UtilsAndCommons;
 import org.springframework.http.HttpStatus;
 
 import java.util.Objects;
@@ -68,9 +67,6 @@ public class InstanceListForm implements NacosForm {
         if (StringUtils.isBlank(groupName)) {
             groupName = Constants.DEFAULT_GROUP;
         }
-        if (StringUtils.isBlank(clusterName)) {
-            clusterName = UtilsAndCommons.DEFAULT_CLUSTER_NAME;
-        }
         if (null == healthyOnly) {
             healthyOnly = false;
         }
