diff --git a/src/node/utils/padDiff.ts b/src/node/utils/padDiff.ts
index 53f79fa..6ab9846 100644
--- a/src/node/utils/padDiff.ts
+++ b/src/node/utils/padDiff.ts
@@ -138,7 +138,7 @@ class PadDiff {
 
     let superChangeset = null;
 
-    for (let rev = this._fromRev + 1; rev <= this._toRev; rev += bulkSize) {
+    for (let rev = Number(this._fromRev) + 1; rev <= Number(this._toRev); rev += bulkSize) {
       // get the bulk
       const {changesets, authors} = await this._getChangesetsInBulk(rev, bulkSize);
 
@@ -209,7 +209,7 @@ class PadDiff {
       await this.getHtml();
     }
 
-    return this.self!._authors;
+    return this._authors;
   }
 
   _extendChangesetWithAuthor(changeset: any, author: any, apool: any){
