diff --git a/include/po/parser.h b/include/po/parser.h
index b143510..9755ef2 100644
--- a/include/po/parser.h
+++ b/include/po/parser.h
@@ -33,9 +33,9 @@ stringToInteger(ConvResultT (&Conv)(const char *, char **, int),
   auto SavedErrNo = std::exchange(errno, 0);
   const auto Result = Conv(CStr, &EndPtr, 10);
   std::swap(SavedErrNo, errno);
-  if (EndPtr == CStr) {
+  if (EndPtr == CStr || *EndPtr != '\0') {
     return cxx20::unexpected<Error>(std::in_place, ErrCode::InvalidArgument,
-                                    ""s);
+                                    "invalid integer value: "s + Value);
   }
   auto InsideRange = [](auto WiderResult) constexpr noexcept {
     using WiderResultT = decltype(WiderResult);
@@ -64,9 +64,9 @@ stringToFloating(ConvResultT (&Conv)(const char *, char **),
   auto SavedErrNo = std::exchange(errno, 0);
   const auto Result = Conv(CStr, &EndPtr);
   std::swap(SavedErrNo, errno);
-  if (EndPtr == CStr) {
+  if (EndPtr == CStr || *EndPtr != '\0') {
     return cxx20::unexpected<Error>(std::in_place, ErrCode::InvalidArgument,
-                                    ""s);
+                                    "invalid floating value: "s + Value);
   }
   if (SavedErrNo == ERANGE) {
     return cxx20::unexpected<Error>(std::in_place, ErrCode::OutOfRange, ""s);
diff --git a/lib/po/argument_parser.cpp b/lib/po/argument_parser.cpp
index d54a5a8..db07db6 100644
--- a/lib/po/argument_parser.cpp
+++ b/lib/po/argument_parser.cpp
@@ -51,7 +51,9 @@ cxx20::expected<bool, Error> ArgumentParser::SubCommandDescriptor::parse(
         }
       }
     } else if (!Escaped && CurrentDesc) {
-      consume_argument(*CurrentDesc, Arg);
+      if (auto Res = consume_argument(*CurrentDesc, Arg); !Res) {
+        return cxx20::unexpected(Res.error());
+      }
       CurrentDesc = nullptr;
     } else {
       // no more options
@@ -277,9 +279,11 @@ ArgumentParser::SubCommandDescriptor::consume_long_option_with_argument(
     } else if (ArgumentDescriptor *CurrentDesc = *Res; !CurrentDesc) {
       return cxx20::unexpected<Error>(std::in_place, ErrCode::InvalidArgument,
                                       "option "s + std::string(Option) +
-                                          "doesn't need arguments."s);
+                                          " doesn't need arguments."s);
     } else {
-      consume_argument(*CurrentDesc, Argument);
+      if (auto Res2 = consume_argument(*CurrentDesc, Argument); !Res2) {
+        return cxx20::unexpected(Res2.error());
+      }
       return nullptr;
     }
   } else {
