diff --git a/src/Riok.Mapperly/Descriptors/DescriptorBuilder.cs b/src/Riok.Mapperly/Descriptors/DescriptorBuilder.cs
index 757be84..153f76e 100644
--- a/src/Riok.Mapperly/Descriptors/DescriptorBuilder.cs
+++ b/src/Riok.Mapperly/Descriptors/DescriptorBuilder.cs
@@ -72,7 +72,7 @@ public class DescriptorBuilder
             _attributeAccessor,
             _unsafeAccessorContext,
             _diagnostics,
-            new MappingBuilder(_mappings, mapperDeclaration),
+            new MappingBuilder(_mappings, mapperDeclaration, genericTypeChecker),
             new ExistingTargetMappingBuilder(_mappings, mapperDeclaration),
             _inlineMappings,
             mapperDeclaration.Syntax.GetLocation()
diff --git a/src/Riok.Mapperly/Descriptors/MappingBodyBuilders/RuntimeTargetTypeMappingBodyBuilder.cs b/src/Riok.Mapperly/Descriptors/MappingBodyBuilders/RuntimeTargetTypeMappingBodyBuilder.cs
index faab2d9..fc9f110 100644
--- a/src/Riok.Mapperly/Descriptors/MappingBodyBuilders/RuntimeTargetTypeMappingBodyBuilder.cs
+++ b/src/Riok.Mapperly/Descriptors/MappingBodyBuilders/RuntimeTargetTypeMappingBodyBuilder.cs
@@ -61,7 +61,7 @@ public static class RuntimeTargetTypeMappingBodyBuilder
     private static IEnumerable<INewInstanceUserMapping> GetUserMappingCandidates(MappingBuilderContext ctx) =>
         ctx
             .NewInstanceMappings.Select(x => x.Value)
-            .Where(x => x is not UserDefinedNewInstanceRuntimeTargetTypeMapping)
+            .Where(x => x is not UserDefinedNewInstanceRuntimeTargetTypeMapping and not UserImplementedGenericTypeMapping { IsClosedGenericMapping: false })
             .OfType<INewInstanceUserMapping>();
 
     private static IEnumerable<IExistingTargetUserMapping> GetExistingTargetUserMappingCandidates(MappingBuilderContext ctx) =>
diff --git a/src/Riok.Mapperly/Descriptors/MappingBuilders/MappingBuilder.cs b/src/Riok.Mapperly/Descriptors/MappingBuilders/MappingBuilder.cs
index e4320dd..118d6a6 100644
--- a/src/Riok.Mapperly/Descriptors/MappingBuilders/MappingBuilder.cs
+++ b/src/Riok.Mapperly/Descriptors/MappingBuilders/MappingBuilder.cs
@@ -1,10 +1,11 @@
 using Riok.Mapperly.Descriptors.Mappings;
 using Riok.Mapperly.Descriptors.Mappings.UserMappings;
+using Riok.Mapperly.Helpers;
 using Riok.Mapperly.Symbols;
 
 namespace Riok.Mapperly.Descriptors.MappingBuilders;
 
-public class MappingBuilder(MappingCollection mappings, MapperDeclaration mapperDeclaration)
+public class MappingBuilder(MappingCollection mappings, MapperDeclaration mapperDeclaration, GenericTypeChecker genericTypeChecker)
 {
     private readonly HashSet<string> _resolvedMappingNames = [];
 
@@ -43,7 +44,7 @@ public class MappingBuilder(MappingCollection mappings, MapperDeclaration mapper
     public IEnumerable<IExistingTargetUserMapping> ExistingTargetUserMappings => mappings.ExistingTargetUserMappings;
 
     public INewInstanceMapping? Find(TypeMappingKey mapping, ParameterScope? scope = null) =>
-        mappings.FindNewInstanceMapping(mapping, scope);
+        mappings.FindNewInstanceMapping(mapping, genericTypeChecker, scope);
 
     public INewInstanceMapping? FindOrResolveNamed(SimpleMappingBuilderContext ctx, string name, out bool ambiguousName)
     {
diff --git a/src/Riok.Mapperly/Descriptors/MappingCollection.cs b/src/Riok.Mapperly/Descriptors/MappingCollection.cs
index a271e04..00acbe0 100644
--- a/src/Riok.Mapperly/Descriptors/MappingCollection.cs
+++ b/src/Riok.Mapperly/Descriptors/MappingCollection.cs
@@ -4,6 +4,7 @@ using Riok.Mapperly.Descriptors.Mappings;
 using Riok.Mapperly.Descriptors.Mappings.ExistingTarget;
 using Riok.Mapperly.Descriptors.Mappings.UserMappings;
 using Riok.Mapperly.Helpers;
+using Riok.Mapperly.Symbols;
 
 namespace Riok.Mapperly.Descriptors;
 
@@ -65,6 +66,12 @@ public class MappingCollection
     public INewInstanceMapping? FindNewInstanceMapping(TypeMappingKey mappingKey, ParameterScope? scope = null) =>
         _newInstanceMappings.Find(mappingKey, scope);
 
+    public INewInstanceMapping? FindNewInstanceMapping(
+        TypeMappingKey mappingKey,
+        GenericTypeChecker genericTypeChecker,
+        ParameterScope? scope = null
+    ) => _newInstanceMappings.Find(mappingKey, genericTypeChecker, scope);
+
     public INewInstanceUserMapping? FindNewInstanceUserMapping(IMethodSymbol method) => _newInstanceMappings.FindUserMapping(method);
 
     public INewInstanceMapping? FindNamedNewInstanceMapping(string name, out bool ambiguousName) =>
@@ -242,6 +249,15 @@ public class MappingCollection
             return mapping;
         }
 
+        public T? Find(TypeMappingKey mappingKey, GenericTypeChecker genericTypeChecker, ParameterScope? scope = null)
+        {
+            var mapping = Find(mappingKey, scope);
+            if (mapping != null)
+                return mapping;
+
+            return FindGenericMapping(mappingKey, genericTypeChecker);
+        }
+
         public T? FindNamed(string name, out bool ambiguousName)
         {
             ambiguousName = false;
@@ -267,6 +283,37 @@ public class MappingCollection
             return default;
         }
 
+        private T? FindGenericMapping(TypeMappingKey mappingKey, GenericTypeChecker genericTypeChecker)
+        {
+            foreach (var userMapping in _userMappingsByMethod.Values.OfType<UserImplementedGenericTypeMapping>())
+            {
+                var genericTypeMappings = genericTypeChecker.InferGenericTypeMappings(
+                    userMapping.Method.TypeParameters,
+                    userMapping.SourceType,
+                    mappingKey.Source,
+                    userMapping.TargetType,
+                    mappingKey.Target
+                );
+
+                foreach (var genericTypeMapping in genericTypeMappings)
+                {
+                    return (T)(object)new UserImplementedGenericTypeMapping(
+                        null,
+                        userMapping.Method,
+                        userMapping.Default,
+                        new MethodParameter(0, "source", mappingKey.Source),
+                        mappingKey.Target,
+                        null,
+                        userMapping.IsExternal,
+                        UserImplementedMethodMapping.TargetNullability.NeverNull,
+                        genericTypeMapping
+                    );
+                }
+            }
+
+            return default;
+        }
+
         public void AddNamedUserMapping(string? name, TUserMapping mapping)
         {
             var isNewUserMappingMethod = _userMappingsByMethod.TryAdd(mapping.Method, mapping);
diff --git a/src/Riok.Mapperly/Descriptors/UserMethodMappingExtractor.cs b/src/Riok.Mapperly/Descriptors/UserMethodMappingExtractor.cs
index 438d9be..7259dbf 100644
--- a/src/Riok.Mapperly/Descriptors/UserMethodMappingExtractor.cs
+++ b/src/Riok.Mapperly/Descriptors/UserMethodMappingExtractor.cs
@@ -167,7 +167,7 @@ public static class UserMethodMappingExtractor
     )
     {
         var userMappingConfig = GetUserMappingConfig(ctx, method, out var hasAttribute);
-        var valid = !method.IsGenericMethod && (allowPartial || !method.IsPartialDefinition) && (!isStatic || method.IsStatic);
+        var valid = (allowPartial || !method.IsPartialDefinition) && (!isStatic || method.IsStatic) && (!method.IsGenericMethod || !method.ReturnsVoid);
 
         if (!valid || !UserMappingMethodParameterExtractor.BuildParameters(ctx, method, out var parameters))
         {
@@ -196,6 +196,21 @@ public static class UserMethodMappingExtractor
         }
 
         var (targetType, targetTypeNullability) = BuildTargetType(ctx, method, parameters.Source.Name);
+        if (method.IsGenericMethod)
+        {
+            return new UserImplementedGenericTypeMapping(
+                receiver,
+                method,
+                userMappingConfig.Default ?? isDefault,
+                parameters.Source,
+                targetType,
+                parameters.ReferenceHandler,
+                isExternal,
+                targetTypeNullability,
+                new GenericTypeChecker.GenericTypeMapping(new Dictionary<ITypeParameterSymbol, ITypeSymbol>(SymbolEqualityComparer.Default))
+            );
+        }
+
         return new UserImplementedMethodMapping(
             receiver,
             method,
diff --git a/src/Riok.Mapperly/Helpers/GenericTypeChecker.cs b/src/Riok.Mapperly/Helpers/GenericTypeChecker.cs
index 83b130c..044b254 100644
--- a/src/Riok.Mapperly/Helpers/GenericTypeChecker.cs
+++ b/src/Riok.Mapperly/Helpers/GenericTypeChecker.cs
@@ -1,3 +1,4 @@
+using System.Collections;
 using System.Diagnostics.CodeAnalysis;
 using Microsoft.CodeAnalysis;
 using Riok.Mapperly.Descriptors;
@@ -39,6 +40,29 @@ public class GenericTypeChecker(SymbolAccessor accessor, WellKnownTypes types)
         return InferAndCheckTypes(state, parameterArguments).Success;
     }
 
+    public IEnumerable<GenericTypeMapping> InferGenericTypeMappings(
+        IEnumerable<ITypeParameterSymbol> typeParameters,
+        ITypeSymbol sourceParameterType,
+        ITypeSymbol sourceType,
+        ITypeSymbol targetParameterType,
+        ITypeSymbol targetType
+    )
+    {
+        var typeParametersArray = typeParameters.ToArray();
+        var state = new InferState(typeParametersArray);
+        if (!InferTypes(state, sourceParameterType, sourceType))
+            return [];
+
+        return InferUnboundTypeParameters(state, typeParametersArray, targetParameterType, targetType)
+            .Where(x =>
+                InferAndCheckTypes(
+                    x.State,
+                    [(sourceParameterType, sourceType), (targetParameterType, targetType)]
+                ).Success
+            )
+            .Select(x => new GenericTypeMapping(x.State.InferredTypes));
+    }
+
     private GenericTypeCheckerResult InferAndCheckTypes(
         InferState state,
         IEnumerable<(ITypeSymbol Parameter, ITypeSymbol Argument)> parameterArguments
@@ -70,6 +94,102 @@ public class GenericTypeChecker(SymbolAccessor accessor, WellKnownTypes types)
         };
     }
 
+    private bool InferTypes(InferState state, ITypeSymbol param, ITypeSymbol arg)
+    {
+        return param switch
+        {
+            ITypeParameterSymbol typeParam => InferType(state, typeParam, arg),
+            IArrayTypeSymbol paramArray => arg is IArrayTypeSymbol argArray
+                && InferTypes(
+                    state,
+                    paramArray.ElementType.WithNullableAnnotation(paramArray.ElementNullableAnnotation),
+                    argArray.ElementType.WithNullableAnnotation(argArray.ElementNullableAnnotation)
+                ),
+            INamedTypeSymbol { IsGenericType: true } paramNamedType => InferTypes(state, paramNamedType, arg),
+            _ => true,
+        };
+    }
+
+    private bool InferTypes(InferState state, INamedTypeSymbol param, ITypeSymbol arg)
+    {
+        if (!arg.ExtendsOrImplementsGeneric(param.OriginalDefinition, out var genericTypedArgument))
+            return false;
+
+        for (var i = 0; i < param.TypeArguments.Length; i++)
+        {
+            if (!InferTypes(state, param.TypeArguments[i], genericTypedArgument.TypeArguments[i]))
+                return false;
+        }
+
+        return true;
+    }
+
+    private bool InferType(InferState state, ITypeParameterSymbol typeParameter, ITypeSymbol arg)
+    {
+        if (state.IsTypeParameterInferred(typeParameter, out var boundType))
+        {
+            return SymbolEqualityComparer.Default.Equals(boundType, arg)
+                && (
+                    arg.NullableAnnotation == boundType.NullableAnnotation
+                    || typeParameter.IsNullable() != false
+                    || !arg.IsNullable()
+                );
+        }
+
+        state.SetInferredType(typeParameter, arg);
+        return true;
+    }
+
+    private IEnumerable<(InferState State, ITypeSymbol InferredType)> InferUnboundTypeParameters(
+        InferState state,
+        IEnumerable<ITypeParameterSymbol> typeParameters,
+        ITypeSymbol param,
+        ITypeSymbol target
+    )
+    {
+        if (param is ITypeParameterSymbol typeParameter && !state.IsTypeParameterInferred(typeParameter, out _))
+        {
+            var inferredState = state;
+            inferredState.SetInferredType(typeParameter, target);
+            yield return (inferredState, target);
+            yield break;
+        }
+
+        if (
+            param is IArrayTypeSymbol paramArray
+            && target is IArrayTypeSymbol targetArray
+            && paramArray.ElementType.WithNullableAnnotation(paramArray.ElementNullableAnnotation) is { } paramElementType
+            && targetArray.ElementType.WithNullableAnnotation(targetArray.ElementNullableAnnotation) is { } targetElementType
+        )
+        {
+            foreach (var result in InferUnboundTypeParameters(state, typeParameters, paramElementType, targetElementType))
+            {
+                yield return result;
+            }
+        }
+
+        if (param is not INamedTypeSymbol { IsGenericType: true } paramNamedType)
+            yield break;
+
+        if (!target.ExtendsOrImplementsGeneric(paramNamedType.OriginalDefinition, out var genericTypedTarget))
+            yield break;
+
+        for (var i = 0; i < paramNamedType.TypeArguments.Length; i++)
+        {
+            foreach (
+                var result in InferUnboundTypeParameters(
+                    state,
+                    typeParameters,
+                    paramNamedType.TypeArguments[i],
+                    genericTypedTarget.TypeArguments[i]
+                )
+            )
+            {
+                yield return result;
+            }
+        }
+    }
+
     private ITypeSymbol? InferAndCheckTypes(InferState state, ITypeParameterSymbol typeParameter, ITypeSymbol arg)
     {
         if (state.IsTypeParameterInferred(typeParameter, out var boundType))
@@ -178,6 +298,27 @@ public class GenericTypeChecker(SymbolAccessor accessor, WellKnownTypes types)
         }
     }
 
+    public readonly record struct GenericTypeMapping(IReadOnlyDictionary<ITypeParameterSymbol, ITypeSymbol> InferredTypes)
+        : IReadOnlyDictionary<ITypeParameterSymbol, ITypeSymbol>
+    {
+        public ITypeSymbol this[ITypeParameterSymbol key] => InferredTypes[key];
+
+        public IEnumerable<ITypeParameterSymbol> Keys => InferredTypes.Keys;
+
+        public IEnumerable<ITypeSymbol> Values => InferredTypes.Values;
+
+        public int Count => InferredTypes.Count;
+
+        public bool ContainsKey(ITypeParameterSymbol key) => InferredTypes.ContainsKey(key);
+
+        public IEnumerator<KeyValuePair<ITypeParameterSymbol, ITypeSymbol>> GetEnumerator() => InferredTypes.GetEnumerator();
+
+        public bool TryGetValue(ITypeParameterSymbol key, [NotNullWhen(true)] out ITypeSymbol value) =>
+            InferredTypes.TryGetValue(key, out value!);
+
+        IEnumerator IEnumerable.GetEnumerator() => GetEnumerator();
+    }
+
     /// <summary>
     /// The result of a type check.
     /// </summary>
