diff --git a/pkg/limatmpl/embed.go b/pkg/limatmpl/embed.go
index c95b29ad..9e7fad81 100644
--- a/pkg/limatmpl/embed.go
+++ b/pkg/limatmpl/embed.go
@@ -160,8 +160,33 @@ func (tmpl *Template) merge(base *Template) error {
 	if err := tmpl.mergeBase(base); err != nil {
 		return tmpl.ClearOnError(err)
 	}
+	if err := tmpl.Unmarshal(); err != nil {
+		return err
+	}
+	if err := base.Unmarshal(); err != nil {
+		return err
+	}
+	dns := base.Config.DNS
+	if len(tmpl.Config.DNS) > 0 {
+		dns = tmpl.Config.DNS
+	}
 	documents := fmt.Sprintf("%s\n---\n%s", string(tmpl.Bytes), string(base.Bytes))
-	return tmpl.evalExprImpl(mergeDocuments, []byte(documents))
+	if err := tmpl.evalExprImpl(mergeDocuments, []byte(documents)); err != nil {
+		return err
+	}
+	if len(dns) > 0 {
+		if err := tmpl.Unmarshal(); err != nil {
+			return err
+		}
+		tmpl.Config.DNS = dns
+		b, err := limayaml.Marshal(tmpl.Config, false)
+		if err != nil {
+			return err
+		}
+		tmpl.Bytes = b
+		tmpl.Config = nil
+	}
+	return nil
 }
 
 // mergeBase generates a yq script to merge the template with a base.
@@ -225,7 +250,7 @@ const mergeDocuments = `
 | $b | select((.base | tag == "!!str") and ($a.base | tag == "!!seq")) | .base = [ "" + .base ]
 
 # Delete base DNS entries if the template list is not empty.
-| $a | select(.dns) | del($b.dns, $c.dns)
+| $a | select(.dns | tag == "!!seq" and length > 0) | del($b.dns, $c.dns)
 
 # Mark all new list fields with a custom tag. This is needed to avoid appending
 # newly copied lists to themselves again when we merge lists.
