diff --git a/springdoc-openapi-starter-common/src/main/java/org/springdoc/core/service/AbstractRequestService.java b/springdoc-openapi-starter-common/src/main/java/org/springdoc/core/service/AbstractRequestService.java
index 681a61570..6c6879988 100644
--- a/springdoc-openapi-starter-common/src/main/java/org/springdoc/core/service/AbstractRequestService.java
+++ b/springdoc-openapi-starter-common/src/main/java/org/springdoc/core/service/AbstractRequestService.java
@@ -471,6 +471,8 @@ public abstract class AbstractRequestService {
 			return false;
 		if (isRequestBodyWithMapType(parameter))
 			return false;
+		if (isRequestPartWithMapType(parameter))
+			return false;
 		return isRequestTypeToIgnore(parameter.getParameterType());
 	}
 
@@ -510,6 +512,27 @@ public abstract class AbstractRequestService {
 		return Map.class.isAssignableFrom(parameterType);
 	}
 
+	/**
+	 * Is request part with map type.
+	 *
+	 * @param parameter the parameter
+	 * @return the boolean
+	 */
+	private boolean isRequestPartWithMapType(MethodParameter parameter) {
+		// Exclude parameters from the Actuator package
+		if (parameter.getContainingClass().getPackageName().startsWith("org.springframework.boot.actuate")) {
+			return false;
+		}
+		// Check for @RequestPart annotation
+		RequestPart requestPart = parameter.getParameterAnnotation(RequestPart.class);
+		if (requestPart == null) {
+			return false;
+		}
+		// Check if the parameter type is assignable to Map
+		Class<?> parameterType = parameter.getParameterType();
+		return Map.class.isAssignableFrom(parameterType);
+	}
+
 	/**
 	 * Sets params.
 	 *
