diff --git a/rewrite-maven/src/main/java/org/openrewrite/maven/UpdateScmFromGitOrigin.java b/rewrite-maven/src/main/java/org/openrewrite/maven/UpdateScmFromGitOrigin.java
index c2d838a..23ad528 100644
--- a/rewrite-maven/src/main/java/org/openrewrite/maven/UpdateScmFromGitOrigin.java
+++ b/rewrite-maven/src/main/java/org/openrewrite/maven/UpdateScmFromGitOrigin.java
@@ -68,7 +68,7 @@ public Xml visitDocument(Xml.Document document, ExecutionContext ctx) {
                 gitOrigin = maybeOrigin.get();
 
                 if(!document.getRoot().getChild("scm").isPresent()) {
-                    if (Boolean.TRUE.equals(addIfMissing)) {
+                    if (Boolean.TRUE.equals(addIfMissing) && !getResolutionResult().parentPomIsProjectPom()) {
                         // Build the SCM tag with all required elements
                         String httpUrl = "https://" + gitOrigin.getHost() + "/" + gitOrigin.getPath();
                         String gitPath = gitOrigin.getPath().endsWith(".git") ?
