diff --git a/src/MudBlazor/Components/Collapse/MudCollapse.razor b/src/MudBlazor/Components/Collapse/MudCollapse.razor
index e9ec3de..fc28ff8 100644
--- a/src/MudBlazor/Components/Collapse/MudCollapse.razor
+++ b/src/MudBlazor/Components/Collapse/MudCollapse.razor
@@ -3,7 +3,8 @@
 
 <div @ontransitionend="this.AsNonRenderingEventHandler(AnimationEndAsync)"
      @attributes="UserAttributes" class="@Classname" style="@Stylename">
-    <div class="mud-collapse-wrapper">
+    <div @ontransitionend="this.AsNonRenderingEventHandler(StopTransitionPropagationAsync)" @ontransitionend:stopPropagation
+         class="mud-collapse-wrapper">
         <div class="mud-collapse-wrapper-inner">
             @ChildContent
         </div>
diff --git a/src/MudBlazor/Components/Collapse/MudCollapse.razor.cs b/src/MudBlazor/Components/Collapse/MudCollapse.razor.cs
index 612a688..2d3a718 100644
--- a/src/MudBlazor/Components/Collapse/MudCollapse.razor.cs
+++ b/src/MudBlazor/Components/Collapse/MudCollapse.razor.cs
@@ -110,5 +110,7 @@ namespace MudBlazor
 
             return OnAnimationEnd.InvokeAsync(_expandedState.Value);
         }
+
+        private static Task StopTransitionPropagationAsync() => Task.CompletedTask;
     }
 }
