diff --git a/packages/svelte/src/internal/client/dom/elements/attributes.js b/packages/svelte/src/internal/client/dom/elements/attributes.js
index 5db685cf3..e14386af0 100644
--- a/packages/svelte/src/internal/client/dom/elements/attributes.js
+++ b/packages/svelte/src/internal/client/dom/elements/attributes.js
@@ -490,8 +490,8 @@ export function attribute_effect(
 		/** @type {Record<string | symbol, any>} */
 		var current = set_attributes(element, prev, next, css_hash, skip_warning);
 
-		if (inited && is_select && 'value' in next) {
-			select_option(/** @type {HTMLSelectElement} */ (element), next.value, false);
+		if (is_select && ('value' in next || !inited)) {
+			select_option(/** @type {HTMLSelectElement} */ (element), next.value, !inited);
 		}
 
 		for (let symbol of Object.getOwnPropertySymbols(effects)) {
