diff --git a/naga/src/front/wgsl/parse/mod.rs b/naga/src/front/wgsl/parse/mod.rs
index 45c5514..b8983a2 100644
--- a/naga/src/front/wgsl/parse/mod.rs
+++ b/naga/src/front/wgsl/parse/mod.rs
@@ -2032,6 +2032,12 @@ impl Parser {
         }
 
         let attrib_span = self.pop_rule_span(lexer);
+
+        // `@must_use` is only valid on function declarations.
+        if must_use.value.is_some() && !matches!(lexer.peek().0, Token::Word("fn")) {
+            return Err(Box::new(Error::UnknownAttribute(must_use.value.unwrap())));
+        }
+
         match (bind_group.value, bind_index.value) {
             (Some(group), Some(index)) => {
                 binding = Some(ast::ResourceBinding {
