diff --git a/processor/sumologicprocessor/config.go b/processor/sumologicprocessor/config.go
index 69226e5a10..f49b6625b1 100644
--- a/processor/sumologicprocessor/config.go
+++ b/processor/sumologicprocessor/config.go
@@ -11,9 +11,9 @@ type Config struct {
 	AddCloudNamespace           bool                      `mapstructure:"add_cloud_namespace"`
 	TranslateAttributes         bool                      `mapstructure:"translate_attributes"`
 	TranslateTelegrafAttributes bool                      `mapstructure:"translate_telegraf_attributes"`
-	NestAttributes              *NestingProcessorConfig   `mapstructure:"nest_attributes"`
-	AggregateAttributes         []aggregationPair         `mapstructure:"aggregate_attributes"`
-	LogFieldsAttributes         *logFieldAttributesConfig `mapstructure:"field_attributes"`
+	NestAttributes              NestingProcessorConfig   `mapstructure:"nest_attributes"`
+	AggregateAttributes         []aggregationPair        `mapstructure:"aggregate_attributes"`
+	LogFieldsAttributes         logFieldAttributesConfig `mapstructure:"field_attributes"`
 	TranslateDockerMetrics      bool                      `mapstructure:"translate_docker_metrics"`
 }
 
@@ -58,7 +58,7 @@ func createDefaultConfig() component.Config {
 		AddCloudNamespace:           defaultAddCloudNamespace,
 		TranslateAttributes:         defaultTranslateAttributes,
 		TranslateTelegrafAttributes: defaultTranslateTelegrafAttributes,
-		NestAttributes: &NestingProcessorConfig{
+		NestAttributes: NestingProcessorConfig{
 			Separator:          defaultNestingSeparator,
 			Enabled:            defaultNestingEnabled,
 			Include:            defaultNestingInclude(),
@@ -66,11 +66,11 @@ func createDefaultConfig() component.Config {
 			SquashSingleValues: defaultNestingSquashSingleValues,
 		},
 		AggregateAttributes: defaultAggregateAttributes(),
-		LogFieldsAttributes: &logFieldAttributesConfig{
-			SeverityNumberAttribute: &logFieldAttribute{defaultAddSeverityNumberAttribute, SeverityNumberAttributeName},
-			SeverityTextAttribute:   &logFieldAttribute{defaultAddSeverityTextAttribute, SeverityTextAttributeName},
-			SpanIDAttribute:         &logFieldAttribute{defaultAddSpanIDAttribute, SpanIDAttributeName},
-			TraceIDAttribute:        &logFieldAttribute{defaultAddTraceIDAttribute, TraceIDAttributeName},
+		LogFieldsAttributes: logFieldAttributesConfig{
+			SeverityNumberAttribute: logFieldAttribute{defaultAddSeverityNumberAttribute, SeverityNumberAttributeName},
+			SeverityTextAttribute:   logFieldAttribute{defaultAddSeverityTextAttribute, SeverityTextAttributeName},
+			SpanIDAttribute:         logFieldAttribute{defaultAddSpanIDAttribute, SpanIDAttributeName},
+			TraceIDAttribute:        logFieldAttribute{defaultAddTraceIDAttribute, TraceIDAttributeName},
 		},
 		TranslateDockerMetrics: defaultTranslateDockerMetrics,
 	}
diff --git a/processor/sumologicprocessor/config_test.go b/processor/sumologicprocessor/config_test.go
index 12d276b0be..03b03456ae 100644
--- a/processor/sumologicprocessor/config_test.go
+++ b/processor/sumologicprocessor/config_test.go
@@ -39,7 +39,7 @@ func TestLoadConfig(t *testing.T) {
 				AddCloudNamespace:           false,
 				TranslateAttributes:         true,
 				TranslateTelegrafAttributes: true,
-				NestAttributes: &NestingProcessorConfig{
+				NestAttributes: NestingProcessorConfig{
 					Enabled:            false,
 					Separator:          ".",
 					Include:            []string{},
@@ -47,11 +47,11 @@ func TestLoadConfig(t *testing.T) {
 					SquashSingleValues: false,
 				},
 				AggregateAttributes: []aggregationPair{},
-				LogFieldsAttributes: &logFieldAttributesConfig{
-					SeverityNumberAttribute: &logFieldAttribute{false, SeverityNumberAttributeName},
-					SeverityTextAttribute:   &logFieldAttribute{false, SeverityTextAttributeName},
-					SpanIDAttribute:         &logFieldAttribute{false, SpanIDAttributeName},
-					TraceIDAttribute:        &logFieldAttribute{false, TraceIDAttributeName},
+				LogFieldsAttributes: logFieldAttributesConfig{
+					SeverityNumberAttribute: logFieldAttribute{false, SeverityNumberAttributeName},
+					SeverityTextAttribute:   logFieldAttribute{false, SeverityTextAttributeName},
+					SpanIDAttribute:         logFieldAttribute{false, SpanIDAttributeName},
+					TraceIDAttribute:        logFieldAttribute{false, TraceIDAttributeName},
 				},
 				TranslateDockerMetrics: false,
 			},
@@ -62,7 +62,7 @@ func TestLoadConfig(t *testing.T) {
 				AddCloudNamespace:           true,
 				TranslateAttributes:         false,
 				TranslateTelegrafAttributes: true,
-				NestAttributes: &NestingProcessorConfig{
+				NestAttributes: NestingProcessorConfig{
 					Enabled:            false,
 					Separator:          ".",
 					Include:            []string{},
@@ -70,11 +70,11 @@ func TestLoadConfig(t *testing.T) {
 					SquashSingleValues: false,
 				},
 				AggregateAttributes: []aggregationPair{},
-				LogFieldsAttributes: &logFieldAttributesConfig{
-					SeverityNumberAttribute: &logFieldAttribute{false, SeverityNumberAttributeName},
-					SeverityTextAttribute:   &logFieldAttribute{false, SeverityTextAttributeName},
-					SpanIDAttribute:         &logFieldAttribute{false, SpanIDAttributeName},
-					TraceIDAttribute:        &logFieldAttribute{false, TraceIDAttributeName},
+				LogFieldsAttributes: logFieldAttributesConfig{
+					SeverityNumberAttribute: logFieldAttribute{false, SeverityNumberAttributeName},
+					SeverityTextAttribute:   logFieldAttribute{false, SeverityTextAttributeName},
+					SpanIDAttribute:         logFieldAttribute{false, SpanIDAttributeName},
+					TraceIDAttribute:        logFieldAttribute{false, TraceIDAttributeName},
 				},
 				TranslateDockerMetrics: false,
 			},
@@ -85,7 +85,7 @@ func TestLoadConfig(t *testing.T) {
 				AddCloudNamespace:           true,
 				TranslateAttributes:         true,
 				TranslateTelegrafAttributes: false,
-				NestAttributes: &NestingProcessorConfig{
+				NestAttributes: NestingProcessorConfig{
 					Enabled:            false,
 					Separator:          ".",
 					Include:            []string{},
@@ -93,11 +93,11 @@ func TestLoadConfig(t *testing.T) {
 					SquashSingleValues: false,
 				},
 				AggregateAttributes: []aggregationPair{},
-				LogFieldsAttributes: &logFieldAttributesConfig{
-					SeverityNumberAttribute: &logFieldAttribute{false, SeverityNumberAttributeName},
-					SeverityTextAttribute:   &logFieldAttribute{false, SeverityTextAttributeName},
-					SpanIDAttribute:         &logFieldAttribute{false, SpanIDAttributeName},
-					TraceIDAttribute:        &logFieldAttribute{false, TraceIDAttributeName},
+				LogFieldsAttributes: logFieldAttributesConfig{
+					SeverityNumberAttribute: logFieldAttribute{false, SeverityNumberAttributeName},
+					SeverityTextAttribute:   logFieldAttribute{false, SeverityTextAttributeName},
+					SpanIDAttribute:         logFieldAttribute{false, SpanIDAttributeName},
+					TraceIDAttribute:        logFieldAttribute{false, TraceIDAttributeName},
 				},
 				TranslateDockerMetrics: false,
 			},
@@ -108,7 +108,7 @@ func TestLoadConfig(t *testing.T) {
 				AddCloudNamespace:           true,
 				TranslateAttributes:         true,
 				TranslateTelegrafAttributes: true,
-				NestAttributes: &NestingProcessorConfig{
+				NestAttributes: NestingProcessorConfig{
 					Enabled:            true,
 					Separator:          "!",
 					Include:            []string{"blep"},
@@ -116,11 +116,11 @@ func TestLoadConfig(t *testing.T) {
 					SquashSingleValues: true,
 				},
 				AggregateAttributes: []aggregationPair{},
-				LogFieldsAttributes: &logFieldAttributesConfig{
-					SeverityNumberAttribute: &logFieldAttribute{false, SeverityNumberAttributeName},
-					SeverityTextAttribute:   &logFieldAttribute{false, SeverityTextAttributeName},
-					SpanIDAttribute:         &logFieldAttribute{false, SpanIDAttributeName},
-					TraceIDAttribute:        &logFieldAttribute{false, TraceIDAttributeName},
+				LogFieldsAttributes: logFieldAttributesConfig{
+					SeverityNumberAttribute: logFieldAttribute{false, SeverityNumberAttributeName},
+					SeverityTextAttribute:   logFieldAttribute{false, SeverityTextAttributeName},
+					SpanIDAttribute:         logFieldAttribute{false, SpanIDAttributeName},
+					TraceIDAttribute:        logFieldAttribute{false, TraceIDAttributeName},
 				},
 				TranslateDockerMetrics: false,
 			},
@@ -131,7 +131,7 @@ func TestLoadConfig(t *testing.T) {
 				AddCloudNamespace:           true,
 				TranslateAttributes:         true,
 				TranslateTelegrafAttributes: true,
-				NestAttributes: &NestingProcessorConfig{
+				NestAttributes: NestingProcessorConfig{
 					Enabled:            false,
 					Separator:          ".",
 					Include:            []string{},
@@ -148,11 +148,11 @@ func TestLoadConfig(t *testing.T) {
 						Prefixes:  []string{"pattern4"},
 					},
 				},
-				LogFieldsAttributes: &logFieldAttributesConfig{
-					SeverityNumberAttribute: &logFieldAttribute{false, SeverityNumberAttributeName},
-					SeverityTextAttribute:   &logFieldAttribute{false, SeverityTextAttributeName},
-					SpanIDAttribute:         &logFieldAttribute{false, SpanIDAttributeName},
-					TraceIDAttribute:        &logFieldAttribute{false, TraceIDAttributeName},
+				LogFieldsAttributes: logFieldAttributesConfig{
+					SeverityNumberAttribute: logFieldAttribute{false, SeverityNumberAttributeName},
+					SeverityTextAttribute:   logFieldAttribute{false, SeverityTextAttributeName},
+					SpanIDAttribute:         logFieldAttribute{false, SpanIDAttributeName},
+					TraceIDAttribute:        logFieldAttribute{false, TraceIDAttributeName},
 				},
 				TranslateDockerMetrics: false,
 			},
@@ -163,7 +163,7 @@ func TestLoadConfig(t *testing.T) {
 				AddCloudNamespace:           true,
 				TranslateAttributes:         true,
 				TranslateTelegrafAttributes: true,
-				NestAttributes: &NestingProcessorConfig{
+				NestAttributes: NestingProcessorConfig{
 					Enabled:            false,
 					Separator:          ".",
 					Include:            []string{},
@@ -171,11 +171,11 @@ func TestLoadConfig(t *testing.T) {
 					SquashSingleValues: false,
 				},
 				AggregateAttributes: []aggregationPair{},
-				LogFieldsAttributes: &logFieldAttributesConfig{
-					SeverityNumberAttribute: &logFieldAttribute{true, SeverityNumberAttributeName},
-					SeverityTextAttribute:   &logFieldAttribute{false, SeverityTextAttributeName},
-					SpanIDAttribute:         &logFieldAttribute{false, SpanIDAttributeName},
-					TraceIDAttribute:        &logFieldAttribute{false, TraceIDAttributeName},
+				LogFieldsAttributes: logFieldAttributesConfig{
+					SeverityNumberAttribute: logFieldAttribute{true, SeverityNumberAttributeName},
+					SeverityTextAttribute:   logFieldAttribute{false, SeverityTextAttributeName},
+					SpanIDAttribute:         logFieldAttribute{false, SpanIDAttributeName},
+					TraceIDAttribute:        logFieldAttribute{false, TraceIDAttributeName},
 				},
 				TranslateDockerMetrics: false,
 			},
@@ -186,7 +186,7 @@ func TestLoadConfig(t *testing.T) {
 				AddCloudNamespace:           true,
 				TranslateAttributes:         true,
 				TranslateTelegrafAttributes: true,
-				NestAttributes: &NestingProcessorConfig{
+				NestAttributes: NestingProcessorConfig{
 					Enabled:            false,
 					Separator:          ".",
 					Include:            []string{},
@@ -194,11 +194,11 @@ func TestLoadConfig(t *testing.T) {
 					SquashSingleValues: false,
 				},
 				AggregateAttributes: []aggregationPair{},
-				LogFieldsAttributes: &logFieldAttributesConfig{
-					SeverityNumberAttribute: &logFieldAttribute{false, SeverityNumberAttributeName},
-					SeverityTextAttribute:   &logFieldAttribute{true, SeverityTextAttributeName},
-					SpanIDAttribute:         &logFieldAttribute{false, SpanIDAttributeName},
-					TraceIDAttribute:        &logFieldAttribute{false, TraceIDAttributeName},
+				LogFieldsAttributes: logFieldAttributesConfig{
+					SeverityNumberAttribute: logFieldAttribute{false, SeverityNumberAttributeName},
+					SeverityTextAttribute:   logFieldAttribute{true, SeverityTextAttributeName},
+					SpanIDAttribute:         logFieldAttribute{false, SpanIDAttributeName},
+					TraceIDAttribute:        logFieldAttribute{false, TraceIDAttributeName},
 				},
 				TranslateDockerMetrics: false,
 			},
@@ -209,7 +209,7 @@ func TestLoadConfig(t *testing.T) {
 				AddCloudNamespace:           true,
 				TranslateAttributes:         true,
 				TranslateTelegrafAttributes: true,
-				NestAttributes: &NestingProcessorConfig{
+				NestAttributes: NestingProcessorConfig{
 					Enabled:            false,
 					Separator:          ".",
 					Include:            []string{},
@@ -217,11 +217,11 @@ func TestLoadConfig(t *testing.T) {
 					SquashSingleValues: false,
 				},
 				AggregateAttributes: []aggregationPair{},
-				LogFieldsAttributes: &logFieldAttributesConfig{
-					SeverityNumberAttribute: &logFieldAttribute{false, SeverityNumberAttributeName},
-					SeverityTextAttribute:   &logFieldAttribute{false, SeverityTextAttributeName},
-					SpanIDAttribute:         &logFieldAttribute{true, SpanIDAttributeName},
-					TraceIDAttribute:        &logFieldAttribute{false, TraceIDAttributeName},
+				LogFieldsAttributes: logFieldAttributesConfig{
+					SeverityNumberAttribute: logFieldAttribute{false, SeverityNumberAttributeName},
+					SeverityTextAttribute:   logFieldAttribute{false, SeverityTextAttributeName},
+					SpanIDAttribute:         logFieldAttribute{true, SpanIDAttributeName},
+					TraceIDAttribute:        logFieldAttribute{false, TraceIDAttributeName},
 				},
 				TranslateDockerMetrics: false,
 			},
@@ -232,7 +232,7 @@ func TestLoadConfig(t *testing.T) {
 				AddCloudNamespace:           true,
 				TranslateAttributes:         true,
 				TranslateTelegrafAttributes: true,
-				NestAttributes: &NestingProcessorConfig{
+				NestAttributes: NestingProcessorConfig{
 					Enabled:            false,
 					Separator:          ".",
 					Include:            []string{},
@@ -240,11 +240,11 @@ func TestLoadConfig(t *testing.T) {
 					SquashSingleValues: false,
 				},
 				AggregateAttributes: []aggregationPair{},
-				LogFieldsAttributes: &logFieldAttributesConfig{
-					SeverityNumberAttribute: &logFieldAttribute{false, SeverityNumberAttributeName},
-					SeverityTextAttribute:   &logFieldAttribute{false, SeverityTextAttributeName},
-					SpanIDAttribute:         &logFieldAttribute{false, SpanIDAttributeName},
-					TraceIDAttribute:        &logFieldAttribute{true, TraceIDAttributeName},
+				LogFieldsAttributes: logFieldAttributesConfig{
+					SeverityNumberAttribute: logFieldAttribute{false, SeverityNumberAttributeName},
+					SeverityTextAttribute:   logFieldAttribute{false, SeverityTextAttributeName},
+					SpanIDAttribute:         logFieldAttribute{false, SpanIDAttributeName},
+					TraceIDAttribute:        logFieldAttribute{true, TraceIDAttributeName},
 				},
 				TranslateDockerMetrics: false,
 			},
@@ -255,7 +255,7 @@ func TestLoadConfig(t *testing.T) {
 				AddCloudNamespace:           true,
 				TranslateAttributes:         true,
 				TranslateTelegrafAttributes: true,
-				NestAttributes: &NestingProcessorConfig{
+				NestAttributes: NestingProcessorConfig{
 					Enabled:            false,
 					Separator:          ".",
 					Include:            []string{},
@@ -263,11 +263,11 @@ func TestLoadConfig(t *testing.T) {
 					SquashSingleValues: false,
 				},
 				AggregateAttributes: []aggregationPair{},
-				LogFieldsAttributes: &logFieldAttributesConfig{
-					SeverityNumberAttribute: &logFieldAttribute{false, SeverityNumberAttributeName},
-					SeverityTextAttribute:   &logFieldAttribute{false, SeverityTextAttributeName},
-					SpanIDAttribute:         &logFieldAttribute{false, SpanIDAttributeName},
-					TraceIDAttribute:        &logFieldAttribute{false, TraceIDAttributeName},
+				LogFieldsAttributes: logFieldAttributesConfig{
+					SeverityNumberAttribute: logFieldAttribute{false, SeverityNumberAttributeName},
+					SeverityTextAttribute:   logFieldAttribute{false, SeverityTextAttributeName},
+					SpanIDAttribute:         logFieldAttribute{false, SpanIDAttributeName},
+					TraceIDAttribute:        logFieldAttribute{false, TraceIDAttributeName},
 				},
 				TranslateDockerMetrics: true,
 			},
diff --git a/processor/sumologicprocessor/log_fields_conversion_processor.go b/processor/sumologicprocessor/log_fields_conversion_processor.go
index 3531f85733..1fc894e13e 100644
--- a/processor/sumologicprocessor/log_fields_conversion_processor.go
+++ b/processor/sumologicprocessor/log_fields_conversion_processor.go
@@ -25,10 +25,10 @@ type logFieldAttribute struct {
 }
 
 type logFieldAttributesConfig struct {
-	SeverityNumberAttribute *logFieldAttribute `mapstructure:"severity_number"`
-	SeverityTextAttribute   *logFieldAttribute `mapstructure:"severity_text"`
-	SpanIDAttribute         *logFieldAttribute `mapstructure:"span_id"`
-	TraceIDAttribute        *logFieldAttribute `mapstructure:"trace_id"`
+	SeverityNumberAttribute logFieldAttribute `mapstructure:"severity_number"`
+	SeverityTextAttribute   logFieldAttribute `mapstructure:"severity_text"`
+	SpanIDAttribute         logFieldAttribute `mapstructure:"span_id"`
+	TraceIDAttribute        logFieldAttribute `mapstructure:"trace_id"`
 }
 
 // spanIDToHexOrEmptyString returns a hex string from SpanID.
@@ -80,12 +80,12 @@ var severityNumberToLevel = map[string]string{
 // logFieldsConversionProcessor converts specific log entries to attributes which leads to presenting them as fields
 // in the backend
 type logFieldsConversionProcessor struct {
-	LogFieldsAttributes *logFieldAttributesConfig
+	LogFieldsAttributes logFieldAttributesConfig
 }
 
-func newLogFieldConversionProcessor(logFieldsAttributes *logFieldAttributesConfig) *logFieldsConversionProcessor {
+func newLogFieldConversionProcessor(logFieldsAttributes logFieldAttributesConfig) *logFieldsConversionProcessor {
 	return &logFieldsConversionProcessor{
-		logFieldsAttributes,
+		LogFieldsAttributes: logFieldsAttributes,
 	}
 }
 
diff --git a/processor/sumologicprocessor/nesting_processor.go b/processor/sumologicprocessor/nesting_processor.go
index 10e3a4659d..dbb54b8935 100644
--- a/processor/sumologicprocessor/nesting_processor.go
+++ b/processor/sumologicprocessor/nesting_processor.go
@@ -28,7 +28,7 @@ type nestingProcessor struct {
 	squashSingleValues bool
 }
 
-func newNestingProcessor(config *NestingProcessorConfig) *nestingProcessor {
+func newNestingProcessor(config NestingProcessorConfig) *nestingProcessor {
 	proc := &nestingProcessor{
 		separator:          config.Separator,
 		enabled:            config.Enabled,
diff --git a/processor/sumologicprocessor/nesting_processor_test.go b/processor/sumologicprocessor/nesting_processor_test.go
index 29b55cab99..3dd00f808e 100644
--- a/processor/sumologicprocessor/nesting_processor_test.go
+++ b/processor/sumologicprocessor/nesting_processor_test.go
@@ -163,7 +163,7 @@ func TestNestingAttributes(t *testing.T) {
 
 	for _, testCase := range testCases {
 		t.Run(testCase.name, func(t *testing.T) {
-			proc := newNestingProcessor(&NestingProcessorConfig{
+			proc := newNestingProcessor(NestingProcessorConfig{
 				Separator: ".", Enabled: true, Include: testCase.include, Exclude: testCase.exclude, SquashSingleValues: false,
 			})
 
@@ -232,7 +232,7 @@ func TestSquashing(t *testing.T) {
 
 	for _, testCase := range testCases {
 		t.Run(testCase.name, func(t *testing.T) {
-			proc := newNestingProcessor(&NestingProcessorConfig{
+			proc := newNestingProcessor(NestingProcessorConfig{
 				Separator: ".", Enabled: true, Include: []string{}, Exclude: []string{}, SquashSingleValues: true,
 			})
 
diff --git a/processor/sumologicprocessor/processor_test.go b/processor/sumologicprocessor/processor_test.go
index 02f31879a8..02260df09e 100644
--- a/processor/sumologicprocessor/processor_test.go
+++ b/processor/sumologicprocessor/processor_test.go
@@ -1351,7 +1351,7 @@ func newCloudNamespaceConfig(addCloudNamespace bool) *Config {
 	config.TranslateAttributes = false
 	config.TranslateTelegrafAttributes = false
 	config.TranslateDockerMetrics = false
-	config.NestAttributes = &NestingProcessorConfig{
+	config.NestAttributes = NestingProcessorConfig{
 		Enabled: false,
 	}
 	return config
@@ -1363,7 +1363,7 @@ func newTranslateAttributesConfig(translateAttributes bool) *Config {
 	config.TranslateAttributes = translateAttributes
 	config.TranslateTelegrafAttributes = false
 	config.TranslateDockerMetrics = false
-	config.NestAttributes = &NestingProcessorConfig{
+	config.NestAttributes = NestingProcessorConfig{
 		Enabled: false,
 	}
 	return config
@@ -1375,7 +1375,7 @@ func newTranslateTelegrafAttributesConfig(translateTelegrafAttributes bool) *Con
 	config.TranslateAttributes = false
 	config.TranslateTelegrafAttributes = translateTelegrafAttributes
 	config.TranslateDockerMetrics = false
-	config.NestAttributes = &NestingProcessorConfig{
+	config.NestAttributes = NestingProcessorConfig{
 		Enabled: false,
 	}
 	return config
@@ -1387,7 +1387,7 @@ func newTranslateDockerMetricsConfig(translateDockerMetrics bool) *Config {
 	config.TranslateAttributes = false
 	config.TranslateTelegrafAttributes = false
 	config.TranslateDockerMetrics = translateDockerMetrics
-	config.NestAttributes = &NestingProcessorConfig{
+	config.NestAttributes = NestingProcessorConfig{
 		Enabled: false,
 	}
 	return config
@@ -1399,7 +1399,7 @@ func newNestAttributesConfig(separator string, enabled bool) *Config {
 	config.TranslateAttributes = false
 	config.TranslateTelegrafAttributes = false
 	config.TranslateDockerMetrics = false
-	config.NestAttributes = &NestingProcessorConfig{
+	config.NestAttributes = NestingProcessorConfig{
 		Separator: separator,
 		Enabled:   enabled,
 	}
@@ -1422,14 +1422,14 @@ func newLogFieldsConversionConfig() *Config {
 	config.TranslateAttributes = false
 	config.TranslateTelegrafAttributes = false
 	config.TranslateDockerMetrics = false
-	config.NestAttributes = &NestingProcessorConfig{
+	config.NestAttributes = NestingProcessorConfig{
 		Enabled: false,
 	}
-	config.LogFieldsAttributes = &logFieldAttributesConfig{
-		&logFieldAttribute{Enabled: true, Name: "definitely_not_default_name"},
-		&logFieldAttribute{Enabled: true, Name: SeverityTextAttributeName},
-		&logFieldAttribute{Enabled: true, Name: SpanIDAttributeName},
-		&logFieldAttribute{Enabled: true, Name: TraceIDAttributeName},
+	config.LogFieldsAttributes = logFieldAttributesConfig{
+		SeverityNumberAttribute: logFieldAttribute{Enabled: true, Name: "definitely_not_default_name"},
+		SeverityTextAttribute:   logFieldAttribute{Enabled: true, Name: SeverityTextAttributeName},
+		SpanIDAttribute:         logFieldAttribute{Enabled: true, Name: SpanIDAttributeName},
+		TraceIDAttribute:        logFieldAttribute{Enabled: true, Name: TraceIDAttributeName},
 	}
 	return config
 }
