diff --git a/plugins/inputs/nftables/types.go b/plugins/inputs/nftables/types.go
index d8a2a5d..12221d2 100644
--- a/plugins/inputs/nftables/types.go
+++ b/plugins/inputs/nftables/types.go
@@ -106,7 +106,5 @@ type namedSet struct {
 	Family string `json:"family"`
 	Name   string `json:"name"`
 	Table  string `json:"table"`
-	Elem   []elem `json:"elem,omitempty"`
+	Elem   []json.RawMessage `json:"elem,omitempty"`
 }
-
-type elem struct{}
