diff --git a/lib/matplotlib/figure.pyi b/lib/matplotlib/figure.pyi
index ade4cfd6f1..a19d0232a5 100644
--- a/lib/matplotlib/figure.pyi
+++ b/lib/matplotlib/figure.pyi
@@ -315,6 +315,7 @@ class Figure(FigureBase):
     transSubfigure: Transform
     patch: Rectangle
     subplotpars: SubplotParams
+    number: int
     def __init__(
         self,
         figsize: tuple[float, float] | None = ...,
diff --git a/reproduce_error.py b/reproduce_error.py
new file mode 100644
index 0000000000..84a1c12664
--- /dev/null
+++ b/reproduce_error.py
@@ -0,0 +1,3 @@
+import matplotlib.pyplot as plt
+screen = plt.figure()
+print(screen.number)
\ No newline at end of file
