diff --git a/multiqc/core/special_case_modules/custom_content.py b/multiqc/core/special_case_modules/custom_content.py
index 23ad327b..82eb9f6a 100644
--- a/multiqc/core/special_case_modules/custom_content.py
+++ b/multiqc/core/special_case_modules/custom_content.py
@@ -599,6 +599,7 @@ class MultiqcModule(BaseMultiqcModule):
                 name=section_name,
                 anchor=section_anchor,
                 id=section_id,
+                description=ccdict.config.get("description", ""),
                 plot=plot,
                 content=content or "",
             )
