diff --git a/gitlab4j-models/src/main/java/org/gitlab4j/api/models/Job.java b/gitlab4j-models/src/main/java/org/gitlab4j/api/models/Job.java
index 151321d9..c6cdc6f0 100644
--- a/gitlab4j-models/src/main/java/org/gitlab4j/api/models/Job.java
+++ b/gitlab4j-models/src/main/java/org/gitlab4j/api/models/Job.java
@@ -35,6 +35,7 @@ public class Job implements Serializable {
     private Float duration;
     private Float queuedDuration;
     private Project project;
+    private List<String> tagList;
 
     public Long getId() {
         return id;
@@ -253,6 +254,14 @@ public class Job implements Serializable {
         this.project = project;
     }
 
+    public List<String> getTagList() {
+        return tagList;
+    }
+
+    public void setTagList(List<String> tagList) {
+        this.tagList = tagList;
+    }
+
     public Job withId(Long id) {
         this.id = id;
         return this;
@@ -363,6 +372,11 @@ public class Job implements Serializable {
         return this;
     }
 
+    public Job withTagList(List<String> tagList) {
+        this.tagList = tagList;
+        return this;
+    }
+
     @Override
     public String toString() {
         return (JacksonJson.toJsonString(this));
