diff --git a/packages/runtime-tags/src/translator/visitors/function.ts b/packages/runtime-tags/src/translator/visitors/function.ts
index 77f6b5902..072ce3b90 100644
--- a/packages/runtime-tags/src/translator/visitors/function.ts
+++ b/packages/runtime-tags/src/translator/visitors/function.ts
@@ -80,10 +80,8 @@ export default {
       } else if (refs.size) {
         getReferencesByFn().set(fnExtra, refs);
       }
-    } else if (shouldAlwaysRegister(markoRoot)) {
-      registerFunction(fnExtra);
     } else {
-      getReferencesByFn().set(fnExtra, new Set([(exprRoot.node.extra ??= {})]));
+      registerFunction(fnExtra);
     }
   },
 } satisfies TemplateVisitor<t.Function>;
@@ -97,9 +95,16 @@ export function finalizeFunctionRegistry() {
       if (seenExtras.has(refExtra)) continue;
       seenExtras.add(refExtra);
 
-      if (couldSerializeDownstream(refExtra.downstream)) {
-        shouldRegister = true;
-        break;
+      if (refExtra.downstream) {
+        const { bindings, excludeProperties } = refExtra.downstream;
+        if (
+          Array.isArray(bindings)
+            ? bindings.some((binding) => couldSerializeBinding(binding))
+            : couldSerializeBinding(bindings, excludeProperties)
+        ) {
+          shouldRegister = true;
+          break;
+        }
       }
     }
 
@@ -109,19 +114,6 @@ export function finalizeFunctionRegistry() {
   }
 }
 
-function couldSerializeDownstream(downstream: t.NodeExtra["downstream"]) {
-  if (downstream) {
-    return Array.isArray(downstream.bindings)
-      ? downstream.bindings.some((binding) => couldSerializeBinding(binding))
-      : couldSerializeBinding(
-          downstream.bindings,
-          downstream.excludeProperties,
-        );
-  }
-
-  return false;
-}
-
 function couldSerializeBinding(
   binding: Binding,
   excludeProperties?: Set<string> | undefined,
@@ -130,20 +122,11 @@ function couldSerializeBinding(
     return true;
   }
 
-  for (const expr of binding.downstreamExpressions) {
-    if (expr.isEffect || couldSerializeDownstream(expr.downstream)) {
-      return true;
-    }
-  }
-
-  for (const alias of binding.aliases) {
-    if (couldSerializeBinding(alias)) {
-      return true;
-    }
-  }
-
   for (const [name, propBinding] of binding.propertyAliases) {
-    if (!excludeProperties?.has(name) && couldSerializeBinding(propBinding)) {
+    if (
+      !excludeProperties?.has(name) &&
+      getBindingSerializeReason(propBinding.section, propBinding)
+    ) {
       return true;
     }
   }
@@ -193,43 +176,37 @@ function getStaticDeclRefs(
             if (getStaticDeclRefs(fnExtra, ref, refs) === true) {
               return true;
             }
-          } else if (shouldAlwaysRegister(markoRoot)) {
-            return true;
-          } else {
-            refs.add((exprRoot.node.extra ??= {}));
+            continue;
           }
-        }
-      }
-    }
-  }
 
-  return refs;
-}
+          const tag = getTagFromMarkoRoot(markoRoot);
+          if (!tag) continue;
+          if (isCoreTagName(tag, "let")) return true;
 
-function shouldAlwaysRegister(markoRoot: MarkoExprRootPath) {
-  const tag = getTagFromMarkoRoot(markoRoot);
-  if (!tag) return false;
-  if (isCoreTagName(tag, "let")) return true;
-  if (isCoreTagName(tag, "return")) return true;
+          switch (analyzeTagNameType(tag)) {
+            case TagNameType.DynamicTag:
+            case TagNameType.NativeTag:
+              // Passing a function to a dynamic tag could always be potentially serialized.
+              // Native tag event handlers are skipped in the `canIgnoreRegister`
+              // if it's anything else we need to unconditionally serialize.
+              return true;
+            case TagNameType.AttributeTag:
+              if (
+                analyzeTagNameType(getAttributeTagParent(tag)) ===
+                TagNameType.DynamicTag
+              ) {
+                return true;
+              }
+              break;
+          }
 
-  switch (analyzeTagNameType(tag)) {
-    case TagNameType.DynamicTag:
-    case TagNameType.NativeTag:
-      // Passing a function to a dynamic tag could always be potentially serialized.
-      // Native tag event handlers are skipped in the `canIgnoreRegister`
-      // if it's anything else we need to unconditionally serialize.
-      return true;
-    case TagNameType.AttributeTag:
-      if (
-        analyzeTagNameType(getAttributeTagParent(tag)) ===
-        TagNameType.DynamicTag
-      ) {
-        return true;
+          refs.add((exprRoot.node.extra ??= {}));
+        }
       }
-      break;
+    }
   }
 
-  return false;
+  return refs;
 }
 
 function getTagFromMarkoRoot(
