diff --git a/Terminal.Gui/Views/SpinnerView/SpinnerView.cs b/Terminal.Gui/Views/SpinnerView/SpinnerView.cs
index 127c0fa..dfddf35 100644
--- a/Terminal.Gui/Views/SpinnerView/SpinnerView.cs
+++ b/Terminal.Gui/Views/SpinnerView/SpinnerView.cs
@@ -18,6 +18,7 @@ public class SpinnerView : View, IDesignable
     private const int DEFAULT_DELAY = 130;
     private static readonly SpinnerStyle DEFAULT_STYLE = new SpinnerStyle.Line ();
 
+    private bool _autoSpin;
     private bool _bounce = DEFAULT_STYLE.SpinBounce;
     private bool _bounceReverse;
     private int _currentIdx;
@@ -46,10 +47,12 @@ public class SpinnerView : View, IDesignable
     /// </summary>
     public bool AutoSpin
     {
-        get => _timeout != null;
+        get => _autoSpin;
         set
         {
-            if (value)
+            _autoSpin = value;
+
+            if (_autoSpin)
             {
                 AddAutoSpinTimeout ();
             }
@@ -110,6 +113,17 @@ public class SpinnerView : View, IDesignable
         set => SetStyle (value);
     }
 
+    /// <inheritdoc/>
+    public override void EndInit ()
+    {
+        base.EndInit ();
+
+        if (AutoSpin)
+        {
+            AddAutoSpinTimeout ();
+        }
+    }
+
     /// <summary>
     ///     Advances the animation frame and notifies main loop that repainting needs to happen. Repeated calls are
     ///     ignored based on <see cref="SpinDelay"/>.
@@ -168,6 +182,28 @@ public class SpinnerView : View, IDesignable
         }
     }
 
+    /// <inheritdoc/>
+    protected override bool OnSuperViewChanging (ValueChangingEventArgs<View?> args)
+    {
+        if (args.NewValue is null)
+        {
+            RemoveAutoSpinTimeout ();
+        }
+
+        return base.OnSuperViewChanging (args);
+    }
+
+    /// <inheritdoc/>
+    protected override void OnSuperViewChanged (ValueChangedEventArgs<View?> args)
+    {
+        base.OnSuperViewChanged (args);
+
+        if (AutoSpin)
+        {
+            AddAutoSpinTimeout ();
+        }
+    }
+
     /// <inheritdoc/>
     protected override bool OnClearingViewport () { return true; }
 
@@ -194,6 +230,7 @@ public class SpinnerView : View, IDesignable
     /// <inheritdoc/>
     protected override void Dispose (bool disposing)
     {
+        _autoSpin = false;
         RemoveAutoSpinTimeout ();
 
         base.Dispose (disposing);
@@ -201,21 +238,20 @@ public class SpinnerView : View, IDesignable
 
     private void AddAutoSpinTimeout ()
     {
-        // Only add timeout if we are initialized and not already spinning
-        if (App is { } && (_timeout is { } || !App.Initialized))
+        if (!AutoSpin || _timeout is { } || App is not { Initialized: true } app)
         {
             return;
         }
 
-        _timeout = App?.AddTimeout (
-                                           TimeSpan.FromMilliseconds (SpinDelay),
-                                           () =>
-                                           {
-                                               App.Invoke ((_) => AdvanceAnimation ());
+        _timeout = app.AddTimeout (
+                                   TimeSpan.FromMilliseconds (SpinDelay),
+                                   () =>
+                                   {
+                                       app.Invoke ((_) => AdvanceAnimation ());
 
-                                               return true;
-                                           }
-                                          );
+                                       return AutoSpin && App == app;
+                                   }
+                                  );
     }
 
     private bool GetIsAsciiOnly ()
