diff --git a/modules/openapi-generator/src/main/resources/kotlin-client/typeInfoAnnotation.mustache b/modules/openapi-generator/src/main/resources/kotlin-client/typeInfoAnnotation.mustache
index 6b3cb83b0f8..0c0e55e5894 100644
--- a/modules/openapi-generator/src/main/resources/kotlin-client/typeInfoAnnotation.mustache
+++ b/modules/openapi-generator/src/main/resources/kotlin-client/typeInfoAnnotation.mustache
@@ -1,4 +1,4 @@
-@JsonTypeInfo(use = JsonTypeInfo.Id.NAME, include = JsonTypeInfo.As.PROPERTY, property = "{{{discriminator.propertyBaseName}}}", visible = true)
+@JsonTypeInfo(use = JsonTypeInfo.Id.NAME, include = JsonTypeInfo.As.EXISTING_PROPERTY, property = "{{{discriminator.propertyBaseName}}}", visible = true)
 @JsonSubTypes(
 {{#discriminator.mappedModels}}
     JsonSubTypes.Type(value = {{modelName}}::class, name = "{{^vendorExtensions.x-discriminator-value}}{{mappingName}}{{/vendorExtensions.x-discriminator-value}}{{#vendorExtensions.x-discriminator-value}}{{{vendorExtensions.x-discriminator-value}}}{{/vendorExtensions.x-discriminator-value}}"){{^-last}},{{/-last}}
diff --git a/modules/openapi-generator/src/main/resources/kotlin-spring/typeInfoAnnotation.mustache b/modules/openapi-generator/src/main/resources/kotlin-spring/typeInfoAnnotation.mustache
index 35d740aa79a..1195cbc7a11 100644
--- a/modules/openapi-generator/src/main/resources/kotlin-spring/typeInfoAnnotation.mustache
+++ b/modules/openapi-generator/src/main/resources/kotlin-spring/typeInfoAnnotation.mustache
@@ -1,6 +1,6 @@
 {{#jackson}}
 
-@JsonTypeInfo(use = JsonTypeInfo.Id.NAME, include = JsonTypeInfo.As.PROPERTY, property = "{{{discriminator.propertyBaseName}}}", visible = true)
+@JsonTypeInfo(use = JsonTypeInfo.Id.NAME, include = JsonTypeInfo.As.EXISTING_PROPERTY, property = "{{{discriminator.propertyBaseName}}}", visible = true)
 @JsonSubTypes(
   {{#discriminator.mappedModels}}
       JsonSubTypes.Type(value = {{modelName}}::class, name = "{{^vendorExtensions.x-discriminator-value}}{{mappingName}}{{/vendorExtensions.x-discriminator-value}}{{#vendorExtensions.x-discriminator-value}}{{{vendorExtensions.x-discriminator-value}}}{{/vendorExtensions.x-discriminator-value}}"){{^-last}},{{/-last}}
