diff --git a/src/cc65/coptstop.c b/src/cc65/coptstop.c
index 7a9416b51..74e946923 100644
--- a/src/cc65/coptstop.c
+++ b/src/cc65/coptstop.c
@@ -1692,7 +1692,9 @@ static int RegAPreCondOk (StackOpData* D)
     LoadInfo* Lhs;
     LoadInfo* Rhs;
     LoadRegInfo* LhsLo;
+    LoadRegInfo* LhsHi;
     LoadRegInfo* RhsLo;
+    LoadRegInfo* RhsHi;
     short LhsLoVal, LhsHiVal;
     short RhsLoVal, RhsHiVal;
     int I;
@@ -1755,28 +1757,30 @@ static int RegAPreCondOk (StackOpData* D)
 
         do {
             LhsLo = &Lhs->A;
+            LhsHi = &Lhs->X;
             RhsLo = &Rhs->A;
+            RhsHi = &Rhs->X;
             /* Currently we have only LHS/RHS checks with identical requirements for A/X,
             ** so we don't need to check twice for now.
             */
 
             if ((Desc->Flags & OP_LHS_LOAD) != 0) {
-                if ((LhsLo->Flags & LI_LOAD_INSN) == 0) {
+                if ((LhsLo->Flags & LhsHi->Flags & LI_LOAD_INSN) == 0) {
                     /* Cannot optimize */
                     break;
                 } else if ((Desc->Flags & OP_LHS_LOAD_DIRECT) != 0) {
-                    if ((LhsLo->Flags & LI_DIRECT) == 0) {
+                    if ((LhsLo->Flags & LhsHi->Flags & LI_DIRECT) == 0) {
                         /* Cannot optimize */
                         break;
                     }
                 }
             }
             if ((Desc->Flags & OP_RHS_LOAD) != 0) {
-                if ((RhsLo->Flags & LI_LOAD_INSN) == 0) {
+                if ((RhsLo->Flags & RhsHi->Flags & LI_LOAD_INSN) == 0) {
                     /* Cannot optimize */
                     break;
                 } else if ((Desc->Flags & OP_RHS_LOAD_DIRECT) != 0) {
-                    if ((RhsLo->Flags & LI_DIRECT) == 0) {
+                    if ((RhsLo->Flags & RhsHi->Flags & LI_DIRECT) == 0) {
                         /* Cannot optimize */
                         break;
                     }
