diff --git a/src/BootstrapBlazor/Components/Radio/RadioList.razor.cs b/src/BootstrapBlazor/Components/Radio/RadioList.razor.cs
index 899f1d494..29f556129 100644
--- a/src/BootstrapBlazor/Components/Radio/RadioList.razor.cs
+++ b/src/BootstrapBlazor/Components/Radio/RadioList.razor.cs
@@ -31,6 +31,12 @@ public partial class RadioList<TValue>
     [Parameter]
     public bool AutoSelectFirstWhenValueIsNull { get; set; } = true;
 
+    /// <summary>
+    /// 获得/设置 Outline 样式 默认 false
+    /// </summary>
+    [Parameter]
+    public bool IsOutline { get; set; }
+
     private string? GroupName => Id;
 
     private string? ClassString => CssBuilder.Default("radio-list form-control")
@@ -46,7 +52,9 @@ public partial class RadioList<TValue>
         .Build();
 
     private string? GetButtonItemClassString(SelectedItem item) => CssBuilder.Default("btn")
-        .AddClass($"active bg-{Color.ToDescriptionString()}", CurrentValueAsString == item.Value)
+        .AddClass($"btn-outline-{Color.ToDescriptionString()}", IsOutline)
+        .AddClass($"btn-{Color.ToDescriptionString()}", !IsOutline && Color != Color.None)
+        .AddClass($"active", CurrentValueAsString == item.Value)
         .Build();
 
     /// <summary>
diff --git a/src/BootstrapBlazor/Components/Radio/RadioList.razor.scss b/src/BootstrapBlazor/Components/Radio/RadioList.razor.scss
index f60f1e5e8..9e37b6c24 100644
--- a/src/BootstrapBlazor/Components/Radio/RadioList.razor.scss
+++ b/src/BootstrapBlazor/Components/Radio/RadioList.razor.scss
@@ -60,4 +60,15 @@
             }
         }
     }
+
+    &.btn-group .btn-outline-primary.active,
+    &.btn-group .btn-outline-secondary.active,
+    &.btn-group .btn-outline-success.active,
+    &.btn-group .btn-outline-danger.active,
+    &.btn-group .btn-outline-warning.active,
+    &.btn-group .btn-outline-info.active,
+    &.btn-group .btn-outline-light.active,
+    &.btn-group .btn-outline-dark.active {
+        color: #fff;
+    }
 }
diff --git a/src/BootstrapBlazor/Components/Radio/RadioListGeneric.razor.cs b/src/BootstrapBlazor/Components/Radio/RadioListGeneric.razor.cs
index 833d8ddc8..6f52285c5 100644
--- a/src/BootstrapBlazor/Components/Radio/RadioListGeneric.razor.cs
+++ b/src/BootstrapBlazor/Components/Radio/RadioListGeneric.razor.cs
@@ -53,6 +53,12 @@ public partial class RadioListGeneric<TValue> : IModelEqualityComparer<TValue>
     [Parameter]
     public Color Color { get; set; }
 
+    /// <summary>
+    /// 获得/设置 Outline 样式 默认 false
+    /// </summary>
+    [Parameter]
+    public bool IsOutline { get; set; }
+
     /// <summary>
     /// 获得/设置 数据源
     /// </summary>
@@ -102,7 +108,9 @@ public partial class RadioListGeneric<TValue> : IModelEqualityComparer<TValue>
         .Build();
 
     private string? GetButtonItemClassString(SelectedItem<TValue> item) => CssBuilder.Default("btn")
-        .AddClass($"active bg-{Color.ToDescriptionString()}", Equals(Value, item.Value))
+        .AddClass($"btn-outline-{Color.ToDescriptionString()}", IsOutline)
+        .AddClass($"btn-{Color.ToDescriptionString()}", !IsOutline && Color != Color.None)
+        .AddClass($"active", Equals(Value, item.Value))
         .Build();
 
     /// <summary>
diff --git a/test/UnitTest/Components/RadioListGenericTest.cs b/test/UnitTest/Components/RadioListGenericTest.cs
index 7d4e8190e..b989e0c9f 100644
--- a/test/UnitTest/Components/RadioListGenericTest.cs
+++ b/test/UnitTest/Components/RadioListGenericTest.cs
@@ -216,14 +216,14 @@ public class RadioListGenericTest : BootstrapBlazorTestBase
         {
             pb.Add(a => a.Color, Color.Danger);
         });
-        cut.Contains("btn active bg-danger");
+        cut.Contains("btn-danger");
 
         var btn = cut.Find(".btn");
         await cut.InvokeAsync(() =>
         {
             btn.Click();
         });
-        cut.Contains("btn active bg-danger");
+        cut.Contains("btn-danger");
     }
 
     [Fact]
