diff --git a/libqtile/layout/spiral.py b/libqtile/layout/spiral.py
index 9ed38292..f28ab5c6 100644
--- a/libqtile/layout/spiral.py
+++ b/libqtile/layout/spiral.py
@@ -372,13 +372,13 @@ class Spiral(_SimpleLayoutBase):
     @expose_command()
     def shuffle_down(self):
         if self.clients:
-            self.clients.rotate_down()
+            self.clients.shuffle_down()
             self.group.layout_all()
 
     @expose_command()
     def shuffle_up(self):
         if self.clients:
-            self.clients.rotate_up()
+            self.clients.shuffle_up()
             self.group.layout_all()
 
     @expose_command()
