diff --git a/lib/node_modules/@stdlib/stats/base/range-by/lib/ndarray.js b/lib/node_modules/@stdlib/stats/base/range-by/lib/ndarray.js
index dbad088a497..079e070b1bc 100644
--- a/lib/node_modules/@stdlib/stats/base/range-by/lib/ndarray.js
+++ b/lib/node_modules/@stdlib/stats/base/range-by/lib/ndarray.js
@@ -21,6 +21,8 @@
 // MODULES //
 
 var isnan = require( '@stdlib/math/base/assert/is-nan' );
+var arraylike2object = require( '@stdlib/array/base/arraylike2object' );
+var accessors = require( './accessors.js' );
 
 
 // MAIN //
@@ -47,22 +49,28 @@ var isnan = require( '@stdlib/math/base/assert/is-nan' );
 * // returns 18.0
 */
 function rangeBy( N, x, stride, offset, clbk, thisArg ) {
-	var max;
-	var min;
-	var ix;
-	var v;
-	var i;
+	var o;
 
 	if ( N <= 0 ) {
 		return NaN;
 	}
+	o = arraylike2object( x );
+	if ( o.accessorProtocol ) {
+		return accessors( N, o, stride, offset, clbk, thisArg );
+	}
 	if ( N === 1 || stride === 0 ) {
-		v = clbk.call( thisArg, x[ 0 ], 0, 0, x );
+		var v = clbk.call( thisArg, x[ offset ], 0, offset, x );
 		if ( v === void 0 || isnan( v ) ) {
 			return NaN;
 		}
 		return 0.0;
 	}
+	var max;
+	var min;
+	var ix;
+	var v;
+	var i;
+
 	ix = offset;
 	for ( i = 0; i < N; i++ ) {
 		min = clbk.call( thisArg, x[ ix ], i, ix, x );
diff --git a/lib/node_modules/@stdlib/stats/base/range-by/test/test.ndarray.js b/lib/node_modules/@stdlib/stats/base/range-by/test/test.ndarray.js
index 4dfd226b799..4d82aec8057 100644
--- a/lib/node_modules/@stdlib/stats/base/range-by/test/test.ndarray.js
+++ b/lib/node_modules/@stdlib/stats/base/range-by/test/test.ndarray.js
@@ -24,6 +24,8 @@ var tape = require( 'tape' );
 var floor = require( '@stdlib/math/base/special/floor' );
 var isnan = require( '@stdlib/math/base/assert/is-nan' );
 var isPositiveZero = require( '@stdlib/math/base/assert/is-positive-zero' );
+var toAccessorArray = require( '@stdlib/array/base/to-accessor-array' );
+var arraylike2object = require( '@stdlib/array/base/arraylike2object' );
 var rangeBy = require( './../lib/ndarray.js' );
