diff --git a/lib/node_modules/@stdlib/stats/base/stdevwd/lib/ndarray.js b/lib/node_modules/@stdlib/stats/base/stdevwd/lib/ndarray.js
index f31809bdc91..e8e45c7ed84 100644
--- a/lib/node_modules/@stdlib/stats/base/stdevwd/lib/ndarray.js
+++ b/lib/node_modules/@stdlib/stats/base/stdevwd/lib/ndarray.js
@@ -20,6 +20,8 @@
 
 // MODULES //
 
+var arraylike2object = require( '@stdlib/array/base/arraylike2object' );
+var accessors = require( './accessors.js' );
 var variancewd = require( '@stdlib/stats/strided/variancewd' ).ndarray;
 var sqrt = require( '@stdlib/math/base/special/sqrt' );
 
@@ -51,6 +53,10 @@ var sqrt = require( '@stdlib/math/base/special/sqrt' );
 * // returns 2.5
 */
 function stdevwd( N, correction, x, stride, offset ) {
+	var o = arraylike2object( x );
+	if ( o.accessorProtocol ) {
+		return accessors( N, correction, o, stride, offset );
+	}
 	return sqrt( variancewd( N, correction, x, stride, offset ) );
 }
