diff --git a/packages/react/src/form/Form.tsx b/packages/react/src/form/Form.tsx
index 89347385e..b8981ca9c 100644
--- a/packages/react/src/form/Form.tsx
+++ b/packages/react/src/form/Form.tsx
@@ -34,7 +34,11 @@ export const Form = React.forwardRef(function Form(
   const onSubmit = useEventCallback(onSubmitProp);
   const onClearErrors = useEventCallback(onClearErrorsProp);
 
-  const focusControl = useEventCallback((control: HTMLElement) => {
+  const focusControl = useEventCallback((control: HTMLElement | null) => {
+    if (!control) {
+      return;
+    }
+
     control.focus();
     if (control.tagName === 'INPUT') {
       (control as HTMLInputElement).select();
