diff --git a/pkg/cmd/server/server.go b/pkg/cmd/server/server.go
index 34e071430..bc8171820 100644
--- a/pkg/cmd/server/server.go
+++ b/pkg/cmd/server/server.go
@@ -590,6 +590,7 @@ func (s *server) runControllers(defaultVolumeSnapshotLocations map[string]string
 		},
 		newPluginManager,
 		backupStoreGetter,
+		s.metrics,
 		s.logger,
 	)
 	if err := bslr.SetupWithManager(s.mgr); err != nil {
diff --git a/pkg/controller/backup_storage_location_controller.go b/pkg/controller/backup_storage_location_controller.go
index a0c9eeb2d..5b798954a 100644
--- a/pkg/controller/backup_storage_location_controller.go
+++ b/pkg/controller/backup_storage_location_controller.go
@@ -32,6 +32,7 @@ import (
 	"github.com/vmware-tanzu/velero/internal/storage"
 	velerov1api "github.com/vmware-tanzu/velero/pkg/apis/velero/v1"
 	"github.com/vmware-tanzu/velero/pkg/constant"
+	"github.com/vmware-tanzu/velero/pkg/metrics"
 	"github.com/vmware-tanzu/velero/pkg/persistence"
 	"github.com/vmware-tanzu/velero/pkg/plugin/clientmgmt"
 	"github.com/vmware-tanzu/velero/pkg/util/kube"
@@ -53,6 +54,7 @@ type backupStorageLocationReconciler struct {
 	// replaced with fakes for testing.
 	newPluginManager  func(logrus.FieldLogger) clientmgmt.Manager
 	backupStoreGetter persistence.ObjectBackupStoreGetter
+	metrics           *metrics.ServerMetrics
 
 	log logrus.FieldLogger
 }
@@ -64,6 +66,7 @@ func NewBackupStorageLocationReconciler(
 	defaultBackupLocationInfo storage.DefaultBackupLocationInfo,
 	newPluginManager func(logrus.FieldLogger) clientmgmt.Manager,
 	backupStoreGetter persistence.ObjectBackupStoreGetter,
+	metrics *metrics.ServerMetrics,
 	log logrus.FieldLogger) *backupStorageLocationReconciler {
 	return &backupStorageLocationReconciler{
 		ctx:                       ctx,
@@ -71,6 +74,7 @@ func NewBackupStorageLocationReconciler(
 		defaultBackupLocationInfo: defaultBackupLocationInfo,
 		newPluginManager:          newPluginManager,
 		backupStoreGetter:         backupStoreGetter,
+		metrics:                   metrics,
 		log:                       log,
 	}
 }
@@ -124,10 +128,16 @@ func (r *backupStorageLocationReconciler) Reconcile(ctx context.Context, req ctr
 				unavailableErrors = append(unavailableErrors, err.Error())
 				location.Status.Phase = velerov1api.BackupStorageLocationPhaseUnavailable
 				location.Status.Message = err.Error()
+				if r.metrics != nil {
+					r.metrics.RegisterBSLLastStatus(location.Name, metrics.BSLLastStatusUnavailable)
+				}
 			} else {
 				log.Info("BackupStorageLocations is valid, marking as available")
 				location.Status.Phase = velerov1api.BackupStorageLocationPhaseAvailable
 				location.Status.Message = ""
+				if r.metrics != nil {
+					r.metrics.RegisterBSLLastStatus(location.Name, metrics.BSLLastStatusAvailable)
+				}
 			}
 			if err := r.client.Patch(r.ctx, &location, client.MergeFrom(original)); err != nil {
 				log.WithError(err).Error("Error updating BackupStorageLocation phase")
diff --git a/pkg/metrics/metrics.go b/pkg/metrics/metrics.go
index 7a477e5f0..05b8c1b70 100644
--- a/pkg/metrics/metrics.go
+++ b/pkg/metrics/metrics.go
@@ -59,6 +59,7 @@ const (
 	csiSnapshotAttemptTotal       = "csi_snapshot_attempt_total"
 	csiSnapshotSuccessTotal       = "csi_snapshot_success_total"
 	csiSnapshotFailureTotal       = "csi_snapshot_failure_total"
+	bslLastStatus                 = "backup_storage_location_last_status"
 
 	// pod volume metrics
 	podVolumeBackupEnqueueTotal           = "pod_volume_backup_enqueue_count"
@@ -80,10 +81,13 @@ const (
 	pvbNameLabel            = "pod_volume_backup"
 	scheduleLabel           = "schedule"
 	backupNameLabel         = "backupName"
+	bslNameLabel            = "backup_storage_location"
 
 	// metrics values
 	BackupLastStatusSucc    int64 = 1
 	BackupLastStatusFailure int64 = 0
+	BSLLastStatusAvailable  int64 = 1
+	BSLLastStatusUnavailable int64 = 0
 )
 
 // NewServerMetrics returns new ServerMetrics
@@ -323,6 +327,14 @@ func NewServerMetrics() *ServerMetrics {
 				},
 				[]string{scheduleLabel, backupNameLabel},
 			),
+			bslLastStatus: prometheus.NewGaugeVec(
+				prometheus.GaugeOpts{
+					Namespace: metricNamespace,
+					Name:      bslLastStatus,
+					Help:      "Last status of the backup storage location. A value of 1 is available, 0 is unavailable",
+				},
+				[]string{bslNameLabel},
+			),
 		},
 	}
 }
@@ -888,3 +900,10 @@ func (m *ServerMetrics) RegisterCSISnapshotFailures(backupSchedule, backupName s
 		c.WithLabelValues(backupSchedule, backupName).Add(float64(csiSnapshotsFailed))
 	}
 }
+
+// RegisterBSLLastStatus records the last status of the backup storage location.
+func (m *ServerMetrics) RegisterBSLLastStatus(bslName string, lastStatus int64) {
+	if g, ok := m.metrics[bslLastStatus].(*prometheus.GaugeVec); ok {
+		g.WithLabelValues(bslName).Set(float64(lastStatus))
+	}
+}
