diff --git a/springdoc-openapi-starter-common/src/main/java/org/springdoc/core/service/AbstractRequestService.java b/springdoc-openapi-starter-common/src/main/java/org/springdoc/core/service/AbstractRequestService.java
index 681a61570..86b4d67fc 100644
--- a/springdoc-openapi-starter-common/src/main/java/org/springdoc/core/service/AbstractRequestService.java
+++ b/springdoc-openapi-starter-common/src/main/java/org/springdoc/core/service/AbstractRequestService.java
@@ -502,12 +502,19 @@ public abstract class AbstractRequestService {
 		}
 		// Check for @RequestBody annotation
 		org.springframework.web.bind.annotation.RequestBody requestBody = parameter.getParameterAnnotation(org.springframework.web.bind.annotation.RequestBody.class);
-		if (requestBody == null) {
-			return false;
+		if (requestBody != null) {
+			// Check if the parameter type is assignable to Map
+			Class<?> parameterType = parameter.getParameterType();
+			return Map.class.isAssignableFrom(parameterType);
+		}
+		// Check for @RequestPart annotation
+		RequestPart requestPart = parameter.getParameterAnnotation(RequestPart.class);
+		if (requestPart != null) {
+			// Check if the parameter type is assignable to Map
+			Class<?> parameterType = parameter.getParameterType();
+			return Map.class.isAssignableFrom(parameterType);
 		}
-		// Check if the parameter type is assignable to Map
-		Class<?> parameterType = parameter.getParameterType();
-		return Map.class.isAssignableFrom(parameterType);
+		return false;
 	}
 
 	/**
