diff --git a/CMakeLists.txt b/CMakeLists.txt
index 7fcb4bc..96a648f 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -10,8 +10,13 @@ set(CMAKE_ENABLE_EXPORTS ON)
 project(chaiscript)
 
 option(MULTITHREAD_SUPPORT_ENABLED "Multithreaded Support Enabled" TRUE)
+option(ENABLE_THREAD_SANITIZER "Enable thread sanitizer testing" FALSE)
 option(DYNLOAD_ENABLED "Dynamic Loading Support Enabled" TRUE)
 
+if(ENABLE_THREAD_SANITIZER)
+  set(MULTITHREAD_SUPPORT_ENABLED ON CACHE BOOL "Multithreaded Support Enabled" FORCE)
+endif()
+
 
 option(BUILD_MODULES "Build Extra Modules (stl)" TRUE)
 option(BUILD_SAMPLES "Build Samples Folder" FALSE)
@@ -35,7 +40,6 @@ if(CMAKE_COMPILER_IS_GNUCC)
 endif()
 
 if(CMAKE_COMPILER_IS_GNUCC OR "${CMAKE_CXX_COMPILER_ID}" STREQUAL "Clang" OR "${CMAKE_CXX_COMPILER_ID}" STREQUAL "AppleClang")
-  option(ENABLE_THREAD_SANITIZER "Enable thread sanitizer testing in gcc/clang" FALSE)
   if(ENABLE_THREAD_SANITIZER)
     add_definitions(-fsanitize=thread -g)
     set(LINKER_FLAGS "${LINKER_FLAGS} -fsanitize=thread")
