diff --git a/pkg/restore/actions/service_action.go b/pkg/restore/actions/service_action.go
index 9de7522..70cd5a1 100644
--- a/pkg/restore/actions/service_action.go
+++ b/pkg/restore/actions/service_action.go
@@ -101,15 +101,21 @@ func deleteHealthCheckNodePort(service *corev1api.Service) error {
 			return errors.WithStack(err)
 		}
 
-		healthCheckNodePort, exist, err := unstructured.NestedFloat64(*appliedServiceUnstructured, "spec", "healthCheckNodePort")
+		healthCheckNodePortValue, exist, err := unstructured.NestedFieldNoCopy(*appliedServiceUnstructured, "spec", "healthCheckNodePort")
 		if err != nil {
 			return errors.WithStack(err)
 		}
 
 		// Found healthCheckNodePort in lastAppliedConfig annotation,
 		// and the value is not 0. No need to delete, return.
-		if exist && healthCheckNodePort != 0 {
-			return nil
+		if exist && healthCheckNodePortValue != nil {
+			healthCheckNodePort, ok := healthCheckNodePortValue.(float64)
+			if !ok {
+				return errors.Errorf(".spec.healthCheckNodePort accessor error: %v is of the type %T, expected float64", healthCheckNodePortValue, healthCheckNodePortValue)
+			}
+			if healthCheckNodePort != 0 {
+				return nil
+			}
 		}
 	}
 
