diff --git a/packages/solid-router/src/Matches.tsx b/packages/solid-router/src/Matches.tsx
index cf83c6b50..a7b922a78 100644
--- a/packages/solid-router/src/Matches.tsx
+++ b/packages/solid-router/src/Matches.tsx
@@ -48,7 +48,7 @@ export function Matches() {
       ? SafeFragment
       : Solid.Suspense
 
-  const inner = (
+  const inner = () => (
     <ResolvedSuspense fallback={pendingElement}>
       {!router.isServer && <Transitioner />}
       <MatchesInner />
@@ -56,9 +56,9 @@ export function Matches() {
   )
 
   return router.options.InnerWrap ? (
-    <router.options.InnerWrap>{inner}</router.options.InnerWrap>
+    <router.options.InnerWrap>{inner()}</router.options.InnerWrap>
   ) : (
-    inner
+    inner()
   )
 }
 
diff --git a/packages/solid-router/src/RouterProvider.tsx b/packages/solid-router/src/RouterProvider.tsx
index 1f579a0a4..7c59a616f 100644
--- a/packages/solid-router/src/RouterProvider.tsx
+++ b/packages/solid-router/src/RouterProvider.tsx
@@ -29,17 +29,17 @@ export function RouterContextProvider<
 
   const routerContext = getRouterContext()
 
-  const provider = (
+  const provider = () => (
     <routerContext.Provider value={router as AnyRouter}>
       {children()}
     </routerContext.Provider>
   )
 
   if (router.options.Wrap) {
-    return <router.options.Wrap>{provider}</router.options.Wrap>
+    return <router.options.Wrap>{provider()}</router.options.Wrap>
   }
 
-  return provider
+  return provider()
 }
 
 export function RouterProvider<
diff --git a/packages/solid-router/src/ssr/RouterClient.tsx b/packages/solid-router/src/ssr/RouterClient.tsx
index d49255312..0e110a330 100644
--- a/packages/solid-router/src/ssr/RouterClient.tsx
+++ b/packages/solid-router/src/ssr/RouterClient.tsx
@@ -9,6 +9,11 @@ let hydrationPromise: Promise<void | Array<Array<void>>> | undefined
 
 const Dummy = (props: { children?: JSXElement }) => <>{props.children}</>
 
+const userInnerWraps = new WeakMap<
+  AnyRouter,
+  AnyRouter['options']['InnerWrap']
+>()
+
 export function RouterClient(props: { router: AnyRouter }) {
   if (!hydrationPromise) {
     if (!props.router.state.matches.length) {
@@ -17,6 +22,12 @@ export function RouterClient(props: { router: AnyRouter }) {
       hydrationPromise = Promise.resolve()
     }
   }
+
+  if (!userInnerWraps.has(props.router)) {
+    userInnerWraps.set(props.router, props.router.options.InnerWrap)
+  }
+  const UserInnerWrap = userInnerWraps.get(props.router)
+
   return (
     <Await
       promise={hydrationPromise}
@@ -25,15 +36,23 @@ export function RouterClient(props: { router: AnyRouter }) {
           <Dummy>
             <RouterProvider
               router={props.router}
-              InnerWrap={(props) => (
-                <Dummy>
+              InnerWrap={(innerProps) => {
+                const inner = () => (
                   <Dummy>
-                    <HeadContent />
-                    {props.children}
+                    <Dummy>
+                      <HeadContent />
+                      {innerProps.children}
+                    </Dummy>
+                    <Dummy />
                   </Dummy>
-                  <Dummy />
-                </Dummy>
-              )}
+                )
+
+                return UserInnerWrap ? (
+                  <UserInnerWrap>{inner()}</UserInnerWrap>
+                ) : (
+                  inner()
+                )
+              }}
             />
           </Dummy>
         </Dummy>
diff --git a/packages/solid-router/src/ssr/RouterServer.tsx b/packages/solid-router/src/ssr/RouterServer.tsx
index 99b4e684b..52de89bfc 100644
--- a/packages/solid-router/src/ssr/RouterServer.tsx
+++ b/packages/solid-router/src/ssr/RouterServer.tsx
@@ -12,6 +12,11 @@ import { RouterProvider } from '../RouterProvider'
 import { Scripts } from '../Scripts'
 import type { AnyRouter } from '@tanstack/router-core'
 
+const userInnerWraps = new WeakMap<
+  AnyRouter,
+  AnyRouter['options']['InnerWrap']
+>()
+
 export function ServerHeadContent() {
   const tags = useTags()
   useAssets(() => {
@@ -31,6 +36,11 @@ const docType = ssr('<!DOCTYPE html>')
 export function RouterServer<TRouter extends AnyRouter>(props: {
   router: TRouter
 }) {
+  if (!userInnerWraps.has(props.router)) {
+    userInnerWraps.set(props.router, props.router.options.InnerWrap)
+  }
+  const UserInnerWrap = userInnerWraps.get(props.router)
+
   return (
     <NoHydration>
       {docType as any}
@@ -42,15 +52,23 @@ export function RouterServer<TRouter extends AnyRouter>(props: {
           <Hydration>
             <RouterProvider
               router={props.router}
-              InnerWrap={(props) => (
-                <NoHydration>
-                  <MetaProvider>
-                    <ServerHeadContent />
-                    <Hydration>{props.children}</Hydration>
-                    <Scripts />
-                  </MetaProvider>
-                </NoHydration>
-              )}
+              InnerWrap={(innerProps) => {
+                const inner = () => (
+                  <NoHydration>
+                    <MetaProvider>
+                      <ServerHeadContent />
+                      <Hydration>{innerProps.children}</Hydration>
+                      <Scripts />
+                    </MetaProvider>
+                  </NoHydration>
+                )
+
+                return UserInnerWrap ? (
+                  <UserInnerWrap>{inner()}</UserInnerWrap>
+                ) : (
+                  inner()
+                )
+              }}
             />
           </Hydration>
         </body>
diff --git a/packages/solid-start-client/src/StartClient.tsx b/packages/solid-start-client/src/StartClient.tsx
index 195632021..6d9cc9d05 100644
--- a/packages/solid-start-client/src/StartClient.tsx
+++ b/packages/solid-start-client/src/StartClient.tsx
@@ -7,6 +7,11 @@ let hydrationPromise: Promise<void | Array<Array<void>>> | undefined
 
 const Dummy = (props: { children?: JSXElement }) => <>{props.children}</>
 
+const userInnerWraps = new WeakMap<
+  AnyRouter,
+  AnyRouter['options']['InnerWrap']
+>()
+
 export function StartClient(props: { router: AnyRouter }) {
   if (!hydrationPromise) {
     if (!props.router.state.matches.length) {
@@ -15,6 +20,12 @@ export function StartClient(props: { router: AnyRouter }) {
       hydrationPromise = Promise.resolve()
     }
   }
+
+  if (!userInnerWraps.has(props.router)) {
+    userInnerWraps.set(props.router, props.router.options.InnerWrap)
+  }
+  const UserInnerWrap = userInnerWraps.get(props.router)
+
   return (
     <Await
       promise={hydrationPromise}
@@ -23,15 +34,23 @@ export function StartClient(props: { router: AnyRouter }) {
           <Dummy>
             <RouterProvider
               router={props.router}
-              InnerWrap={(props) => (
-                <Dummy>
+              InnerWrap={(innerProps) => {
+                const inner = () => (
                   <Dummy>
-                    <HeadContent />
-                    {props.children}
+                    <Dummy>
+                      <HeadContent />
+                      {innerProps.children}
+                    </Dummy>
+                    <Dummy />
                   </Dummy>
-                  <Dummy />
-                </Dummy>
-              )}
+                )
+
+                return UserInnerWrap ? (
+                  <UserInnerWrap>{inner()}</UserInnerWrap>
+                ) : (
+                  inner()
+                )
+              }}
             />
           </Dummy>
         </Dummy>
diff --git a/packages/solid-start-server/src/StartServer.tsx b/packages/solid-start-server/src/StartServer.tsx
index 823d46b14..03fe8834a 100644
--- a/packages/solid-start-server/src/StartServer.tsx
+++ b/packages/solid-start-server/src/StartServer.tsx
@@ -9,6 +9,11 @@ import {
 import { MetaProvider } from '@solidjs/meta'
 import type { AnyRouter } from '@tanstack/router-core'
 
+const userInnerWraps = new WeakMap<
+  AnyRouter,
+  AnyRouter['options']['InnerWrap']
+>()
+
 export function ServerHeadContent() {
   const tags = useTags()
   useAssets(() => {
@@ -28,6 +33,11 @@ const docType = ssr('<!DOCTYPE html>')
 export function StartServer<TRouter extends AnyRouter>(props: {
   router: TRouter
 }) {
+  if (!userInnerWraps.has(props.router)) {
+    userInnerWraps.set(props.router, props.router.options.InnerWrap)
+  }
+  const UserInnerWrap = userInnerWraps.get(props.router)
+
   return (
     <NoHydration>
       {docType as any}
@@ -39,15 +49,23 @@ export function StartServer<TRouter extends AnyRouter>(props: {
           <Hydration>
             <RouterProvider
               router={props.router}
-              InnerWrap={(props) => (
-                <NoHydration>
-                  <MetaProvider>
-                    <ServerHeadContent />
-                    <Hydration>{props.children}</Hydration>
-                    <Scripts />
-                  </MetaProvider>
-                </NoHydration>
-              )}
+              InnerWrap={(innerProps) => {
+                const inner = () => (
+                  <NoHydration>
+                    <MetaProvider>
+                      <ServerHeadContent />
+                      <Hydration>{innerProps.children}</Hydration>
+                      <Scripts />
+                    </MetaProvider>
+                  </NoHydration>
+                )
+
+                return UserInnerWrap ? (
+                  <UserInnerWrap>{inner()}</UserInnerWrap>
+                ) : (
+                  inner()
+                )
+              }}
             />
           </Hydration>
         </body>
