diff --git a/src/hb-font.hh b/src/hb-font.hh
index 842d0ea..58c2fb8 100644
--- a/src/hb-font.hh
+++ b/src/hb-font.hh
@@ -668,7 +668,9 @@ struct hb_font_t
   hb_bool_t get_glyph_name (hb_codepoint_t glyph,
 			    char *name, unsigned int size)
   {
-    if (size) *name = '\0';
+    if (unlikely (!size)) return false;
+
+    *name = '\0';
     return klass->get.f.glyph_name (this, user_data,
 				    glyph,
 				    name, size,
diff --git a/src/rust/font.rs b/src/rust/font.rs
index 8e6b526..c9820a1 100644
--- a/src/rust/font.rs
+++ b/src/rust/font.rs
@@ -980,6 +980,10 @@ extern "C" fn _hb_fontations_glyph_name(
 ) -> hb_bool_t {
     let data = unsafe { &mut *(font_data as *mut FontationsData) };
 
+    if size == 0 {
+        return false as hb_bool_t;
+    }
+
     if let Some(glyph_name) = data.glyph_names.get(GlyphId::new(glyph)) {
         let glyph_name = glyph_name.as_str();
         // Copy the glyph name into the buffer, up to size-1 bytes
