diff --git a/internal/components/processors/k8sattribute.go b/internal/components/processors/k8sattribute.go
index 84a76c36..3e84d1de 100644
--- a/internal/components/processors/k8sattribute.go
+++ b/internal/components/processors/k8sattribute.go
@@ -53,7 +53,7 @@ func GenerateK8SAttrRbacRules(_ logr.Logger, config K8sAttributeConfig) ([]rbacv
 	addedReplicasetPolicy := false
 	for _, m := range config.Extract.Metadata {
 		metadataField := fmt.Sprint(m)
-		if (metadataField == "k8s.deployment.uid" || metadataField == "k8s.deployment.name") && !addedReplicasetPolicy {
+		if (metadataField == "k8s.deployment.uid" || metadataField == "k8s.deployment.name" || metadataField == "service.name") && !addedReplicasetPolicy {
 			prs = append(prs, replicasetPolicy)
 			addedReplicasetPolicy = true
 		} else if strings.Contains(metadataField, "k8s.node") {
diff --git a/internal/components/processors/k8sattribute_test.go b/internal/components/processors/k8sattribute_test.go
index 95db1c1b..45eaf204 100644
--- a/internal/components/processors/k8sattribute_test.go
+++ b/internal/components/processors/k8sattribute_test.go
@@ -126,6 +126,31 @@ func TestGenerateK8SAttrRbacRules(t *testing.T) {
 			},
 			wantErr: assert.NoError,
 		},
+		{
+			name: "config with service.name metadata",
+			args: args{
+				config: map[string]interface{}{
+					"extract": map[string]interface{}{
+						"metadata":    []string{"service.name"},
+						"labels":      []interface{}{},
+						"annotations": []interface{}{},
+					},
+				},
+			},
+			want: []rbacv1.PolicyRule{
+				{
+					APIGroups: []string{""},
+					Resources: []string{"pods", "namespaces"},
+					Verbs:     []string{"get", "watch", "list"},
+				},
+				{
+					APIGroups: []string{"apps"},
+					Resources: []string{"replicasets"},
+					Verbs:     []string{"get", "watch", "list"},
+				},
+			},
+			wantErr: assert.NoError,
+		},
 	}
 	for _, tt := range tests {
 		t.Run(tt.name, func(t *testing.T) {
