diff --git a/crates/ide-completion/src/tests/expression.rs b/crates/ide-completion/src/tests/expression.rs
index 33f729f016..dcc958e77b 100644
--- a/crates/ide-completion/src/tests/expression.rs
+++ b/crates/ide-completion/src/tests/expression.rs
@@ -2275,3 +2275,32 @@ fn foo() {
         "#]],
     );
 }
+
+#[test]
+fn deref_assign_completion_does_not_suggest_mutated_self_variants() {
+    check(
+        r#"
+enum State {
+    Stop(()),
+}
+
+impl State {
+    fn get_or_init(&mut self) -> &mut () {
+        match self {
+            State::Stop(_) => {
+                let this = self;
+                let Self::Stop(f) = this else { unreachable!() };
+                *self = $0;
+            },
+        }
+    }
+}
+"#,
+        expect![[r#"
+            en State       State
+            ev Stop(…)  State::Stop
+            sp Self       State
+            bt u32         u32
+        "#]],
+    )
+}
