diff --git a/hexpm/src/lib.rs b/hexpm/src/lib.rs
index 328aafa..26bbf02 100644
--- a/hexpm/src/lib.rs
+++ b/hexpm/src/lib.rs
@@ -1142,7 +1142,7 @@ impl OAuthDeviceAuthorisation {
         &mut self,
         response: http::Response<Vec<u8>>,
     ) -> Result<PollStep, ApiError> {
-        if self.start_time.elapsed() > Duration::from_mins(10) {
+        if self.start_time.elapsed() > Duration::from_secs(60 * 10) {
             return Err(ApiError::OAuthTimeout);
         }
