diff --git a/internal/cmd/readdcmd.go b/internal/cmd/readdcmd.go
index 8f7f91c..766aa39 100644
--- a/internal/cmd/readdcmd.go
+++ b/internal/cmd/readdcmd.go
@@ -198,12 +198,16 @@ TARGET_REL_PATH:
 		destAbsPathInfos := map[chezmoi.AbsPath]fs.FileInfo{
 			destAbsPath: destAbsPathInfo,
 		}
+		preAddFunc := c.defaultPreAddFunc
+		if sourceStateFile.Attr().Encrypted {
+			preAddFunc = nil
+		}
 		if err := sourceState.Add(c.sourceSystem, c.persistentState, c.destSystem, destAbsPathInfos, &chezmoi.AddOptions{
 			Encrypt:         sourceStateFile.Attr().Encrypted,
 			EncryptedSuffix: c.encryption.EncryptedSuffix(),
 			Errorf:          c.errorf,
 			Filter:          c.reAdd.filter,
-			PreAddFunc:      c.defaultPreAddFunc,
+			PreAddFunc:      preAddFunc,
 		}); err != nil {
 			return err
 		}
