diff --git a/src/ts_catalog/continuous_agg.c b/src/ts_catalog/continuous_agg.c
index f375b4d14..4512080d2 100644
--- a/src/ts_catalog/continuous_agg.c
+++ b/src/ts_catalog/continuous_agg.c
@@ -1519,10 +1519,17 @@ ts_compute_inscribed_bucketed_refresh_window_variable(int64 *start, int64 *end,
 	 * the CAGG. For more details see ts_internal_to_time_value() implementation.
 	 */
 	start_old = ts_internal_to_time_value(*start, TIMESTAMPOID);
-	end_old = ts_internal_to_time_value(*end, TIMESTAMPOID);
-
 	start_aligned = generic_time_bucket(bf, start_old);
-	end_aliged = generic_time_bucket(bf, end_old);
+
+	if (*end == ts_time_get_end_or_max(TIMESTAMPOID))
+	{
+		end_aliged = ts_internal_to_time_value(*end, TIMESTAMPOID);
+	}
+	else
+	{
+		end_old = ts_internal_to_time_value(*end, TIMESTAMPOID);
+		end_aliged = generic_time_bucket(bf, end_old);
+	}
 
 	if (DatumGetTimestamp(start_aligned) != DatumGetTimestamp(start_old))
 	{
