diff --git a/src/js/components/FileInput/FileInput.js b/src/js/components/FileInput/FileInput.js
index 98ae9e121..1c2e6dd18 100644
--- a/src/js/components/FileInput/FileInput.js
+++ b/src/js/components/FileInput/FileInput.js
@@ -353,6 +353,7 @@ const FileInput = forwardRef(
                           id: 'fileInput.browse',
                           messages,
                         })}
+                        disabled={disabled}
                         onClick={() => {
                           inputRef.current.click();
                           inputRef.current.focus();
@@ -365,6 +366,7 @@ const FileInput = forwardRef(
                         alignSelf="center"
                         ref={controlRef}
                         margin="small"
+                        disabled={disabled}
                         onClick={() => {
                           inputRef.current.click();
                           inputRef.current.focus();
@@ -427,6 +429,7 @@ const FileInput = forwardRef(
                         id: 'fileInput.browse',
                         messages,
                       })}
+                      disabled={disabled}
                       onClick={() => {
                         inputRef.current.click();
                         inputRef.current.focus();
@@ -439,6 +442,7 @@ const FileInput = forwardRef(
                       alignSelf="center"
                       ref={controlRef}
                       margin="small"
+                      disabled={disabled}
                       onClick={() => {
                         inputRef.current.click();
                         inputRef.current.focus();
@@ -524,6 +528,7 @@ const FileInput = forwardRef(
                             id: 'fileInput.browse',
                             messages,
                           })}
+                          disabled={disabled}
                           onClick={() => {
                             inputRef.current.click();
                             inputRef.current.focus();
@@ -535,6 +540,7 @@ const FileInput = forwardRef(
                           tabIndex={-1}
                           ref={controlRef}
                           margin="small"
+                          disabled={disabled}
                           onClick={() => {
                             inputRef.current.click();
                             inputRef.current.focus();
