diff --git a/datafusion/substrait/src/logical_plan/consumer/rel/join_rel.rs b/datafusion/substrait/src/logical_plan/consumer/rel/join_rel.rs
index 3604630..ed87d25 100644
--- a/datafusion/substrait/src/logical_plan/consumer/rel/join_rel.rs
+++ b/datafusion/substrait/src/logical_plan/consumer/rel/join_rel.rs
@@ -93,43 +93,59 @@ fn split_eq_and_noneq_join_predicate_with_nulls_equality(
 ) -> (Vec<(Column, Column)>, bool, Option<Expr>) {
     let exprs = split_conjunction(filter);
 
-    let mut accum_join_keys: Vec<(Column, Column)> = vec![];
+    let mut eq_join_keys: Vec<(Column, Column)> = vec![];
+    let mut is_not_distinct_from_join_keys: Vec<(Column, Column)> = vec![];
+    let mut eq_filters: Vec<Expr> = vec![];
+    let mut is_not_distinct_from_filters: Vec<Expr> = vec![];
     let mut accum_filters: Vec<Expr> = vec![];
-    let mut nulls_equal_nulls = false;
 
     for expr in exprs {
         #[expect(clippy::collapsible_match)]
         match expr {
             Expr::BinaryExpr(binary_expr) => match binary_expr {
-                x @ (BinaryExpr {
+                BinaryExpr {
                     left,
                     op: Operator::Eq,
                     right,
-                }
-                | BinaryExpr {
+                } => match (left.as_ref(), right.as_ref()) {
+                    (Expr::Column(l), Expr::Column(r)) => {
+                        eq_join_keys.push((l.clone(), r.clone()));
+                    }
+                    _ => eq_filters.push(expr.clone()),
+                },
+                BinaryExpr {
                     left,
                     op: Operator::IsNotDistinctFrom,
                     right,
-                }) => {
-                    nulls_equal_nulls = match x.op {
-                        Operator::Eq => false,
-                        Operator::IsNotDistinctFrom => true,
-                        _ => unreachable!(),
-                    };
-
-                    match (left.as_ref(), right.as_ref()) {
-                        (Expr::Column(l), Expr::Column(r)) => {
-                            accum_join_keys.push((l.clone(), r.clone()));
-                        }
-                        _ => accum_filters.push(expr.clone()),
+                } => match (left.as_ref(), right.as_ref()) {
+                    (Expr::Column(l), Expr::Column(r)) => {
+                        is_not_distinct_from_join_keys.push((l.clone(), r.clone()));
                     }
-                }
+                    _ => is_not_distinct_from_filters.push(expr.clone()),
+                },
                 _ => accum_filters.push(expr.clone()),
             },
             _ => accum_filters.push(expr.clone()),
         }
     }
 
+    let (accum_join_keys, nulls_equal_nulls) = if eq_join_keys.is_empty() {
+        accum_filters.extend(eq_filters);
+        accum_filters.extend(is_not_distinct_from_filters);
+        (is_not_distinct_from_join_keys, true)
+    } else {
+        accum_filters.extend(eq_filters);
+        accum_filters.extend(is_not_distinct_from_filters);
+        accum_filters.extend(is_not_distinct_from_join_keys.into_iter().map(|(l, r)| {
+            Expr::BinaryExpr(BinaryExpr::new(
+                Box::new(Expr::Column(l)),
+                Operator::IsNotDistinctFrom,
+                Box::new(Expr::Column(r)),
+            ))
+        }));
+        (eq_join_keys, false)
+    };
+
     let join_filter = accum_filters.into_iter().reduce(Expr::and);
     (accum_join_keys, nulls_equal_nulls, join_filter)
 }
