diff --git a/src/main/java/org/json/XML.java b/src/main/java/org/json/XML.java
index 4bf4759..641be2a 100644
--- a/src/main/java/org/json/XML.java
+++ b/src/main/java/org/json/XML.java
@@ -366,6 +366,11 @@ public class XML {
                                         config.isKeepNumberAsString()
                                                 ? ((String) token)
                                                 : obj);
+                            } else if (obj == JSONObject.NULL) {
+                                jsonObject.accumulate(string,
+                                        (config.isKeepNumberAsString() && config.isKeepBooleanAsString())
+                                                ? ((String) token)
+                                                : obj);
                             } else {
                                 jsonObject.accumulate(string, stringToValue((String) token));
                             }
@@ -428,6 +433,11 @@ public class XML {
                                                 config.isKeepNumberAsString()
                                                         ? ((String) token)
                                                         : obj);
+                                    } else if (obj == JSONObject.NULL) {
+                                        jsonObject.accumulate(config.getcDataTagName(),
+                                                (config.isKeepNumberAsString() && config.isKeepBooleanAsString())
+                                                        ? ((String) token)
+                                                        : obj);
                                     } else {
                                         jsonObject.accumulate(config.getcDataTagName(), stringToValue((String) token));
                                     }
