diff --git a/crd-generator/src/main/java/io/strimzi/crdgenerator/DocGenerator.java b/crd-generator/src/main/java/io/strimzi/crdgenerator/DocGenerator.java
index b1b3123..db9117b 100644
--- a/crd-generator/src/main/java/io/strimzi/crdgenerator/DocGenerator.java
+++ b/crd-generator/src/main/java/io/strimzi/crdgenerator/DocGenerator.java
@@ -13,6 +13,7 @@ import io.strimzi.crdgenerator.annotations.AddedIn;
 import io.strimzi.crdgenerator.annotations.Crd;
 import io.strimzi.crdgenerator.annotations.Description;
 import io.strimzi.crdgenerator.annotations.DescriptionFile;
+import io.strimzi.crdgenerator.annotations.ExternalLink;
 import io.strimzi.crdgenerator.annotations.KubeLink;
 import io.strimzi.crdgenerator.annotations.PresentInVersions;
 
@@ -89,7 +90,8 @@ class DocGenerator {
         Set<Property> memorableProperties = new HashSet<>();
 
         for (Property property : properties(crApiVersion, cls).values()) {
-            if (property.isAnnotationPresent(KubeLink.class)) {
+            if (property.isAnnotationPresent(KubeLink.class)
+                    || property.isAnnotationPresent(ExternalLink.class)) {
                 continue;
             }
 
@@ -151,9 +153,15 @@ class DocGenerator {
             final Property property = entry.getValue();
             PropertyType propertyType = property.getType();
         
-            // Set the external link to Kubernetes docs or the link for fields distinguished by `type`
+            // Set the external link to Kubernetes docs, an external URL, or the link for fields distinguished by `type`
             KubeLink kubeLink = property.getAnnotation(KubeLink.class);
-            String externalUrl = linker != null && kubeLink != null ? linker.link(kubeLink) : null;
+            ExternalLink externalLink = property.getAnnotation(ExternalLink.class);
+            String externalUrl = null;
+            if (linker != null && kubeLink != null) {
+                externalUrl = linker.link(kubeLink);
+            } else if (linker != null && externalLink != null) {
+                externalUrl = linker.link(externalLink);
+            }
         
             // Add the property name
             out.append("|").append(propertyName);
diff --git a/crd-generator/src/main/java/io/strimzi/crdgenerator/KubeLinker.java b/crd-generator/src/main/java/io/strimzi/crdgenerator/KubeLinker.java
index def3362..07c33e3 100644
--- a/crd-generator/src/main/java/io/strimzi/crdgenerator/KubeLinker.java
+++ b/crd-generator/src/main/java/io/strimzi/crdgenerator/KubeLinker.java
@@ -4,6 +4,7 @@
  */
 package io.strimzi.crdgenerator;
 
+import io.strimzi.crdgenerator.annotations.ExternalLink;
 import io.strimzi.crdgenerator.annotations.KubeLink;
 
 /**
@@ -34,4 +35,16 @@ public class KubeLinker implements Linker {
         // E.g. https://kubernetes.io/docs/reference/generated/kubernetes-api/v1.11/#networkpolicyingressrule-v1-networking-k8s-io
         return baseUrl + "#" + kubeLink.kind() + "-" + kubeLink.version() + "-" + kubeLink.group().replace(".", "-");
     }
+
+    /**
+     * Generates an URL to external documentation. Returns the URL from the annotation directly.
+     *
+     * @param externalLink  Specifies the external URL which should the link point to
+     *
+     * @return  An HTTP link to the external documentation
+     */
+    @Override
+    public String link(ExternalLink externalLink) {
+        return externalLink.url();
+    }
 }
diff --git a/crd-generator/src/main/java/io/strimzi/crdgenerator/Linker.java b/crd-generator/src/main/java/io/strimzi/crdgenerator/Linker.java
index 64909db..fd0cb3e 100644
--- a/crd-generator/src/main/java/io/strimzi/crdgenerator/Linker.java
+++ b/crd-generator/src/main/java/io/strimzi/crdgenerator/Linker.java
@@ -4,6 +4,7 @@
  */
 package io.strimzi.crdgenerator;
 
+import io.strimzi.crdgenerator.annotations.ExternalLink;
 import io.strimzi.crdgenerator.annotations.KubeLink;
 
 /**
@@ -18,4 +19,13 @@ interface Linker {
      * @return  An HTTP link deep-linking to the Kubernetes documentation
      */
     String link(KubeLink kubeLink);
+
+    /**
+     * Generates URL to some external documentation. This returns the URL directly from the annotation.
+     *
+     * @param externalLink  Specifies the external URL which should the link point to
+     *
+     * @return  An HTTP link to the external documentation
+     */
+    String link(ExternalLink externalLink);
 }
