diff --git a/pkg/models/kanban_task_bucket.go b/pkg/models/kanban_task_bucket.go
index 35c623a..28a2d63 100644
--- a/pkg/models/kanban_task_bucket.go
+++ b/pkg/models/kanban_task_bucket.go
@@ -132,8 +132,6 @@ func updateTaskBucket(s *xorm.Session, a web.Auth, b *TaskBucket) (err error) {
 		bucket.Count = taskCount
 	}
 
-	var updateBucket = true
-
 	// mark task done if moved into or out of the done bucket
 	// Only change the done state if the task's done value actually changes
 	var doneChanged bool
@@ -145,8 +143,10 @@ func updateTaskBucket(s *xorm.Session, a web.Auth, b *TaskBucket) (err error) {
 				oldTask := *task
 				oldTask.Done = false
 				updateDone(&oldTask, task)
-				updateBucket = false
-				b.BucketID = oldTaskBucket.BucketID
+				b.BucketID, err = getDefaultBucketID(s, view)
+				if err != nil {
+					return err
+				}
 			}
 		}
 
@@ -204,13 +204,11 @@ func updateTaskBucket(s *xorm.Session, a web.Auth, b *TaskBucket) (err error) {
 		}
 	}
 
-	if updateBucket {
-		err = b.upsert(s)
-		if err != nil {
-			return
-		}
-		bucket.Count++
+	err = b.upsert(s)
+	if err != nil {
+		return
 	}
+	bucket.Count++
 
 	b.Task = task
 	b.Bucket = bucket
