diff --git a/packages/solid-router/src/Matches.tsx b/packages/solid-router/src/Matches.tsx
index cf83c6b50..6251db550 100644
--- a/packages/solid-router/src/Matches.tsx
+++ b/packages/solid-router/src/Matches.tsx
@@ -48,18 +48,24 @@ export function Matches() {
       ? SafeFragment
       : Solid.Suspense
 
-  const inner = (
+  const Inner = () => (
     <ResolvedSuspense fallback={pendingElement}>
       {!router.isServer && <Transitioner />}
       <MatchesInner />
     </ResolvedSuspense>
   )
 
-  return router.options.InnerWrap ? (
-    <router.options.InnerWrap>{inner}</router.options.InnerWrap>
-  ) : (
-    inner
-  )
+  if (router.options.InnerWrap) {
+    const InnerWrap = router.options.InnerWrap
+
+    return (
+      <InnerWrap>
+        <Inner />
+      </InnerWrap>
+    )
+  }
+
+  return <Inner />
 }
 
 function MatchesInner() {
diff --git a/packages/solid-router/src/RouterProvider.tsx b/packages/solid-router/src/RouterProvider.tsx
index 1f579a0a4..135e6edfd 100644
--- a/packages/solid-router/src/RouterProvider.tsx
+++ b/packages/solid-router/src/RouterProvider.tsx
@@ -1,6 +1,6 @@
+import * as Solid from 'solid-js'
 import { Matches } from './Matches'
 import { getRouterContext } from './routerContext'
-import type * as Solid from 'solid-js'
 import type {
   AnyRouter,
   RegisteredRouter,
@@ -29,17 +29,25 @@ export function RouterContextProvider<
 
   const routerContext = getRouterContext()
 
-  const provider = (
+  const Provider = (props: { children: Solid.JSX.Element }) => (
     <routerContext.Provider value={router as AnyRouter}>
-      {children()}
+      {props.children}
     </routerContext.Provider>
   )
 
+  const Inner = () => <Provider>{children()}</Provider>
+
   if (router.options.Wrap) {
-    return <router.options.Wrap>{provider}</router.options.Wrap>
+    const Wrap = router.options.Wrap
+
+    return (
+      <Wrap>
+        <Inner />
+      </Wrap>
+    )
   }
 
-  return provider
+  return <Inner />
 }
 
 export function RouterProvider<
