diff --git a/src/BootstrapBlazor/Components/Radio/RadioList.razor.cs b/src/BootstrapBlazor/Components/Radio/RadioList.razor.cs
index 899f1d494..2157303a2 100644
--- a/src/BootstrapBlazor/Components/Radio/RadioList.razor.cs
+++ b/src/BootstrapBlazor/Components/Radio/RadioList.razor.cs
@@ -46,7 +46,8 @@ public partial class RadioList<TValue>
         .Build();
 
     private string? GetButtonItemClassString(SelectedItem item) => CssBuilder.Default("btn")
-        .AddClass($"active bg-{Color.ToDescriptionString()}", CurrentValueAsString == item.Value)
+        .AddClass($"btn-outline-{Color.ToDescriptionString()}")
+        .AddClass("active", CurrentValueAsString == item.Value)
         .Build();
 
     /// <summary>
diff --git a/src/BootstrapBlazor/Components/Radio/RadioList.razor.scss b/src/BootstrapBlazor/Components/Radio/RadioList.razor.scss
index f60f1e5e8..62c192ed0 100644
--- a/src/BootstrapBlazor/Components/Radio/RadioList.razor.scss
+++ b/src/BootstrapBlazor/Components/Radio/RadioList.razor.scss
@@ -38,6 +38,10 @@
             color: var(--bb-radio-group-item-active-color);
         }
 
+        .btn-outline-info {
+            --bs-btn-active-color: var(--bb-radio-group-item-active-color);
+        }
+
         &.disabled > span {
             opacity: var(--bs-btn-disabled-opacity);
 
diff --git a/src/BootstrapBlazor/Components/Radio/RadioListGeneric.razor.cs b/src/BootstrapBlazor/Components/Radio/RadioListGeneric.razor.cs
index 833d8ddc8..349e2ad02 100644
--- a/src/BootstrapBlazor/Components/Radio/RadioListGeneric.razor.cs
+++ b/src/BootstrapBlazor/Components/Radio/RadioListGeneric.razor.cs
@@ -102,7 +102,8 @@ public partial class RadioListGeneric<TValue> : IModelEqualityComparer<TValue>
         .Build();
 
     private string? GetButtonItemClassString(SelectedItem<TValue> item) => CssBuilder.Default("btn")
-        .AddClass($"active bg-{Color.ToDescriptionString()}", Equals(Value, item.Value))
+        .AddClass($"btn-outline-{Color.ToDescriptionString()}")
+        .AddClass("active", Equals(Value, item.Value))
         .Build();
 
     /// <summary>
diff --git a/test/UnitTest/Components/RadioTest.cs b/test/UnitTest/Components/RadioTest.cs
index 3cbc56d27..cfc6c0c04 100644
--- a/test/UnitTest/Components/RadioTest.cs
+++ b/test/UnitTest/Components/RadioTest.cs
@@ -232,13 +232,15 @@ public class RadioTest : BootstrapBlazorTestBase
         {
             pb.Add(a => a.Color, Color.Danger);
         });
-        cut.Contains("btn active bg-danger");
+        cut.Contains("btn-outline-danger");
+        cut.Contains(" active");
 
         cut.InvokeAsync(() =>
         {
             var btn = cut.Find(".btn");
             btn.Click();
-            cut.Contains("btn active bg-danger");
+            cut.Contains("btn-outline-danger");
+            cut.Contains(" active");
         });
     }
