diff --git a/pkg/util/overridemanager/overridemanager.go b/pkg/util/overridemanager/overridemanager.go
index 7fb4e3d..9cb3add 100644
--- a/pkg/util/overridemanager/overridemanager.go
+++ b/pkg/util/overridemanager/overridemanager.go
@@ -433,15 +433,19 @@ func applyFieldOverriders(rawObj *unstructured.Unstructured, FieldOverriders []p
 		if len(FieldOverriders[index].YAML) > 0 {
 			appliedRawData, err = applyRawYAMLPatch(dataBytes, parseYAMLPatchesByField(FieldOverriders[index].YAML))
 			if err != nil {
-				klog.Errorf("Error applying raw JSON patch: %v", err)
+				klog.Errorf("Error applying raw YAML patch: %v", err)
 				return err
 			}
 		} else if len(FieldOverriders[index].JSON) > 0 {
 			appliedRawData, err = applyRawJSONPatch(dataBytes, parseJSONPatchesByField(FieldOverriders[index].JSON))
 			if err != nil {
-				klog.Errorf("Error applying raw YAML patch: %v", err)
+				klog.Errorf("Error applying raw JSON patch: %v", err)
 				return err
 			}
+		} else {
+			errMsg := fmt.Sprintf("FieldOverriders[%d] must set either YAML or JSON", index)
+			klog.Error(errMsg)
+			return errors.New(errMsg)
 		}
 		_, err = pointer.Set(rawObj.Object, string(appliedRawData))
 		if err != nil {
diff --git a/pkg/util/validation/validation.go b/pkg/util/validation/validation.go
index 9e47448..8496c94 100644
--- a/pkg/util/validation/validation.go
+++ b/pkg/util/validation/validation.go
@@ -435,6 +435,9 @@ func ValidateOverrideRules(overrideRules []policyv1alpha1.RuleWithCluster, fldPa
 			if len(fieldOverrider.YAML) > 0 && len(fieldOverrider.JSON) > 0 {
 				allErrs = append(allErrs, field.Invalid(fieldPath, fieldOverrider, "FieldOverrider has both YAML and JSON set. Only one is allowed"))
 			}
+			if len(fieldOverrider.YAML) == 0 && len(fieldOverrider.JSON) == 0 {
+				allErrs = append(allErrs, field.Invalid(fieldPath, fieldOverrider, "FieldOverrider must set either YAML or JSON"))
+			}
 			// validates the field path.
 			if _, err := jsonpointer.New(fieldOverrider.FieldPath); err != nil {
 				allErrs = append(allErrs, field.Invalid(fieldPath.Child("fieldPath"), fieldOverrider.FieldPath, err.Error()))
