diff --git a/clap_builder/src/output/help_template.rs b/clap_builder/src/output/help_template.rs
index 5ac11fe7..21b02f3f 100644
--- a/clap_builder/src/output/help_template.rs
+++ b/clap_builder/src/output/help_template.rs
@@ -1005,9 +1005,14 @@ impl HelpTemplate<'_, '_> {
             .get_visible_short_flag_aliases()
             .map(|a| format!("{ctx_val}-{a}{ctx_val:#}"))
             .collect::<Vec<_>>();
+        let long_als = a
+            .get_visible_long_flag_aliases()
+            .map(|s| format!("{ctx_val}--{s}{ctx_val:#}"))
+            .collect::<Vec<_>>();
         let als = a
             .get_visible_aliases()
             .map(|s| format!("{ctx_val}{s}{ctx_val:#}"));
+        short_als.extend(long_als);
         short_als.extend(als);
         let all_als = short_als.join(&val_sep);
         if !all_als.is_empty() {
@@ -1019,6 +1024,10 @@ impl HelpTemplate<'_, '_> {
                 "HelpTemplate::spec_vals: Found short flag aliases...{:?}",
                 a.get_all_short_flag_aliases().collect::<Vec<_>>()
             );
+            debug!(
+                "HelpTemplate::spec_vals: Found long flag aliases...{:?}",
+                a.get_all_long_flag_aliases().collect::<Vec<_>>()
+            );
             spec_vals.push(format!("{ctx}[aliases: {ctx:#}{all_als}{ctx}]{ctx:#}"));
         }
