diff --git a/packages/cli-harnesses/harness-fullstack-with-optional-tokio/Cargo.toml b/packages/cli-harnesses/harness-fullstack-with-optional-tokio/Cargo.toml
index 5b0d69a..e9ae65c 100644
--- a/packages/cli-harnesses/harness-fullstack-with-optional-tokio/Cargo.toml
+++ b/packages/cli-harnesses/harness-fullstack-with-optional-tokio/Cargo.toml
@@ -7,7 +7,7 @@ publish = false
 
 [dependencies]
 dioxus = { workspace = true, features = ["fullstack"] }
-serde = { workspace = true }
+serde = "1.0.219"
 tokio = { workspace = true, features = ["full"], optional = true }
 
 
@@ -16,4 +16,4 @@ default = []
 server = ["dioxus/server", "dep:tokio"]
 web = ["dioxus/web"]
 
-    
+    
\ No newline at end of file
