diff --git a/packages/renderer/src/validate-even-dimensions-with-codec.ts b/packages/renderer/src/validate-even-dimensions-with-codec.ts
index 430dce99ba..7ae96454c4 100644
--- a/packages/renderer/src/validate-even-dimensions-with-codec.ts
+++ b/packages/renderer/src/validate-even-dimensions-with-codec.ts
@@ -63,6 +63,24 @@ export const validateEvenDimensionsWithCodec = ({
 		actualHeight = Math.round(actualHeight);
 	}
 
+	// Be lenient with scale factors: If scaling results in an uneven dimension,
+	// we round down by 1px to get an even number (required by H264/H265).
+	if (scale !== 1 && actualWidth % 2 !== 0) {
+		Log.verbose(
+			{indent, logLevel},
+			`Rounding down width to an even number from ${actualWidth} to ${actualWidth - 1}`,
+		);
+		actualWidth -= 1;
+	}
+
+	if (scale !== 1 && actualHeight % 2 !== 0) {
+		Log.verbose(
+			{indent, logLevel},
+			`Rounding down height to an even number from ${actualHeight} to ${actualHeight - 1}`,
+		);
+		actualHeight -= 1;
+	}
+
 	const displayName = codec === 'h265' ? 'H265' : 'H264';
 
 	if (actualWidth % 2 !== 0) {
