diff --git a/src/function/cast/map_cast.cpp b/src/function/cast/map_cast.cpp
index f6c2d70..1320758 100644
--- a/src/function/cast/map_cast.cpp
+++ b/src/function/cast/map_cast.cpp
@@ -21,6 +21,32 @@ unique_ptr<BoundCastData> MapBoundCastData::BindMapToMapCast(BindCastInput &inpu
 	return make_uniq<MapBoundCastData>(std::move(key_cast), std::move(value_cast));
 }
 
+static bool MapToMapCast(Vector &source, Vector &result, idx_t count, CastParameters &parameters) {
+	bool all_succeeded = ListCast::ListToListCast(source, result, count, parameters);
+
+	auto valid_check = MapVector::CheckMapValidity(result, count);
+	if (valid_check == MapInvalidReason::VALID) {
+		return all_succeeded;
+	}
+	if (!parameters.error_message) {
+		MapVector::EvalMapInvalidReason(valid_check);
+	}
+
+	SelectionVector row_sel(1);
+	result.Flatten(count);
+	auto &result_validity = FlatVector::ValidityMutable(result);
+	for (idx_t row_idx = 0; row_idx < count; row_idx++) {
+		if (!result_validity.RowIsValid(row_idx)) {
+			continue;
+		}
+		row_sel.set_index(0, row_idx);
+		if (MapVector::CheckMapValidity(result, 1, row_sel) != MapInvalidReason::VALID) {
+			result_validity.SetInvalid(row_idx);
+		}
+	}
+	return false;
+}
+
 static bool MapToVarcharCast(Vector &source, Vector &result, idx_t count, CastParameters &parameters) {
 	auto constant = source.GetVectorType() == VectorType::CONSTANT_VECTOR;
 	auto varchar_type = LogicalType::MAP(LogicalType::VARCHAR, LogicalType::VARCHAR);
@@ -152,7 +178,7 @@ static bool MapToVarcharCast(Vector &source, Vector &result, idx_t count, CastPa
 BoundCastInfo DefaultCasts::MapCastSwitch(BindCastInput &input, const LogicalType &source, const LogicalType &target) {
 	switch (target.id()) {
 	case LogicalTypeId::MAP:
-		return BoundCastInfo(ListCast::ListToListCast, ListBoundCastData::BindListToListCast(input, source, target),
+		return BoundCastInfo(MapToMapCast, ListBoundCastData::BindListToListCast(input, source, target),
 		                     ListBoundCastData::InitListLocalState);
 	case LogicalTypeId::VARCHAR: {
 		auto varchar_type = LogicalType::MAP(LogicalType::VARCHAR, LogicalType::VARCHAR);
