diff --git a/src/parser/mod.rs b/src/parser/mod.rs
index 279d650..400d228 100644
--- a/src/parser/mod.rs
+++ b/src/parser/mod.rs
@@ -411,10 +411,12 @@ fn _indirect_object<'a>(
 
 pub fn header(input: ParserInput) -> Option<String> {
     strip_nom(map_res(
-        delimited(
+        preceded(
             tag(&b"%PDF-"[..]),
-            take_while(|c: u8| !b"\r\n".contains(&c)),
-            pair(eol, many0_count(comment)),
+            terminated(
+                take_while_m_n(3, 7, |c: u8| c.is_dec_digit() || c == b'.'),
+                pair(take_while(|c: u8| !b"\r\n".contains(&c)), pair(eol, many0_count(comment))),
+            ),
         ),
         |v: ParserInput| str::from_utf8(&v).map(Into::into),
     ).parse(input))
