diff --git a/conan/tools/env/environment.py b/conan/tools/env/environment.py
index c42d9f060..99bc3f14f 100644
--- a/conan/tools/env/environment.py
+++ b/conan/tools/env/environment.py
@@ -253,14 +253,15 @@ class Environment:
         """
         self._values.setdefault(name, _EnvValue(name, _EnvVarPlaceHolder)).append(value, separator)
 
-    def append_path(self, name, value):
+    def append_path(self, name, value, separator=None):
         """
         Similar to "append" method but indicating that the variable is a filesystem path. It will automatically handle the path separators depending on the operating system.
 
         :param name: Name of the variable to append a new value
         :param value: New value
+        :param separator: The character to separate the appended value with the previous value
         """
-        self._values.setdefault(name, _EnvValue(name, _EnvVarPlaceHolder, path=True)).append(value)
+        self._values.setdefault(name, _EnvValue(name, _EnvVarPlaceHolder, path=True)).append(value, separator)
 
     def prepend(self, name, value, separator=None):
         """
@@ -272,14 +273,15 @@ class Environment:
         """
         self._values.setdefault(name, _EnvValue(name, _EnvVarPlaceHolder)).prepend(value, separator)
 
-    def prepend_path(self, name, value):
+    def prepend_path(self, name, value, separator=None):
         """
         Similar to "prepend" method but indicating that the variable is a filesystem path. It will automatically handle the path separators depending on the operating system.
 
         :param name: Name of the variable to prepend a new value
         :param value: New value
+        :param separator: The character to separate the prepended value with the previous value
         """
-        self._values.setdefault(name, _EnvValue(name, _EnvVarPlaceHolder, path=True)).prepend(value)
+        self._values.setdefault(name, _EnvValue(name, _EnvVarPlaceHolder, path=True)).prepend(value, separator)
 
     def remove(self, name, value):
         """
@@ -645,11 +647,16 @@ class ProfileEnvironment:
                 if method == "unset":
                     env.unset(name)
                 else:
-                    if value.strip().startswith("(path)"):
-                        value = value.strip()
+                    value = value.strip()
+                    separator = None
+                    if value.startswith("(path)"):
                         value = value[6:]
                         method = method + "_path"
-                    getattr(env, method)(name, value)
+                    elif value.startswith("(win_path)"):
+                        value = value[10:]
+                        method = method + "_path"
+                        separator = ";" if method == "append_path" or method == "prepend_path" else None
+                    getattr(env, method)(name, value, separator)
 
                 existing = result._environments.get(pattern)
                 if existing is None:
