diff --git a/packages/survey-core/src/panel.ts b/packages/survey-core/src/panel.ts
index 4a7c94c..b00b161 100644
--- a/packages/survey-core/src/panel.ts
+++ b/packages/survey-core/src/panel.ts
@@ -993,7 +993,7 @@ export class PanelModelBase extends SurveyElement<Question>
   }
 
   private validateRequired(context: ValidationContext, errors: Array<SurveyError>): void {
-    if (!this.isRequired) return;
+    if (!this.isRequired || (this.isReadOnly && !settings.readOnly.enableValidation)) return;
     var visQuestions = <Array<any>>[];
     this.addQuestionsToList(visQuestions, true);
     if (visQuestions.length == 0) return;
