diff --git a/packages/webcodecs/src/reencode-video-track.ts b/packages/webcodecs/src/reencode-video-track.ts
index 16ff9fd0de..29048197fb 100644
--- a/packages/webcodecs/src/reencode-video-track.ts
+++ b/packages/webcodecs/src/reencode-video-track.ts
@@ -52,7 +52,7 @@ export const reencodeVideoTrack = async ({
 		);
 	}
 
-	const rotation = (videoOperation.rotate ?? rotate) - track.rotation;
+	const rotation = videoOperation.rotate ?? rotate;
 
 	const {height: newHeight, width: newWidth} =
 		calculateNewDimensionsFromRotateAndScale({
