diff --git a/packages/router-plugin/src/core/code-splitter/compilers.ts b/packages/router-plugin/src/core/code-splitter/compilers.ts
index 650c8d3a1..3d2626d0b 100644
--- a/packages/router-plugin/src/core/code-splitter/compilers.ts
+++ b/packages/router-plugin/src/core/code-splitter/compilers.ts
@@ -209,12 +209,14 @@ export function compileCodeSplitReferenceRoute(
                         return
                       }
 
-                      // Exit early if the value is undefined
-                      // Since we don't need to run an import just to get the value of `undefined`
-                      // This is useful for cases like: `createFileRoute('/')({ component: undefined })`
+                      // Exit early if the value is undefined, null, or a boolean
+                      // Since we don't need to run an import just to get the value of `undefined`, `null`, or `false`
+                      // This is useful for cases like: `createFileRoute('/')({ component: undefined })` or `errorComponent: null` or `errorComponent: false`
                       if (
-                        t.isIdentifier(prop.value) &&
-                        prop.value.name === 'undefined'
+                        (t.isIdentifier(prop.value) &&
+                          prop.value.name === 'undefined') ||
+                        t.isNullLiteral(prop.value) ||
+                        t.isBooleanLiteral(prop.value)
                       ) {
                         return
                       }
@@ -470,11 +472,15 @@ export function compileCodeSplitVirtualRoute(
 
                       const value = prop.value
 
-                      // If the value for the `key` is `undefined`, then we don't need to include it
+                      // If the value for the `key` is `undefined`, `null`, or a boolean, then we don't need to include it
                       // in the split file, so we can just return, since it will kept in-place in the
                       // reference file
-                      // This is useful for cases like: `createFileRoute('/')({ component: undefined })`
-                      if (t.isIdentifier(value) && value.name === 'undefined') {
+                      // This is useful for cases like: `createFileRoute('/')({ component: undefined })` or `errorComponent: null` or `errorComponent: false`
+                      if (
+                        (t.isIdentifier(value) && value.name === 'undefined') ||
+                        t.isNullLiteral(value) ||
+                        t.isBooleanLiteral(value)
+                      ) {
                         return
                       }
