diff --git a/Zend/zend_compile.c b/Zend/zend_compile.c
index 0a7ab7da1bd..46e7f49ba0a 100644
--- a/Zend/zend_compile.c
+++ b/Zend/zend_compile.c
@@ -7792,6 +7792,10 @@ static void zend_compile_params(zend_ast *ast, zend_ast *return_type_ast, uint32
 				ZSTR_VAL(name));
 		}
 
+		if (zend_string_equals_literal(name, "http_response_header")) {
+			CG(context).has_assigned_to_http_response_header = true;
+		}
+
 		var_node.op_type = IS_CV;
 		var_node.u.op.var = lookup_cv(name);
