diff --git a/core/src/main/java/org/web3j/protocol/core/filters/BlockFilter.java b/core/src/main/java/org/web3j/protocol/core/filters/BlockFilter.java
index c411b63..c3ead7a 100644
--- a/core/src/main/java/org/web3j/protocol/core/filters/BlockFilter.java
+++ b/core/src/main/java/org/web3j/protocol/core/filters/BlockFilter.java
@@ -35,8 +35,8 @@ public class BlockFilter extends Filter<String> {
     }
 
     @Override
-    protected void process(List<EthLog.LogResult> logResults) {
-        for (EthLog.LogResult logResult : logResults) {
+    protected void process(List<EthLog.LogResult<?>> logResults) {
+        for (EthLog.LogResult<?> logResult : logResults) {
             if (logResult instanceof EthLog.Hash) {
                 String blockHash = ((EthLog.Hash) logResult).get();
                 callback.onEvent(blockHash);
diff --git a/core/src/main/java/org/web3j/protocol/core/filters/BlocksFilter.java b/core/src/main/java/org/web3j/protocol/core/filters/BlocksFilter.java
index 6fc51bc..ac4a648 100644
--- a/core/src/main/java/org/web3j/protocol/core/filters/BlocksFilter.java
+++ b/core/src/main/java/org/web3j/protocol/core/filters/BlocksFilter.java
@@ -37,10 +37,10 @@ public class BlocksFilter extends Filter<List<String>> {
     }
 
     @Override
-    protected void process(List<LogResult> logResults) {
+    protected void process(List<LogResult<?>> logResults) {
         List<String> blockHashes = new ArrayList<>(logResults.size());
 
-        for (EthLog.LogResult logResult : logResults) {
+        for (EthLog.LogResult<?> logResult : logResults) {
             if (!(logResult instanceof EthLog.Hash)) {
                 throw new FilterException(
                         "Unexpected result type: " + logResult.get() + ", required Hash");
diff --git a/core/src/main/java/org/web3j/protocol/core/filters/Filter.java b/core/src/main/java/org/web3j/protocol/core/filters/Filter.java
index d083caf..1ceb279 100644
--- a/core/src/main/java/org/web3j/protocol/core/filters/Filter.java
+++ b/core/src/main/java/org/web3j/protocol/core/filters/Filter.java
@@ -157,7 +157,7 @@ public abstract class Filter<T> {
 
     protected abstract EthFilter sendRequest() throws IOException;
 
-    protected abstract void process(List<EthLog.LogResult> logResults);
+    protected abstract void process(List<EthLog.LogResult<?>> logResults);
 
     private void reinstallFilter() {
         log.warn(
diff --git a/core/src/main/java/org/web3j/protocol/core/filters/LogFilter.java b/core/src/main/java/org/web3j/protocol/core/filters/LogFilter.java
index 417e7ce..550cd2d 100644
--- a/core/src/main/java/org/web3j/protocol/core/filters/LogFilter.java
+++ b/core/src/main/java/org/web3j/protocol/core/filters/LogFilter.java
@@ -42,8 +42,8 @@ public class LogFilter extends Filter<Log> {
     }
 
     @Override
-    protected void process(List<EthLog.LogResult> logResults) {
-        for (EthLog.LogResult logResult : logResults) {
+    protected void process(List<EthLog.LogResult<?>> logResults) {
+        for (EthLog.LogResult<?> logResult : logResults) {
             if (logResult instanceof EthLog.LogObject) {
                 Log log = ((EthLog.LogObject) logResult).get();
                 callback.onEvent(log);
diff --git a/core/src/main/java/org/web3j/protocol/core/filters/LogsFilter.java b/core/src/main/java/org/web3j/protocol/core/filters/LogsFilter.java
index b75ca88..4620dff 100644
--- a/core/src/main/java/org/web3j/protocol/core/filters/LogsFilter.java
+++ b/core/src/main/java/org/web3j/protocol/core/filters/LogsFilter.java
@@ -44,10 +44,10 @@ public class LogsFilter extends Filter<List<Log>> {
     }
 
     @Override
-    protected void process(List<LogResult> logResults) {
+    protected void process(List<LogResult<?>> logResults) {
         List<Log> logs = new ArrayList<>(logResults.size());
 
-        for (EthLog.LogResult logResult : logResults) {
+        for (EthLog.LogResult<?> logResult : logResults) {
             if (!(logResult instanceof EthLog.LogObject)) {
                 throw new FilterException(
                         "Unexpected result type: " + logResult.get() + " required LogObject");
diff --git a/core/src/main/java/org/web3j/protocol/core/filters/PendingTransactionFilter.java b/core/src/main/java/org/web3j/protocol/core/filters/PendingTransactionFilter.java
index 8dcccbb..33ba5d1 100644
--- a/core/src/main/java/org/web3j/protocol/core/filters/PendingTransactionFilter.java
+++ b/core/src/main/java/org/web3j/protocol/core/filters/PendingTransactionFilter.java
@@ -35,8 +35,8 @@ public class PendingTransactionFilter extends Filter<String> {
     }
 
     @Override
-    protected void process(List<EthLog.LogResult> logResults) {
-        for (EthLog.LogResult logResult : logResults) {
+    protected void process(List<EthLog.LogResult<?>> logResults) {
+        for (EthLog.LogResult<?> logResult : logResults) {
             if (logResult instanceof EthLog.Hash) {
                 String transactionHash = ((EthLog.Hash) logResult).get();
                 callback.onEvent(transactionHash);
diff --git a/core/src/main/java/org/web3j/protocol/core/filters/PendingTransactionsFilter.java b/core/src/main/java/org/web3j/protocol/core/filters/PendingTransactionsFilter.java
index bde8028..6f40572 100644
--- a/core/src/main/java/org/web3j/protocol/core/filters/PendingTransactionsFilter.java
+++ b/core/src/main/java/org/web3j/protocol/core/filters/PendingTransactionsFilter.java
@@ -36,10 +36,10 @@ public class PendingTransactionsFilter extends Filter<List<String>> {
     }
 
     @Override
-    protected void process(List<EthLog.LogResult> logResults) {
+    protected void process(List<EthLog.LogResult<?>> logResults) {
         List<String> logs = new ArrayList<>(logResults.size());
 
-        for (EthLog.LogResult logResult : logResults) {
+        for (EthLog.LogResult<?> logResult : logResults) {
             if (!(logResult instanceof EthLog.Hash)) {
                 throw new FilterException(
                         "Unexpected result type: " + logResult.get() + ", required Hash");
diff --git a/core/src/main/java/org/web3j/protocol/core/methods/response/EthLog.java b/core/src/main/java/org/web3j/protocol/core/methods/response/EthLog.java
index c80ca90..f350e72 100644
--- a/core/src/main/java/org/web3j/protocol/core/methods/response/EthLog.java
+++ b/core/src/main/java/org/web3j/protocol/core/methods/response/EthLog.java
@@ -39,16 +39,17 @@ import org.web3j.protocol.core.Response;
  * <p>See <a href="https://github.com/ethereum/wiki/wiki/JSON-RPC#eth_getfilterchanges">docs</a> for
  * further details.
  */
-public class EthLog extends Response<List<EthLog.LogResult>> {
+public class EthLog extends Response<List<EthLog.LogResult<?>>> {
 
     @Override
     @JsonDeserialize(using = LogResultDeserialiser.class)
-    public void setResult(List<LogResult> result) {
+    public void setResult(List<LogResult<?>> result) {
         super.setResult(result);
     }
 
-    public List<LogResult> getLogs() {
-        return getResult();
+    @SuppressWarnings("unchecked")
+    public <T extends LogResult<?>> List<T> getLogs() {
+        return (List<T>) getResult();
     }
 
     public interface LogResult<T> {
@@ -127,16 +128,16 @@ public class EthLog extends Response<List<EthLog.LogResult>> {
         }
     }
 
-    public static class LogResultDeserialiser extends JsonDeserializer<List<LogResult>> {
+    public static class LogResultDeserialiser extends JsonDeserializer<List<LogResult<?>>> {
 
         private ObjectReader objectReader = ObjectMapperFactory.getObjectReader();
 
         @Override
-        public List<LogResult> deserialize(
+        public List<LogResult<?>> deserialize(
                 JsonParser jsonParser, DeserializationContext deserializationContext)
                 throws IOException {
 
-            List<LogResult> logResults = new ArrayList<>();
+            List<LogResult<?>> logResults = new ArrayList<>();
             JsonToken nextToken = jsonParser.nextToken();
 
             if (nextToken == JsonToken.START_OBJECT) {
