diff --git a/harper-wasm/src/lib.rs b/harper-wasm/src/lib.rs
index 15d6cd1..ae3c590 100644
--- a/harper-wasm/src/lib.rs
+++ b/harper-wasm/src/lib.rs
@@ -649,6 +649,11 @@ impl Lint {
         self.inner.suggestions.len()
     }
 
+    /// Get the lint priority. Lower values are more important.
+    pub fn priority(&self) -> u8 {
+        self.inner.priority
+    }
+
     /// Get an array of any suggestions that may resolve the issue.
     pub fn suggestions(&self) -> Vec<Suggestion> {
         self.inner
diff --git a/packages/lint-framework/src/lint/LintFramework.ts b/packages/lint-framework/src/lint/LintFramework.ts
index 3652360..c709f84 100644
--- a/packages/lint-framework/src/lint/LintFramework.ts
+++ b/packages/lint-framework/src/lint/LintFramework.ts
@@ -306,13 +306,15 @@ export default class LintFramework {
 		requestAnimationFrame(() => {
 			const boxes = this.lastLints.flatMap(({ target, lints }) =>
 				target
-					? Object.entries(lints).flatMap(([ruleName, ls]) =>
+					? Object.entries(lints)
+						.flatMap(([ruleName, ls]) =>
 							ls.flatMap((l) =>
 								computeLintBoxes(target, l as any, ruleName, {
 									ignoreLint: this.actions.ignoreLint,
 								}),
 							),
 						)
+						.sort((a, b) => a.lint.priority - b.lint.priority)
 					: [],
 			);
 			this.lastLintBoxes = boxes;
diff --git a/packages/lint-framework/src/lint/unpackLint.ts b/packages/lint-framework/src/lint/unpackLint.ts
index 71e47c8..0ea3f46 100644
--- a/packages/lint-framework/src/lint/unpackLint.ts
+++ b/packages/lint-framework/src/lint/unpackLint.ts
@@ -15,6 +15,7 @@ export type UnpackedLint = {
 	suggestions: UnpackedSuggestion[];
 	context_hash: string;
 	source: string;
+	priority: number;
 };
 
 export type UnpackedLintGroups = Record<string, UnpackedLint[]>;
@@ -43,6 +44,7 @@ export default async function unpackLint(
 		}),
 		context_hash: (await linter.contextHash(text, lint)).toString(),
 		source: text,
+		priority: lint.priority(),
 	};
 }
