diff --git a/client/resource_group/controller/controller.go b/client/resource_group/controller/controller.go
index be3094a1a..5db183b9f 100644
--- a/client/resource_group/controller/controller.go
+++ b/client/resource_group/controller/controller.go
@@ -1293,6 +1293,7 @@ func initCounterNotify(counter *tokenCounter) {
 func (gc *groupCostController) collectRequestAndConsumption(selectTyp selectType) *rmpb.TokenBucketRequest {
 	req := &rmpb.TokenBucketRequest{
 		ResourceGroupName: gc.name,
+		KeyspaceId:        gc.getMeta().GetKeyspaceId(),
 	}
 	// collect request resource
 	selected := gc.run.requestInProgress
