diff --git a/src/xdocs-examples/resources/com/puppycrawl/tools/checkstyle/checks/indentation/indentation/Example4.java b/src/xdocs-examples/resources/com/puppycrawl/tools/checkstyle/checks/indentation/indentation/Example4.java
index cc54314..c51ae60 100644
--- a/src/xdocs-examples/resources/com/puppycrawl/tools/checkstyle/checks/indentation/indentation/Example4.java
+++ b/src/xdocs-examples/resources/com/puppycrawl/tools/checkstyle/checks/indentation/indentation/Example4.java
@@ -1,7 +1,9 @@
 /*xml
 <module name="Checker">
   <module name="TreeWalker">
-    <module name="Indentation"/>
+    <module name="Indentation">
+      <property name="basicOffset" value="4"/>
+    </module>
   </module>
 </module>
 */
@@ -10,15 +12,15 @@ package com.puppycrawl.tools.checkstyle.checks.indentation.indentation;
 
 // xdoc section -- start
 class Example4 {
-    String field = "example";                   // basicOffset
-    int[] values = {                            // basicOffset
+    String field = "example";                  // basicOffset
+    int[] values = {                           // basicOffset
         10,
         20,
         30
     };
 
     void processValues() throws Exception {
-        handleValue("Test String", 42);
+        handleValue("Test String", 42);          // basicOffset
     }
 
     void handleValue(String aFooString,
@@ -46,8 +48,8 @@ class Example4 {
 
     void demonstrateSwitch() throws Exception {
         switch (field) {
-            case "EXAMPLE": processValues();                            // caseIndent
-            case "COMPLETED": handleValue("Completed Case", 456);       // caseIndent
+            case "EXAMPLE": processValues();                        // caseIndent
+            case "COMPLETED": handleValue("Completed Case", 456);   // caseIndent
         }
     }
 }
diff --git a/src/xdocs-examples/resources/com/puppycrawl/tools/checkstyle/checks/indentation/indentation/Example9.java b/src/xdocs-examples/resources/com/puppycrawl/tools/checkstyle/checks/indentation/indentation/Example9.java
index a60d872..de665ba 100644
--- a/src/xdocs-examples/resources/com/puppycrawl/tools/checkstyle/checks/indentation/indentation/Example9.java
+++ b/src/xdocs-examples/resources/com/puppycrawl/tools/checkstyle/checks/indentation/indentation/Example9.java
@@ -11,20 +11,21 @@
 package com.puppycrawl.tools.checkstyle.checks.indentation.indentation;
 
 // xdoc section -- start
-class Example9 {
-    String field = "example";                  // basicOffset
-    int[] values = {                           // basicOffset
+class Example9
+  {                                        // braceAdjustment
+    String field = "example";              // basicOffset
+    int[] values = {                       // basicOffset
         10,
         20,
         30
     };
 
     void processValues() throws Exception {
-        handleValue("Test String", 42);          // basicOffset
+        handleValue("Test String", 42);        // basicOffset
     }
 
     void handleValue(String aFooString,
-                     int aFooInt) {             // indent:8 ; expected: > 4;
+                     int aFooInt) {
 
         boolean cond1,cond2,cond3,cond4,cond5,cond6;
         cond1=cond2=cond3=cond4=cond5=cond6=false;
@@ -36,11 +37,11 @@ class Example9 {
         }
 
         if ((cond1 && cond2)
-                || (cond3 && cond4)          // ok, lineWrappingIndentation
-                || !(cond5 && cond6)) {      // ok, lineWrappingIndentation
+                || (cond3 && cond4)
+                || !(cond5 && cond6)) {
             field.toUpperCase()
-                 .concat(" TASK")             // ok, lineWrappingIndentation
-                 .chars().forEach(c -> {      // ok, lineWrappingIndentation
+                 .concat(" TASK")
+                 .chars().forEach(c -> {
                      System.out.println((char) c);
                  });
         }
@@ -48,9 +49,9 @@ class Example9 {
 
     void demonstrateSwitch() throws Exception {
         switch (field) {
-            case "EXAMPLE": processValues();                        // caseIndent
-            case "COMPLETED": handleValue("Completed Case", 456);   // caseIndent
+            case "EXAMPLE": processValues();
+            case "COMPLETED": handleValue("Completed Case", 456);
         }
     }
-}
+  }
 // xdoc section -- end
