diff --git a/gitlab4j-models/src/main/java/org/gitlab4j/api/models/Job.java b/gitlab4j-models/src/main/java/org/gitlab4j/api/models/Job.java
index 151321d9..0211493e 100644
--- a/gitlab4j-models/src/main/java/org/gitlab4j/api/models/Job.java
+++ b/gitlab4j-models/src/main/java/org/gitlab4j/api/models/Job.java
@@ -25,6 +25,7 @@ public class Job implements Serializable {
     private ArtifactsFile artifactsFile;
     private List<Artifact> artifacts;
     private Boolean tag;
+    private List<String> tagList;
     private String webUrl;
     private String stage;
     private JobStatus status;
@@ -141,6 +142,14 @@ public class Job implements Serializable {
         this.tag = tag;
     }
 
+    public List<String> getTagList() {
+        return tagList;
+    }
+
+    public void setTagList(List<String> tagList) {
+        this.tagList = tagList;
+    }
+
     public String getWebUrl() {
         return webUrl;
     }
@@ -323,6 +332,11 @@ public class Job implements Serializable {
         return this;
     }
 
+    public Job withTagList(List<String> tagList) {
+        this.tagList = tagList;
+        return this;
+    }
+
     public Job withStage(String stage) {
         this.stage = stage;
         return this;
