diff --git a/src/MudBlazor/Components/Collapse/MudCollapse.razor.cs b/src/MudBlazor/Components/Collapse/MudCollapse.razor.cs
index 612a688..b680bf8 100644
--- a/src/MudBlazor/Components/Collapse/MudCollapse.razor.cs
+++ b/src/MudBlazor/Components/Collapse/MudCollapse.razor.cs
@@ -101,14 +101,16 @@ namespace MudBlazor
             {
                 _state = CollapseState.Entered;
                 StateHasChanged();
+                return OnAnimationEnd.InvokeAsync(_expandedState.Value);
             }
             else if (_state == CollapseState.Exiting)
             {
                 _state = CollapseState.Exited;
                 StateHasChanged();
+                return OnAnimationEnd.InvokeAsync(_expandedState.Value);
             }
 
-            return OnAnimationEnd.InvokeAsync(_expandedState.Value);
+            return Task.CompletedTask;
         }
     }
 }
