diff --git a/src/I18nProvider.tsx b/src/I18nProvider.tsx
index 8230eb3..336bc79 100644
--- a/src/I18nProvider.tsx
+++ b/src/I18nProvider.tsx
@@ -2,7 +2,6 @@ import React, { createContext, useContext } from 'react'
 import { useRouter } from 'next/router'
 import I18nContext from './context'
 import transCore from './transCore'
-import useTranslation from './useTranslation'
 import { I18nDictionary, I18nProviderProps } from '.'
 
 export const InternalContext = createContext({ ns: {}, config: {} })
@@ -13,7 +12,7 @@ export default function I18nProvider({
   children,
   config: newConfig = {},
 }: I18nProviderProps) {
-  const { lang: parentLang } = useTranslation()
+  const { lang: parentLang } = useContext(I18nContext)
   const { locale, defaultLocale } = useRouter() || {}
   const internal = useContext(InternalContext)
   const allNamespaces: Record<string, I18nDictionary> = {
diff --git a/src/useTranslation.tsx b/src/useTranslation.tsx
index 33f98ac..8f0df57 100644
--- a/src/useTranslation.tsx
+++ b/src/useTranslation.tsx
@@ -16,7 +16,16 @@ function useTranslationInPages(defaultNS?: string): I18n {
 }
 
 export default function useTranslation(defaultNS?: string): I18n {
+  const ctx = useContext(I18nContext)
   const appDir = globalThis.__NEXT_TRANSLATE__
-  const useT = appDir?.config ? createTranslation : useTranslationInPages
-  return useT(defaultNS)
+
+  if (appDir?.config && !ctx.lang) return createTranslation(defaultNS)
+
+  return useMemo(
+    () => ({
+      ...ctx,
+      t: wrapTWithDefaultNs(ctx.t, defaultNS),
+    }),
+    [ctx, defaultNS]
+  )
 }
