diff --git a/dev/io.openliberty.microprofile.openapi.2.0.internal/src/io/openliberty/microprofile/openapi20/internal/merge/MergeProcessorImpl.java b/dev/io.openliberty.microprofile.openapi.2.0.internal/src/io/openliberty/microprofile/openapi20/internal/merge/MergeProcessorImpl.java
index 3a74cb905f4..b3eb58bbdc0 100644
--- a/dev/io.openliberty.microprofile.openapi.2.0.internal/src/io/openliberty/microprofile/openapi20/internal/merge/MergeProcessorImpl.java
+++ b/dev/io.openliberty.microprofile.openapi.2.0.internal/src/io/openliberty/microprofile/openapi20/internal/merge/MergeProcessorImpl.java
@@ -20,6 +20,7 @@ import java.io.IOException;
 import java.util.ArrayList;
 import java.util.Collections;
 import java.util.HashMap;
+import java.util.LinkedHashMap;
 import java.util.List;
 import java.util.Map;
 import java.util.Map.Entry;
@@ -386,7 +387,7 @@ public class MergeProcessorImpl implements MergeProcessor {
                 return null;
             }
 
-            HashMap<String, T> newMap = new HashMap<>();
+            Map<String, T> newMap = new LinkedHashMap<>();
             for (Entry<String, T> entry : componentMap.entrySet()) {
                 String key = entry.getKey();
                 newMap.put(documentNameProcessor.createUniqueName(nameType, key, entry.getValue()), entry.getValue());
@@ -499,7 +500,7 @@ public class MergeProcessorImpl implements MergeProcessor {
                 removeContextRoot(server, contextRoot);
             }
 
-            Map<String, PathItem> newPathItems = new HashMap<>();
+            Map<String, PathItem> newPathItems = new LinkedHashMap<>();
             for (Entry<String, PathItem> entry : pathItems.entrySet()) {
                 PathItem pathItem = entry.getValue();
 
diff --git a/dev/io.openliberty.microprofile.openapi.2.0.internal/src/io/openliberty/microprofile/openapi20/internal/merge/RenameReferenceVisitor.java b/dev/io.openliberty.microprofile.openapi.2.0.internal/src/io/openliberty/microprofile/openapi20/internal/merge/RenameReferenceVisitor.java
index 15b1ffd8989..64f5e42ca49 100644
--- a/dev/io.openliberty.microprofile.openapi.2.0.internal/src/io/openliberty/microprofile/openapi20/internal/merge/RenameReferenceVisitor.java
+++ b/dev/io.openliberty.microprofile.openapi.2.0.internal/src/io/openliberty/microprofile/openapi20/internal/merge/RenameReferenceVisitor.java
@@ -14,6 +14,7 @@ package io.openliberty.microprofile.openapi20.internal.merge;
 
 import java.util.ArrayList;
 import java.util.HashMap;
+import java.util.LinkedHashMap;
 import java.util.List;
 import java.util.Map;
 import java.util.Map.Entry;
@@ -193,7 +194,7 @@ public class RenameReferenceVisitor implements OpenAPIModelVisitor {
     public SecurityRequirement visitSecurityRequirement(Context context, SecurityRequirement secReq) {
         Map<String, List<String>> schemes = secReq.getSchemes();
         if (schemes != null) {
-            Map<String, List<String>> newSchemes = new HashMap<>();
+            Map<String, List<String>> newSchemes = new LinkedHashMap<>();
             for (Entry<String, List<String>> entry : schemes.entrySet()) {
                 String newName = documentNameProcessor.lookupName(NameType.SECURITY_SCHEMES, entry.getKey());
                 newSchemes.put(newName, entry.getValue());
