diff --git a/src/BootstrapBlazor/Components/Radio/RadioList.razor.cs b/src/BootstrapBlazor/Components/Radio/RadioList.razor.cs
index 899f1d494..8c6416b76 100644
--- a/src/BootstrapBlazor/Components/Radio/RadioList.razor.cs
+++ b/src/BootstrapBlazor/Components/Radio/RadioList.razor.cs
@@ -31,6 +31,12 @@ public partial class RadioList<TValue>
     [Parameter]
     public bool AutoSelectFirstWhenValueIsNull { get; set; } = true;
 
+    /// <summary>
+    /// 获得/设置 是否为 Outline 样式 默认 false
+    /// </summary>
+    [Parameter]
+    public bool IsOutline { get; set; }
+
     private string? GroupName => Id;
 
     private string? ClassString => CssBuilder.Default("radio-list form-control")
@@ -46,7 +52,9 @@ public partial class RadioList<TValue>
         .Build();
 
     private string? GetButtonItemClassString(SelectedItem item) => CssBuilder.Default("btn")
-        .AddClass($"active bg-{Color.ToDescriptionString()}", CurrentValueAsString == item.Value)
+        .AddClass($"btn-outline-{Color.ToDescriptionString()}", IsOutline)
+        .AddClass($"active bg-{Color.ToDescriptionString()}", !IsOutline && CurrentValueAsString == item.Value)
+        .AddClass($"active", IsOutline && CurrentValueAsString == item.Value)
         .Build();
 
     /// <summary>
diff --git a/src/BootstrapBlazor/Components/Radio/RadioListGeneric.razor.cs b/src/BootstrapBlazor/Components/Radio/RadioListGeneric.razor.cs
index 833d8ddc8..623404a38 100644
--- a/src/BootstrapBlazor/Components/Radio/RadioListGeneric.razor.cs
+++ b/src/BootstrapBlazor/Components/Radio/RadioListGeneric.razor.cs
@@ -35,6 +35,12 @@ public partial class RadioListGeneric<TValue> : IModelEqualityComparer<TValue>
     [Parameter]
     public bool IsButton { get; set; }
 
+    /// <summary>
+    /// 获得/设置 是否为 Outline 样式 默认 false
+    /// </summary>
+    [Parameter]
+    public bool IsOutline { get; set; }
+
     /// <summary>
     /// 获得/设置 是否显示边框 默认为 true
     /// </summary>
@@ -102,7 +108,9 @@ public partial class RadioListGeneric<TValue> : IModelEqualityComparer<TValue>
         .Build();
 
     private string? GetButtonItemClassString(SelectedItem<TValue> item) => CssBuilder.Default("btn")
-        .AddClass($"active bg-{Color.ToDescriptionString()}", Equals(Value, item.Value))
+        .AddClass($"btn-outline-{Color.ToDescriptionString()}", IsOutline)
+        .AddClass($"active bg-{Color.ToDescriptionString()}", !IsOutline && Equals(Value, item.Value))
+        .AddClass($"active", IsOutline && Equals(Value, item.Value))
         .Build();
 
     /// <summary>
