diff --git a/pkg/controller/git/work_tree.go b/pkg/controller/git/work_tree.go
index 6c5349e..90e964c 100644
--- a/pkg/controller/git/work_tree.go
+++ b/pkg/controller/git/work_tree.go
@@ -416,7 +416,9 @@ func (w *workTree) Fetch(opts *FetchOptions) error {
 }
 
 func (w *workTree) GetDiffPathsForCommitID(commitID string) ([]string, error) {
-	resBytes, err := libExec.Exec(w.buildGitCommand("show", "--pretty=", "--name-only", commitID))
+	resBytes, err := libExec.Exec(w.buildGitCommand(
+		"show", "--pretty=", "--name-only", "--first-parent", commitID,
+	))
 	if err != nil {
 		return nil, fmt.Errorf("error getting diff paths for commit %q: %w", commitID, err)
 	}
