diff --git a/multiqc/core/special_case_modules/custom_content.py b/multiqc/core/special_case_modules/custom_content.py
index 23ad327b..1146505f 100644
--- a/multiqc/core/special_case_modules/custom_content.py
+++ b/multiqc/core/special_case_modules/custom_content.py
@@ -429,7 +429,9 @@ class MultiqcModule(BaseMultiqcModule):
         if "parent_name" in cc_dict.config:
             assert isinstance(cc_dict.config["parent_name"], str)
             modname = cc_dict.config["parent_name"]
-            mod_info = cc_dict.config.get("parent_description")
+            # If this section is grouped under a parent, prefer parent_description when provided,
+            # but fall back to the child section description if not.
+            mod_info = cc_dict.config.get("parent_description", mod_info)
         elif "section_name" in cc_dict.config:
             assert isinstance(cc_dict.config["section_name"], str)
             modname = cc_dict.config["section_name"]
