diff --git a/pvlib/pvsystem.py b/pvlib/pvsystem.py
index f99f027..d3aa12c 100644
--- a/pvlib/pvsystem.py
+++ b/pvlib/pvsystem.py
@@ -29,11 +29,10 @@ import pvlib.tools as tools
 # a dict of required parameter names for each DC power model
 _DC_MODEL_PARAMS = {
     'sapm': {
-        'A0', 'A1', 'A2', 'A3', 'A4', 'B0', 'B1', 'B2', 'B3',
-        'B4', 'B5', 'C0', 'C1', 'C2', 'C3', 'C4', 'C5', 'C6',
+        'C0', 'C1', 'C2', 'C3', 'C4', 'C5', 'C6',
         'C7', 'Isco', 'Impo', 'Voco', 'Vmpo', 'Aisc', 'Aimp', 'Bvoco',
         'Mbvoc', 'Bvmpo', 'Mbvmp', 'N', 'Cells_in_Series',
-        'IXO', 'IXXO', 'FD'},
+        'IXO', 'IXXO'},
     'desoto': {
         'alpha_sc', 'a_ref', 'I_L_ref', 'I_o_ref',
         'R_sh_ref', 'R_s'},
diff --git a/tests/test_modelchain.py b/tests/test_modelchain.py
index 836fd3c..a8cf8d6 100644
--- a/tests/test_modelchain.py
+++ b/tests/test_modelchain.py
@@ -1730,7 +1730,7 @@ def test_invalid_dc_model_params(sapm_dc_snl_ac_system, cec_dc_snl_ac_system,
               'aoi_model': 'no_loss', 'spectral_model': 'no_loss',
               'temperature_model': 'sapm', 'losses_model': 'no_loss'}
     for array in sapm_dc_snl_ac_system.arrays:
-        array.module_parameters.pop('A0')  # remove a parameter
+        array.module_parameters.pop('C0')  # remove a required parameter
     with pytest.raises(ValueError):
         ModelChain(sapm_dc_snl_ac_system, location, **kwargs)
 
