diff --git a/lib/get-html-headers.js b/lib/get-html-headers.js
index 0c1077b..b661692 100644
--- a/lib/get-html-headers.js
+++ b/lib/get-html-headers.js
@@ -3,13 +3,18 @@
 var htmlparser = require("htmlparser2"),
   md = require("@textlint/markdown-to-ast");
 
+function escapeRegExp(string) {
+  if (string == null) return '';
+  return string.replace(/[.*+?^${}()|[\]\\]/g, '\\$&');
+}
+
 function addLinenos(lines, headers) {
   var current = 0, line;
 
   return headers.map(function (x) {
     for (var lineno = current; lineno < lines.length; lineno++) {
       line = lines[lineno];
-      if (new RegExp(x.text[0]).test(line)) {
+      if (new RegExp(escapeRegExp(x.text[0])).test(line)) {
         current = lineno;
         x.line = lineno;
         x.name = x.text.join("");
