diff --git a/pkg/internal/source/kind.go b/pkg/internal/source/kind.go
index a28aeb1..8ae80a8 100644
--- a/pkg/internal/source/kind.go
+++ b/pkg/internal/source/kind.go
@@ -116,10 +116,13 @@ func (ks *Kind[object, request]) Start(ctx context.Context, queue workqueue.Type
 }
 
 func (ks *Kind[object, request]) String() string {
-	if !isNil(ks.Type) {
-		return fmt.Sprintf("kind source: %T", ks.Type)
+	if isNil(ks.Type) {
+		return "kind source: unknown type"
+	}
+	if gvk := ks.Type.GetObjectKind().GroupVersionKind(); !gvk.Empty() {
+		return fmt.Sprintf("kind source: %T %s", ks.Type, gvk)
 	}
-	return "kind source: unknown type"
+	return fmt.Sprintf("kind source: %T", ks.Type)
 }
 
 // WaitForSync implements SyncingSource to allow controllers to wait with starting
