diff --git a/lib/node_modules/@stdlib/stats/base/nanstdevwd/lib/nanstdevwd.js b/lib/node_modules/@stdlib/stats/base/nanstdevwd/lib/nanstdevwd.js
index 2a4392866b8..6adf73849b5 100644
--- a/lib/node_modules/@stdlib/stats/base/nanstdevwd/lib/nanstdevwd.js
+++ b/lib/node_modules/@stdlib/stats/base/nanstdevwd/lib/nanstdevwd.js
@@ -20,8 +20,7 @@
 
 // MODULES //
 
-var nanvariancewd = require( '@stdlib/stats/base/nanvariancewd' );
-var sqrt = require( '@stdlib/math/base/special/sqrt' );
+var ndarray = require( './ndarray.js' );
 
 
 // MAIN //
@@ -42,7 +41,13 @@ var sqrt = require( '@stdlib/math/base/special/sqrt' );
 * // returns ~2.0817
 */
 function nanstdevwd( N, correction, x, stride ) {
-	return sqrt( nanvariancewd( N, correction, x, stride ) );
+	var offset;
+	if ( stride < 0 ) {
+		offset = (1-N) * stride;
+	} else {
+		offset = 0;
+	}
+	return ndarray( N, correction, x, stride, offset );
 }
 
 
diff --git a/lib/node_modules/@stdlib/stats/base/nanstdevwd/lib/ndarray.js b/lib/node_modules/@stdlib/stats/base/nanstdevwd/lib/ndarray.js
index 2a1d789fa56..1908632872a 100644
--- a/lib/node_modules/@stdlib/stats/base/nanstdevwd/lib/ndarray.js
+++ b/lib/node_modules/@stdlib/stats/base/nanstdevwd/lib/ndarray.js
@@ -20,8 +20,10 @@
 
 // MODULES //
 
+var arraylike2object = require( '@stdlib/array/base/arraylike2object' );
 var nanvariancewd = require( '@stdlib/stats/base/nanvariancewd' ).ndarray;
 var sqrt = require( '@stdlib/math/base/special/sqrt' );
+var accessors = require( './accessors.js' );
 
 
 // MAIN //
@@ -46,6 +48,10 @@ var sqrt = require( '@stdlib/math/base/special/sqrt' );
 * // returns 2.5
 */
 function nanstdevwd( N, correction, x, stride, offset ) {
+	var o = arraylike2object( x );
+	if ( o.accessorProtocol ) {
+		return accessors( N, correction, o, stride, offset );
+	}
 	return sqrt( nanvariancewd( N, correction, x, stride, offset ) );
 }
 
diff --git a/lib/node_modules/@stdlib/stats/base/nanstdevwd/test/test.ndarray.js b/lib/node_modules/@stdlib/stats/base/nanstdevwd/test/test.ndarray.js
index cd085c15467..10f72eee710 100644
--- a/lib/node_modules/@stdlib/stats/base/nanstdevwd/test/test.ndarray.js
+++ b/lib/node_modules/@stdlib/stats/base/nanstdevwd/test/test.ndarray.js
@@ -24,6 +24,7 @@ var tape = require( 'tape' );
 var floor = require( '@stdlib/math/base/special/floor' );
 var sqrt = require( '@stdlib/math/base/special/sqrt' );
 var isnan = require( '@stdlib/math/base/assert/is-nan' );
+var toAccessorArray = require( '@stdlib/array/base/to-accessor-array' );
 var nanstdevwd = require( './../lib/ndarray.js' );
 
 
@@ -316,3 +317,66 @@ tape( 'the function supports an `offset` parameter', function test( t ) {
 
 	t.end();
 });
+
+tape( 'the function supports accessor arrays', function test( t ) {
+	var x;
+	var v;
+
+	x = toAccessorArray( [ 1.0, -2.0, NaN, 2.0 ] );
+
+	v = nanstdevwd( x.length, 1, x, 1, 0 );
+	t.strictEqual( v, sqrt( 4.333333333333333 ), 'returns expected value' );
+
+	t.end();
+});
+
+tape( 'the function supports accessor arrays with a stride parameter', function test( t ) {
+	var N;
+	var x;
+	var v;
+
+	x = toAccessorArray([
+		1.0,  // 0
+		2.0,
+		2.0,  // 1
+		-7.0,
+		-2.0, // 2
+		3.0,
+		4.0,  // 3
+		2.0,
+		NaN,  // 4
+		NaN
+	]);
+
+	N = floor( x.length / 2 );
+	v = nanstdevwd( N, 1, x, 2, 0 );
+
+	t.strictEqual( v, 2.5, 'returns expected value' );
+	t.end();
+});
+
+tape( 'the function supports accessor arrays with an offset parameter', function test( t ) {
+	var N;
+	var x;
+	var v;
+
+	x = toAccessorArray([
+		2.0,
+		1.0,  // 0
+		2.0,
+		-2.0, // 1
+		-2.0,
+		2.0,  // 2
+		3.0,
+		4.0,  // 3
+		NaN,
+		NaN   // 4
+	]);
+	N = floor( x.length / 2 );
+
+	v = nanstdevwd( N, 1, x, 2, 1 );
+	t.strictEqual( v, 2.5, 'returns expected value' );
+
+	t.end();
+});
+
