diff --git a/src/MudBlazor/Components/FileUpload/MudFileUpload.razor b/src/MudBlazor/Components/FileUpload/MudFileUpload.razor
index 1358681..1aecb7c 100644
--- a/src/MudBlazor/Components/FileUpload/MudFileUpload.razor
+++ b/src/MudBlazor/Components/FileUpload/MudFileUpload.razor
@@ -19,11 +19,11 @@
                        multiple="@(typeof(T) == typeof(IReadOnlyList<IBrowserFile>) ? "" : null)"
                        accept="@Accept"
                        disabled="@(GetDisabledState())"
-                       @attributes="UserAttributes"
                        @ondrop="OnDragResetAsync"
                        @ondragleave="OnDragResetAsync"
                        @ondragend="OnDragResetAsync"
                        id="@GetInputId(localIndex)"
+                       @attributes="UserAttributes"
                        required="@Required"
                        aria-required="@Required.ToString().ToLowerInvariant()" />
         }
