diff --git a/notify/jira/jira.go b/notify/jira/jira.go
index 259a5e62..f2f34963 100644
--- a/notify/jira/jira.go
+++ b/notify/jira/jira.go
@@ -205,6 +205,12 @@ func (n *Notifier) searchExistingIssue(ctx context.Context, logger *slog.Logger,
 	if firing {
 		if n.conf.ReopenTransition == "" {
 			jql.WriteString(`statusCategory != Done and `)
+		} else {
+			// When a reopen transition is configured, only consider recently resolved issues.
+			reopenDuration := int64(time.Duration(n.conf.ReopenDuration).Minutes())
+			if reopenDuration != 0 {
+				jql.WriteString(fmt.Sprintf(`(resolutiondate is EMPTY OR resolutiondate >= -%dm) and `, reopenDuration))
+			}
 		}
 	} else {
 		reopenDuration := int64(time.Duration(n.conf.ReopenDuration).Minutes())
