diff --git a/cluster-operator/src/main/java/io/strimzi/operator/cluster/model/CruiseControl.java b/cluster-operator/src/main/java/io/strimzi/operator/cluster/model/CruiseControl.java
index 8b965894d..21d2179bd 100644
--- a/cluster-operator/src/main/java/io/strimzi/operator/cluster/model/CruiseControl.java
+++ b/cluster-operator/src/main/java/io/strimzi/operator/cluster/model/CruiseControl.java
@@ -303,7 +303,7 @@ public class CruiseControl extends AbstractModel implements SupportsMetrics, Sup
                 labels,
                 ownerReference,
                 templateService,
-                List.of(ServiceUtils.createServicePort(REST_API_PORT_NAME, REST_API_PORT, REST_API_PORT, "TCP"))
+                List.of(ServiceUtils.createServicePort(REST_API_PORT_NAME, REST_API_PORT, REST_API_PORT_NAME, "TCP"))
         );
     }
 
diff --git a/cluster-operator/src/main/java/io/strimzi/operator/cluster/model/KafkaBridgeCluster.java b/cluster-operator/src/main/java/io/strimzi/operator/cluster/model/KafkaBridgeCluster.java
index 0ddb0ce38..4417b51f3 100644
--- a/cluster-operator/src/main/java/io/strimzi/operator/cluster/model/KafkaBridgeCluster.java
+++ b/cluster-operator/src/main/java/io/strimzi/operator/cluster/model/KafkaBridgeCluster.java
@@ -234,7 +234,7 @@ public class KafkaBridgeCluster extends AbstractModel implements SupportsLogging
                 labels,
                 ownerReference,
                 templateService,
-                List.of(ServiceUtils.createServicePort(REST_API_PORT_NAME, port, port, "TCP")),
+                List.of(ServiceUtils.createServicePort(REST_API_PORT_NAME, port, REST_API_PORT_NAME, "TCP")),
                 labels.strimziSelectorLabels(),
                 ModelUtils.getCustomLabelsOrAnnotations(CO_ENV_VAR_CUSTOM_SERVICE_LABELS),
                 Util.mergeLabelsOrAnnotations(getDiscoveryAnnotation(port), ModelUtils.getCustomLabelsOrAnnotations(CO_ENV_VAR_CUSTOM_SERVICE_ANNOTATIONS))
diff --git a/cluster-operator/src/main/java/io/strimzi/operator/cluster/model/KafkaCluster.java b/cluster-operator/src/main/java/io/strimzi/operator/cluster/model/KafkaCluster.java
index d981a8882..75aaa17de 100644
--- a/cluster-operator/src/main/java/io/strimzi/operator/cluster/model/KafkaCluster.java
+++ b/cluster-operator/src/main/java/io/strimzi/operator/cluster/model/KafkaCluster.java
@@ -650,10 +650,10 @@ public class KafkaCluster extends AbstractModel implements SupportsMetrics, Supp
         List<GenericKafkaListener> internalListeners = ListenersUtils.internalListeners(listeners);
 
         List<ServicePort> ports = new ArrayList<>(internalListeners.size() + 1);
-        ports.add(ServiceUtils.createServicePort(REPLICATION_PORT_NAME, REPLICATION_PORT, REPLICATION_PORT, "TCP"));
+        ports.add(ServiceUtils.createServicePort(REPLICATION_PORT_NAME, REPLICATION_PORT, REPLICATION_PORT_NAME, "TCP"));
 
         for (GenericKafkaListener listener : internalListeners) {
-            ports.add(ServiceUtils.createServicePort(ListenersUtils.backwardsCompatiblePortName(listener), listener.getPort(), listener.getPort(), "TCP"));
+            ports.add(ServiceUtils.createServicePort(ListenersUtils.backwardsCompatiblePortName(listener), listener.getPort(), ListenersUtils.backwardsCompatiblePortName(listener), "TCP"));
         }
 
         return ports;
@@ -669,12 +669,12 @@ public class KafkaCluster extends AbstractModel implements SupportsMetrics, Supp
         List<GenericKafkaListener> internalListeners = ListenersUtils.internalListeners(listeners);
 
         List<ServicePort> ports = new ArrayList<>(internalListeners.size() + 3);
-        ports.add(ServiceUtils.createServicePort(CONTROLPLANE_PORT_NAME, CONTROLPLANE_PORT, CONTROLPLANE_PORT, "TCP"));
-        ports.add(ServiceUtils.createServicePort(REPLICATION_PORT_NAME, REPLICATION_PORT, REPLICATION_PORT, "TCP"));
-        ports.add(ServiceUtils.createServicePort(KAFKA_AGENT_PORT_NAME, KAFKA_AGENT_PORT, KAFKA_AGENT_PORT, "TCP"));
+        ports.add(ServiceUtils.createServicePort(CONTROLPLANE_PORT_NAME, CONTROLPLANE_PORT, CONTROLPLANE_PORT_NAME, "TCP"));
+        ports.add(ServiceUtils.createServicePort(REPLICATION_PORT_NAME, REPLICATION_PORT, REPLICATION_PORT_NAME, "TCP"));
+        ports.add(ServiceUtils.createServicePort(KAFKA_AGENT_PORT_NAME, KAFKA_AGENT_PORT, KAFKA_AGENT_PORT_NAME, "TCP"));
 
         for (GenericKafkaListener listener : internalListeners) {
-            ports.add(ServiceUtils.createServicePort(ListenersUtils.backwardsCompatiblePortName(listener), listener.getPort(), listener.getPort(), "TCP"));
+            ports.add(ServiceUtils.createServicePort(ListenersUtils.backwardsCompatiblePortName(listener), listener.getPort(), ListenersUtils.backwardsCompatiblePortName(listener), "TCP"));
         }
 
         ports.addAll(jmx.servicePorts());
@@ -747,7 +747,7 @@ public class KafkaCluster extends AbstractModel implements SupportsMetrics, Supp
             List<ServicePort> ports = Collections.singletonList(
                     ServiceUtils.createServicePort(ListenersUtils.backwardsCompatiblePortName(listener),
                             listener.getPort(),
-                            listener.getPort(),
+                            ListenersUtils.backwardsCompatiblePortName(listener),
                             ListenersUtils.bootstrapNodePort(listener),
                             "TCP")
             );
@@ -838,7 +838,7 @@ public class KafkaCluster extends AbstractModel implements SupportsMetrics, Supp
                                 ServiceUtils.createServicePort(
                                         ListenersUtils.backwardsCompatiblePortName(listener),
                                         listener.getPort(),
-                                        listener.getPort(),
+                                        ListenersUtils.backwardsCompatiblePortName(listener),
                                         ListenersUtils.brokerNodePort(listener, node.nodeId()),
                                         "TCP")
                         );
@@ -940,7 +940,7 @@ public class KafkaCluster extends AbstractModel implements SupportsMetrics, Supp
                             .withName(serviceName)
                         .endTo()
                         .withNewPort()
-                            .withNewTargetPort(listener.getPort())
+                            .withTargetPort(new io.fabric8.kubernetes.api.model.IntOrString(ListenersUtils.backwardsCompatiblePortName(listener)))
                         .endPort()
                         .withNewTls()
                             .withTermination("passthrough")
@@ -987,7 +987,7 @@ public class KafkaCluster extends AbstractModel implements SupportsMetrics, Supp
                                         .withName(routeName)
                                     .endTo()
                                     .withNewPort()
-                                        .withNewTargetPort(listener.getPort())
+                                        .withTargetPort(new io.fabric8.kubernetes.api.model.IntOrString(ListenersUtils.backwardsCompatiblePortName(listener)))
                                     .endPort()
                                     .withNewTls()
                                         .withTermination("passthrough")
diff --git a/cluster-operator/src/main/java/io/strimzi/operator/cluster/model/KafkaConnectCluster.java b/cluster-operator/src/main/java/io/strimzi/operator/cluster/model/KafkaConnectCluster.java
index 8b82270f6..336cc7482 100644
--- a/cluster-operator/src/main/java/io/strimzi/operator/cluster/model/KafkaConnectCluster.java
+++ b/cluster-operator/src/main/java/io/strimzi/operator/cluster/model/KafkaConnectCluster.java
@@ -342,7 +342,7 @@ public class KafkaConnectCluster extends AbstractModel implements SupportsMetric
      */
     public Service generateService() {
         List<ServicePort> ports = new ArrayList<>(1);
-        ports.add(ServiceUtils.createServicePort(REST_API_PORT_NAME, REST_API_PORT, REST_API_PORT, "TCP"));
+        ports.add(ServiceUtils.createServicePort(REST_API_PORT_NAME, REST_API_PORT, REST_API_PORT_NAME, "TCP"));
 
         ports.addAll(jmx.servicePorts());
 
@@ -362,7 +362,7 @@ public class KafkaConnectCluster extends AbstractModel implements SupportsMetric
      * @return The generated Service
      */
     public Service generateHeadlessService() {
-        List<ServicePort> ports = List.of(ServiceUtils.createServicePort(REST_API_PORT_NAME, REST_API_PORT, REST_API_PORT, "TCP"));
+        List<ServicePort> ports = List.of(ServiceUtils.createServicePort(REST_API_PORT_NAME, REST_API_PORT, REST_API_PORT_NAME, "TCP"));
 
         return ServiceUtils.createHeadlessService(
                 componentName,
diff --git a/cluster-operator/src/main/java/io/strimzi/operator/cluster/model/ServiceUtils.java b/cluster-operator/src/main/java/io/strimzi/operator/cluster/model/ServiceUtils.java
index ac2592f2a..e816af48b 100644
--- a/cluster-operator/src/main/java/io/strimzi/operator/cluster/model/ServiceUtils.java
+++ b/cluster-operator/src/main/java/io/strimzi/operator/cluster/model/ServiceUtils.java
@@ -219,12 +219,12 @@ public class ServiceUtils {
      *
      * @param name          Name of the port
      * @param port          The port on the service which can be accessed by clients
-     * @param targetPort    The port on the container / Pod where the connections will be routed
+     * @param targetPort    The named port on the container / Pod where the connections will be routed
      * @param protocol      Protocol used by this port
      *
      * @return  Created port
      */
-    public static ServicePort createServicePort(String name, int port, int targetPort, String protocol)   {
+    public static ServicePort createServicePort(String name, int port, String targetPort, String protocol)   {
         return createServicePort(name, port, targetPort, null, protocol);
     }
 
@@ -233,18 +233,18 @@ public class ServiceUtils {
      *
      * @param name          Name of the port
      * @param port          The port on the service which can be accessed by clients
-     * @param targetPort    The port on the container / Pod where the connections will be routed
+     * @param targetPort    The named port on the container / Pod where the connections will be routed
      * @param nodePort      The desired node port number
      * @param protocol      Protocol used by this port
      *
      * @return  Created port
      */
-    public static ServicePort createServicePort(String name, int port, int targetPort, Integer nodePort, String protocol)   {
+    public static ServicePort createServicePort(String name, int port, String targetPort, Integer nodePort, String protocol)   {
         return new ServicePortBuilder()
                 .withName(name)
                 .withProtocol(protocol)
                 .withPort(port)
-                .withNewTargetPort(targetPort)
+                .withTargetPort(new io.fabric8.kubernetes.api.model.IntOrString(targetPort))
                 .withNodePort(nodePort)
                 .build();
     }
diff --git a/cluster-operator/src/main/java/io/strimzi/operator/cluster/model/jmx/JmxModel.java b/cluster-operator/src/main/java/io/strimzi/operator/cluster/model/jmx/JmxModel.java
index 24e77f05e..cb0e25570 100644
--- a/cluster-operator/src/main/java/io/strimzi/operator/cluster/model/jmx/JmxModel.java
+++ b/cluster-operator/src/main/java/io/strimzi/operator/cluster/model/jmx/JmxModel.java
@@ -115,7 +115,7 @@ public class JmxModel {
      */
     public List<ServicePort> servicePorts()    {
         if (isEnabled)  {
-            return List.of(ServiceUtils.createServicePort(JMX_PORT_NAME, JMX_PORT, JMX_PORT, "TCP"));
+            return List.of(ServiceUtils.createServicePort(JMX_PORT_NAME, JMX_PORT, JMX_PORT_NAME, "TCP"));
         } else {
             return List.of();
         }
diff --git a/cluster-operator/src/test/java/io/strimzi/operator/cluster/model/ServiceUtilsTest.java b/cluster-operator/src/test/java/io/strimzi/operator/cluster/model/ServiceUtilsTest.java
index f43cc1d8c..305a7673c 100644
--- a/cluster-operator/src/test/java/io/strimzi/operator/cluster/model/ServiceUtilsTest.java
+++ b/cluster-operator/src/test/java/io/strimzi/operator/cluster/model/ServiceUtilsTest.java
@@ -42,7 +42,7 @@ public class ServiceUtilsTest {
             .withStrimziComponentType("my-component-type")
             .withAdditionalLabels(Map.of("label-1", "value-1", "label-2", "value-2"));
     private static final String PORT_NAME = "my-port";
-    private static final ServicePort PORT = ServiceUtils.createServicePort(PORT_NAME, 1234, 5678, "HTTP");
+    private static final ServicePort PORT = ServiceUtils.createServicePort(PORT_NAME, 1234, PORT_NAME, "HTTP");
     private static final InternalServiceTemplate TEMPLATE = new InternalServiceTemplateBuilder()
             .withNewMetadata()
                 .withLabels(Map.of("label-3", "value-3", "label-4", "value-4"))
@@ -54,22 +54,22 @@ public class ServiceUtilsTest {
 
     @ParallelTest
     public void testCreateServicePort() {
-        ServicePort port = ServiceUtils.createServicePort(PORT_NAME, 1234, 5678, "HTTP");
+        ServicePort port = ServiceUtils.createServicePort(PORT_NAME, 1234, PORT_NAME, "HTTP");
 
         assertThat(port.getName(), is(PORT_NAME));
         assertThat(port.getPort(), is(1234));
-        assertThat(port.getTargetPort().getIntVal(), is(5678));
+        assertThat(port.getTargetPort().getStrVal(), is(PORT_NAME));
         assertThat(port.getNodePort(), is(nullValue()));
         assertThat(port.getProtocol(), is("HTTP"));
     }
 
     @ParallelTest
     public void testCreateServiceWithNodePort() {
-        ServicePort port = ServiceUtils.createServicePort(PORT_NAME, 1234, 5678, 30000, "HTTP");
+        ServicePort port = ServiceUtils.createServicePort(PORT_NAME, 1234, PORT_NAME, 30000, "HTTP");
 
         assertThat(port.getName(), is(PORT_NAME));
         assertThat(port.getPort(), is(1234));
-        assertThat(port.getTargetPort().getIntVal(), is(5678));
+        assertThat(port.getTargetPort().getStrVal(), is(PORT_NAME));
         assertThat(port.getNodePort(), is(30000));
         assertThat(port.getProtocol(), is("HTTP"));
     }
