diff --git a/internal/codegen/golang/query.go b/internal/codegen/golang/query.go
index 3b4fb2f..3484610 100644
--- a/internal/codegen/golang/query.go
+++ b/internal/codegen/golang/query.go
@@ -101,13 +101,6 @@ func (v *QueryValue) DefineType() string {
 	return t
 }
 
-func (v *QueryValue) ReturnName() string {
-	if v.IsPointer() {
-		return "&" + escape(v.Name)
-	}
-	return escape(v.Name)
-}
-
 func (v QueryValue) UniqueFields() []Field {
 	seen := map[string]struct{}{}
 	fields := make([]Field, 0, len(v.Struct.Fields))
@@ -190,13 +183,28 @@ func (v QueryValue) HasSqlcSlices() bool {
 	return false
 }
 
+func (v QueryValue) LocalName() string {
+	return escape(v.Name)
+}
+
+func (v *QueryValue) ReturnName() string {
+	if v.IsPointer() {
+		return "&" + v.LocalName()
+	}
+	return v.LocalName()
+}
+
 func (v QueryValue) Scan() string {
+	return v.ScanForName(v.LocalName())
+}
+
+func (v QueryValue) ScanForName(name string) string {
 	var out []string
 	if v.Struct == nil {
 		if strings.HasPrefix(v.Typ, "[]") && v.Typ != "[]byte" && !v.SQLDriver.IsPGX() {
-			out = append(out, "pq.Array(&"+v.Name+")")
+			out = append(out, "pq.Array(&"+name+")")
 		} else {
-			out = append(out, "&"+v.Name)
+			out = append(out, "&"+name)
 		}
 	} else {
 		for _, f := range v.Struct.Fields {
diff --git a/internal/codegen/golang/templates/pgx/batchCode.tmpl b/internal/codegen/golang/templates/pgx/batchCode.tmpl
index 35bd701..bb3b384 100644
--- a/internal/codegen/golang/templates/pgx/batchCode.tmpl
+++ b/internal/codegen/golang/templates/pgx/batchCode.tmpl
@@ -90,7 +90,7 @@ func (b *{{.MethodName}}BatchResults) Query(f func(int, []{{.Ret.DefineType}}, e
        }
        defer rows.Close()
        for rows.Next() {
-           var {{.Ret.Name}} {{.Ret.Type}}
+           var {{.Ret.LocalName}} {{.Ret.Type}}
            if err := rows.Scan({{.Ret.Scan}}); err != nil {
              return err
            }
@@ -109,10 +109,10 @@ func (b *{{.MethodName}}BatchResults) Query(f func(int, []{{.Ret.DefineType}}, e
 func (b *{{.MethodName}}BatchResults) QueryRow(f func(int, {{.Ret.DefineType}}, error)) {
 	defer b.br.Close()
    for t := 0; t < b.tot; t++ {
-     var {{.Ret.Name}} {{.Ret.Type}}
+     var {{.Ret.LocalName}} {{.Ret.Type}}
      if b.closed {
         if f != nil {
-          f(t, {{if .Ret.IsPointer}}nil{{else}}{{.Ret.Name}}{{end}}, ErrBatchAlreadyClosed)
+          f(t, {{if .Ret.IsPointer}}nil{{else}}{{.Ret.LocalName}}{{end}}, ErrBatchAlreadyClosed)
         }
         continue
      }
diff --git a/internal/codegen/golang/templates/pgx/queryCode.tmpl b/internal/codegen/golang/templates/pgx/queryCode.tmpl
index 59a88c8..6d940a8 100644
--- a/internal/codegen/golang/templates/pgx/queryCode.tmpl
+++ b/internal/codegen/golang/templates/pgx/queryCode.tmpl
@@ -34,15 +34,17 @@ func (q *Queries) {{.MethodName}}(ctx context.Context, {{.Arg.Pair}}) ({{.Ret.De
 	row := q.db.QueryRow(ctx, {{.ConstantName}}, {{.Arg.Params}})
 {{- end}}
 	{{- if or (ne .Arg.Pair .Ret.Pair) (ne .Arg.DefineType .Ret.DefineType) }}
-	var {{.Ret.Name}} {{.Ret.Type}}
+	var {{.Ret.LocalName}} {{.Ret.Type}}
+	{{- else }}
+	var {{.Ret.LocalName}}Result {{.Ret.Type}}
 	{{- end}}
-	err := row.Scan({{.Ret.Scan}})
+	err := row.Scan({{if and (eq .Arg.Pair .Ret.Pair) (eq .Arg.DefineType .Ret.DefineType)}}{{.Ret.ScanForName (print .Ret.LocalName "Result")}}{{else}}{{.Ret.Scan}}{{end}})
 	{{- if $.WrapErrors}}
 	if err != nil {
 		err = fmt.Errorf("query {{.MethodName}}: %w", err)
 	}
 	{{- end}}
-	return {{.Ret.ReturnName}}, err
+	return {{if and (eq .Arg.Pair .Ret.Pair) (eq .Arg.DefineType .Ret.DefineType)}}{{.Ret.LocalName}}Result{{else}}{{.Ret.ReturnName}}{{end}}, err
 }
 {{end}}
 
@@ -66,7 +68,7 @@ func (q *Queries) {{.MethodName}}(ctx context.Context, {{.Arg.Pair}}) ([]{{.Ret.
 	var items []{{.Ret.DefineType}}
 	{{end -}}
 	for rows.Next() {
-		var {{.Ret.Name}} {{.Ret.Type}}
+		var {{.Ret.LocalName}} {{.Ret.Type}}
 		if err := rows.Scan({{.Ret.Scan}}); err != nil {
 			return nil, {{if $.WrapErrors}}fmt.Errorf("query {{.MethodName}}: %w", err){{else}}err{{end}}
 		}
diff --git a/internal/codegen/golang/templates/stdlib/queryCode.tmpl b/internal/codegen/golang/templates/stdlib/queryCode.tmpl
index 1e7f4e2..f790972 100644
--- a/internal/codegen/golang/templates/stdlib/queryCode.tmpl
+++ b/internal/codegen/golang/templates/stdlib/queryCode.tmpl
@@ -25,15 +25,17 @@ type {{.Ret.Type}} struct { {{- range .Ret.Struct.Fields}}
 func (q *Queries) {{.MethodName}}(ctx context.Context, {{ dbarg }} {{.Arg.Pair}}) ({{.Ret.DefineType}}, error) {
     {{- template "queryCodeStdExec" . }}
 	{{- if or (ne .Arg.Pair .Ret.Pair) (ne .Arg.DefineType .Ret.DefineType) }}
-	var {{.Ret.Name}} {{.Ret.Type}}
+	var {{.Ret.LocalName}} {{.Ret.Type}}
+	{{- else }}
+	var {{.Ret.LocalName}}Result {{.Ret.Type}}
 	{{- end}}
-	err := row.Scan({{.Ret.Scan}})
+	err := row.Scan({{if and (eq .Arg.Pair .Ret.Pair) (eq .Arg.DefineType .Ret.DefineType)}}{{.Ret.ScanForName (print .Ret.LocalName "Result")}}{{else}}{{.Ret.Scan}}{{end}})
 	{{- if $.WrapErrors}}
 	if err != nil {
 		err = fmt.Errorf("query {{.MethodName}}: %w", err)
 	}
 	{{- end}}
-	return {{.Ret.ReturnName}}, err
+	return {{if and (eq .Arg.Pair .Ret.Pair) (eq .Arg.DefineType .Ret.DefineType)}}{{.Ret.LocalName}}Result{{else}}{{.Ret.ReturnName}}{{end}}, err
 }
 {{end}}
 
@@ -52,7 +54,7 @@ func (q *Queries) {{.MethodName}}(ctx context.Context, {{ dbarg }} {{.Arg.Pair}}
     var items []{{.Ret.DefineType}}
     {{end -}}
     for rows.Next() {
-        var {{.Ret.Name}} {{.Ret.Type}}
+        var {{.Ret.LocalName}} {{.Ret.Type}}
         if err := rows.Scan({{.Ret.Scan}}); err != nil {
             return nil, {{if $.WrapErrors}}fmt.Errorf("query {{.MethodName}}: %w", err){{else}}err{{end}}
         }
