diff --git a/packages/svelte/src/compiler/phases/3-transform/client/visitors/ConstTag.js b/packages/svelte/src/compiler/phases/3-transform/client/visitors/ConstTag.js
index 34acdd6bb..b95eae098 100644
--- a/packages/svelte/src/compiler/phases/3-transform/client/visitors/ConstTag.js
+++ b/packages/svelte/src/compiler/phases/3-transform/client/visitors/ConstTag.js
@@ -17,7 +17,14 @@ export function ConstTag(node, context) {
 	// TODO we can almost certainly share some code with $derived(...)
 	if (declaration.id.type === 'Identifier') {
 		const init = build_expression(context, declaration.init, node.metadata.expression);
-		let expression = create_derived(context.state, b.thunk(init));
+		const thunk = b.thunk(init, node.metadata.expression.has_await);
+		let expression = create_derived(context.state, thunk);
+
+		// if the initializer contains an await, the derived callback must be async
+		// (otherwise codegen will contain `await` in a non-async function)
+		if (node.metadata.expression.has_await) {
+			/** @type {import('estree').ArrowFunctionExpression} */ (thunk).async = true;
+		}
 
 		if (dev) {
 			expression = b.call('$.tag', expression, b.literal(declaration.id.name));
@@ -63,6 +70,12 @@ export function ConstTag(node, context) {
 
 		let expression = create_derived(context.state, fn);
 
+		// if the initializer contains an await, the derived callback must be async
+		// (otherwise codegen will contain `await` in a non-async function)
+		if (node.metadata.expression.has_await) {
+			fn.async = true;
+		}
+
 		if (dev) {
 			expression = b.call('$.tag', expression, b.literal('[@const]'));
 		}
