diff --git a/crates/core/src/operations/merge/mod.rs b/crates/core/src/operations/merge/mod.rs
index 42af43b..f68c25a 100644
--- a/crates/core/src/operations/merge/mod.rs
+++ b/crates/core/src/operations/merge/mod.rs
@@ -43,6 +43,7 @@ use datafusion::common::{
 use datafusion::datasource::provider_as_source;
 use datafusion::error::Result as DataFusionResult;
 use datafusion::execution::session_state::SessionStateBuilder;
+use datafusion::functions_aggregate::count::count;
 use datafusion::logical_expr::build_join_schema;
 use datafusion::logical_expr::execution_props::ExecutionProps;
 use datafusion::logical_expr::simplify::SimplifyContext;
@@ -949,6 +950,32 @@ async fn execute(
     let target = target.with_column(TARGET_COLUMN, lit(true))?;
 
     let join = source.join(target, JoinType::Full, &[], &[], Some(predicate.clone()))?;
+
+    let duplicate_target_matches = join
+        .clone()
+        .filter(
+            col(SOURCE_COLUMN)
+                .is_true()
+                .and(col(TARGET_COLUMN).is_true()),
+        )?
+        .aggregate(
+            vec![col(file_column.as_str())],
+            vec![count(lit(1)).alias("count")],
+        )?
+        .filter(col("count").gt(lit(1i64)))?
+        .limit(0, Some(1))?
+        .collect()
+        .await?;
+
+    if duplicate_target_matches
+        .iter()
+        .any(|batch| batch.num_rows() > 0)
+    {
+        return Err(DeltaTableError::Generic(
+            "MERGE operation matched multiple source rows with the same target row".to_string(),
+        ));
+    }
+
     let join_schema_df = join.schema().to_owned();
 
     let match_operations: Vec<MergeOperation> = match_operations
