diff --git a/aiogram/fsm/context.py b/aiogram/fsm/context.py
index 53a8ea46..2016377f 100644
--- a/aiogram/fsm/context.py
+++ b/aiogram/fsm/context.py
@@ -1,7 +1,9 @@
-from typing import Any, Dict, Optional
+from typing import Any, Dict, Optional, TypeVar
 
 from aiogram.fsm.storage.base import BaseStorage, StateType, StorageKey
 
+T = TypeVar("T")
+
 
 class FSMContext:
     def __init__(self, storage: BaseStorage, key: StorageKey) -> None:
@@ -20,6 +22,16 @@ class FSMContext:
     async def get_data(self) -> Dict[str, Any]:
         return await self.storage.get_data(key=self.key)
 
+    async def get_value(self, key: str) -> Any:
+        """
+        Get value from state data by key.
+
+        :param key: Key to get value from state data
+        :return: Value from state data
+        """
+        data = await self.get_data()
+        return data[key]
+
     async def update_data(
         self, data: Optional[Dict[str, Any]] = None, **kwargs: Any
     ) -> Dict[str, Any]:
diff --git a/tests/test_fsm/test_context.py b/tests/test_fsm/test_context.py
index f0c29911..db3792b7 100644
--- a/tests/test_fsm/test_context.py
+++ b/tests/test_fsm/test_context.py
@@ -49,3 +49,12 @@ class TestFSMContext:
         assert await state.get_data() == {}
 
         assert await state2.get_state() == "experiments"
+        
+    async def test_get_value(self, state):
+        # Test getting an existing value
+        value = await state.get_value("foo")
+        assert value == "bar"
+        
+        # Test getting a non-existent value
+        with pytest.raises(KeyError):
+            await state.get_value("non_existent_key")
