diff --git a/hugolib/page__content.go b/hugolib/page__content.go
index 7377936..93c4b24 100644
--- a/hugolib/page__content.go
+++ b/hugolib/page__content.go
@@ -1103,10 +1103,13 @@ func (c *cachedContentScope) RenderShortcodes(ctx context.Context) (template.HTM
 	}
 
 	if tpl.Context.IsInGoldmark.Get(ctx) {
-		// This content will be parsed and rendered by Goldmark.
-		// Wrap it in a special Hugo markup to assign the correct Page from
-		// the stack.
-		return template.HTML(hugocontext.Wrap(cc, pco.po.p.pid)), nil
+		if cb != nil {
+			// This content will be parsed and rendered by Goldmark.
+			// Wrap it in a special Hugo markup to assign the correct Page from
+			// the stack.
+			return template.HTML(hugocontext.Wrap(cc, pco.po.p.pid)), nil
+		}
+		cc = hugocontext.Unwrap(cc)
 	}
 
 	return helpers.BytesToHTML(cc), nil
diff --git a/markup/goldmark/hugocontext/hugocontext.go b/markup/goldmark/hugocontext/hugocontext.go
index 8afe64b..69fa2a3 100644
--- a/markup/goldmark/hugocontext/hugocontext.go
+++ b/markup/goldmark/hugocontext/hugocontext.go
@@ -57,6 +57,13 @@ func Wrap(b []byte, pid uint64) string {
 	return buf.String()
 }
 
+func Unwrap(b []byte) []byte {
+	if !bytes.Contains(b, hugoCtxPrefix) {
+		return b
+	}
+	return hugoCtxRe.ReplaceAll(b, nil)
+}
+
 var kindHugoContext = ast.NewNodeKind("HugoContext")
 
 // HugoContext is a node that represents a Hugo context.
