diff --git a/src/UI/Features/Ocr/Engines/PaddleOcrResultParser.cs b/src/UI/Features/Ocr/Engines/PaddleOcrResultParser.cs
index 3a8cce6..8b85272 100644
--- a/src/UI/Features/Ocr/Engines/PaddleOcrResultParser.cs
+++ b/src/UI/Features/Ocr/Engines/PaddleOcrResultParser.cs
@@ -66,7 +66,9 @@ public class PaddleOcrResultParser
     {
         // Extract text using regex
         var textMatch = Regex.Match(input, @"\([""'](.*)[""'],");
-        var text = textMatch.Groups[1].Value;
+        var text = textMatch.Groups[1].Value
+            .Replace(@"\'", "'")
+            .Replace(@"\""", "\"");
 
         // Extract confidence using regex
         input = input.Replace(" ", string.Empty);
diff --git a/src/libse/Common/StringExtensions.cs b/src/libse/Common/StringExtensions.cs
index b44810e..0683515 100644
--- a/src/libse/Common/StringExtensions.cs
+++ b/src/libse/Common/StringExtensions.cs
@@ -57,12 +57,13 @@ namespace Nikse.SubtitleEdit.Core.Common
             }
 
             var newLineIdx = text.IndexOf(Environment.NewLine, StringComparison.Ordinal);
-            if (newLineIdx < 0 || text.Length < newLineIdx + 5)
+            var newLineLength = Environment.NewLine.Length;
+            if (newLineIdx < 0 || text.Length < newLineIdx + newLineLength + 3)
             {
                 return false;
             }
 
-            text = text.Substring(newLineIdx + 2);
+            text = text.Substring(newLineIdx + newLineLength);
             return StartsWithHtmlTag(text, threeLengthTag, includeFont);
         }
 
diff --git a/src/libse/VobSub/Ocr/Service/GoogleCloudVisionApi.cs b/src/libse/VobSub/Ocr/Service/GoogleCloudVisionApi.cs
index c9b240a..cac758f 100644
--- a/src/libse/VobSub/Ocr/Service/GoogleCloudVisionApi.cs
+++ b/src/libse/VobSub/Ocr/Service/GoogleCloudVisionApi.cs
@@ -315,6 +315,7 @@ namespace Nikse.SubtitleEdit.Core.VobSub.Ocr.Service
                     var ocrResult = sb.ToString().Trim();
                     if (ocrResult.Length > 0)
                     {
+                        ocrResult = ocrResult.Replace("\r\n", "\n").Replace("\n", "\r\n");
                         return new List<string> { ocrResult };
                     }
                 }
