diff --git a/caddyconfig/caddyfile/parse.go b/caddyconfig/caddyfile/parse.go
index d04a1ac4..d889afb5 100644
--- a/caddyconfig/caddyfile/parse.go
+++ b/caddyconfig/caddyfile/parse.go
@@ -383,18 +383,22 @@ func (p *parser) doImport(nesting int) error {
 			// see if we can grab a key
 			var currentMappingKey string
 			if bd.Val() == "{" {
-				return p.Err("anonymous blocks are not supported")
+				// treat this as an anonymous block; it maps to {block}
+				currentMappingKey = "block"
+			} else {
+				currentMappingKey = bd.Val()
 			}
-			currentMappingKey = bd.Val()
 			currentMappingTokens := []Token{}
-			// read all args until end of line / {
-			if bd.NextArg() {
+
+			// If this was an anonymous block, we're currently on the opening "{" token.
+			// Otherwise, attempt to parse any args on the same line before an optional block.
+			if currentMappingKey != "block" && bd.NextArg() {
 				currentMappingTokens = append(currentMappingTokens, bd.Token())
 				for bd.NextArg() {
 					currentMappingTokens = append(currentMappingTokens, bd.Token())
 				}
-				// TODO(elee1766): we don't enter another mapping here because it's annoying to extract the { and } properly.
-				// maybe someone can do that in the future
+				// If args were present, we intentionally do not attempt to also parse a nested block.
+				// This mirrors the previous behavior.
 			} else {
 				// attempt to enter a block and add tokens to the currentMappingTokens
 				for mappingNesting := bd.Nesting(); bd.NextBlock(mappingNesting); {
