diff --git a/packages/agents-core/src/result.ts b/packages/agents-core/src/result.ts
index 12332a2..083d635 100644
--- a/packages/agents-core/src/result.ts
+++ b/packages/agents-core/src/result.ts
@@ -1,4 +1,4 @@
-import { Agent, AgentOutputType } from './agent';
+import { Agent } from './agent';
 import { Handoff } from './handoff';
 import {
   ResolvedAgentOutput,
@@ -201,7 +201,7 @@ class RunResultBase<TContext, TAgent extends Agent<TContext, any>>
  */
 export class RunResult<
   TContext,
-  TAgent extends Agent<TContext, AgentOutputType>,
+  TAgent extends Agent<TContext, any>,
 > extends RunResultBase<TContext, TAgent> {
   constructor(state: RunState<TContext, TAgent>) {
     super(state);
@@ -213,7 +213,7 @@ export class RunResult<
  */
 export class StreamedRunResult<
     TContext,
-    TAgent extends Agent<TContext, AgentOutputType>,
+    TAgent extends Agent<TContext, any>,
   >
   extends RunResultBase<TContext, TAgent>
   implements AsyncIterable<RunStreamEvent>
diff --git a/packages/agents-core/src/runState.ts b/packages/agents-core/src/runState.ts
index a18e649..958c2e0 100644
--- a/packages/agents-core/src/runState.ts
+++ b/packages/agents-core/src/runState.ts
@@ -32,6 +32,7 @@ import { AgentInputItem, UnknownContext } from './types';
 import type { InputGuardrailResult, OutputGuardrailResult } from './guardrail';
 import { safeExecute } from './utils/safeExecute';
 import { HostedMCPTool } from './tool';
+import { getTurnInput } from './run';
 
 /**
  * The schema version of the serialized run state. This is used to ensure that the serialized
@@ -375,6 +376,16 @@ export class RunState<TContext, TAgent extends Agent<any, any>> {
     this._context.rejectTool(approvalItem, options);
   }
 
+  /**
+   * The history of the agent run. This includes the input items and the new items generated during
+   * the agent run.
+   *
+   * This can be used as inputs for the next agent run.
+   */
+  get history(): AgentInputItem[] {
+    return getTurnInput(this._originalInput, this._generatedItems);
+  }
+
   /**
    * Serializes the run state to a JSON object.
    *
