diff --git a/plugins/inputs/nftables/types.go b/plugins/inputs/nftables/types.go
index d8a2a5d..9e576e7 100644
--- a/plugins/inputs/nftables/types.go
+++ b/plugins/inputs/nftables/types.go
@@ -103,10 +103,8 @@ type namedCounter struct {
 }
 
 type namedSet struct {
-	Family string `json:"family"`
-	Name   string `json:"name"`
-	Table  string `json:"table"`
-	Elem   []elem `json:"elem,omitempty"`
+	Family string        `json:"family"`
+	Name   string        `json:"name"`
+	Table  string        `json:"table"`
+	Elem   []interface{} `json:"elem,omitempty"`
 }
-
-type elem struct{}
