diff --git a/src/Quartz/Impl/AdoJobStore/StdAdoDelegate.Triggers.cs b/src/Quartz/Impl/AdoJobStore/StdAdoDelegate.Triggers.cs
index af3300e..5dad072 100644
--- a/src/Quartz/Impl/AdoJobStore/StdAdoDelegate.Triggers.cs
+++ b/src/Quartz/Impl/AdoJobStore/StdAdoDelegate.Triggers.cs
@@ -756,6 +756,16 @@ public virtual async ValueTask<List<IOperableTrigger>> SelectTriggersForRecoveri
             if (await rs2.ReadAsync(cancellationToken).ConfigureAwait(false))
             {
                 trigger = await GetObjectFromBlob<IOperableTrigger>(rs2, 0, cancellationToken).ConfigureAwait(false);
+                if (trigger is not null)
+                {
+                    trigger.SetNextFireTimeUtc(nextFireTimeUtc);
+                    trigger.SetPreviousFireTimeUtc(previousFireTimeUtc);
+
+                    if (misfireOrigFireTime.HasValue && trigger is AbstractTrigger at)
+                    {
+                        at.MisfiredFromFireTimeUtc = misfireOrigFireTime;
+                    }
+                }
             }
         }
         else
