diff --git a/pkg/cidata/cidata.TEMPLATE.d/user-data b/pkg/cidata/cidata.TEMPLATE.d/user-data
index 3842f8d4..2f9847a4 100644
--- a/pkg/cidata/cidata.TEMPLATE.d/user-data
+++ b/pkg/cidata/cidata.TEMPLATE.d/user-data
@@ -21,6 +21,8 @@ mounts:
 - [{{$m.Tag}}, {{$m.MountPoint}}, {{$m.Type}}, "{{$m.Options}}", "0", "0"]
     {{- end }}
   {{- end }}
+{{- else }}
+mounts: []
 {{- end }}
 
 {{- if .TimeZone }}
@@ -29,7 +31,7 @@ timezone: {{.TimeZone}}
 
 users:
   - name: "{{.User}}"
-    uid: "{{.UID}}"
+    uid: {{.UID}}
 {{- if .Comment }}
     gecos: {{ printf "%q" .Comment }}
 {{- end }}
@@ -37,7 +39,7 @@ users:
     shell: {{.Shell}}
     sudo: ALL=(ALL) NOPASSWD:ALL
     lock_passwd: true
-    ssh-authorized-keys:
+    ssh_authorized_keys:
     {{- range $val := .SSHPubKeys }}
       - {{ printf "%q" $val }}
     {{- end }}
diff --git a/pkg/cidata/cidata.go b/pkg/cidata/cidata.go
index 0bf39d3f..37126068 100644
--- a/pkg/cidata/cidata.go
+++ b/pkg/cidata/cidata.go
@@ -338,7 +338,7 @@ func GenerateCloudConfig(instDir, name string, instConfig *limayaml.LimaYAML) er
 		return err
 	}
 	// mounts are not included here
-	args.Mounts = nil
+	args.Mounts = []Mount{}
 	// resolv_conf is not included here
 	args.DNSAddresses = nil
 
diff --git a/pkg/cidata/template_test.go b/pkg/cidata/template_test.go
index 31f7b2df..f805734b 100644
--- a/pkg/cidata/template_test.go
+++ b/pkg/cidata/template_test.go
@@ -13,24 +13,6 @@ import (
 
 var defaultRemoveDefaults = false
 
-func TestConfig(t *testing.T) {
-	args := &TemplateArgs{
-		Name:    "default",
-		User:    "foo",
-		UID:     501,
-		Comment: "Foo",
-		Home:    "/home/foo.linux",
-		Shell:   "/bin/bash",
-		SSHPubKeys: []string{
-			"ssh-rsa dummy foo@example.com",
-		},
-		MountType: "reverse-sshfs",
-	}
-	config, err := ExecuteTemplateCloudConfig(args)
-	assert.NilError(t, err)
-	t.Log(string(config))
-	assert.Assert(t, !strings.Contains(string(config), "ca_certs:"))
-}
 
 func TestConfigCACerts(t *testing.T) {
 	args := &TemplateArgs{
@@ -44,6 +26,7 @@ func TestConfigCACerts(t *testing.T) {
 			"ssh-rsa dummy foo@example.com",
 		},
 		MountType: "reverse-sshfs",
+		Mounts:    []Mount{},
 		CACerts: CACerts{
 			RemoveDefaults: &defaultRemoveDefaults,
 		},
@@ -82,8 +65,9 @@ func TestTemplate(t *testing.T) {
 		assert.NilError(t, err)
 		t.Log(string(b))
 		if f.Path == "user-data" {
-			// mounted later
-			assert.Assert(t, !strings.Contains(string(b), "mounts:"))
+			// For reverse-sshfs, mounts are handled by lima-guestagent. The cloud-config should
+			// still include a valid (array-typed) mounts key to satisfy cloud-init schema.
+			assert.Assert(t, strings.Contains(string(b), "mounts: []"))
 			// ca_certs:
 			assert.Assert(t, !strings.Contains(string(b), "trusted:"))
 		}
