diff --git a/crates/turborepo-scope/src/target_selector.rs b/crates/turborepo-scope/src/target_selector.rs
index 3462b01..19ffa05 100644
--- a/crates/turborepo-scope/src/target_selector.rs
+++ b/crates/turborepo-scope/src/target_selector.rs
@@ -190,6 +190,19 @@ impl FromStr for TargetSelector {
                     allow_unknown_objects: false,
                     merge_base: true,
                 }
+            } else if let Some((a, b)) = commits_str.split_once("..") {
+                if a.is_empty() || b.is_empty() {
+                    return Err(InvalidSelectorError::InvalidGitRange(
+                        commits_str.to_string(),
+                    ));
+                }
+                GitRange {
+                    from_ref: Some(a.to_string()),
+                    to_ref: Some(b.to_string()),
+                    include_uncommitted: false,
+                    allow_unknown_objects: false,
+                    merge_base: false,
+                }
             } else {
                 // If only the start of the range is specified, we assume that
                 // we want to include uncommitted changes
