diff --git a/src/MudBlazor/Components/Collapse/MudCollapse.razor b/src/MudBlazor/Components/Collapse/MudCollapse.razor
index e9ec3de..b78e1c4 100644
--- a/src/MudBlazor/Components/Collapse/MudCollapse.razor
+++ b/src/MudBlazor/Components/Collapse/MudCollapse.razor
@@ -1,4 +1,4 @@
-﻿@namespace MudBlazor
+@namespace MudBlazor
 @inherits MudComponentBase
 
 <div @ontransitionend="this.AsNonRenderingEventHandler(AnimationEndAsync)"
@@ -8,4 +8,4 @@
             @ChildContent
         </div>
     </div>
-</div>
+</div>
\ No newline at end of file
diff --git a/src/MudBlazor/Components/Collapse/MudCollapse.razor.cs b/src/MudBlazor/Components/Collapse/MudCollapse.razor.cs
index 612a688..bf5c351 100644
--- a/src/MudBlazor/Components/Collapse/MudCollapse.razor.cs
+++ b/src/MudBlazor/Components/Collapse/MudCollapse.razor.cs
@@ -1,4 +1,4 @@
-﻿using Microsoft.AspNetCore.Components;
+using Microsoft.AspNetCore.Components;
 using MudBlazor.State;
 using MudBlazor.Utilities;
 
@@ -101,14 +101,17 @@ namespace MudBlazor
             {
                 _state = CollapseState.Entered;
                 StateHasChanged();
+                return OnAnimationEnd.InvokeAsync(_expandedState.Value);
             }
             else if (_state == CollapseState.Exiting)
             {
                 _state = CollapseState.Exited;
                 StateHasChanged();
+                return OnAnimationEnd.InvokeAsync(_expandedState.Value);
             }
 
-            return OnAnimationEnd.InvokeAsync(_expandedState.Value);
+            // Ignore transitionend events from child elements
+            return Task.CompletedTask;
         }
     }
-}
+}
\ No newline at end of file
