diff --git a/lib/node_modules/@stdlib/stats/base/stdevwd/lib/ndarray.js b/lib/node_modules/@stdlib/stats/base/stdevwd/lib/ndarray.js
index f31809bdc91..4afdcd52870 100644
--- a/lib/node_modules/@stdlib/stats/base/stdevwd/lib/ndarray.js
+++ b/lib/node_modules/@stdlib/stats/base/stdevwd/lib/ndarray.js
@@ -20,7 +20,9 @@
 
 // MODULES //
 
+var arraylike2object = require( '@stdlib/array/base/arraylike2object' );
 var variancewd = require( '@stdlib/stats/strided/variancewd' ).ndarray;
+var accessors = require( './accessors.js' );
 var sqrt = require( '@stdlib/math/base/special/sqrt' );
 
 
@@ -51,6 +53,11 @@ var sqrt = require( '@stdlib/math/base/special/sqrt' );
 * // returns 2.5
 */
 function stdevwd( N, correction, x, stride, offset ) {
+	var o;
+	o = arraylike2object( x );
+	if ( o.accessorProtocol ) {
+		return accessors( N, correction, o, stride, offset );
+	}
 	return sqrt( variancewd( N, correction, x, stride, offset ) );
 }
 
diff --git a/lib/node_modules/@stdlib/stats/base/stdevwd/test/test.ndarray.js b/lib/node_modules/@stdlib/stats/base/stdevwd/test/test.ndarray.js
index 0b5022acab7..9654cd2d78e 100644
--- a/lib/node_modules/@stdlib/stats/base/stdevwd/test/test.ndarray.js
+++ b/lib/node_modules/@stdlib/stats/base/stdevwd/test/test.ndarray.js
@@ -200,3 +200,4 @@ tape( 'the function supports an `offset` parameter', function test( t ) {
 
 	t.end();
 });
+
