diff --git a/Zend/zend_compile.c b/Zend/zend_compile.c
index 5527f55a25c..e62090dc188 100644
--- a/Zend/zend_compile.c
+++ b/Zend/zend_compile.c
@@ -7550,27 +7550,49 @@ static void zend_compile_attributes(
 	if (*attributes != NULL) {
 		/* Validate attributes in a secondary loop (needed to detect repeated attributes). */
 		ZEND_HASH_PACKED_FOREACH_PTR(*attributes, attr) {
-			if (attr->offset != offset || NULL == (config = zend_internal_attribute_get(attr->lcname))) {
+			if (attr->offset != offset) {
 				continue;
 			}
 
-			if (!(target & (config->flags & ZEND_ATTRIBUTE_TARGET_ALL))) {
-				zend_string *location = zend_get_attribute_target_names(target);
-				zend_string *allowed = zend_get_attribute_target_names(config->flags);
+			config = zend_internal_attribute_get(attr->lcname);
 
-				zend_error_noreturn(E_ERROR, "Attribute \"%s\" cannot target %s (allowed targets: %s)",
-					ZSTR_VAL(attr->name), ZSTR_VAL(location), ZSTR_VAL(allowed)
-				);
-			}
+			if (config != NULL) {
+				/* Validate internal attributes */
+				if (!(target & (config->flags & ZEND_ATTRIBUTE_TARGET_ALL))) {
+					zend_string *location = zend_get_attribute_target_names(target);
+					zend_string *allowed = zend_get_attribute_target_names(config->flags);
 
-			if (!(config->flags & ZEND_ATTRIBUTE_IS_REPEATABLE)) {
-				if (zend_is_attribute_repeated(*attributes, attr)) {
-					zend_error_noreturn(E_ERROR, "Attribute \"%s\" must not be repeated", ZSTR_VAL(attr->name));
+					zend_error_noreturn(E_ERROR, "Attribute \"%s\" cannot target %s (allowed targets: %s)",
+						ZSTR_VAL(attr->name), ZSTR_VAL(location), ZSTR_VAL(allowed)
+					);
+				}
+
+				if (!(config->flags & ZEND_ATTRIBUTE_IS_REPEATABLE)) {
+					if (zend_is_attribute_repeated(*attributes, attr)) {
+						zend_error_noreturn(E_ERROR, "Attribute \"%s\" must not be repeated", ZSTR_VAL(attr->name));
+					}
 				}
-			}
 
-			if (config->validator != NULL) {
-				config->validator(attr, target, CG(active_class_entry));
+				if (config->validator != NULL) {
+					config->validator(attr, target, CG(active_class_entry));
+				}
+			} else {
+				/* Validate user-defined attributes */
+				zend_class_entry *ce = zend_hash_find_ptr(CG(class_table), attr->lcname);
+				if (ce != NULL) {
+					if (ce->ce_flags & ZEND_ACC_TRAIT) {
+						zend_error_noreturn(E_ERROR, "Cannot use trait %s as attribute", ZSTR_VAL(attr->name));
+					}
+					if (ce->ce_flags & ZEND_ACC_INTERFACE) {
+						zend_error_noreturn(E_ERROR, "Cannot use interface %s as attribute", ZSTR_VAL(attr->name));
+					}
+					if (ce->ce_flags & ZEND_ACC_ENUM) {
+						zend_error_noreturn(E_ERROR, "Cannot use enum %s as attribute", ZSTR_VAL(attr->name));
+					}
+					if (ce->ce_flags & ZEND_ACC_EXPLICIT_ABSTRACT_CLASS) {
+						zend_error_noreturn(E_ERROR, "Cannot use abstract class %s as attribute", ZSTR_VAL(attr->name));
+					}
+				}
 			}
 		} ZEND_HASH_FOREACH_END();
 	}
