diff --git a/src/http-proxy-middleware.ts b/src/http-proxy-middleware.ts
index eb369f2..d2abc0d 100644
--- a/src/http-proxy-middleware.ts
+++ b/src/http-proxy-middleware.ts
@@ -103,7 +103,7 @@ export class HttpProxyMiddleware<TReq, TRes> {
       this.serverOnCloseSubscribed = true;
     }
 
-    if (this.proxyOptions.ws === true) {
+    if (this.proxyOptions.ws === true && server) {
       // use initial request to access the server object to subscribe to http upgrade event
       this.catchUpgradeRequest(server);
     }
