diff --git a/tools/api-checker/config/gwt212_213userapi.conf b/tools/api-checker/config/gwt212_213userapi.conf
index 1c09dc514..1ee1b20e1 100644
--- a/tools/api-checker/config/gwt212_213userapi.conf
+++ b/tools/api-checker/config/gwt212_213userapi.conf
@@ -164,7 +164,5 @@ java.lang.annotation.Native MISSING
 java.lang.annotation.Repeatable MISSING
 java.lang.annotation.Retention MISSING
 java.lang.annotation.Target MISSING
-javax.annotation MISSING
-javax.annotation.processing MISSING
 org.junit.runner.RunWith MISSING
 org.junit.runners.Suite.SuiteClasses MISSING
