diff --git a/modules/openapi-generator/src/main/resources/kotlin-spring/enumClass.mustache b/modules/openapi-generator/src/main/resources/kotlin-spring/enumClass.mustache
index fafbb66708d..7c931469198 100644
--- a/modules/openapi-generator/src/main/resources/kotlin-spring/enumClass.mustache
+++ b/modules/openapi-generator/src/main/resources/kotlin-spring/enumClass.mustache
@@ -12,5 +12,11 @@ enum class {{classname}}(@get:JsonValue val value: {{dataType}}) {
         fun forValue(value: {{dataType}}): {{classname}} {
                 return values().first{it -> it.value == value}
         }
+
+        /**
+         * Used by Spring's default String-to-Enum conversion in query/path params.
+         */
+        @JvmStatic
+        fun fromString(value: {{dataType}}): {{classname}} = forValue(value)
     }
 }
diff --git a/modules/openapi-generator/src/test/java/org/openapitools/codegen/kotlin/spring/KotlinSpringServerCodegenTest.java b/modules/openapi-generator/src/test/java/org/openapitools/codegen/kotlin/spring/KotlinSpringServerCodegenTest.java
index ba7b73e78ee..723ac069ded 100644
--- a/modules/openapi-generator/src/test/java/org/openapitools/codegen/kotlin/spring/KotlinSpringServerCodegenTest.java
+++ b/modules/openapi-generator/src/test/java/org/openapitools/codegen/kotlin/spring/KotlinSpringServerCodegenTest.java
@@ -366,7 +366,8 @@ public class KotlinSpringServerCodegenTest {
     public void arrayItemsCanBeNullable() throws IOException {
         File output = Files.createTempDirectory("test").toFile().getCanonicalFile();
         output.deleteOnExit();
-        String outputPath = output.getAbsolutePath().replace('\\', '/');
+        String outputPath = output.getAbsolutePath().replace("\\", "/");
+
 
         OpenAPI openAPI = new OpenAPIParser()
                 .readLocation("src/test/resources/3_0/array-nullable-items.yaml", null, new ParseOptions()).getOpenAPI();
@@ -397,7 +398,8 @@ public class KotlinSpringServerCodegenTest {
     public void doNotGenerateRequestParamForObjectQueryParam() throws IOException {
         File output = Files.createTempDirectory("test").toFile().getCanonicalFile();
         output.deleteOnExit();
-        String outputPath = output.getAbsolutePath().replace('\\', '/');
+        String outputPath = output.getAbsolutePath().replace("\\", "/");
+
 
         OpenAPI openAPI = new OpenAPIParser()
                 .readLocation("src/test/resources/3_0/objectQueryParam.yaml", null, new ParseOptions()).getOpenAPI();
@@ -427,7 +429,8 @@ public class KotlinSpringServerCodegenTest {
     public void doGenerateRequestParamForSimpleParam() throws IOException {
         File output = Files.createTempDirectory("test").toFile().getCanonicalFile();
         output.deleteOnExit();
-        String outputPath = output.getAbsolutePath().replace('\\', '/');
+        String outputPath = output.getAbsolutePath().replace("\\", "/");
+
 
         OpenAPI openAPI = new OpenAPIParser()
                 .readLocation("src/test/resources/3_0/issue_3248.yaml", null, new ParseOptions()).getOpenAPI();
@@ -464,7 +467,8 @@ public class KotlinSpringServerCodegenTest {
     public void generateFormatForDateAndDateTimeQueryParam() throws IOException {
         File output = Files.createTempDirectory("test").toFile().getCanonicalFile();
         output.deleteOnExit();
-        String outputPath = output.getAbsolutePath().replace('\\', '/');
+        String outputPath = output.getAbsolutePath().replace("\\", "/");
+
 
         OpenAPI openAPI = new OpenAPIParser()
                 .readLocation("src/test/resources/3_0/issue_2053.yaml", null, new ParseOptions()).getOpenAPI();
@@ -500,7 +504,8 @@ public class KotlinSpringServerCodegenTest {
     public void beanQualifiers() throws Exception {
         File output = Files.createTempDirectory("test").toFile().getCanonicalFile();
         output.deleteOnExit();
-        String outputPath = output.getAbsolutePath().replace('\\', '/');
+        String outputPath = output.getAbsolutePath().replace("\\", "/");
+
 
         KotlinSpringServerCodegen codegen = new KotlinSpringServerCodegen();
         codegen.setOutputDir(output.getAbsolutePath());
@@ -521,7 +526,8 @@ public class KotlinSpringServerCodegenTest {
     public void skipDefaultInterface() throws Exception {
         File output = Files.createTempDirectory("test").toFile().getCanonicalFile();
         output.deleteOnExit();
-        String outputPath = output.getAbsolutePath().replace('\\', '/');
+        String outputPath = output.getAbsolutePath().replace("\\", "/");
+
 
         KotlinSpringServerCodegen codegen = new KotlinSpringServerCodegen();
         codegen.setOutputDir(output.getAbsolutePath());
@@ -543,7 +549,8 @@ public class KotlinSpringServerCodegenTest {
     public void cookieParameterGenerationApis() throws Exception {
         File output = Files.createTempDirectory("test").toFile().getCanonicalFile();
         output.deleteOnExit();
-        String outputPath = output.getAbsolutePath().replace('\\', '/');
+        String outputPath = output.getAbsolutePath().replace("\\", "/");
+
 
         KotlinSpringServerCodegen codegen = new KotlinSpringServerCodegen();
         codegen.setOutputDir(output.getAbsolutePath());
@@ -565,7 +572,8 @@ public class KotlinSpringServerCodegenTest {
     public void cookieParameterGenerationControllers() throws Exception {
         File output = Files.createTempDirectory("test").toFile().getCanonicalFile();
         output.deleteOnExit();
-        String outputPath = output.getAbsolutePath().replace('\\', '/');
+        String outputPath = output.getAbsolutePath().replace("\\", "/");
+
 
         KotlinSpringServerCodegen codegen = new KotlinSpringServerCodegen();
         codegen.setOutputDir(output.getAbsolutePath());
@@ -585,7 +593,8 @@ public class KotlinSpringServerCodegenTest {
     public void useSpringBoot3() throws Exception {
         File output = Files.createTempDirectory("test").toFile().getCanonicalFile();
         output.deleteOnExit();
-        String outputPath = output.getAbsolutePath().replace('\\', '/');
+        String outputPath = output.getAbsolutePath().replace("\\", "/");
+
 
         KotlinSpringServerCodegen codegen = new KotlinSpringServerCodegen();
         codegen.setOutputDir(output.getAbsolutePath());
@@ -623,7 +632,8 @@ public class KotlinSpringServerCodegenTest {
     private static void testMultiLineOperationDescription(final boolean isInterfaceOnly) throws IOException {
         File output = Files.createTempDirectory("test").toFile().getCanonicalFile();
         output.deleteOnExit();
-        String outputPath = output.getAbsolutePath().replace('\\', '/');
+        String outputPath = output.getAbsolutePath().replace("\\", "/");
+
 
         KotlinSpringServerCodegen codegen = new KotlinSpringServerCodegen();
         codegen.setOutputDir(output.getAbsolutePath());
@@ -658,7 +668,8 @@ public class KotlinSpringServerCodegenTest {
     public void useTargetOnInterfaceAnnotations() throws IOException {
         File output = Files.createTempDirectory("test").toFile().getCanonicalFile();
         output.deleteOnExit();
-        String outputPath = output.getAbsolutePath().replace('\\', '/');
+        String outputPath = output.getAbsolutePath().replace("\\", "/");
+
 
         KotlinSpringServerCodegen codegen = new KotlinSpringServerCodegen();
         codegen.setOutputDir(output.getAbsolutePath());
@@ -690,7 +701,8 @@ public class KotlinSpringServerCodegenTest {
     public void useTargetOnInterfaceAnnotationsWithSwagger1() throws IOException {
         File output = Files.createTempDirectory("test").toFile().getCanonicalFile();
         output.deleteOnExit();
-        String outputPath = output.getAbsolutePath().replace('\\', '/');
+        String outputPath = output.getAbsolutePath().replace("\\", "/");
+
 
         KotlinSpringServerCodegen codegen = new KotlinSpringServerCodegen();
         codegen.setOutputDir(output.getAbsolutePath());
@@ -752,7 +764,8 @@ public class KotlinSpringServerCodegenTest {
     public void givenMultipartFormArray_whenGenerateDelegateAndService_thenParameterIsCreatedAsListOfMultipartFile() throws IOException {
         File output = Files.createTempDirectory("test").toFile().getCanonicalFile();
         output.deleteOnExit();
-        String outputPath = output.getAbsolutePath().replace('\\', '/');
+        String outputPath = output.getAbsolutePath().replace("\\", "/");
+
 
         final OpenAPI openAPI = TestUtils.parseFlattenSpec("src/test/resources/3_0/kotlin/petstore-with-tags.yaml");
         final KotlinSpringServerCodegen codegen = new KotlinSpringServerCodegen();
@@ -791,7 +804,8 @@ public class KotlinSpringServerCodegenTest {
     public void givenOctetStreamResponseType_whenGenerateServer_thenReturnTypeIsResource() throws IOException {
         File output = Files.createTempDirectory("test").toFile().getCanonicalFile();
         output.deleteOnExit();
-        String outputPath = output.getAbsolutePath().replace('\\', '/');
+        String outputPath = output.getAbsolutePath().replace("\\", "/");
+
 
         final OpenAPI openAPI = TestUtils.parseFlattenSpec("src/test/resources/3_0/kotlin/petstore-with-tags.yaml");
         final KotlinSpringServerCodegen codegen = new KotlinSpringServerCodegen();
@@ -821,7 +835,8 @@ public class KotlinSpringServerCodegenTest {
     public void givenMultipartForm_whenGenerateReactiveServer_thenParameterAreCreatedAsRequestPart() throws IOException {
         File output = Files.createTempDirectory("test").toFile().getCanonicalFile();
         output.deleteOnExit();
-        String outputPath = output.getAbsolutePath().replace('\\', '/');
+        String outputPath = output.getAbsolutePath().replace("\\", "/");
+
 
         final OpenAPI openAPI = TestUtils.parseFlattenSpec("src/test/resources/3_0/kotlin/petstore-with-tags.yaml");
         final KotlinSpringServerCodegen codegen = new KotlinSpringServerCodegen();
@@ -856,7 +871,8 @@ public class KotlinSpringServerCodegenTest {
 
         File output = Files.createTempDirectory("test").toFile().getCanonicalFile();
         output.deleteOnExit();
-        String outputPath = output.getAbsolutePath().replace('\\', '/');
+        String outputPath = output.getAbsolutePath().replace("\\", "/");
+
 
         KotlinSpringServerCodegen codegen = new KotlinSpringServerCodegen();
         codegen.setOutputDir(output.getAbsolutePath());
@@ -885,7 +901,8 @@ public class KotlinSpringServerCodegenTest {
     public void generateSerializableModel() throws Exception {
         File output = Files.createTempDirectory("test").toFile().getCanonicalFile();
         output.deleteOnExit();
-        String outputPath = output.getAbsolutePath().replace('\\', '/');
+        String outputPath = output.getAbsolutePath().replace("\\", "/");
+
 
         KotlinSpringServerCodegen codegen = new KotlinSpringServerCodegen();
         codegen.setOutputDir(output.getAbsolutePath());
