diff --git a/src/instructlab/configuration.py b/src/instructlab/configuration.py
index 3e353953..4b7703ec 100644
--- a/src/instructlab/configuration.py
+++ b/src/instructlab/configuration.py
@@ -154,7 +154,7 @@ class _serve_vllm(BaseModel):
     llm_family: str = Field(
         default="",  # TODO: convert to None and use a pattern to validate
         description="Large Language Model Family",
-        examples=["merlinite", "granite"],
+        examples=["instructlab", "granite"],
     )
     max_startup_attempts: int | None = Field(
         default=120,
@@ -186,7 +186,7 @@ class _serve_llama_cpp(BaseModel):
     llm_family: str = Field(
         default="",  # TODO: convert to None and use a pattern to validate
         description="Large Language Model Family",
-        examples=["merlinite", "granite"],
+        examples=["instructlab", "granite"],
     )
 
 
diff --git a/src/instructlab/defaults.py b/src/instructlab/defaults.py
index 930ccc0b..380cd82f 100644
--- a/src/instructlab/defaults.py
+++ b/src/instructlab/defaults.py
@@ -11,11 +11,13 @@ CONFIG_FILENAME = "config.yaml"
 CONFIG_VERSION = "1.0.0"
 
 # Model families understood by ilab
-MODEL_FAMILIES = {"merlinite", "mixtral"}
+MODEL_FAMILIES = {"instructlab", "merlinite", "mixtral"}
 
 # Map model names to their family
 MODEL_FAMILY_MAPPINGS = {
-    "granite": "merlinite",
+    "merlinite": "instructlab",
+    "granite": "instructlab",
+    "mixtral": "mixtral",
 }
 
 # Indentation constant
@@ -57,10 +59,11 @@ class _InstructlabDefaults:
     ILAB_SYSTEM_PROFILE_DIR = "ILAB_SYSTEM_PROFILE_DIR"
 
     # TODO: Consolidate --model and --model-path into one --model-path flag since we always need a path now
-    MODEL_NAME_OLD = "merlinite-7b-lab-Q4_K_M"
+    MODEL_NAME_OLD = "granite-7b-lab-Q4_K_M"
     MERLINITE_GGUF_REPO = "instructlab/merlinite-7b-lab-GGUF"
+    GRANITE_GGUF_REPO = "instructlab/granite-7b-lab-GGUF"
     MISTRAL_GGUF_REPO = "TheBloke/Mistral-7B-Instruct-v0.2-GGUF"
-    GGUF_MODEL_NAME = "merlinite-7b-lab-Q4_K_M.gguf"
+    GGUF_MODEL_NAME = "granite-7b-lab-Q4_K_M.gguf"
     MISTRAL_GGUF_MODEL_NAME = "mistral-7b-instruct-v0.2.Q4_K_M.gguf"
     MODEL_REPO = "instructlab/granite-7b-lab"
     JUDGE_MODEL_MT = "prometheus-eval/prometheus-8x7b-v2.0"
@@ -77,7 +80,7 @@ class _InstructlabDefaults:
     SDG_SCALE_FACTOR = 30
 
     # When otherwise unknown, ilab uses this as the default family
-    MODEL_FAMILY = "merlinite"
+    MODEL_FAMILY = "granite"
     ADDITIONAL_ARGS_DEFAULTS = {
         "learning_rate": 2e-5,
         "warmup_steps": 25,
diff --git a/src/instructlab/model/backends/common.py b/src/instructlab/model/backends/common.py
index d8b642a9..c5d35278 100644
--- a/src/instructlab/model/backends/common.py
+++ b/src/instructlab/model/backends/common.py
@@ -19,7 +19,7 @@ LLAMA_CPP = "llama-cpp"
 VLLM = "vllm"
 templates = [
     {
-        "model": "merlinite",
+        "model": "instructlab",
         "template": "{% for message in messages %}\n{% if message['role'] == 'user' %}\n{{ '<|user|>\n' + message['content'] }}\n{% elif message['role'] == 'system' %}\n{{ '<|system|>\n' + message['content'] }}\n{% elif message['role'] == 'assistant' %}\n{{ '<|assistant|>\n' + message['content'] + eos_token }}\n{% endif %}\n{% if loop.last and add_generation_prompt %}\n{{ '<|assistant|>' }}\n{% endif %}\n{% endfor %}",
     },
     {
