diff --git a/application/src/main/java/org/opentripplanner/routing/algorithm/raptoradapter/transit/request/TripTimesForDaysIndex.java b/application/src/main/java/org/opentripplanner/routing/algorithm/raptoradapter/transit/request/TripTimesForDaysIndex.java
index 9613af9002..62a977666c 100644
--- a/application/src/main/java/org/opentripplanner/routing/algorithm/raptoradapter/transit/request/TripTimesForDaysIndex.java
+++ b/application/src/main/java/org/opentripplanner/routing/algorithm/raptoradapter/transit/request/TripTimesForDaysIndex.java
@@ -50,6 +50,14 @@ final class TripTimesForDaysIndex {
     // 'list' is an alias to make the logic below easier to read
     final List<int[]> list = firstStopDepartureTimesPerDay;
     this.tripIndex = new int[list.stream().mapToInt(a -> a.length).sum() * 2];
+
+    // No departures on any day. This can happen for patterns that are active on a set of dates,
+    // but where the service window contains no trips (or trips are filtered away).
+    // In this case, there is nothing to index.
+    if (this.tripIndex.length == 0 || list.isEmpty() || list.get(0).length == 0) {
+      return;
+    }
+
     int[] a;
 
     // 'day' is the current day index
@@ -139,6 +147,9 @@ final class TripTimesForDaysIndex {
 
   @Override
   public String toString() {
+    if (tripIndex.length == 0) {
+      return "";
+    }
     var buf = new StringBuilder();
     for (int i = 0; i < tripIndex.length; i += 2) {
       buf.append(tripIndex[i]).append(':').append(tripIndex[i + 1]).append(' ');
