diff --git a/src/options/file_name.rs b/src/options/file_name.rs
index 700b908..5df9b1e 100644
--- a/src/options/file_name.rs
+++ b/src/options/file_name.rs
@@ -95,10 +95,10 @@ impl QuoteStyle {
 
 impl EmbedHyperlinks {
     fn deduce(matches: &ArgMatches) -> Self {
-        if matches.get_flag("hyperlink") {
-            Self::On
-        } else {
-            Self::Off
+        match matches.get_one("hyperlink") {
+            Some(ShowWhen::Always) => Self::On,
+            Some(ShowWhen::Auto) => Self::Auto,
+            None | Some(ShowWhen::Never) => Self::Off,
         }
     }
 }
diff --git a/src/options/parser.rs b/src/options/parser.rs
index 526188a..1eb2fb4 100644
--- a/src/options/parser.rs
+++ b/src/options/parser.rs
@@ -85,7 +85,11 @@ pub fn get_command() -> clap::Command {
             .num_args(0..=1)
             .value_parser(value_parser!(ShowWhen))
             .default_missing_value("auto"))
-        .arg(arg!(--hyperlink "display entries as hyperlinks"))
+        .arg(arg!(--hyperlink <WHEN> "display entries as hyperlinks")
+            .num_args(0..=1)
+            .value_parser(value_parser!(ShowWhen))
+            .default_missing_value("always")
+            .require_equals(true))
         .arg(arg!(--"no-quotes" "don't quote file names with spaces"))
 
         .next_help_heading("FILTERING OPTIONS")
diff --git a/src/output/file_name.rs b/src/output/file_name.rs
index 6e14e20..eb23644 100644
--- a/src/output/file_name.rs
+++ b/src/output/file_name.rs
@@ -123,6 +123,7 @@ pub enum ShowIcons {
 pub enum EmbedHyperlinks {
     Off,
     On,
+    Auto,
 }
 
 /// Whether to show absolute paths
@@ -413,7 +414,10 @@ impl<C: Colours> FileName<'_, '_, C> {
         let mut bits = Vec::new();
 
         let mut display_hyperlink = false;
-        if self.options.embed_hyperlinks == EmbedHyperlinks::On
+        let hyperlink_active = self.options.embed_hyperlinks == EmbedHyperlinks::On
+            || (self.options.embed_hyperlinks == EmbedHyperlinks::Auto
+                && self.options.is_a_tty);
+        if hyperlink_active
             && let Some(abs_path) = self
                 .file
                 .absolute_path()
