diff --git a/src/Spectre.Console/Prompts/TextPrompt.cs b/src/Spectre.Console/Prompts/TextPrompt.cs
index 5418ef8..9a5b6fd 100644
--- a/src/Spectre.Console/Prompts/TextPrompt.cs
+++ b/src/Spectre.Console/Prompts/TextPrompt.cs
@@ -7,7 +7,7 @@ namespace Spectre.Console;
 public sealed class TextPrompt<T> : IPrompt<T>, IHasCulture
 {
     private readonly string _prompt;
-    private readonly StringComparer? _comparer;
+    private readonly StringComparer _comparer;
 
     /// <summary>
     /// Gets or sets the prompt style.
@@ -105,11 +105,11 @@ public sealed class TextPrompt<T> : IPrompt<T>, IHasCulture
     /// Initializes a new instance of the <see cref="TextPrompt{T}"/> class.
     /// </summary>
     /// <param name="prompt">The prompt markup text.</param>
-    /// <param name="comparer">The comparer used for choices.</param>
+    /// <param name="comparer">The comparer used for choices. Defaults to <see cref="StringComparer.CurrentCultureIgnoreCase"/>.</param>
     public TextPrompt(string prompt, StringComparer? comparer = null)
     {
         _prompt = prompt ?? throw new System.ArgumentNullException(nameof(prompt));
-        _comparer = comparer;
+        _comparer = comparer ?? StringComparer.CurrentCultureIgnoreCase;
     }
 
     /// <summary>
@@ -293,4 +293,4 @@ public sealed class TextPrompt<T> : IPrompt<T>, IHasCulture
         message = null;
         return true;
     }
-}
+}
\ No newline at end of file
