diff --git a/mesa/experimental/cell_space/discrete_space.py b/mesa/experimental/cell_space/discrete_space.py
index 7ed06a84..802574bf 100644
--- a/mesa/experimental/cell_space/discrete_space.py
+++ b/mesa/experimental/cell_space/discrete_space.py
@@ -2,7 +2,7 @@
 
 from __future__ import annotations
 
-from collections.abc import Callable
+from collections.abc import Callable, Iterable
 from functools import cached_property
 from random import Random
 from typing import Any, Generic, TypeVar
@@ -73,6 +73,11 @@ class DiscreteSpace(Generic[T]):
     def empties(self) -> CellCollection[T]:
         """Return all empty in spaces."""
         return self.all_cells.select(lambda cell: cell.is_empty)
+        
+    @property
+    def agents(self) -> Iterable["CellAgent"]:
+        """Returns an iterator of all agents in the space."""
+        return self.all_cells.agents
 
     def select_random_empty_cell(self) -> T:
         """Select random empty cell."""
diff --git a/mesa/space.py b/mesa/space.py
index e3991fcc..5b29c79b 100644
--- a/mesa/space.py
+++ b/mesa/space.py
@@ -151,6 +151,12 @@ class _Grid:
         if not self._empties_built:
             self.build_empties()
         return self._empties
+        
+    @property
+    def agents(self) -> Iterator[Agent]:
+        """Returns an iterator of all agents in the grid."""
+        coordinates = list(itertools.product(range(self.width), range(self.height)))
+        return self.iter_cell_list_contents(coordinates)
 
     def build_empties(self) -> None:
         self._empties = set(
@@ -1343,6 +1349,11 @@ class ContinuousSpace:
         self._agent_points: npt.NDArray[FloatCoordinate] | None = None
         self._index_to_agent: dict[int, Agent] = {}
         self._agent_to_index: dict[Agent, int | None] = {}
+        
+    @property
+    def agents(self) -> Iterator[Agent]:
+        """Returns an iterator of all agents in the space."""
+        return iter(self._agent_to_index)
 
     def _build_agent_cache(self):
         """Cache agents positions to speed up neighbors calculations."""
@@ -1521,6 +1532,11 @@ class NetworkGrid:
     def default_val() -> list:
         """Default value for a new node."""
         return []
+        
+    @property
+    def agents(self) -> Iterator[Agent]:
+        """Returns an iterator of all agents in the network."""
+        return self.iter_cell_list_contents(self.G.nodes)
 
     @warn_if_agent_has_position_already
     def place_agent(self, agent: Agent, node_id: int) -> None:
