diff --git a/haystack/components/converters/azure.py b/haystack/components/converters/azure.py
index edcc348d..0d0569f2 100644
--- a/haystack/components/converters/azure.py
+++ b/haystack/components/converters/azure.py
@@ -61,7 +61,7 @@ class AzureOCRDocumentConverter:
         merge_multiple_column_headers: bool = True,
         page_layout: Literal["natural", "single_column"] = "natural",
         threshold_y: Optional[float] = 0.05,
-        store_full_path: bool = True,
+        store_full_path: bool = False,
     ):
         """
         Creates an AzureOCRDocumentConverter component.
@@ -143,12 +143,6 @@ class AzureOCRDocumentConverter:
             azure_output.append(result.to_dict())
 
             merged_metadata = {**bytestream.meta, **metadata}
-            warnings.warn(
-                "The `store_full_path` parameter defaults to True, storing full file paths in metadata. "
-                "In the 2.9.0 release, the default value for `store_full_path` will change to False, "
-                "storing only file names to improve privacy.",
-                DeprecationWarning,
-            )
 
             if not self.store_full_path and (file_path := bytestream.meta.get("file_path")):
                 merged_metadata["file_path"] = os.path.basename(file_path)
diff --git a/haystack/components/converters/csv.py b/haystack/components/converters/csv.py
index 2b36d4cc..8259f43c 100644
--- a/haystack/components/converters/csv.py
+++ b/haystack/components/converters/csv.py
@@ -36,7 +36,7 @@ class CSVToDocument:
     ```
     """
 
-    def __init__(self, encoding: str = "utf-8", store_full_path: bool = True):
+    def __init__(self, encoding: str = "utf-8", store_full_path: bool = False):
         """
         Creates a CSVToDocument component.
 
@@ -94,13 +94,6 @@ class CSVToDocument:
 
             merged_metadata = {**bytestream.meta, **metadata}
 
-            warnings.warn(
-                "The `store_full_path` parameter defaults to True, storing full file paths in metadata. "
-                "In the 2.9.0 release, the default value for `store_full_path` will change to False, "
-                "storing only file names to improve privacy.",
-                DeprecationWarning,
-            )
-
             if not self.store_full_path and "file_path" in bytestream.meta:
                 file_path = bytestream.meta.get("file_path")
                 if file_path:  # Ensure the value is not None for pylint
diff --git a/haystack/components/converters/docx.py b/haystack/components/converters/docx.py
index 604b1a51..54cf9e58 100644
--- a/haystack/components/converters/docx.py
+++ b/haystack/components/converters/docx.py
@@ -109,7 +109,7 @@ class DOCXToDocument:
     ```
     """
 
-    def __init__(self, table_format: Union[str, DOCXTableFormat] = DOCXTableFormat.CSV, store_full_path: bool = True):
+    def __init__(self, table_format: Union[str, DOCXTableFormat] = DOCXTableFormat.CSV, store_full_path: bool = False):
         """
         Create a DOCXToDocument component.
 
@@ -189,13 +189,6 @@ class DOCXToDocument:
                 )
                 continue
 
-            warnings.warn(
-                "The `store_full_path` parameter defaults to True, storing full file paths in metadata. "
-                "In the 2.9.0 release, the default value for `store_full_path` will change to False, "
-                "storing only file names to improve privacy.",
-                DeprecationWarning,
-            )
-
             docx_metadata = self._get_docx_metadata(document=docx_document)
             merged_metadata = {**bytestream.meta, **metadata, "docx": docx_metadata}
 
diff --git a/haystack/components/converters/html.py b/haystack/components/converters/html.py
index 4c226ab9..227fb902 100644
--- a/haystack/components/converters/html.py
+++ b/haystack/components/converters/html.py
@@ -35,7 +35,7 @@ class HTMLToDocument:
     ```
     """
 
-    def __init__(self, extraction_kwargs: Optional[Dict[str, Any]] = None, store_full_path: bool = True):
+    def __init__(self, extraction_kwargs: Optional[Dict[str, Any]] = None, store_full_path: bool = False):
         """
         Create an HTMLToDocument component.
 
@@ -122,13 +122,6 @@ class HTMLToDocument:
                 continue
 
             merged_metadata = {**bytestream.meta, **metadata}
-
-            warnings.warn(
-                "The `store_full_path` parameter defaults to True, storing full file paths in metadata. "
-                "In the 2.9.0 release, the default value for `store_full_path` will change to False, "
-                "storing only file names to improve privacy.",
-                DeprecationWarning,
-            )
             if not self.store_full_path and "file_path" in bytestream.meta:
                 file_path = bytestream.meta.get("file_path")
                 if file_path:  # Ensure the value is not None for pylint
diff --git a/haystack/components/converters/json.py b/haystack/components/converters/json.py
index e6fc5646..4066466b 100644
--- a/haystack/components/converters/json.py
+++ b/haystack/components/converters/json.py
@@ -95,7 +95,7 @@ class JSONConverter:
         jq_schema: Optional[str] = None,
         content_key: Optional[str] = None,
         extra_meta_fields: Optional[Union[Set[str], Literal["*"]]] = None,
-        store_full_path: bool = True,
+        store_full_path: bool = False,
     ):
         """
         Creates a JSONConverter component.
@@ -279,13 +279,6 @@ class JSONConverter:
                 continue
 
             data = self._get_content_and_meta(bytestream)
-
-            warnings.warn(
-                "The `store_full_path` parameter defaults to True, storing full file paths in metadata. "
-                "In the 2.9.0 release, the default value for `store_full_path` will change to False, "
-                "storing only file names to improve privacy.",
-                DeprecationWarning,
-            )
             for text, extra_meta in data:
                 merged_metadata = {**bytestream.meta, **metadata, **extra_meta}
 
diff --git a/haystack/components/converters/markdown.py b/haystack/components/converters/markdown.py
index 0bd722c7..f078d91d 100644
--- a/haystack/components/converters/markdown.py
+++ b/haystack/components/converters/markdown.py
@@ -40,7 +40,7 @@ class MarkdownToDocument:
     ```
     """
 
-    def __init__(self, table_to_single_line: bool = False, progress_bar: bool = True, store_full_path: bool = True):
+    def __init__(self, table_to_single_line: bool = False, progress_bar: bool = True, store_full_path: bool = False):
         """
         Create a MarkdownToDocument component.
 
@@ -112,13 +112,6 @@ class MarkdownToDocument:
 
             merged_metadata = {**bytestream.meta, **metadata}
 
-            warnings.warn(
-                "The `store_full_path` parameter defaults to True, storing full file paths in metadata. "
-                "In the 2.9.0 release, the default value for `store_full_path` will change to False, "
-                "storing only file names to improve privacy.",
-                DeprecationWarning,
-            )
-
             if not self.store_full_path and (file_path := bytestream.meta.get("file_path")):
                 merged_metadata["file_path"] = os.path.basename(file_path)
 
diff --git a/haystack/components/converters/pdfminer.py b/haystack/components/converters/pdfminer.py
index a66fde92..71b90b76 100644
--- a/haystack/components/converters/pdfminer.py
+++ b/haystack/components/converters/pdfminer.py
@@ -48,7 +48,7 @@ class PDFMinerToDocument:
         boxes_flow: Optional[float] = 0.5,
         detect_vertical: bool = True,
         all_texts: bool = False,
-        store_full_path: bool = True,
+        store_full_path: bool = False,
     ) -> None:
         """
         Create a PDFMinerToDocument component.
@@ -172,12 +172,6 @@ class PDFMinerToDocument:
                 )
 
             merged_metadata = {**bytestream.meta, **metadata}
-            warnings.warn(
-                "The `store_full_path` parameter defaults to True, storing full file paths in metadata. "
-                "In the 2.9.0 release, the default value for `store_full_path` will change to False, "
-                "storing only file names to improve privacy.",
-                DeprecationWarning,
-            )
 
             if not self.store_full_path and (file_path := bytestream.meta.get("file_path")):
                 merged_metadata["file_path"] = os.path.basename(file_path)
diff --git a/haystack/components/converters/pptx.py b/haystack/components/converters/pptx.py
index f671acab..c19c2906 100644
--- a/haystack/components/converters/pptx.py
+++ b/haystack/components/converters/pptx.py
@@ -37,7 +37,7 @@ class PPTXToDocument:
     ```
     """
 
-    def __init__(self, store_full_path: bool = True):
+    def __init__(self, store_full_path: bool = False):
         """
         Create an PPTXToDocument component.
 
@@ -104,12 +104,6 @@ class PPTXToDocument:
                 continue
 
             merged_metadata = {**bytestream.meta, **metadata}
-            warnings.warn(
-                "The `store_full_path` parameter defaults to True, storing full file paths in metadata. "
-                "In the 2.9.0 release, the default value for `store_full_path` will change to False, "
-                "storing only file names to improve privacy.",
-                DeprecationWarning,
-            )
 
             if not self.store_full_path and (file_path := bytestream.meta.get("file_path")):
                 merged_metadata["file_path"] = os.path.basename(file_path)
diff --git a/haystack/components/converters/pypdf.py b/haystack/components/converters/pypdf.py
index 2086873c..ee5ad97b 100644
--- a/haystack/components/converters/pypdf.py
+++ b/haystack/components/converters/pypdf.py
@@ -79,7 +79,7 @@ class PyPDFToDocument:
         layout_mode_scale_weight: float = 1.25,
         layout_mode_strip_rotated: bool = True,
         layout_mode_font_height_weight: float = 1.0,
-        store_full_path: bool = True,
+        store_full_path: bool = False,
     ):
         """
         Create an PyPDFToDocument component.
@@ -220,12 +220,6 @@ class PyPDFToDocument:
                 )
 
             merged_metadata = {**bytestream.meta, **metadata}
-            warnings.warn(
-                "The `store_full_path` parameter defaults to True, storing full file paths in metadata. "
-                "In the 2.9.0 release, the default value for `store_full_path` will change to False, "
-                "storing only file names to improve privacy.",
-                DeprecationWarning,
-            )
             if not self.store_full_path and (file_path := bytestream.meta.get("file_path")):
                 merged_metadata["file_path"] = os.path.basename(file_path)
             document.meta = merged_metadata
diff --git a/haystack/components/converters/tika.py b/haystack/components/converters/tika.py
index 7a8fdb36..7386cbb5 100644
--- a/haystack/components/converters/tika.py
+++ b/haystack/components/converters/tika.py
@@ -75,7 +75,7 @@ class TikaDocumentConverter:
     ```
     """
 
-    def __init__(self, tika_url: str = "http://localhost:9998/tika", store_full_path: bool = True):
+    def __init__(self, tika_url: str = "http://localhost:9998/tika", store_full_path: bool = False):
         """
         Create a TikaDocumentConverter component.
 
@@ -139,12 +139,6 @@ class TikaDocumentConverter:
                 continue
 
             merged_metadata = {**bytestream.meta, **metadata}
-            warnings.warn(
-                "The `store_full_path` parameter defaults to True, storing full file paths in metadata. "
-                "In the 2.9.0 release, the default value for `store_full_path` will change to False, "
-                "storing only file names to improve privacy.",
-                DeprecationWarning,
-            )
 
             if not self.store_full_path and (file_path := bytestream.meta.get("file_path")):
                 merged_metadata["file_path"] = os.path.basename(file_path)
diff --git a/haystack/components/converters/txt.py b/haystack/components/converters/txt.py
index 8d7a12a4..6720347c 100644
--- a/haystack/components/converters/txt.py
+++ b/haystack/components/converters/txt.py
@@ -36,7 +36,7 @@ class TextFileToDocument:
     ```
     """
 
-    def __init__(self, encoding: str = "utf-8", store_full_path: bool = True):
+    def __init__(self, encoding: str = "utf-8", store_full_path: bool = False):
         """
         Creates a TextFileToDocument component.
 
@@ -93,12 +93,6 @@ class TextFileToDocument:
                 continue
 
             merged_metadata = {**bytestream.meta, **metadata}
-            warnings.warn(
-                "The `store_full_path` parameter defaults to True, storing full file paths in metadata. "
-                "In the 2.9.0 release, the default value for `store_full_path` will change to False, "
-                "storing only file names to improve privacy.",
-                DeprecationWarning,
-            )
 
             if not self.store_full_path and (file_path := bytestream.meta.get("file_path")):
                 merged_metadata["file_path"] = os.path.basename(file_path)
