diff --git a/src/Uno.UI.RemoteControl.Messaging/IdeChannel/AddMenuItemRequestIdeMessage.cs b/src/Uno.UI.RemoteControl.Messaging/IdeChannel/AddMenuItemRequestIdeMessage.cs
deleted file mode 100644
index fbe3728f..00000000
--- a/src/Uno.UI.RemoteControl.Messaging/IdeChannel/AddMenuItemRequestIdeMessage.cs
+++ /dev/null
@@ -1,11 +0,0 @@
-﻿#nullable enable
-using System;
-using System.Linq;
-
-namespace Uno.UI.RemoteControl.Messaging.IdeChannel;
-
-/// <summary>
-/// Request to the IDE to add a menu item.
-/// </summary>
-/// <param name="Command">The command to add in the IDE menus.</param>
-public record AddMenuItemRequestIdeMessage(Command Command) : IdeMessage(WellKnownScopes.Ide);
diff --git a/src/Uno.UI.RemoteControl.Messaging/IdeChannel/Command.cs b/src/Uno.UI.RemoteControl.Messaging/IdeChannel/Command.cs
deleted file mode 100644
index c7388e86..00000000
--- a/src/Uno.UI.RemoteControl.Messaging/IdeChannel/Command.cs
+++ /dev/null
@@ -1,23 +0,0 @@
-﻿#nullable enable
-using System;
-using System.Linq;
-
-namespace Uno.UI.RemoteControl.Messaging.IdeChannel;
-
-/// <summary>
-/// Description of a command to execute.
-/// </summary>
-/// <param name="Text">Text content to display to the user.</param>
-/// <param name="Name">The name/id of the command (e.g. uno.hotreload.open_hotreload_window).</param>
-/// <param name="Parameter">A json serialized parameter to execute the command.</param>
-public record Command(string Text, string Name, string? Parameter = null)
-{
-	/// <summary>
-	/// Creates a command that requests to the IDE to open a link in the browser (internal or external).
-	/// </summary>
-	/// <param name="Text">User-friendly text for the link.</param>
-	/// <param name="Uri">Uri to navigate to.</param>
-	/// <returns></returns>
-	public static Command OpenBrowser(string Text, Uri Uri)
-		=> new(Text, "ide.open_browser", Uri.ToString());
-}
diff --git a/src/Uno.UI.RemoteControl.Messaging/IdeChannel/IIdeChannelServer.cs b/src/Uno.UI.RemoteControl.Messaging/IdeChannel/IIdeChannelServer.cs
deleted file mode 100644
index 7b0a24e1..00000000
--- a/src/Uno.UI.RemoteControl.Messaging/IdeChannel/IIdeChannelServer.cs
+++ /dev/null
@@ -1,68 +0,0 @@
-﻿#nullable enable
-
-using System;
-using System.Threading;
-using System.Threading.Tasks;
-using Uno.UI.RemoteControl.Messaging.IdeChannel;
-
-namespace Uno.UI.RemoteControl.Messaging.IdeChannel
-{
-	/// <summary>
-	/// Interface for the IDE to communicate with the dev-server.
-	/// </summary>
-	internal interface IIdeChannelServer // This is internal as it should be used only by IDE extensions
-	{
-		// Note: This interface define the contract of the SHARED OBJECT for the communication channel between the IDE and the dev-server.
-		// 		 It is not meant to be used by the user code, and uses Envelopes (while public IIdeChannel uses IdeMessage).
-		//		 Envelopes are needed for JsonRpc as it is designed for strongly typed communication, not generic like we have.
-
-		/// <summary>
-		/// Fired when a message is received from the dev-server.
-		/// </summary>
-		event EventHandler<IdeMessageEnvelope>? MessageFromDevServer;
-
-		/// <summary>
-		/// Sends a message to the dev-server.
-		/// </summary>
-		/// <param name="envelope">The message to send to the dev-server.</param>
-		/// <param name="ct">CancellationToken to cancel the async operation.</param>
-		/// <returns>An async operation representing the fact that the message has been sent to the dev-server (does not include any form of processing by dev-server).</returns>
-		Task SendToDevServerAsync(IdeMessageEnvelope envelope, CancellationToken ct);
-	}
-}
-
-namespace Uno.UI.RemoteControl.Services
-{
-	// Unlike the IIdeChannelServer, this is to be used by the user code.
-	// Its abstract the notion of sending messages to the IDE, a service offered by the dev-server.
-
-	// Note: We put those interfaces in the same file to make it easier to understand the (non-)relationship between them.
-
-	/// <summary>
-	/// Interface to communicate with the IDE.
-	/// </summary>
-	public interface IIdeChannel
-	{
-		/// <summary>
-		/// Fired when a message is received from the IDE.
-		/// </summary>
-		event EventHandler<IdeMessage>? MessageFromIde;
-
-		/// <summary>
-		/// DO NOT USE - Prefer the TrySendToIdeAsync instead
-		/// Sends a message to the IDE.
-		/// </summary>
-		/// <param name="message">The message to send to the IDE.</param>
-		/// <param name="ct">CancellationToken to cancel the async operation.</param>
-		/// <returns>An async operation representing the fact that the message has been sent to the IDE (does not include any form of processing by IDE).</returns>
-		Task SendToIdeAsync(IdeMessage message, CancellationToken ct);
-
-		/// <summary>
-		/// Sends a message to the IDE.
-		/// </summary>
-		/// <param name="message">The message to send to the IDE.</param>
-		/// <param name="ct">CancellationToken to cancel the async operation.</param>
-		/// <returns>An async bool representing the fact that the message has been sent to the IDE or not (does not include any form of processing by IDE).</returns>
-		Task<bool> TrySendToIdeAsync(IdeMessage message, CancellationToken ct);
-	}
-}
diff --git a/src/Uno.UI.RemoteControl.Messaging/IdeChannel/NotificationRequestIdeMessage.cs b/src/Uno.UI.RemoteControl.Messaging/IdeChannel/NotificationRequestIdeMessage.cs
deleted file mode 100644
index 482d0b6d..00000000
--- a/src/Uno.UI.RemoteControl.Messaging/IdeChannel/NotificationRequestIdeMessage.cs
+++ /dev/null
@@ -1,23 +0,0 @@
-﻿#nullable enable
-using System;
-using System.Linq;
-
-namespace Uno.UI.RemoteControl.Messaging.IdeChannel;
-
-/// <summary>
-/// A message sent by the sev-server to the IDE to notify the user.
-/// </summary>
-/// <param name="Title">Kind of the notification.</param>
-/// <param name="Title">Title of the notification (might not be visible on all IDEs).</param>
-/// <param name="Message">The message of the notification.</param>
-/// <param name="Commands">For call-to-action notification, set of commands to show with the notification.</param>
-public record NotificationRequestIdeMessage(NotificationKind Kind, string Title, string Message, Command[]? Commands = default) : IdeMessage(WellKnownScopes.Ide);
-
-/// <summary>
-/// Kind of <see cref="NotificationRequestIdeMessage"/>
-/// </summary>
-public enum NotificationKind
-{
-	Information,
-	Error
-}
diff --git a/src/Uno.UI.Runtime.Skia.X11/Storage/Pickers/LinuxFilePickerExtension.cs b/src/Uno.UI.Runtime.Skia.X11/Storage/Pickers/LinuxFilePickerExtension.cs
index 44ebca96..a050762d 100644
--- a/src/Uno.UI.Runtime.Skia.X11/Storage/Pickers/LinuxFilePickerExtension.cs
+++ b/src/Uno.UI.Runtime.Skia.X11/Storage/Pickers/LinuxFilePickerExtension.cs
@@ -185,7 +185,7 @@ internal class LinuxFilePickerExtension(IFilePicker picker) : IFileOpenPickerExt
 				return ImmutableList<string>.Empty;
 			}
 
-			return results["uris"].GetArray<string>().Select(s => new Uri(s).AbsolutePath).ToImmutableList();
+			return results["uris"].GetArray<string>().Select(GetPathFromFileUri).ToImmutableList();
 		}
 		catch (Exception e)
 		{
@@ -198,6 +198,8 @@ internal class LinuxFilePickerExtension(IFilePicker picker) : IFileOpenPickerExt
 		}
 	}
 
+	internal static string GetPathFromFileUri(string uri) => new Uri(uri).LocalPath;
+
 	private Array<Struct<string, Array<Struct<uint, string>>>> GetPortalFilters()
 	{
 		// Except from the API reference
