diff --git a/src/I18nProvider.tsx b/src/I18nProvider.tsx
index 8230eb3..c828ec0 100644
--- a/src/I18nProvider.tsx
+++ b/src/I18nProvider.tsx
@@ -28,6 +28,10 @@ export default function I18nProvider({
   const pluralRules = new Intl.PluralRules(ignoreLang ? undefined : lang)
   const t = transCore({ config, allNamespaces, pluralRules, lang })
 
+  if (typeof window !== 'undefined' && globalThis.__NEXT_TRANSLATE__) {
+    globalThis.__NEXT_TRANSLATE__ = { lang, namespaces: allNamespaces, config }
+  }
+
   return (
     <I18nContext.Provider value={{ lang, t }}>
       <InternalContext.Provider value={{ ns: allNamespaces, config }}>
