diff --git a/src/Spectre.Console/Prompts/TextPrompt.cs b/src/Spectre.Console/Prompts/TextPrompt.cs
index 5418ef8..a11a973 100644
--- a/src/Spectre.Console/Prompts/TextPrompt.cs
+++ b/src/Spectre.Console/Prompts/TextPrompt.cs
@@ -133,7 +133,8 @@ public sealed class TextPrompt<T> : IPrompt<T>, IHasCulture
             var promptStyle = PromptStyle ?? Style.Plain;
             var converter = Converter ?? TypeConverterHelper.ConvertToString;
             var choices = Choices.Select(choice => converter(choice)).ToList();
-            var choiceMap = Choices.ToDictionary(choice => converter(choice), choice => choice, _comparer);
+            var comparer = _comparer ?? StringComparer.OrdinalIgnoreCase;
+            var choiceMap = Choices.ToDictionary(choice => converter(choice), choice => choice, comparer);
 
             WritePrompt(console);
