diff --git a/pkg/plugins/golang/v4/scaffolds/api.go b/pkg/plugins/golang/v4/scaffolds/api.go
index a5a7e41..cb537a2 100644
--- a/pkg/plugins/golang/v4/scaffolds/api.go
+++ b/pkg/plugins/golang/v4/scaffolds/api.go
@@ -84,7 +84,7 @@ func (s *apiScaffolder) Scaffold() error {
 	// Initialize the machinery.Scaffold that will write the files to disk
 	scaffold := machinery.NewScaffold(s.fs,
 		machinery.WithConfig(s.config),
-		machinery.WithBoilerplate(string(boilerplate)),
+		machinery.WithBoilerplate(replaceYearPlaceholder(string(boilerplate))),
 		machinery.WithResource(&s.resource),
 	)
 
diff --git a/pkg/plugins/golang/v4/scaffolds/init.go b/pkg/plugins/golang/v4/scaffolds/init.go
index f7ca206..3c0ad9e 100644
--- a/pkg/plugins/golang/v4/scaffolds/init.go
+++ b/pkg/plugins/golang/v4/scaffolds/init.go
@@ -21,6 +21,7 @@ import (
 	"fmt"
 	log "log/slog"
 	"strings"
+	"time"
 
 	"github.com/spf13/afero"
 
@@ -93,6 +94,10 @@ func getControllerRuntimeReleaseBranch() string {
 	return releaseBranch
 }
 
+func replaceYearPlaceholder(boilerplate string) string {
+	return strings.ReplaceAll(boilerplate, "YEAR", fmt.Sprintf("%d", time.Now().Year()))
+}
+
 // Scaffold implements cmdutil.Scaffolder
 func (s *initScaffolder) Scaffold() error {
 	log.Info("Writing scaffold for you to edit...")
@@ -131,7 +136,7 @@ func (s *initScaffolder) Scaffold() error {
 		// Initialize the machinery.Scaffold that will write the files to disk
 		scaffold = machinery.NewScaffold(s.fs,
 			machinery.WithConfig(s.config),
-			machinery.WithBoilerplate(string(boilerplate)),
+			machinery.WithBoilerplate(replaceYearPlaceholder(string(boilerplate))),
 		)
 	} else {
 		s.boilerplatePath = ""
diff --git a/pkg/plugins/golang/v4/scaffolds/internal/templates/hack/boilerplate.go b/pkg/plugins/golang/v4/scaffolds/internal/templates/hack/boilerplate.go
index b200fca..d871662 100644
--- a/pkg/plugins/golang/v4/scaffolds/internal/templates/hack/boilerplate.go
+++ b/pkg/plugins/golang/v4/scaffolds/internal/templates/hack/boilerplate.go
@@ -19,7 +19,6 @@ package hack
 import (
 	"fmt"
 	"path/filepath"
-	"time"
 
 	"sigs.k8s.io/kubebuilder/v4/pkg/machinery"
 )
@@ -80,7 +79,7 @@ func (f *Boilerplate) SetTemplateDefaults() error {
 	}
 
 	if f.Year == "" {
-		f.Year = fmt.Sprintf("%v", time.Now().Year())
+		f.Year = "YEAR"
 	}
 
 	// Boilerplate given
diff --git a/pkg/plugins/golang/v4/scaffolds/internal/templates/makefile.go b/pkg/plugins/golang/v4/scaffolds/internal/templates/makefile.go
index 49253d2..84a2bcf 100644
--- a/pkg/plugins/golang/v4/scaffolds/internal/templates/makefile.go
+++ b/pkg/plugins/golang/v4/scaffolds/internal/templates/makefile.go
@@ -89,6 +89,9 @@ endif
 # tools. (i.e. podman)
 CONTAINER_TOOL ?= docker
 
+# YEAR defines the current year for replacing YEAR in the boilerplate.
+YEAR ?= $(shell date +%Y)
+
 # Setting SHELL to bash allows bash commands to be executed by recipes.
 # Options are set to exit when a recipe line exits non-zero or a piped command fails.
 SHELL = /usr/bin/env bash -o pipefail
@@ -123,7 +126,7 @@ manifests: controller-gen ## Generate WebhookConfiguration, ClusterRole and Cust
 .PHONY: generate
 generate: controller-gen ## Generate code containing DeepCopy, DeepCopyInto, and DeepCopyObject method implementations.
 	{{ if .BoilerplatePath -}}
-	"$(CONTROLLER_GEN)" object:headerFile={{printf "%q" .BoilerplatePath}} paths="./..."
+	"$(CONTROLLER_GEN)" object:headerFile={{printf "%q" .BoilerplatePath}},year=$(YEAR) paths="./..."
 	{{- else -}}
 	"$(CONTROLLER_GEN)" object paths="./..."
 	{{- end }}
diff --git a/pkg/plugins/golang/v4/scaffolds/webhook.go b/pkg/plugins/golang/v4/scaffolds/webhook.go
index 00d526f..a4b68d0 100644
--- a/pkg/plugins/golang/v4/scaffolds/webhook.go
+++ b/pkg/plugins/golang/v4/scaffolds/webhook.go
@@ -90,7 +90,7 @@ func (s *webhookScaffolder) Scaffold() error {
 	// Initialize the machinery.Scaffold that will write the files to disk
 	scaffold := machinery.NewScaffold(s.fs,
 		machinery.WithConfig(s.config),
-		machinery.WithBoilerplate(string(boilerplate)),
+		machinery.WithBoilerplate(replaceYearPlaceholder(string(boilerplate))),
 		machinery.WithResource(&s.resource),
 	)
