diff --git a/crates/passes/src/type_checking/program.rs b/crates/passes/src/type_checking/program.rs
index e92081b..b590692 100644
--- a/crates/passes/src/type_checking/program.rs
+++ b/crates/passes/src/type_checking/program.rs
@@ -350,6 +350,9 @@ impl ProgramVisitor for TypeCheckingVisitor<'_> {
         self.assert_type_is_valid(&input.key_type, input.span);
         // Check that a mapping's key type is not a future, tuple, record, or mapping.
         match input.key_type.clone() {
+            Type::Array(_) if !input.key_type.is_empty() => {
+                self.emit_err(TypeCheckerError::invalid_mapping_type("key", "non-empty array", input.span));
+            }
             Type::Future(_) => self.emit_err(TypeCheckerError::invalid_mapping_type("key", "future", input.span)),
             Type::Tuple(_) => self.emit_err(TypeCheckerError::invalid_mapping_type("key", "tuple", input.span)),
             Type::Composite(composite_type) => {
