diff --git a/packages/runtime-tags/src/translator/util/references.ts b/packages/runtime-tags/src/translator/util/references.ts
index 169f5994e..d38845663 100644
--- a/packages/runtime-tags/src/translator/util/references.ts
+++ b/packages/runtime-tags/src/translator/util/references.ts
@@ -361,6 +361,13 @@ function trackReferencesForBinding(babelBinding: t.Binding, binding: Binding) {
   }
 
   for (const ref of constantViolations) {
+    // Only allow assignments to let variables, not const variables
+    if (binding.type !== BindingType.let) {
+      throw ref.buildCodeFrameError(
+        `Cannot assign to "${binding.name}" because it is a constant`
+      );
+    }
+
     if (ref.isUpdateExpression()) {
       trackAssignment(ref.get("argument"), binding);
     } else if (ref.isAssignmentExpression()) {
