diff --git a/internal/tools/semconvkit/templates/error_type.go.tmpl b/internal/tools/semconvkit/templates/error_type.go.tmpl
index 193c43a..d2c6741 100644
--- a/internal/tools/semconvkit/templates/error_type.go.tmpl
+++ b/internal/tools/semconvkit/templates/error_type.go.tmpl
@@ -5,6 +5,7 @@ package semconv // import "go.opentelemetry.io/otel/semconv/{{.TagVer}}"
 
 import (
 	"errors"
+	"fmt"
 	"reflect"
 
 	"go.opentelemetry.io/otel/attribute"
@@ -49,6 +50,7 @@ func errorType(err error) string {
 	if s == "" {
 		// Fallback to reflection if the ErrorType method is not supported or
 		// returns an empty value.
+		err = unwrapFmtWrapErrors(err)
 
 		t := reflect.TypeOf(err)
 		pkg, name := t.PkgPath(), t.Name()
@@ -64,3 +66,16 @@ func errorType(err error) string {
 	}
 	return s
 }
+
+var fmtWrapErrorType = reflect.TypeOf(fmt.Errorf("%w", errors.New("")))
+
+func unwrapFmtWrapErrors(err error) error {
+	for reflect.TypeOf(err) == fmtWrapErrorType {
+		unwrapped := errors.Unwrap(err)
+		if unwrapped == nil {
+			return err
+		}
+		err = unwrapped
+	}
+	return err
+}
diff --git a/semconv/v1.34.0/error_type.go b/semconv/v1.34.0/error_type.go
index 19bf022..f9679ec 100644
--- a/semconv/v1.34.0/error_type.go
+++ b/semconv/v1.34.0/error_type.go
@@ -4,6 +4,7 @@
 package semconv // import "go.opentelemetry.io/otel/semconv/v1.34.0"
 
 import (
+	"errors"
 	"fmt"
 	"reflect"
 
@@ -15,6 +16,7 @@ func ErrorType(err error) attribute.KeyValue {
 	if err == nil {
 		return ErrorTypeOther
 	}
+	err = unwrapFmtWrapErrors(err)
 	t := reflect.TypeOf(err)
 	var value string
 	if t.PkgPath() == "" && t.Name() == "" {
@@ -29,3 +31,16 @@ func ErrorType(err error) attribute.KeyValue {
 	}
 	return ErrorTypeKey.String(value)
 }
+
+var fmtWrapErrorType = reflect.TypeOf(fmt.Errorf("%w", errors.New("")))
+
+func unwrapFmtWrapErrors(err error) error {
+	for reflect.TypeOf(err) == fmtWrapErrorType {
+		unwrapped := errors.Unwrap(err)
+		if unwrapped == nil {
+			return err
+		}
+		err = unwrapped
+	}
+	return err
+}
diff --git a/semconv/v1.36.0/error_type.go b/semconv/v1.36.0/error_type.go
index c1fa7af..4b69dcf 100644
--- a/semconv/v1.36.0/error_type.go
+++ b/semconv/v1.36.0/error_type.go
@@ -4,6 +4,7 @@
 package semconv // import "go.opentelemetry.io/otel/semconv/v1.36.0"
 
 import (
+	"errors"
 	"fmt"
 	"reflect"
 
@@ -15,6 +16,7 @@ func ErrorType(err error) attribute.KeyValue {
 	if err == nil {
 		return ErrorTypeOther
 	}
+	err = unwrapFmtWrapErrors(err)
 	t := reflect.TypeOf(err)
 	var value string
 	if t.PkgPath() == "" && t.Name() == "" {
@@ -29,3 +31,16 @@ func ErrorType(err error) attribute.KeyValue {
 	}
 	return ErrorTypeKey.String(value)
 }
+
+var fmtWrapErrorType = reflect.TypeOf(fmt.Errorf("%w", errors.New("")))
+
+func unwrapFmtWrapErrors(err error) error {
+	for reflect.TypeOf(err) == fmtWrapErrorType {
+		unwrapped := errors.Unwrap(err)
+		if unwrapped == nil {
+			return err
+		}
+		err = unwrapped
+	}
+	return err
+}
diff --git a/semconv/v1.37.0/error_type.go b/semconv/v1.37.0/error_type.go
index 267979c..f7edb66 100644
--- a/semconv/v1.37.0/error_type.go
+++ b/semconv/v1.37.0/error_type.go
@@ -4,6 +4,8 @@
 package semconv // import "go.opentelemetry.io/otel/semconv/v1.37.0"
 
 import (
+	"errors"
+	"fmt"
 	"reflect"
 
 	"go.opentelemetry.io/otel/attribute"
@@ -39,6 +41,7 @@ func errorType(err error) string {
 	if s == "" {
 		// Fallback to reflection if the ErrorType method is not supported or
 		// returns an empty value.
+		err = unwrapFmtWrapErrors(err)
 
 		t := reflect.TypeOf(err)
 		pkg, name := t.PkgPath(), t.Name()
@@ -54,3 +57,16 @@ func errorType(err error) string {
 	}
 	return s
 }
+
+var fmtWrapErrorType = reflect.TypeOf(fmt.Errorf("%w", errors.New("")))
+
+func unwrapFmtWrapErrors(err error) error {
+	for reflect.TypeOf(err) == fmtWrapErrorType {
+		unwrapped := errors.Unwrap(err)
+		if unwrapped == nil {
+			return err
+		}
+		err = unwrapped
+	}
+	return err
+}
diff --git a/semconv/v1.38.0/error_type.go b/semconv/v1.38.0/error_type.go
index 84d909c..35d74e3 100644
--- a/semconv/v1.38.0/error_type.go
+++ b/semconv/v1.38.0/error_type.go
@@ -4,6 +4,8 @@
 package semconv // import "go.opentelemetry.io/otel/semconv/v1.38.0"
 
 import (
+	"errors"
+	"fmt"
 	"reflect"
 
 	"go.opentelemetry.io/otel/attribute"
@@ -39,6 +41,7 @@ func errorType(err error) string {
 	if s == "" {
 		// Fallback to reflection if the ErrorType method is not supported or
 		// returns an empty value.
+		err = unwrapFmtWrapErrors(err)
 
 		t := reflect.TypeOf(err)
 		pkg, name := t.PkgPath(), t.Name()
@@ -54,3 +57,16 @@ func errorType(err error) string {
 	}
 	return s
 }
+
+var fmtWrapErrorType = reflect.TypeOf(fmt.Errorf("%w", errors.New("")))
+
+func unwrapFmtWrapErrors(err error) error {
+	for reflect.TypeOf(err) == fmtWrapErrorType {
+		unwrapped := errors.Unwrap(err)
+		if unwrapped == nil {
+			return err
+		}
+		err = unwrapped
+	}
+	return err
+}
diff --git a/semconv/v1.39.0/error_type.go b/semconv/v1.39.0/error_type.go
index 84cf636..02f394a 100644
--- a/semconv/v1.39.0/error_type.go
+++ b/semconv/v1.39.0/error_type.go
@@ -4,6 +4,8 @@
 package semconv // import "go.opentelemetry.io/otel/semconv/v1.39.0"
 
 import (
+	"errors"
+	"fmt"
 	"reflect"
 
 	"go.opentelemetry.io/otel/attribute"
@@ -39,6 +41,7 @@ func errorType(err error) string {
 	if s == "" {
 		// Fallback to reflection if the ErrorType method is not supported or
 		// returns an empty value.
+		err = unwrapFmtWrapErrors(err)
 
 		t := reflect.TypeOf(err)
 		pkg, name := t.PkgPath(), t.Name()
@@ -54,3 +57,16 @@ func errorType(err error) string {
 	}
 	return s
 }
+
+var fmtWrapErrorType = reflect.TypeOf(fmt.Errorf("%w", errors.New("")))
+
+func unwrapFmtWrapErrors(err error) error {
+	for reflect.TypeOf(err) == fmtWrapErrorType {
+		unwrapped := errors.Unwrap(err)
+		if unwrapped == nil {
+			return err
+		}
+		err = unwrapped
+	}
+	return err
+}
diff --git a/semconv/v1.40.0/error_type.go b/semconv/v1.40.0/error_type.go
index 6d26e52..3a9e49f 100644
--- a/semconv/v1.40.0/error_type.go
+++ b/semconv/v1.40.0/error_type.go
@@ -5,6 +5,7 @@ package semconv // import "go.opentelemetry.io/otel/semconv/v1.40.0"
 
 import (
 	"errors"
+	"fmt"
 	"reflect"
 
 	"go.opentelemetry.io/otel/attribute"
@@ -49,6 +50,7 @@ func errorType(err error) string {
 	if s == "" {
 		// Fallback to reflection if the ErrorType method is not supported or
 		// returns an empty value.
+		err = unwrapFmtWrapErrors(err)
 
 		t := reflect.TypeOf(err)
 		pkg, name := t.PkgPath(), t.Name()
@@ -64,3 +66,16 @@ func errorType(err error) string {
 	}
 	return s
 }
+
+var fmtWrapErrorType = reflect.TypeOf(fmt.Errorf("%w", errors.New("")))
+
+func unwrapFmtWrapErrors(err error) error {
+	for reflect.TypeOf(err) == fmtWrapErrorType {
+		unwrapped := errors.Unwrap(err)
+		if unwrapped == nil {
+			return err
+		}
+		err = unwrapped
+	}
+	return err
+}
