diff --git a/datafusion/core/src/datasource/listing_table_factory.rs b/datafusion/core/src/datasource/listing_table_factory.rs
index 3ca388af0..b761a5ea8 100644
--- a/datafusion/core/src/datasource/listing_table_factory.rs
+++ b/datafusion/core/src/datasource/listing_table_factory.rs
@@ -383,35 +383,6 @@ mod tests {
         );
     }
 
-    #[tokio::test]
-    async fn test_odd_directory_names() {
-        let dir = tempfile::tempdir().unwrap();
-        let mut path = PathBuf::from(dir.path());
-        path.extend(["odd.v1", "odd.v2"]);
-        fs::create_dir_all(&path).unwrap();
-
-        let factory = ListingTableFactory::new();
-        let context = SessionContext::new();
-        let state = context.state();
-        let name = TableReference::bare("foo");
-
-        let cmd = CreateExternalTable::builder(
-            name,
-            String::from(path.to_str().unwrap()),
-            "parquet",
-            Arc::new(DFSchema::empty()),
-        )
-        .build();
-        let table_provider = factory.create(&state, &cmd).await.unwrap();
-        let listing_table = table_provider
-            .as_any()
-            .downcast_ref::<ListingTable>()
-            .unwrap();
-
-        let listing_options = listing_table.options();
-        assert_eq!("", listing_options.file_extension);
-    }
-
     #[tokio::test]
     async fn test_create_with_hive_partitions() {
         let dir = tempfile::tempdir().unwrap();
