diff --git a/esphome/components/substitutions/__init__.py b/esphome/components/substitutions/__init__.py
index 94aebbb..0eef5b5 100644
--- a/esphome/components/substitutions/__init__.py
+++ b/esphome/components/substitutions/__init__.py
@@ -429,6 +429,12 @@ def do_substitution_pass(
     # Use merge_dicts_ordered to preserve OrderedDict type for move_to_end()
     substitutions = config.pop(CONF_SUBSTITUTIONS, {})
     with cv.prepend_path(CONF_SUBSTITUTIONS):
+        if isinstance(substitutions, IncludeFile):
+            substitutions, _ = resolve_include(
+                substitutions,
+                [CONF_SUBSTITUTIONS],
+                ContextVars(command_line_substitutions or {}),
+            )
         if not isinstance(substitutions, dict):
             raise cv.Invalid(
                 f"Substitutions must be a key to value mapping, got {type(substitutions)}"
