diff --git a/internal/api/ui/login/login.go b/internal/api/ui/login/login.go
index 5ff27c14f..82ac94328 100644
--- a/internal/api/ui/login/login.go
+++ b/internal/api/ui/login/login.go
@@ -183,8 +183,7 @@ func (l *Login) getClaimedUserIDsOfOrgDomain(ctx context.Context, orgName string
 
 func setContext(ctx context.Context, resourceOwner string) context.Context {
 	data := authz.CtxData{
-		UserID: login,
-		OrgID:  resourceOwner,
+		OrgID: resourceOwner,
 	}
 	return authz.SetCtxData(ctx, data)
 }
