diff --git a/datafusion/physical-optimizer/src/output_requirements.rs b/datafusion/physical-optimizer/src/output_requirements.rs
index 0dc6a25fb..3cb1342e2 100644
--- a/datafusion/physical-optimizer/src/output_requirements.rs
+++ b/datafusion/physical-optimizer/src/output_requirements.rs
@@ -185,7 +185,12 @@ impl DisplayAs for OutputRequirementExec {
                 )
             }
             DisplayFormatType::TreeRender => {
-                write!(f, "")
+                // Include useful information in tree displays (e.g. EXPLAIN output)
+                write!(
+                    f,
+                    "order_requirement={:?}, dist_requirement={}",
+                    self.order_requirement, self.dist_requirement
+                )
             }
         }
     }
