diff --git a/pkg/controller/admissionchecks/provisioning/constants.go b/pkg/controller/admissionchecks/provisioning/constants.go
index 551aae676..cec4fe66a 100644
--- a/pkg/controller/admissionchecks/provisioning/constants.go
+++ b/pkg/controller/admissionchecks/provisioning/constants.go
@@ -17,11 +17,9 @@ limitations under the License.
 package provisioning
 
 const (
-	ConfigKind                       = "ProvisioningRequestConfig"
-	DeprecatedConsumesAnnotationKey  = "cluster-autoscaler.kubernetes.io/consume-provisioning-request"
-	DeprecatedClassNameAnnotationKey = "cluster-autoscaler.kubernetes.io/provisioning-class-name"
-	ConsumesAnnotationKey            = "autoscaling.x-k8s.io/consume-provisioning-request"
-	ClassNameAnnotationKey           = "autoscaling.x-k8s.io/provisioning-class-name"
+	ConfigKind             = "ProvisioningRequestConfig"
+	ConsumesAnnotationKey  = "autoscaling.x-k8s.io/consume-provisioning-request"
+	ClassNameAnnotationKey = "autoscaling.x-k8s.io/provisioning-class-name"
 
 	CheckInactiveMessage = "the check is not active"
 	NoRequestNeeded      = "the provisioning request is not needed"
diff --git a/pkg/controller/admissionchecks/provisioning/controller.go b/pkg/controller/admissionchecks/provisioning/controller.go
index 7feca94b8..86f2522dd 100644
--- a/pkg/controller/admissionchecks/provisioning/controller.go
+++ b/pkg/controller/admissionchecks/provisioning/controller.go
@@ -624,10 +624,8 @@ func podSetUpdates(log logr.Logger, wl *kueue.Workload, pr *autoscaling.Provisio
 		podSetUpdate := kueue.PodSetUpdate{
 			Name: refMap[ps.PodTemplateRef.Name],
 			Annotations: map[string]string{
-				DeprecatedConsumesAnnotationKey:  pr.Name,
-				DeprecatedClassNameAnnotationKey: pr.Spec.ProvisioningClassName,
-				ConsumesAnnotationKey:            pr.Name,
-				ClassNameAnnotationKey:           pr.Spec.ProvisioningClassName},
+				ConsumesAnnotationKey:  pr.Name,
+				ClassNameAnnotationKey: pr.Spec.ProvisioningClassName},
 		}
 		if psUpdate := prc.Spec.PodSetUpdates; psUpdate != nil {
 			podSetUpdate.NodeSelector = make(map[string]string, len(psUpdate.NodeSelector))
diff --git a/pkg/controller/admissionchecks/provisioning/controller_test.go b/pkg/controller/admissionchecks/provisioning/controller_test.go
index d71634987..ed7c229d1 100644
--- a/pkg/controller/admissionchecks/provisioning/controller_test.go
+++ b/pkg/controller/admissionchecks/provisioning/controller_test.go
@@ -616,19 +616,15 @@ func TestReconcile(t *testing.T) {
 							{
 								Name: "ps1",
 								Annotations: map[string]string{
-									DeprecatedConsumesAnnotationKey:  "wl-check1-1",
-									DeprecatedClassNameAnnotationKey: "class1",
-									ConsumesAnnotationKey:            "wl-check1-1",
-									ClassNameAnnotationKey:           "class1",
+									ConsumesAnnotationKey:  "wl-check1-1",
+									ClassNameAnnotationKey: "class1",
 								},
 							},
 							{
 								Name: "ps2",
 								Annotations: map[string]string{
-									DeprecatedConsumesAnnotationKey:  "wl-check1-1",
-									DeprecatedClassNameAnnotationKey: "class1",
-									ConsumesAnnotationKey:            "wl-check1-1",
-									ClassNameAnnotationKey:           "class1",
+									ConsumesAnnotationKey:  "wl-check1-1",
+									ClassNameAnnotationKey: "class1",
 								},
 							},
 						},
@@ -1181,10 +1177,8 @@ func TestReconcile(t *testing.T) {
 							{
 								Name: "ps1",
 								Annotations: map[string]string{
-									DeprecatedConsumesAnnotationKey:  "wl-check1-1",
-									DeprecatedClassNameAnnotationKey: "class1",
-									ConsumesAnnotationKey:            "wl-check1-1",
-									ClassNameAnnotationKey:           "class1",
+									ConsumesAnnotationKey:  "wl-check1-1",
+									ClassNameAnnotationKey: "class1",
 								},
 								NodeSelector: map[string]string{
 									"node-selector-key": "nodes-selector-xyz",
@@ -1193,10 +1187,8 @@ func TestReconcile(t *testing.T) {
 							{
 								Name: "ps2",
 								Annotations: map[string]string{
-									DeprecatedConsumesAnnotationKey:  "wl-check1-1",
-									DeprecatedClassNameAnnotationKey: "class1",
-									ConsumesAnnotationKey:            "wl-check1-1",
-									ClassNameAnnotationKey:           "class1",
+									ConsumesAnnotationKey:  "wl-check1-1",
+									ClassNameAnnotationKey: "class1",
 								},
 								NodeSelector: map[string]string{
 									"node-selector-key": "nodes-selector-xyz",
@@ -1234,19 +1226,15 @@ func TestReconcile(t *testing.T) {
 							{
 								Name: "ps1",
 								Annotations: map[string]string{
-									DeprecatedConsumesAnnotationKey:  "wl-check1-1",
-									DeprecatedClassNameAnnotationKey: "class1",
-									ConsumesAnnotationKey:            "wl-check1-1",
-									ClassNameAnnotationKey:           "class1",
+									ConsumesAnnotationKey:  "wl-check1-1",
+									ClassNameAnnotationKey: "class1",
 								},
 							},
 							{
 								Name: "ps2",
 								Annotations: map[string]string{
-									DeprecatedConsumesAnnotationKey:  "wl-check1-1",
-									DeprecatedClassNameAnnotationKey: "class1",
-									ConsumesAnnotationKey:            "wl-check1-1",
-									ClassNameAnnotationKey:           "class1",
+									ConsumesAnnotationKey:  "wl-check1-1",
+									ClassNameAnnotationKey: "class1",
 								},
 							},
 						},
diff --git a/test/integration/singlecluster/controller/admissionchecks/provisioning/provisioning_test.go b/test/integration/singlecluster/controller/admissionchecks/provisioning/provisioning_test.go
index 111ff3105..2f17866a5 100644
--- a/test/integration/singlecluster/controller/admissionchecks/provisioning/provisioning_test.go
+++ b/test/integration/singlecluster/controller/admissionchecks/provisioning/provisioning_test.go
@@ -302,19 +302,15 @@ var _ = ginkgo.Describe("Provisioning", ginkgo.Ordered, ginkgo.ContinueOnFailure
 						{
 							Name: "ps1",
 							Annotations: map[string]string{
-								provisioning.ConsumesAnnotationKey:            provReqKey.Name,
-								provisioning.DeprecatedConsumesAnnotationKey:  provReqKey.Name,
-								provisioning.ClassNameAnnotationKey:           prc.Spec.ProvisioningClassName,
-								provisioning.DeprecatedClassNameAnnotationKey: prc.Spec.ProvisioningClassName,
+								provisioning.ConsumesAnnotationKey:  provReqKey.Name,
+								provisioning.ClassNameAnnotationKey: prc.Spec.ProvisioningClassName,
 							},
 						},
 						{
 							Name: "ps2",
 							Annotations: map[string]string{
-								provisioning.ConsumesAnnotationKey:            provReqKey.Name,
-								provisioning.DeprecatedConsumesAnnotationKey:  provReqKey.Name,
-								provisioning.ClassNameAnnotationKey:           prc.Spec.ProvisioningClassName,
-								provisioning.DeprecatedClassNameAnnotationKey: prc.Spec.ProvisioningClassName,
+								provisioning.ConsumesAnnotationKey:  provReqKey.Name,
+								provisioning.ClassNameAnnotationKey: prc.Spec.ProvisioningClassName,
 							},
 						},
 					}))
@@ -747,19 +743,15 @@ var _ = ginkgo.Describe("Provisioning", ginkgo.Ordered, ginkgo.ContinueOnFailure
 						{
 							Name: "ps1",
 							Annotations: map[string]string{
-								provisioning.ConsumesAnnotationKey:            provReqKey.Name,
-								provisioning.DeprecatedConsumesAnnotationKey:  provReqKey.Name,
-								provisioning.ClassNameAnnotationKey:           prc.Spec.ProvisioningClassName,
-								provisioning.DeprecatedClassNameAnnotationKey: prc.Spec.ProvisioningClassName,
+								provisioning.ConsumesAnnotationKey:  provReqKey.Name,
+								provisioning.ClassNameAnnotationKey: prc.Spec.ProvisioningClassName,
 							},
 						},
 						{
 							Name: "ps2",
 							Annotations: map[string]string{
-								provisioning.ConsumesAnnotationKey:            provReqKey.Name,
-								provisioning.DeprecatedConsumesAnnotationKey:  provReqKey.Name,
-								provisioning.ClassNameAnnotationKey:           prc.Spec.ProvisioningClassName,
-								provisioning.DeprecatedClassNameAnnotationKey: prc.Spec.ProvisioningClassName,
+								provisioning.ConsumesAnnotationKey:  provReqKey.Name,
+								provisioning.ClassNameAnnotationKey: prc.Spec.ProvisioningClassName,
 							},
 						},
 					}))
