diff --git a/lib/mpl_toolkits/mplot3d/art3d.py b/lib/mpl_toolkits/mplot3d/art3d.py
index 0467d2e96e..0fd63c15d3 100644
--- a/lib/mpl_toolkits/mplot3d/art3d.py
+++ b/lib/mpl_toolkits/mplot3d/art3d.py
@@ -1381,6 +1381,7 @@ def _generate_normals(polygons):
         v1 = np.empty((len(polygons), 3))
         v2 = np.empty((len(polygons), 3))
         for poly_i, ps in enumerate(polygons):
+            ps = np.asarray(ps)
             n = len(ps)
             i1, i2, i3 = 0, n//3, 2*n//3
             v1[poly_i, :] = ps[i1, :] - ps[i2, :]
