diff --git a/lib/get-html-headers.js b/lib/get-html-headers.js
index 0c1077b..7e9d94a 100644
--- a/lib/get-html-headers.js
+++ b/lib/get-html-headers.js
@@ -9,7 +9,7 @@ function addLinenos(lines, headers) {
   return headers.map(function (x) {
     for (var lineno = current; lineno < lines.length; lineno++) {
       line = lines[lineno];
-      if (new RegExp(x.text[0]).test(line)) {
+      if (line.indexOf(x.text[0]) !== -1) {
         current = lineno;
         x.line = lineno;
         x.name = x.text.join("");
