diff --git a/packages/compiler/src/babel-types/generator/patch.js b/packages/compiler/src/babel-types/generator/patch.js
index 6a5ccb294..424a84dac 100644
--- a/packages/compiler/src/babel-types/generator/patch.js
+++ b/packages/compiler/src/babel-types/generator/patch.js
@@ -1,4 +1,4 @@
-import "../types/patch";
+import "../types/patch.js";
 
 import Printer from "@babel/generator/lib/printer";
 import * as t from "@babel/types";
diff --git a/packages/compiler/src/babel-types/index.js b/packages/compiler/src/babel-types/index.js
index b51514155..fd3f6a7f4 100644
--- a/packages/compiler/src/babel-types/index.js
+++ b/packages/compiler/src/babel-types/index.js
@@ -1,5 +1,5 @@
-import "./types/patch";
-import "./traverse/patch";
-import "./generator/patch";
-export { MARKO_TYPES } from "./types/definitions";
+import "./types/patch.js";
+import "./traverse/patch.js";
+import "./generator/patch.js";
+export { MARKO_TYPES } from "./types/definitions.js";
 export * from "@babel/types";
diff --git a/packages/compiler/src/babel-types/traverse/patch.js b/packages/compiler/src/babel-types/traverse/patch.js
index 482cfdcf5..8b6fe186f 100644
--- a/packages/compiler/src/babel-types/traverse/patch.js
+++ b/packages/compiler/src/babel-types/traverse/patch.js
@@ -1,9 +1,9 @@
-import "../types/patch";
+import "../types/patch.js";
 
 import traverse, { NodePath, Scope } from "@babel/traverse";
 import * as t from "@babel/types";
 
-import { MARKO_ALIAS_TYPES, MARKO_TYPES } from "../types/definitions";
+import { MARKO_ALIAS_TYPES, MARKO_TYPES } from "../types/definitions.js";
 
 MARKO_TYPES.forEach((typeName) => {
   const checkKey = `is${typeName}`;
diff --git a/packages/compiler/src/babel-types/types/definitions.js b/packages/compiler/src/babel-types/types/definitions.js
index 61bd0f5a5..c4e7b04db 100644
--- a/packages/compiler/src/babel-types/types/definitions.js
+++ b/packages/compiler/src/babel-types/types/definitions.js
@@ -4,7 +4,7 @@ import {
   assertOneOf,
   assertValueType,
   chain,
-} from "@babel/types/lib/definitions/utils";
+} from "@babel/types/lib/definitions/utils.js";
 
 const valueFieldCommon = {
   value: {
diff --git a/packages/compiler/src/babel-types/types/patch.js b/packages/compiler/src/babel-types/types/patch.js
index ad9760022..214b8a7c5 100644
--- a/packages/compiler/src/babel-types/types/patch.js
+++ b/packages/compiler/src/babel-types/types/patch.js
@@ -1,11 +1,11 @@
 /* eslint-disable no-import-assign */
 import * as babelTypes from "@babel/types";
-import * as definitionUtils from "@babel/types/lib/definitions/utils";
-import * as generatedValidators from "@babel/types/lib/validators/generated";
-import * as referencedValidators from "@babel/types/lib/validators/isReferenced";
-import validate from "@babel/types/lib/validators/validate";
+import * as definitionUtils from "@babel/types/lib/definitions/utils.js";
+import * as generatedValidators from "@babel/types/lib/validators/generated/index.js";
+import * as referencedValidators from "@babel/types/lib/validators/isReferenced.js";
+import validate from "@babel/types/lib/validators/validate.js";
 
-import definitions, { MARKO_ALIAS_TYPES, MARKO_TYPES } from "./definitions";
+import definitions, { MARKO_ALIAS_TYPES, MARKO_TYPES } from "./definitions.js";
 
 const {
   TYPES,
diff --git a/packages/compiler/src/index.js b/packages/compiler/src/index.js
index 8ce6d21f6..e7f778140 100644
--- a/packages/compiler/src/index.js
+++ b/packages/compiler/src/index.js
@@ -1,4 +1,4 @@
-export * as types from "./babel-types";
+export * as types from "./babel-types/index.js";
 import * as babel from "@babel/core";
 import tsSyntaxPlugin from "@babel/plugin-syntax-typescript";
 import cjsPlugin from "@babel/plugin-transform-modules-commonjs";
