diff --git a/compat/src/Children.js b/compat/src/Children.js
index 0295d936..79f7c635 100644
--- a/compat/src/Children.js
+++ b/compat/src/Children.js
@@ -1,8 +1,10 @@
 import { toChildArray } from 'preact';
 
-const mapFn = (children, fn) => {
+const mapFn = (children, fn, context) => {
 	if (children == null) return null;
-	return toChildArray(toChildArray(children).map(fn));
+	// React.Children.map/forEach accept an optional 3rd argument that is used as
+	// the callback "this" binding.
+	return toChildArray(toChildArray(children).map(fn, context));
 };
 
 // This API is completely unnecessary for Preact, so it's basically passthrough.
diff --git a/compat/test/browser/Children.test.js b/compat/test/browser/Children.test.js
index 53c0c776..7dc4ce86 100644
--- a/compat/test/browser/Children.test.js
+++ b/compat/test/browser/Children.test.js
@@ -83,6 +83,7 @@ describe('Children', () => {
 			return <div>{children}</div>;
 		}
 
+
 		it('should iterate over children', () => {
 			render(
 				<Foo>
@@ -171,6 +172,7 @@ describe('Children', () => {
 			return <div>{children}</div>;
 		}
 
+
 		it('should iterate over children', () => {
 			render(
 				<Foo>
