diff --git a/api/src/main/java/io/grpc/EquivalentAddressGroup.java b/api/src/main/java/io/grpc/EquivalentAddressGroup.java
index 18151e8..811f5eb 100644
--- a/api/src/main/java/io/grpc/EquivalentAddressGroup.java
+++ b/api/src/main/java/io/grpc/EquivalentAddressGroup.java
@@ -50,6 +50,13 @@ public final class EquivalentAddressGroup {
   @ExperimentalApi("https://github.com/grpc/grpc-java/issues/6138")
   public static final Attributes.Key<String> ATTR_AUTHORITY_OVERRIDE =
       Attributes.Key.create("io.grpc.EquivalentAddressGroup.ATTR_AUTHORITY_OVERRIDE");
+  /**
+   * The backend service that this EquivalentAddressGroup belongs to.
+   */
+  @Attr
+  @ExperimentalApi("https://github.com/grpc/grpc-java/issues/11989")
+  public static final Attributes.Key<String> ATTR_BACKEND_SERVICE =
+      Attributes.Key.create("io.grpc.EquivalentAddressGroup.ATTR_BACKEND_SERVICE");
   /**
    * The name of the locality that this EquivalentAddressGroup is in.
    */
diff --git a/core/src/main/java/io/grpc/internal/InternalSubchannel.java b/core/src/main/java/io/grpc/internal/InternalSubchannel.java
index 7a48bf6..85471dd 100644
--- a/core/src/main/java/io/grpc/internal/InternalSubchannel.java
+++ b/core/src/main/java/io/grpc/internal/InternalSubchannel.java
@@ -603,8 +603,7 @@ final class InternalSubchannel implements InternalInstrumented<ChannelStats>, Tr
             connectedAddressAttributes = addressIndex.getCurrentEagAttributes();
             gotoNonErrorState(READY);
             subchannelMetrics.recordConnectionAttemptSucceeded(/* target= */ target,
-                /* backendService= */ getAttributeOrDefault(
-                    addressIndex.getCurrentEagAttributes(), NameResolver.ATTR_BACKEND_SERVICE),
+                /* backendService= */ getBackendService(addressIndex.getCurrentEagAttributes()),
                 /* locality= */ getAttributeOrDefault(addressIndex.getCurrentEagAttributes(),
                     EquivalentAddressGroup.ATTR_LOCALITY_NAME),
                 /* securityLevel= */ extractSecurityLevel(addressIndex.getCurrentEagAttributes()
@@ -635,8 +634,7 @@ final class InternalSubchannel implements InternalInstrumented<ChannelStats>, Tr
             addressIndex.reset();
             gotoNonErrorState(IDLE);
             subchannelMetrics.recordDisconnection(/* target= */ target,
-                /* backendService= */ getAttributeOrDefault(addressIndex.getCurrentEagAttributes(),
-                    NameResolver.ATTR_BACKEND_SERVICE),
+                /* backendService= */ getBackendService(addressIndex.getCurrentEagAttributes()),
                 /* locality= */ getAttributeOrDefault(addressIndex.getCurrentEagAttributes(),
                     EquivalentAddressGroup.ATTR_LOCALITY_NAME),
                 /* disconnectError= */ disconnectError.toErrorString(),
@@ -644,8 +642,7 @@ final class InternalSubchannel implements InternalInstrumented<ChannelStats>, Tr
                     .get(GrpcAttributes.ATTR_SECURITY_LEVEL)));
           } else if (pendingTransport == transport) {
             subchannelMetrics.recordConnectionAttemptFailed(/* target= */ target,
-                /* backendService= */getAttributeOrDefault(addressIndex.getCurrentEagAttributes(),
-                    NameResolver.ATTR_BACKEND_SERVICE),
+                /* backendService= */ getBackendService(addressIndex.getCurrentEagAttributes()),
                 /* locality= */ getAttributeOrDefault(addressIndex.getCurrentEagAttributes(),
                     EquivalentAddressGroup.ATTR_LOCALITY_NAME));
             Preconditions.checkState(state.getState() == CONNECTING,
@@ -704,6 +701,14 @@ final class InternalSubchannel implements InternalInstrumented<ChannelStats>, Tr
       }
     }
 
+    private String getBackendService(Attributes attributes) {
+      String backendService = attributes.get(EquivalentAddressGroup.ATTR_BACKEND_SERVICE);
+      if (backendService == null) {
+        backendService = attributes.get(NameResolver.ATTR_BACKEND_SERVICE);
+      }
+      return backendService == null ? "" : backendService;
+    }
+
     private String getAttributeOrDefault(Attributes attributes, Attributes.Key<String> key) {
       String value = attributes.get(key);
       return value == null ? "" : value;
diff --git a/xds/src/main/java/io/grpc/xds/CdsLoadBalancer2.java b/xds/src/main/java/io/grpc/xds/CdsLoadBalancer2.java
index 5a59b47..393d283 100644
--- a/xds/src/main/java/io/grpc/xds/CdsLoadBalancer2.java
+++ b/xds/src/main/java/io/grpc/xds/CdsLoadBalancer2.java
@@ -369,6 +369,7 @@ final class CdsLoadBalancer2 extends LoadBalancer {
             String localityName = localityName(locality);
             Attributes attr =
                 endpoint.eag().getAttributes().toBuilder()
+                    .set(EquivalentAddressGroup.ATTR_BACKEND_SERVICE, clusterName)
                     .set(io.grpc.xds.XdsAttributes.ATTR_LOCALITY, locality)
                     .set(EquivalentAddressGroup.ATTR_LOCALITY_NAME, localityName)
                     .set(io.grpc.xds.XdsAttributes.ATTR_LOCALITY_WEIGHT,
