diff --git a/Terminal.Gui/Views/Selectors/SelectorBase.cs b/Terminal.Gui/Views/Selectors/SelectorBase.cs
index f1f7ed1..5362f6d 100644
--- a/Terminal.Gui/Views/Selectors/SelectorBase.cs
+++ b/Terminal.Gui/Views/Selectors/SelectorBase.cs
@@ -529,6 +529,13 @@ public abstract class SelectorBase : View, IOrientation, IValue<int?>
                 SubViews.ElementAt (i).Width = Dim.Auto ();
             }
         }
+
+        foreach (View subview in SubViews)
+        {
+            subview.SetRelativeLayout (GetContentSize ());
+        }
+
+        SetNeedsLayout ();
     }
 
     /// <summary>
