diff --git a/inlong-sdk/transform-sdk/src/main/java/org/apache/inlong/sdk/transform/process/function/string/UrlDecodeFunction.java b/inlong-sdk/transform-sdk/src/main/java/org/apache/inlong/sdk/transform/process/function/string/UrlDecodeFunction.java
index ff26ee1bf..27a953cf5 100644
--- a/inlong-sdk/transform-sdk/src/main/java/org/apache/inlong/sdk/transform/process/function/string/UrlDecodeFunction.java
+++ b/inlong-sdk/transform-sdk/src/main/java/org/apache/inlong/sdk/transform/process/function/string/UrlDecodeFunction.java
@@ -24,34 +24,46 @@ import org.apache.inlong.sdk.transform.process.function.TransformFunction;
 import org.apache.inlong.sdk.transform.process.operator.OperatorTools;
 import org.apache.inlong.sdk.transform.process.parser.ValueParser;
 
+import net.sf.jsqlparser.expression.Expression;
 import net.sf.jsqlparser.expression.Function;
 
 import java.net.URLDecoder;
+import java.nio.charset.Charset;
 import java.nio.charset.StandardCharsets;
+import java.util.List;
 
 /**
- * UrlDecodeFunction  ->  url_decode(str)
+ * UrlDecodeFunction  ->  url_decode(str[, charset])
  * description:
  * - Return NULL if 'str' is NULL, or there is an issue with the decoding process(such as encountering an illegal
  *          escape pattern), or the encoding scheme is not supported;
- * - Return the result of decoding a given 'str' in 'application/x-www-form-urlencoded' format using the UTF-8 encoding scheme.
+ * - Return the result of decoding a given 'str' in 'application/x-www-form-urlencoded' format using the specified encoding scheme.
+ *          If charset is not specified, UTF-8 will be used.
  */
 @TransformFunction(type = FunctionConstant.STRING_TYPE, names = {
-        "url_decode"}, parameter = "(String str)", descriptions = {
+        "url_decode"}, parameter = "(String str[, String charset])", descriptions = {
                 "- Return \"\" if 'str' is NULL, or there is an issue with the decoding process(such as encountering an "
                         +
                         "illegal escape pattern), or the encoding scheme is not supported;",
                 "- Return the result of decoding a given 'str' in 'application/x-www-form-urlencoded' format using the "
                         +
-                        "UTF-8 encoding scheme."
+                        "specified encoding scheme. If charset is not specified, UTF-8 will be used."
         }, examples = {
-                "url_decode('https%3A%2F%2Fapache.inlong.com%2Fsearch%3Fq%3Djava+url+encode') = \"https://apache.inlong.com/search?q=java url encode\""})
+                "url_decode('https%3A%2F%2Fapache.inlong.com%2Fsearch%3Fq%3Djava+url+encode') = \"https://apache.inlong.com/search?q=java url encode\"",
+                "url_decode('%B2%E2%CA%D4%D6%D0%CE%C4', 'GBK') = '测试中文'"})
 public class UrlDecodeFunction implements ValueParser {
 
     private final ValueParser stringParser;
+    private final ValueParser charsetParser;
 
     public UrlDecodeFunction(Function expr) {
-        stringParser = OperatorTools.buildParser(expr.getParameters().getExpressions().get(0));
+        List<Expression> expressions = expr.getParameters().getExpressions();
+        stringParser = OperatorTools.buildParser(expressions.get(0));
+        if (expressions.size() == 2) {
+            charsetParser = OperatorTools.buildParser(expressions.get(1));
+        } else {
+            charsetParser = null;
+        }
     }
 
     @Override
@@ -66,7 +78,14 @@ public class UrlDecodeFunction implements ValueParser {
         }
 
         try {
-            return URLDecoder.decode(string, StandardCharsets.UTF_8.toString());
+            Charset charset = StandardCharsets.UTF_8;
+            if (charsetParser != null) {
+                Object charsetObj = charsetParser.parse(sourceData, rowIndex, context);
+                if (charsetObj != null) {
+                    charset = Charset.forName(OperatorTools.parseString(charsetObj));
+                }
+            }
+            return URLDecoder.decode(string, charset.toString());
         } catch (Exception e) {
             return null;
         }
diff --git a/inlong-sdk/transform-sdk/src/main/java/org/apache/inlong/sdk/transform/process/function/string/UrlEncodeFunction.java b/inlong-sdk/transform-sdk/src/main/java/org/apache/inlong/sdk/transform/process/function/string/UrlEncodeFunction.java
index e7bf57ebc..ca8c7530d 100644
--- a/inlong-sdk/transform-sdk/src/main/java/org/apache/inlong/sdk/transform/process/function/string/UrlEncodeFunction.java
+++ b/inlong-sdk/transform-sdk/src/main/java/org/apache/inlong/sdk/transform/process/function/string/UrlEncodeFunction.java
@@ -24,33 +24,45 @@ import org.apache.inlong.sdk.transform.process.function.TransformFunction;
 import org.apache.inlong.sdk.transform.process.operator.OperatorTools;
 import org.apache.inlong.sdk.transform.process.parser.ValueParser;
 
+import net.sf.jsqlparser.expression.Expression;
 import net.sf.jsqlparser.expression.Function;
 
 import java.net.URLEncoder;
+import java.nio.charset.Charset;
 import java.nio.charset.StandardCharsets;
+import java.util.List;
 
 /**
- * UrlEncodeFunction  ->  url_encode(str)
+ * UrlEncodeFunction  ->  url_encode(str[, charset])
  * description:
  * - Return NULL if 'str' is NULL, or there is an issue with the decoding process(such as encountering an illegal
  *          escape pattern), or the encoding scheme is not supported;
- * - Return the result of translating 'str' into ‘application/x-www-form-urlencoded’ format using the UTF-8 encoding scheme.
+ * - Return the result of translating 'str' into ‘application/x-www-form-urlencoded’ format using the specified encoding scheme.
+ *          If charset is not specified, UTF-8 will be used.
  */
 @TransformFunction(type = FunctionConstant.STRING_TYPE, names = {
-        "url_encode"}, parameter = "(String str)", descriptions = {
+        "url_encode"}, parameter = "(String str[, String charset])", descriptions = {
                 "- Return \"\" if 'str' is NULL, or there is an issue with the decoding process(such as encountering an "
                         +
                         "illegal escape pattern), or the encoding scheme is not supported;",
                 "- Return the result of translating 'str' into 'application/x-www-form-urlencoded' format using the " +
-                        "UTF-8 encoding scheme."
+                        "specified encoding scheme. If charset is not specified, UTF-8 will be used."
         }, examples = {
-                "url_encode('https://apache.inlong.com/search?q=java url encode') = \"https%3A%2F%2Fapache.inlong.com%2Fsearch%3Fq%3Djava+url+encode\""})
+                "url_encode('https://apache.inlong.com/search?q=java url encode') = \"https%3A%2F%2Fapache.inlong.com%2Fsearch%3Fq%3Djava+url+encode\"",
+                "url_encode('测试中文', 'GBK') = \"%B2%E2%CA%D4%D6%D0%CE%C4\""})
 public class UrlEncodeFunction implements ValueParser {
 
     private final ValueParser stringParser;
+    private final ValueParser charsetParser;
 
     public UrlEncodeFunction(Function expr) {
-        stringParser = OperatorTools.buildParser(expr.getParameters().getExpressions().get(0));
+        List<Expression> expressions = expr.getParameters().getExpressions();
+        stringParser = OperatorTools.buildParser(expressions.get(0));
+        if (expressions.size() == 2) {
+            charsetParser = OperatorTools.buildParser(expressions.get(1));
+        } else {
+            charsetParser = null;
+        }
     }
 
     @Override
@@ -66,7 +78,14 @@ public class UrlEncodeFunction implements ValueParser {
         }
 
         try {
-            return URLEncoder.encode(string, StandardCharsets.UTF_8.toString());
+            Charset charset = StandardCharsets.UTF_8;
+            if (charsetParser != null) {
+                Object charsetObj = charsetParser.parse(sourceData, rowIndex, context);
+                if (charsetObj != null) {
+                    charset = Charset.forName(OperatorTools.parseString(charsetObj));
+                }
+            }
+            return URLEncoder.encode(string, charset.toString());
         } catch (Exception e) {
             return null;
         }
