diff --git a/packages/renderer/src/validate-even-dimensions-with-codec.ts b/packages/renderer/src/validate-even-dimensions-with-codec.ts
index 430dce99ba..9183f14cbf 100644
--- a/packages/renderer/src/validate-even-dimensions-with-codec.ts
+++ b/packages/renderer/src/validate-even-dimensions-with-codec.ts
@@ -66,27 +66,19 @@ export const validateEvenDimensionsWithCodec = ({
 	const displayName = codec === 'h265' ? 'H265' : 'H264';
 
 	if (actualWidth % 2 !== 0) {
-		const message = [
-			`Codec error: You are trying to render a video with a ${displayName} codec that has a width of ${actualWidth}px, which is an uneven number.`,
-			`The ${displayName} codec does only support dimensions that are evenly divisible by two.`,
-			scale === 1
-				? `Change the width to ${Math.floor(width - 1)}px to fix this issue.`
-				: `You have used the "scale" option which might be the reason for the problem: The original width is ${width} and the scale is ${scale}x, which was multiplied to get the actual width.`,
-		]
-			.filter(truthy)
-			.join(' ');
-		throw new Error(message);
+		Log.verbose(
+			{indent, logLevel},
+			`Rounding width down to an even number from ${actualWidth} to ${Math.floor(actualWidth / 2) * 2}`,
+		);
+		actualWidth = Math.floor(actualWidth / 2) * 2;
 	}
 
 	if (actualHeight % 2 !== 0) {
-		const message = [
-			`Codec error: You are trying to render a video with a ${displayName} codec that has a height of ${actualHeight}px, which is an uneven number.`,
-			`The ${displayName} codec does only support dimensions that are evenly divisible by two. `,
-			scale === 1
-				? `Change the height to ${Math.floor(actualHeight - 1)}px to fix this issue.`
-				: `You have used the "scale" option which might be the reason for the problem: The original height is ${height} and the scale is ${scale}x, which was multiplied to get the actual height.`,
-		].join(' ');
-		throw new Error(message);
+		Log.verbose(
+			{indent, logLevel},
+			`Rounding height down to an even number from ${actualHeight} to ${Math.floor(actualHeight / 2) * 2}`,
+		);
+		actualHeight = Math.floor(actualHeight / 2) * 2;
 	}
 
 	return {
