diff --git a/internal/codegen/golang/result.go b/internal/codegen/golang/result.go
index 0820488..bff308b 100644
--- a/internal/codegen/golang/result.go
+++ b/internal/codegen/golang/result.go
@@ -268,8 +268,15 @@ func buildQueries(req *plugin.GenerateRequest, options *opts.Options, structs []
 			c := query.Columns[0]
 			name := columnName(c, 0)
 			name = strings.Replace(name, "$", "_", -1)
+			retName := escape(name)
+			// If the return value has the same name as an argument, we need to
+			// use a different variable name for the return value to avoid
+			// shadowing the argument parameter.
+			if retName == gq.Arg.Name && !gq.Arg.IsStruct() {
+				retName = "_" + retName
+			}
 			gq.Ret = QueryValue{
-				Name:      escape(name),
+				Name:      retName,
 				DBName:    name,
 				Typ:       goType(req, options, c),
 				SQLDriver: sqlpkg,
