diff --git a/src/Quartz/Impl/AdoJobStore/StdAdoDelegate.Triggers.cs b/src/Quartz/Impl/AdoJobStore/StdAdoDelegate.Triggers.cs
index af3300e..854120b 100644
--- a/src/Quartz/Impl/AdoJobStore/StdAdoDelegate.Triggers.cs
+++ b/src/Quartz/Impl/AdoJobStore/StdAdoDelegate.Triggers.cs
@@ -757,6 +757,12 @@ public virtual async ValueTask<List<IOperableTrigger>> SelectTriggersForRecoveri
             {
                 trigger = await GetObjectFromBlob<IOperableTrigger>(rs2, 0, cancellationToken).ConfigureAwait(false);
             }
+
+            if (trigger is not null)
+            {
+                trigger.SetNextFireTimeUtc(nextFireTimeUtc);
+                trigger.SetPreviousFireTimeUtc(previousFireTimeUtc);
+            }
         }
         else
         {
