diff --git a/packages/core/src/extensions/tiptap-extensions/KeyboardShortcuts/KeyboardShortcutsExtension.ts b/packages/core/src/extensions/tiptap-extensions/KeyboardShortcuts/KeyboardShortcutsExtension.ts
index c0578d7..5e67dbb 100644
--- a/packages/core/src/extensions/tiptap-extensions/KeyboardShortcuts/KeyboardShortcutsExtension.ts
+++ b/packages/core/src/extensions/tiptap-extensions/KeyboardShortcuts/KeyboardShortcutsExtension.ts
@@ -136,6 +136,22 @@ export const KeyboardShortcutsExtension = Extension.create<{
               return false;
             }
 
+            const selectionAtBlockStart =
+              state.selection.from === blockInfo.blockContent.beforePos + 1;
+            const selectionEmpty = state.selection.empty;
+
+            if (!selectionAtBlockStart || !selectionEmpty) {
+              return false;
+            }
+
+            const blockEmpty =
+              blockInfo.blockContent.node.childCount === 0 &&
+              blockInfo.blockContent.node.type.spec.content === "inline*";
+
+            if (blockEmpty) {
+              return false;
+            }
+
             if (dispatch) {
               const columnAfterPos = prevBlockInfo.bnBlock.afterPos - 1;
               const $blockAfterPos = tr.doc.resolve(columnAfterPos - 1);
