diff --git a/crd-generator/src/main/java/io/strimzi/crdgenerator/DocGenerator.java b/crd-generator/src/main/java/io/strimzi/crdgenerator/DocGenerator.java
index b1b3123..2ca50ef 100644
--- a/crd-generator/src/main/java/io/strimzi/crdgenerator/DocGenerator.java
+++ b/crd-generator/src/main/java/io/strimzi/crdgenerator/DocGenerator.java
@@ -13,6 +13,7 @@ import io.strimzi.crdgenerator.annotations.AddedIn;
 import io.strimzi.crdgenerator.annotations.Crd;
 import io.strimzi.crdgenerator.annotations.Description;
 import io.strimzi.crdgenerator.annotations.DescriptionFile;
+import io.strimzi.crdgenerator.annotations.ExternalLink;
 import io.strimzi.crdgenerator.annotations.KubeLink;
 import io.strimzi.crdgenerator.annotations.PresentInVersions;
 
@@ -89,7 +90,8 @@ class DocGenerator {
         Set<Property> memorableProperties = new HashSet<>();
 
         for (Property property : properties(crApiVersion, cls).values()) {
-            if (property.isAnnotationPresent(KubeLink.class)) {
+            if (property.isAnnotationPresent(KubeLink.class)
+                    || property.isAnnotationPresent(ExternalLink.class)) {
                 continue;
             }
 
@@ -151,9 +153,11 @@ class DocGenerator {
             final Property property = entry.getValue();
             PropertyType propertyType = property.getType();
         
-            // Set the external link to Kubernetes docs or the link for fields distinguished by `type`
+            // Set the external link to Kubernetes docs or other external docs
             KubeLink kubeLink = property.getAnnotation(KubeLink.class);
-            String externalUrl = linker != null && kubeLink != null ? linker.link(kubeLink) : null;
+            ExternalLink externalLink = property.getAnnotation(ExternalLink.class);
+            String externalUrl = linker != null && kubeLink != null ? linker.link(kubeLink)
+                    : externalLink != null ? externalLink.url() : null;
         
             // Add the property name
             out.append("|").append(propertyName);
diff --git a/crd-generator/src/main/java/io/strimzi/crdgenerator/annotations/ExternalLink.java b/crd-generator/src/main/java/io/strimzi/crdgenerator/annotations/ExternalLink.java
new file mode 100644
index 0000000..41edc64
--- /dev/null
+++ b/crd-generator/src/main/java/io/strimzi/crdgenerator/annotations/ExternalLink.java
@@ -0,0 +1,24 @@
+/*
+ * Copyright Strimzi authors.
+ * License: Apache License 2.0 (see the file LICENSE or http://apache.org/licenses/LICENSE-2.0.html).
+ */
+package io.strimzi.crdgenerator.annotations;
+
+import java.lang.annotation.ElementType;
+import java.lang.annotation.Retention;
+import java.lang.annotation.RetentionPolicy;
+import java.lang.annotation.Target;
+
+/**
+ * Configure an external link to documentation that should be used for this property
+ */
+@Retention(RetentionPolicy.RUNTIME)
+@Target({ElementType.METHOD, ElementType.FIELD})
+public @interface ExternalLink {
+    /**
+     * Gets the URL to which the link should point.
+     *
+     * @return  URL to which the link should point
+     */
+    String url();
+}
\ No newline at end of file
