diff --git a/harper-core/src/lib.rs b/harper-core/src/lib.rs
index b1e845c..b6f66e2 100644
--- a/harper-core/src/lib.rs
+++ b/harper-core/src/lib.rs
@@ -81,7 +81,7 @@ pub fn remove_overlaps(lints: &mut Vec<Lint>) {
     }
 
     let mut remove_indices = VecDeque::new();
-    lints.sort_by_key(|l| l.priority);
+    lints.sort_by_key(|l| std::cmp::Reverse(l.priority));
     lints.sort_by_key(|l| (l.span.start, !0 - l.span.end));
 
     let mut cur = 0;
@@ -133,7 +133,7 @@ pub fn remove_overlaps_map<K: Ord>(lint_map: &mut BTreeMap<K, Vec<Lint>>) {
         }
     }
 
-    spans.sort_by_key(|span| span.priority);
+    spans.sort_by_key(|span| std::cmp::Reverse(span.priority));
     spans.sort_by_key(|span| (span.start, usize::MAX - span.end));
 
     let mut cur = 0;
diff --git a/harper-core/src/linting/split_words.rs b/harper-core/src/linting/split_words.rs
index df69afa..88d3791 100644
--- a/harper-core/src/linting/split_words.rs
+++ b/harper-core/src/linting/split_words.rs
@@ -135,7 +135,7 @@ impl ExprLinter for SplitWords {
                 lint_kind: LintKind::Typo,
                 suggestions,
                 message: message?,
-                priority: 31,
+                priority: 127,
             });
         }
 
diff --git a/harper-core/tests/text/linters/Alice's Adventures in Wonderland.snap.yml b/harper-core/tests/text/linters/Alice's Adventures in Wonderland.snap.yml
index 777c402..5088491 100644
--- a/harper-core/tests/text/linters/Alice's Adventures in Wonderland.snap.yml	
+++ b/harper-core/tests/text/linters/Alice's Adventures in Wonderland.snap.yml	
@@ -1074,7 +1074,7 @@ Suggest:
 
 
 
-Lint:    Typo (31 priority)
+Lint:    Typo (127 priority)
 Message: |
      668 | “Sure, it’s an arm, yer honour!” (He pronounced it “arrum.”)
          |                                                     ^~~~~ `arrum` should probably be written as `arr um`.
@@ -1395,7 +1395,7 @@ Suggest:
 
 
 
-Lint:    Typo (31 priority)
+Lint:    Typo (127 priority)
 Message: |
      961 | her foot, that there was hardly room to open her mouth; but she did it at last,
      962 | and managed to swallow a morsel of the lefthand bit.
@@ -1513,7 +1513,7 @@ Suggest:
 
 
 
-Lint:    Typo (31 priority)
+Lint:    Typo (127 priority)
 Message: |
     1055 | this size: why, I should frighten them out of their wits!” So she began nibbling
     1056 | at the righthand bit again, and did not venture to go near the house till she
@@ -1740,7 +1740,7 @@ Suggest:
 
 
 
-Lint:    Typo (31 priority)
+Lint:    Typo (127 priority)
 Message: |
     1349 | did not like to go nearer till she had nibbled some more of the lefthand bit of
          |                                                                 ^~~~~~~~ `lefthand` should probably be written as `left hand`.
@@ -3304,7 +3304,7 @@ Suggest:
 
 
 
-Lint:    Typo (31 priority)
+Lint:    Typo (127 priority)
 Message: |
     2490 | > beautiful Soup? Beau—ootiful Soo—oop! Beau—ootiful Soo—oop! Soo—oop of the
     2491 | > e—e—evening, Beautiful, beauti—FUL SOUP!”
diff --git a/harper-core/tests/text/linters/Computer science.snap.yml b/harper-core/tests/text/linters/Computer science.snap.yml
index 581e43c..4a0e4c1 100644
--- a/harper-core/tests/text/linters/Computer science.snap.yml	
+++ b/harper-core/tests/text/linters/Computer science.snap.yml	
@@ -428,7 +428,7 @@ Suggest:
 
 
 
-Lint:    Typo (31 priority)
+Lint:    Typo (127 priority)
 Message: |
      131 | expression "automatic information" (e.g. "informazione automatica" in Italian)
          |                                                        ^~~~~~~~~~ `automatica` should probably be written as `automatic a`.
@@ -1327,7 +1327,7 @@ Suggest:
 
 
 
-Lint:    Typo (31 priority)
+Lint:    Typo (127 priority)
 Message: |
      468 |     Boehm's and Jacopini's insight can be further simplified with the use of
      469 |     goto (which means it is more elementary than structured programming).
diff --git a/harper-core/tests/text/linters/The Constitution of the United States.snap.yml b/harper-core/tests/text/linters/The Constitution of the United States.snap.yml
index b31db15..903e630 100644
--- a/harper-core/tests/text/linters/The Constitution of the United States.snap.yml	
+++ b/harper-core/tests/text/linters/The Constitution of the United States.snap.yml	
@@ -269,7 +269,7 @@ Suggest:
 
 
 
-Lint:    Typo (31 priority)
+Lint:    Typo (127 priority)
 Message: |
      103 | The Senate shall chuse their other Officers, and also a President pro tempore,
          |                                                                       ^~~~~~~ `tempore` should probably be written as `temp ore`.
@@ -1095,7 +1095,7 @@ Suggest:
 
 
 
-Lint:    Typo (31 priority)
+Lint:    Typo (127 priority)
 Message: |
      415 | Whenever the President transmits to the President pro tempore of the Senate and
          |                                                       ^~~~~~~ `tempore` should probably be written as `temp ore`.
@@ -1134,7 +1134,7 @@ Suggest:
 
 
 
-Lint:    Typo (31 priority)
+Lint:    Typo (127 priority)
 Message: |
      422 | the executive departments or of such other body as Congress may by law provide,
      423 | transmit to the President pro tempore of the Senate and the Speaker of the
@@ -1177,7 +1177,7 @@ Suggest:
 
 
 
-Lint:    Typo (31 priority)
+Lint:    Typo (127 priority)
 Message: |
      428 | Thereafter, when the President transmits to the President pro tempore of the
          |                                                               ^~~~~~~ `tempore` should probably be written as `temp ore`.
@@ -1199,7 +1199,7 @@ Suggest:
 
 
 
-Lint:    Typo (31 priority)
+Lint:    Typo (127 priority)
 Message: |
      433 | transmit within four days to the President pro tempore of the Senate and the
          |                                                ^~~~~~~ `tempore` should probably be written as `temp ore`.
diff --git a/harper-core/tests/text/linters/The Great Gatsby.snap.yml b/harper-core/tests/text/linters/The Great Gatsby.snap.yml
index 10db0e2..2a4f65c 100644
--- a/harper-core/tests/text/linters/The Great Gatsby.snap.yml	
+++ b/harper-core/tests/text/linters/The Great Gatsby.snap.yml	
@@ -213,7 +213,7 @@ Suggest:
 
 
 
-Lint:    Typo (31 priority)
+Lint:    Typo (127 priority)
 Message: |
       72 | sounded like a great idea. He found the house, a weatherbeaten cardboard
          |                                                  ^~~~~~~~~~~~~ `weatherbeaten` should probably be written as `weather beaten`.
@@ -638,7 +638,7 @@ Suggest:
 
 
 
-Lint:    Typo (31 priority)
+Lint:    Typo (127 priority)
 Message: |
      529 | the whole evening had been a trick of some sort to exact a contributary emotion
          |                                                            ^~~~~~~~~~~~ `contributary` should probably be written as `con tributary`.
@@ -709,7 +709,7 @@ Suggest:
 
 
 
-Lint:    Typo (31 priority)
+Lint:    Typo (127 priority)
 Message: |
      628 | Already it was deep summer on roadhouse roofs and in front of wayside garages,
      629 | where new red gaspumps sat out in pools of light, and when I reached my estate
@@ -875,7 +875,7 @@ Suggest:
 
 
 
-Lint:    Typo (31 priority)
+Lint:    Typo (127 priority)
 Message: |
      683 | up to New York with Tom on the train one afternoon, and when we stopped by the
      684 | ashheaps he jumped to his feet and, taking hold of my elbow, literally forced me
@@ -1352,7 +1352,7 @@ Suggest:
 
 
 
-Lint:    Typo (31 priority)
+Lint:    Typo (127 priority)
 Message: |
     1172 | York—every Monday these same oranges and lemons left his back door in a pyramid
     1173 | of pulpless halves. There was a machine in the kitchen which could extract the
@@ -1723,7 +1723,7 @@ Suggest:
 
 
 
-Lint:    Typo (31 priority)
+Lint:    Typo (127 priority)
 Message: |
     1413 | trembling a little to the stiff, tinny drip of the banjoes on the lawn.
          |                                                    ^~~~~~~ `banjoes` should probably be written as `banjo es`.
@@ -1984,7 +1984,7 @@ Suggest:
 
 
 
-Lint:    Typo (31 priority)
+Lint:    Typo (127 priority)
 Message: |
     1693 | “Wha’s matter?” he inquired calmly. “Did we run outa gas?”
          |                                                 ^~~~ `outa` should probably be written as `out a`.
@@ -3572,7 +3572,7 @@ Suggest:
 
 
 
-Lint:    Typo (31 priority)
+Lint:    Typo (127 priority)
 Message: |
     2319 | That was nineteen-seventeen. By the next year I had a few beaux myself, and I
          |                                                           ^~~~~ `beaux` has a missing space between words.
@@ -3654,7 +3654,7 @@ Suggest:
 
 
 
-Lint:    Typo (31 priority)
+Lint:    Typo (127 priority)
 Message: |
     2349 | “Here, deares’.” She groped around in a wastebasket she had with her on the bed
          |        ^~~~~~ `deares` should probably be written as `dear es`.
@@ -3903,7 +3903,7 @@ Suggest:
 
 
 
-Lint:    Typo (31 priority)
+Lint:    Typo (127 priority)
 Message: |
     2607 | Clay’s “Economics,” starting at the Finnish tread that shook the kitchen floor,
     2608 | and peering toward the bleared windows from time to time as if a series of
@@ -4169,7 +4169,7 @@ Suggest:
 
 
 
-Lint:    Typo (31 priority)
+Lint:    Typo (127 priority)
 Message: |
     2878 | brought more and the soft rich heap mounted higher—shirts with stripes and
     2879 | scrolls and plaids in coral and applegreen and lavender and faint orange, with
@@ -4443,7 +4443,7 @@ Suggest:
 
 
 
-Lint:    Typo (31 priority)
+Lint:    Typo (127 priority)
 Message: |
     3089 | in turn steward, mate, skipper, secretary, and even jailor, for Dan Cody sober
          |                                                     ^~~~~~ `jailor` should probably be written as `jail or`.
@@ -4655,7 +4655,7 @@ Suggest:
 
 
 
-Lint:    Typo (31 priority)
+Lint:    Typo (127 priority)
 Message: |
     3349 | But the rest offended her—and inarguably, because it wasn’t a gesture but an
          |                               ^~~~~~~~~~ `inarguably` should probably be written as `in arguably`.
@@ -5006,7 +5006,7 @@ Suggest:
 
 
 
-Lint:    Typo (31 priority)
+Lint:    Typo (127 priority)
 Message: |
     3646 | “The bles-sed pre-cious! Did mother get powder on your old yellowy hair? Stand
          |      ^~~~ `bles` should probably be written as `bl es`.
@@ -5215,7 +5215,7 @@ Suggest:
 
 
 
-Lint:    Typo (31 priority)
+Lint:    Typo (127 priority)
 Message: |
     3948 | behind. Over the ashheaps the giant eyes of Doctor T. J. Eckleburg kept their
          |                  ^~~~~~~~ `ashheaps` should probably be written as `ash heaps`.
@@ -5693,7 +5693,7 @@ Suggest:
 
 
 
-Lint:    Typo (31 priority)
+Lint:    Typo (127 priority)
 Message: |
     4391 | The young Greek, Michaelis, who ran the coffee joint beside the ashheaps was the
          |                                                                 ^~~~~~~~ `ashheaps` should probably be written as `ash heaps`.
@@ -5871,7 +5871,7 @@ Suggest:
 
 
 
-Lint:    Typo (31 priority)
+Lint:    Typo (127 priority)
 Message: |
     4470 | a little book. At first I couldn’t find the source of the high, groaning words
     4471 | that echoed clamorously through the bare garage—then I saw Wilson standing on
@@ -6068,7 +6068,7 @@ Suggest:
 
 
 
-Lint:    Typo (31 priority)
+Lint:    Typo (127 priority)
 Message: |
     4503 | “She ran out ina road. Son-of-a-bitch didn’t even stopus car.”
          |                                                   ^~~~~~ `stopus` has a missing space between words.
@@ -6100,7 +6100,7 @@ Suggest:
 
 
 
-Lint:    Typo (31 priority)
+Lint:    Typo (127 priority)
 Message: |
     4505 | “There was two cars,” said Michaelis, “one comin’, one goin’, see?”
          |                                            ^~~~~ `comin` has a missing space between words.
@@ -6121,7 +6121,7 @@ Suggest:
 
 
 
-Lint:    Typo (31 priority)
+Lint:    Typo (127 priority)
 Message: |
     4505 | “There was two cars,” said Michaelis, “one comin’, one goin’, see?”
          |                                                        ^~~~ `goin` should probably be written as `go in`.
@@ -6141,7 +6141,7 @@ Suggest:
 
 
 
-Lint:    Typo (31 priority)
+Lint:    Typo (127 priority)
 Message: |
     4509 | “One goin’ each way. Well, she”—his hand rose toward the blankets but stopped
          |      ^~~~ `goin` should probably be written as `go in`.
@@ -6180,7 +6180,7 @@ Suggest:
 
 
 
-Lint:    Typo (31 priority)
+Lint:    Typo (127 priority)
 Message: |
     4510 | half way and fell to his side—“she ran out there an’ the one comin’ from N’York
          |                                                              ^~~~~ `comin` has a missing space between words.
@@ -6213,7 +6213,7 @@ Suggest:
 
 
 
-Lint:    Typo (31 priority)
+Lint:    Typo (127 priority)
 Message: |
     4510 | half way and fell to his side—“she ran out there an’ the one comin’ from N’York
     4511 | knock right into her, goin’ thirty or forty miles an hour.”
@@ -6509,7 +6509,7 @@ Suggest:
 
 
 
-Lint:    Typo (31 priority)
+Lint:    Typo (127 priority)
 Message: |
     5015 | When I passed the ashheaps on the train that morning I had crossed deliberately
          |                   ^~~~~~~~ `ashheaps` should probably be written as `ash heaps`.
@@ -6764,7 +6764,7 @@ Suggest:
 
 
 
-Lint:    Typo (31 priority)
+Lint:    Typo (127 priority)
 Message: |
     5146 | Wilson’s glazed eyes turned out to the ashheaps, where small gray clouds took on
          |                                        ^~~~~~~~ `ashheaps` should probably be written as `ash heaps`.
@@ -7938,7 +7938,7 @@ Suggest:
 
 
 
-Lint:    Typo (31 priority)
+Lint:    Typo (127 priority)
 Message: |
     5714 | leaf, her face the same brown tint as the fingerless glove on her knee. When I
          |                                           ^~~~~~~~~~ `fingerless` should probably be written as `finger less`.
