diff --git a/sphinx/writers/html5.py b/sphinx/writers/html5.py
index d65823026..a0064c748 100644
--- a/sphinx/writers/html5.py
+++ b/sphinx/writers/html5.py
@@ -233,7 +233,8 @@ def depart_desc_parameter(self, node: Element) -> None:
                 and self.list_is_required_param[self.param_group_index + 1]
             )
             opt_param_left_at_level = self.params_left_at_level > 0
-            if (
+            # Don't add a comma after the last parameter in a multi-line signature
+            if not is_last_group and (
                 opt_param_left_at_level
                 or is_required
                 and (is_last_group or next_is_required)
diff --git a/sphinx/writers/latex.py b/sphinx/writers/latex.py
index 1004b716e..c82ebd22f 100644
--- a/sphinx/writers/latex.py
+++ b/sphinx/writers/latex.py
@@ -1010,7 +1010,8 @@ def _depart_sig_parameter(self, node: Element) -> None:
                 and self.list_is_required_param[self.param_group_index + 1]
             )
             opt_param_left_at_level = self.params_left_at_level > 0
-            if (
+            # Don't add a comma after the last parameter in a multi-line signature
+            if not is_last_group and (
                 opt_param_left_at_level
                 or is_required
                 and (is_last_group or next_is_required)
diff --git a/sphinx/writers/text.py b/sphinx/writers/text.py
index 748603d71..14c950505 100644
--- a/sphinx/writers/text.py
+++ b/sphinx/writers/text.py
@@ -694,7 +694,8 @@ def visit_desc_parameter(self, node: Element) -> None:
                 and self.list_is_required_param[self.param_group_index + 1]
             )
             opt_param_left_at_level = self.params_left_at_level > 0
-            if (
+            # Don't add a comma after the last parameter in a multi-line signature
+            if not is_last_group and (
                 opt_param_left_at_level
                 or is_required
                 and (is_last_group or next_is_required)
