diff --git a/playwright/src/main/java/com/microsoft/playwright/impl/BrowserContextImpl.java b/playwright/src/main/java/com/microsoft/playwright/impl/BrowserContextImpl.java
index b1f21d88..5f7db466 100644
--- a/playwright/src/main/java/com/microsoft/playwright/impl/BrowserContextImpl.java
+++ b/playwright/src/main/java/com/microsoft/playwright/impl/BrowserContextImpl.java
@@ -65,6 +65,7 @@ class BrowserContextImpl extends ChannelOwner implements BrowserContext {
     result.put(EventType.RESPONSE, "response");
     result.put(EventType.REQUESTFINISHED, "requestFinished");
     result.put(EventType.REQUESTFAILED, "requestFailed");
+    result.put(EventType.RECORDER_EVENT, "recorderEvent");
     return result;
   }
   private final ListenerCollection<EventType> listeners = new ListenerCollection<>(eventSubscriptions(), this);
@@ -92,6 +93,7 @@ class BrowserContextImpl extends ChannelOwner implements BrowserContext {
     REQUESTFAILED,
     REQUESTFINISHED,
     RESPONSE,
+    RECORDER_EVENT,
   }
 
   BrowserContextImpl(ChannelOwner parent, String type, String guid, JsonObject initializer) {
@@ -568,6 +570,17 @@ class BrowserContextImpl extends ChannelOwner implements BrowserContext {
     sendMessage("setOffline", params, NO_TIMEOUT);
   }
 
+  void enableRecorder(JsonObject params, Consumer<JsonObject> eventSink) {
+    sendMessage("enableRecorder", params, NO_TIMEOUT);
+    if (eventSink != null) {
+      listeners.add(EventType.RECORDER_EVENT, eventSink);
+    }
+  }
+
+  void disableRecorder() {
+    sendMessage("disableRecorder", null, NO_TIMEOUT);
+  }
+
   @Override
   public String storageState(StorageStateOptions options) {
     if (options == null) {
