diff --git a/iwyu.cc b/iwyu.cc
index ab2eef7..d9eeed4 100644
--- a/iwyu.cc
+++ b/iwyu.cc
@@ -5165,6 +5165,19 @@ class IwyuAstConsumer
   bool VisitFunctionDecl(FunctionDecl* decl) {
     if (CanIgnoreCurrentASTNode())
       return true;
+
+    // If the function is an explicit specialization of a function template,
+    // we need to report the primary template declaration as used.
+    // e.g.
+    //   template <typename T> void print(T);  // required by specialization
+    //   template <> void print(A) = delete;   // explicit specialization
+    if (const FunctionTemplateDecl* ft_decl = decl->getPrimaryTemplate()) {
+      if (decl->isFunctionTemplateSpecialization() &&
+          !IsExplicitInstantiation(decl)) {
+        ReportDeclForwardDeclareUse(CurrentLoc(), ft_decl);
+      }
+    }
+
     // If a function declaration specifies default arguments but not for
     // the last parameters, like this:
     // void Fn(int = 0, int);
