diff --git a/harper-core/src/linting/lint_group.rs b/harper-core/src/linting/lint_group.rs
index 1a7c3ad..8c81086 100644
--- a/harper-core/src/linting/lint_group.rs
+++ b/harper-core/src/linting/lint_group.rs
@@ -77,6 +77,7 @@ use super::feel_fell::FeelFell;
 use super::few_units_of_time_ago::FewUnitsOfTimeAgo;
 use super::filler_words::FillerWords;
 use super::find_fine::FindFine;
+use super::flaunt_flout::FlauntFlout;
 use super::first_aid_kit::FirstAidKit;
 use super::flesh_out_vs_full_fledged::FleshOutVsFullFledged;
 use super::for_noun::ForNoun;
@@ -663,6 +664,7 @@ impl LintGroup {
         insert_expr_rule!(FewUnitsOfTimeAgo, true);
         insert_expr_rule!(FillerWords, true);
         insert_struct_rule!(FindFine, true);
+        insert_expr_rule!(FlauntFlout, true);
         insert_expr_rule!(FirstAidKit, true);
         insert_expr_rule!(FleshOutVsFullFledged, true);
         insert_expr_rule!(ForNoun, true);
diff --git a/harper-core/src/linting/mod.rs b/harper-core/src/linting/mod.rs
index 0b7af3e..25dd9a5 100644
--- a/harper-core/src/linting/mod.rs
+++ b/harper-core/src/linting/mod.rs
@@ -72,6 +72,7 @@ mod feel_fell;
 mod few_units_of_time_ago;
 mod filler_words;
 mod find_fine;
+mod flaunt_flout;
 mod first_aid_kit;
 mod flesh_out_vs_full_fledged;
 mod for_noun;
