diff --git a/hugolib/page__content.go b/hugolib/page__content.go
index 7377936..ce864b3 100644
--- a/hugolib/page__content.go
+++ b/hugolib/page__content.go
@@ -1102,7 +1102,7 @@ func (c *cachedContentScope) RenderShortcodes(ctx context.Context) (template.HTM
 		cb(pco, ct)
 	}
 
-	if tpl.Context.IsInGoldmark.Get(ctx) {
+	if cb != nil && tpl.Context.IsInGoldmark.Get(ctx) {
 		// This content will be parsed and rendered by Goldmark.
 		// Wrap it in a special Hugo markup to assign the correct Page from
 		// the stack.
