diff --git a/packages/expect/src/index.ts b/packages/expect/src/index.ts
index 934d5711a3..1228cd4cfb 100644
--- a/packages/expect/src/index.ts
+++ b/packages/expect/src/index.ts
@@ -165,7 +165,7 @@ const makeResolveMatcher =
     matcherName: string,
     matcher: RawMatcherFn,
     isNot: boolean,
-    actual: Promise<any>,
+    actual: Promise<any> | (() => Promise<any>),
     outerErr: JestAssertionError,
   ): PromiseMatcherFn =>
   (...args) => {
@@ -174,11 +174,16 @@ const makeResolveMatcher =
       promise: 'resolves',
     };
 
-    if (!isPromise(actual)) {
+    const actualWrapper: Promise<any> =
+      typeof actual === 'function' ? actual() : actual;
+
+    if (!isPromise(actualWrapper)) {
       throw new JestAssertionError(
         matcherUtils.matcherErrorMessage(
           matcherUtils.matcherHint(matcherName, undefined, '', options),
-          `${matcherUtils.RECEIVED_COLOR('received')} value must be a promise`,
+          `${matcherUtils.RECEIVED_COLOR(
+            'received',
+          )} value must be a promise or a function returning a promise`,
           matcherUtils.printWithType(
             'Received',
             actual,
@@ -190,7 +195,7 @@ const makeResolveMatcher =
 
     const innerErr = new JestAssertionError();
 
-    return actual.then(
+    return actualWrapper.then(
       result =>
         makeThrowingMatcher(matcher, isNot, 'resolves', result, innerErr).apply(
           null,
