diff --git a/bundles/org.openhab.binding.avmfritz/src/main/java/org/openhab/binding/avmfritz/internal/dto/HeatingModel.java b/bundles/org.openhab.binding.avmfritz/src/main/java/org/openhab/binding/avmfritz/internal/dto/HeatingModel.java
index 2f5f374781..8081082fcf 100644
--- a/bundles/org.openhab.binding.avmfritz/src/main/java/org/openhab/binding/avmfritz/internal/dto/HeatingModel.java
+++ b/bundles/org.openhab.binding.avmfritz/src/main/java/org/openhab/binding/avmfritz/internal/dto/HeatingModel.java
@@ -107,6 +107,10 @@ public class HeatingModel implements BatteryModel {
             return MODE_UNKNOWN;
         } else if (TEMP_FRITZ_ON.compareTo(tsoll) == 0) {
             return MODE_ON;
+        } else if (BigDecimal.ONE.equals(getBoostactive()) || TEMP_FRITZ_MAX.compareTo(tsoll) == 0) {
+            // Boost mode should be shown even if the target temperature still reports OFF (e.g. 6°C/"off")
+            // when the boost button is pressed on the device.
+            return MODE_BOOST;
         } else if (TEMP_FRITZ_OFF.compareTo(tsoll) == 0) {
             return MODE_OFF;
         } else if (BigDecimal.ONE.equals(getWindowopenactiv())) {
@@ -115,8 +119,6 @@ public class HeatingModel implements BatteryModel {
             return MODE_COMFORT;
         } else if (absenk != null && absenk.compareTo(tsoll) == 0) {
             return MODE_ECO;
-        } else if (BigDecimal.ONE.equals(getBoostactive()) || TEMP_FRITZ_MAX.compareTo(tsoll) == 0) {
-            return MODE_BOOST;
         } else {
             return MODE_ON;
         }
@@ -171,6 +173,10 @@ public class HeatingModel implements BatteryModel {
         return boostactive;
     }
 
+    public void setBoostactive(BigDecimal boostactive) {
+        this.boostactive = boostactive;
+    }
+
     public @Nullable BigDecimal getBoostactiveendtime() {
         return boostactiveendtime;
     }
