diff --git a/src/npcmove.cpp b/src/npcmove.cpp
index d88cb01..984bc6c 100644
--- a/src/npcmove.cpp
+++ b/src/npcmove.cpp
@@ -4088,6 +4088,10 @@ void npc::worker_downtime()
     //  should walk to the chair and then change their destination due to the seat being taken.
     tripoint_bub_ms local_chair_pos = chair_pos ? here.get_bub( *chair_pos ) :
                                       tripoint_bub_ms::zero;
+    if( assigned_camp && chair_pos &&
+        project_to<coords::omt>( pos_abs() ) != project_to<coords::omt>( *chair_pos ) ) {
+        chair_pos = std::nullopt;
+    }
     if( chair_pos && !creatures.creature_at( local_chair_pos ) ) {
         if( here.has_flag_furn( ter_furn_flag::TFLAG_CAN_SIT, local_chair_pos ) ) {
             update_path( local_chair_pos );
@@ -4108,11 +4112,10 @@ void npc::worker_downtime()
             for( const tripoint_bub_ms &elem : here.points_in_radius( pos_bub(), 30 ) ) {
                 if( here.has_flag_furn( ter_furn_flag::TFLAG_CAN_SIT, elem ) && !creatures.creature_at( elem ) &&
                     could_move_onto( elem ) && !!assigned_camp ) {
-                    std::optional<basecamp *> camp = overmap_buffer.find_camp( assigned_camp->xy() );
-                    if( !!camp && !!( *camp ) &&
-                        ( *camp )->point_within_camp( project_to<coords::omt>( here.get_abs( elem ) ) ) ) {
+                    const tripoint_abs_ms elem_abs = here.get_abs( elem );
+                    if( project_to<coords::omt>( elem_abs ) == pos_abs_omt() ) {
                         // this one will do
-                        chair_pos = here.get_abs( elem );
+                        chair_pos = elem_abs;
                         return;
                     }
                 }
