diff --git a/naga/src/front/wgsl/parse/conv.rs b/naga/src/front/wgsl/parse/conv.rs
index 6270d4b..3bfd70c 100644
--- a/naga/src/front/wgsl/parse/conv.rs
+++ b/naga/src/front/wgsl/parse/conv.rs
@@ -550,7 +550,10 @@ pub fn map_predeclared_type(
         "texture_storage_2d_array" => TypeGenerator::StorageTexture { dim: ImageDimension::D2,   arrayed: true  }.into(),
         "texture_storage_3d" =>       TypeGenerator::StorageTexture { dim: ImageDimension::D3,   arrayed: false }.into(),
         // binding array
-        "binding_array" => TypeGenerator::BindingArray.into(),
+        "binding_array" => {
+            enable_extensions.require(ImplementedEnableExtension::WgpuBindingArray, span)?;
+            TypeGenerator::BindingArray.into()
+        },
         // acceleration structure
         "acceleration_structure" => TypeGenerator::AccelerationStructure.into(),
         // ray query
diff --git a/naga/src/front/wgsl/parse/directive/enable_extension.rs b/naga/src/front/wgsl/parse/directive/enable_extension.rs
index a6d7285..2a46bc5 100644
--- a/naga/src/front/wgsl/parse/directive/enable_extension.rs
+++ b/naga/src/front/wgsl/parse/directive/enable_extension.rs
@@ -10,6 +10,7 @@ use alloc::boxed::Box;
 /// Tracks the status of every enable-extension known to Naga.
 #[derive(Clone, Copy, Debug, Eq, PartialEq)]
 pub(crate) struct EnableExtensions {
+    wgpu_binding_array: bool,
     wgpu_mesh_shader: bool,
     wgpu_ray_query: bool,
     wgpu_ray_query_vertex_return: bool,
@@ -27,6 +28,7 @@ pub(crate) struct EnableExtensions {
 impl EnableExtensions {
     pub(crate) const fn empty() -> Self {
         Self {
+            wgpu_binding_array: false,
             wgpu_mesh_shader: false,
             wgpu_ray_query: false,
             wgpu_ray_query_vertex_return: false,
@@ -44,6 +46,7 @@ impl EnableExtensions {
     /// Add an enable-extension to the set requested by a module.
     pub(crate) const fn add(&mut self, ext: ImplementedEnableExtension) {
         let field = match ext {
+            ImplementedEnableExtension::WgpuBindingArray => &mut self.wgpu_binding_array,
             ImplementedEnableExtension::WgpuMeshShader => &mut self.wgpu_mesh_shader,
             ImplementedEnableExtension::WgpuRayQuery => &mut self.wgpu_ray_query,
             ImplementedEnableExtension::WgpuRayQueryVertexReturn => {
@@ -66,6 +69,7 @@ impl EnableExtensions {
     /// Query whether an enable-extension tracked here has been requested.
     pub(crate) const fn contains(&self, ext: ImplementedEnableExtension) -> bool {
         match ext {
+            ImplementedEnableExtension::WgpuBindingArray => self.wgpu_binding_array,
             ImplementedEnableExtension::WgpuMeshShader => self.wgpu_mesh_shader,
             ImplementedEnableExtension::WgpuRayQuery => self.wgpu_ray_query,
             ImplementedEnableExtension::WgpuRayQueryVertexReturn => {
@@ -123,6 +127,7 @@ impl EnableExtension {
     const F16: &'static str = "f16";
     const CLIP_DISTANCES: &'static str = "clip_distances";
     const DUAL_SOURCE_BLENDING: &'static str = "dual_source_blending";
+    const BINDING_ARRAY: &'static str = "wgpu_binding_array";
     const MESH_SHADER: &'static str = "wgpu_mesh_shader";
     const RAY_QUERY: &'static str = "wgpu_ray_query";
     const RAY_QUERY_VERTEX_RETURN: &'static str = "wgpu_ray_query_vertex_return";
@@ -141,6 +146,7 @@ impl EnableExtension {
             Self::DUAL_SOURCE_BLENDING => {
                 Self::Implemented(ImplementedEnableExtension::DualSourceBlending)
             }
+            Self::BINDING_ARRAY => Self::Implemented(ImplementedEnableExtension::WgpuBindingArray),
             Self::MESH_SHADER => Self::Implemented(ImplementedEnableExtension::WgpuMeshShader),
             Self::RAY_QUERY => Self::Implemented(ImplementedEnableExtension::WgpuRayQuery),
             Self::RAY_QUERY_VERTEX_RETURN => {
@@ -164,6 +170,7 @@ impl EnableExtension {
     pub const fn to_ident(self) -> &'static str {
         match self {
             Self::Implemented(kind) => match kind {
+                ImplementedEnableExtension::WgpuBindingArray => Self::BINDING_ARRAY,
                 ImplementedEnableExtension::WgpuMeshShader => Self::MESH_SHADER,
                 ImplementedEnableExtension::WgpuRayQuery => Self::RAY_QUERY,
                 ImplementedEnableExtension::WgpuRayQueryVertexReturn => {
@@ -207,6 +214,8 @@ pub enum ImplementedEnableExtension {
     ///
     /// [`enable clip_distances;`]: https://www.w3.org/TR/WGSL/#extension-clip_distances
     ClipDistances,
+    /// Enables the `wgpu_binding_array` extension, native only.
+    WgpuBindingArray,
     /// Enables the `wgpu_mesh_shader` extension, native only
     WgpuMeshShader,
     /// Enables the `wgpu_ray_query` extension, native only.
@@ -235,6 +244,7 @@ impl ImplementedEnableExtension {
         Self::F16,
         Self::DualSourceBlending,
         Self::ClipDistances,
+        Self::WgpuBindingArray,
         Self::WgpuMeshShader,
         Self::WgpuRayQuery,
         Self::WgpuRayQueryVertexReturn,
@@ -257,6 +267,7 @@ impl ImplementedEnableExtension {
             Self::F16 => C::SHADER_FLOAT16,
             Self::DualSourceBlending => C::DUAL_SOURCE_BLENDING,
             Self::ClipDistances => C::CLIP_DISTANCES,
+            Self::WgpuBindingArray => C::empty(),
             Self::WgpuMeshShader => C::MESH_SHADER,
             Self::WgpuRayQuery => C::RAY_QUERY,
             Self::WgpuRayQueryVertexReturn => C::RAY_HIT_VERTEX_POSITION,
diff --git a/naga/tests/in/wgsl/binding-arrays.wgsl b/naga/tests/in/wgsl/binding-arrays.wgsl
index d1bf121..0f7e955 100644
--- a/naga/tests/in/wgsl/binding-arrays.wgsl
+++ b/naga/tests/in/wgsl/binding-arrays.wgsl
@@ -1,3 +1,5 @@
+enable wgpu_binding_array;
+
 struct UniformIndex {
     index: u32
 };
diff --git a/naga/tests/in/wgsl/binding-buffer-arrays.wgsl b/naga/tests/in/wgsl/binding-buffer-arrays.wgsl
index fb25623..7a8748c 100644
--- a/naga/tests/in/wgsl/binding-buffer-arrays.wgsl
+++ b/naga/tests/in/wgsl/binding-buffer-arrays.wgsl
@@ -1,3 +1,5 @@
+enable wgpu_binding_array;
+
 struct UniformIndex {
     index: u32
 }
