diff --git a/clients/drcachesim/CMakeLists.txt b/clients/drcachesim/CMakeLists.txt
index cf0fbf939..449b1bc56 100644
--- a/clients/drcachesim/CMakeLists.txt
+++ b/clients/drcachesim/CMakeLists.txt
@@ -581,7 +581,8 @@ macro(add_drmemtrace name type static_DR no_main)
     target_link_libraries(${name} atomic)
   endif ()
   add_dependencies(${name} api_headers)
-  install_target(${name} ${INSTALL_CLIENTS_LIB})
+  install_exported_target(${name} ${INSTALL_CLIENTS_LIB})
+  DR_export_target(${name})
 endmacro()
 
 add_drmemtrace(drmemtrace SHARED OFF OFF)
