diff --git a/src/Quartz/Impl/Triggers/DailyTimeIntervalTriggerImpl.cs b/src/Quartz/Impl/Triggers/DailyTimeIntervalTriggerImpl.cs
index 7627b0a..729a11b 100644
--- a/src/Quartz/Impl/Triggers/DailyTimeIntervalTriggerImpl.cs
+++ b/src/Quartz/Impl/Triggers/DailyTimeIntervalTriggerImpl.cs
@@ -734,11 +734,10 @@ public override void SetPreviousFireTimeUtc(DateTimeOffset? previousFireTimeUtc)
 
         // f. Continue to calculate the fireTime by incremental unit of intervals.
         // recall that if fireTime was less that fireTimeStartDate, we didn't get this far
-        DateTimeOffset startOfDayUtc = TimeZoneUtil.ConvertTime(fireTimeStartDate, TimeZone);
-        long secondsAfterStart = (long) (fireTime.Value - startOfDayUtc).TotalSeconds;
+        long secondsAfterStart = (long) (fireTime.Value.DateTime - fireTimeStartDate.DateTime).TotalSeconds;
         long repeatLong = RepeatInterval;
 
-        DateTimeOffset sTime = fireTimeStartDate.ToUniversalTime();
+        DateTime fireTimeLocal = fireTimeStartDate.DateTime;
         IntervalUnit repeatUnit = RepeatIntervalUnit;
         long jumpCount = 0;
         if (repeatUnit == IntervalUnit.Second)
@@ -749,8 +748,8 @@ public override void SetPreviousFireTimeUtc(DateTimeOffset? previousFireTimeUtc)
                 jumpCount++;
             }
 
-            sTime = sTime.AddSeconds(RepeatInterval * (int) jumpCount);
-            fireTime = TimeZoneUtil.ConvertTime(sTime, TimeZone);
+            fireTimeLocal = fireTimeLocal.AddSeconds(RepeatInterval * (int) jumpCount);
+            fireTime = new DateTimeOffset(fireTimeLocal, TimeZoneUtil.GetUtcOffset(fireTimeLocal, TimeZone));
         }
         else if (repeatUnit == IntervalUnit.Minute)
         {
@@ -759,8 +758,8 @@ public override void SetPreviousFireTimeUtc(DateTimeOffset? previousFireTimeUtc)
             {
                 jumpCount++;
             }
-            sTime = sTime.AddMinutes(RepeatInterval * (int) jumpCount);
-            fireTime = TimeZoneUtil.ConvertTime(sTime, TimeZone);
+            fireTimeLocal = fireTimeLocal.AddMinutes(RepeatInterval * (int) jumpCount);
+            fireTime = new DateTimeOffset(fireTimeLocal, TimeZoneUtil.GetUtcOffset(fireTimeLocal, TimeZone));
         }
         else if (repeatUnit == IntervalUnit.Hour)
         {
@@ -769,8 +768,8 @@ public override void SetPreviousFireTimeUtc(DateTimeOffset? previousFireTimeUtc)
             {
                 jumpCount++;
             }
-            sTime = sTime.AddHours(RepeatInterval * (int) jumpCount);
-            fireTime = TimeZoneUtil.ConvertTime(sTime, TimeZone);
+            fireTimeLocal = fireTimeLocal.AddHours(RepeatInterval * (int) jumpCount);
+            fireTime = new DateTimeOffset(fireTimeLocal, TimeZoneUtil.GetUtcOffset(fireTimeLocal, TimeZone));
         }
 
         // g. Check if we've exceeded the per-day repeat count.
