diff --git a/src/MudBlazor/Components/FileUpload/MudFileUpload.razor b/src/MudBlazor/Components/FileUpload/MudFileUpload.razor
index 1358681..8ea9f13 100644
--- a/src/MudBlazor/Components/FileUpload/MudFileUpload.razor
+++ b/src/MudBlazor/Components/FileUpload/MudFileUpload.razor
@@ -54,7 +54,9 @@
         else
         {
             <CascadingValue Name="ParentDisabled" Value="GetDisabledState()">
-                <div class="mud-file-upload-custom-content">
+                <div class="mud-file-upload-custom-content"
+                     @ondragenter="OnDragEnterAsync"
+                     @ondragleave="OnDragResetAsync">
                     @CustomContent(this)
                 </div>
             </CascadingValue>
