diff --git a/caddyconfig/caddyfile/parse.go b/caddyconfig/caddyfile/parse.go
index d04a1ac4..26135008 100644
--- a/caddyconfig/caddyfile/parse.go
+++ b/caddyconfig/caddyfile/parse.go
@@ -382,9 +382,6 @@ func (p *parser) doImport(nesting int) error {
 		for bd.Next() {
 			// see if we can grab a key
 			var currentMappingKey string
-			if bd.Val() == "{" {
-				return p.Err("anonymous blocks are not supported")
-			}
 			currentMappingKey = bd.Val()
 			currentMappingTokens := []Token{}
 			// read all args until end of line / {
