diff --git a/mesa/batchrunner.py b/mesa/batchrunner.py
index bc35d1cf..266692d6 100644
--- a/mesa/batchrunner.py
+++ b/mesa/batchrunner.py
@@ -120,7 +120,11 @@ def _make_model_kwargs(
             all_values = [(param, values)]
         else:
             try:
-                all_values = [(param, value) for value in values]
+                # Check if values is an empty list
+                if hasattr(values, "__len__") and len(values) == 0:
+                    all_values = [(param, values)]
+                else:
+                    all_values = [(param, value) for value in values]
             except TypeError:
                 all_values = [(param, values)]
         parameter_list.append(all_values)
