diff --git a/bundles/org.openhab.binding.avmfritz/src/main/java/org/openhab/binding/avmfritz/internal/dto/HeatingModel.java b/bundles/org.openhab.binding.avmfritz/src/main/java/org/openhab/binding/avmfritz/internal/dto/HeatingModel.java
index 2f5f374781..4be5034121 100644
--- a/bundles/org.openhab.binding.avmfritz/src/main/java/org/openhab/binding/avmfritz/internal/dto/HeatingModel.java
+++ b/bundles/org.openhab.binding.avmfritz/src/main/java/org/openhab/binding/avmfritz/internal/dto/HeatingModel.java
@@ -107,16 +107,16 @@ public class HeatingModel implements BatteryModel {
             return MODE_UNKNOWN;
         } else if (TEMP_FRITZ_ON.compareTo(tsoll) == 0) {
             return MODE_ON;
-        } else if (TEMP_FRITZ_OFF.compareTo(tsoll) == 0) {
-            return MODE_OFF;
         } else if (BigDecimal.ONE.equals(getWindowopenactiv())) {
             return MODE_WINDOW_OPEN;
+        } else if (BigDecimal.ONE.equals(getBoostactive()) || TEMP_FRITZ_MAX.compareTo(tsoll) == 0) {
+            return MODE_BOOST;
+        } else if (TEMP_FRITZ_OFF.compareTo(tsoll) == 0) {
+            return MODE_OFF;
         } else if (komfort != null && komfort.compareTo(tsoll) == 0) {
             return MODE_COMFORT;
         } else if (absenk != null && absenk.compareTo(tsoll) == 0) {
             return MODE_ECO;
-        } else if (BigDecimal.ONE.equals(getBoostactive()) || TEMP_FRITZ_MAX.compareTo(tsoll) == 0) {
-            return MODE_BOOST;
         } else {
             return MODE_ON;
         }
