diff --git a/src/ast/nodes/BinaryExpression.ts b/src/ast/nodes/BinaryExpression.ts
index 5fd97123e..b09a920e7 100644
--- a/src/ast/nodes/BinaryExpression.ts
+++ b/src/ast/nodes/BinaryExpression.ts
@@ -149,7 +149,7 @@ export default class BinaryExpression extends NodeBase implements DeoptimizableE
 		includeChildrenRecursively: IncludeChildren,
 		_options?: InclusionOptions
 	) {
-		this.included = true;
+		if (!this.included) this.includeNode(context);
 		if (typeof this.getRenderedLiteralValue() === 'symbol') {
 			super.include(context, includeChildrenRecursively, _options);
 		}
