diff --git a/parquet-testing b/parquet-testing
index 107b36603..92d45b075 160000
--- a/parquet-testing
+++ b/parquet-testing
@@ -1 +1 @@
-Subproject commit 107b36603e051aee26bd93e04b871034f6c756c0
+Subproject commit 92d45b0752487a4b55fb7f1581c8126ee3e73b0d
diff --git a/testing b/testing
index 0d60ccae4..da559b541 160000
--- a/testing
+++ b/testing
@@ -1 +1 @@
-Subproject commit 0d60ccae40d0e8f2d22c15fafb01c5d4be8c63a6
+Subproject commit da559b5412551c490a97c67f6779fcefd4352aff
