diff --git a/lib/Proj.js b/lib/Proj.js
index ca1575b..53cde99 100644
--- a/lib/Proj.js
+++ b/lib/Proj.js
@@ -66,6 +66,9 @@ function Projection(srsCode, callback) {
 
   var sphere_ = dc_sphere(json.a, json.b, json.rf, json.ellps, json.sphere);
   var ecc = dc_eccentricity(sphere_.a, sphere_.b, sphere_.rf, json.R_A);
+  if (json.datumCode && json.datumCode.toLowerCase().indexOf('not specified') === 0) {
+    json.datumCode = 'none';
+  }
   var nadgrids = getNadgrids(json.nadgrids);
   /** @type {DatumDefinition} */
   var datumObj = json.datum || datum(json.datumCode, json.datum_params, sphere_.a, sphere_.b, ecc.es, ecc.ep2,
diff --git a/lib/projections/sinu.js b/lib/projections/sinu.js
index f869d82..f9037f4 100644
--- a/lib/projections/sinu.js
+++ b/lib/projections/sinu.js
@@ -22,6 +22,9 @@ import asinz from '../common/asinz';
 export function init() {
   /* Place parameters in static storage for common use
     ------------------------------------------------- */
+  this.x0 = this.x0 !== undefined ? this.x0 : 0;
+  this.y0 = this.y0 !== undefined ? this.y0 : 0;
+  this.long0 = this.long0 !== undefined ? this.long0 : this.longc || 0;
 
   if (!this.sphere) {
     this.en = pj_enfn(this.es);
