diff --git a/packages/webcodecs/src/convert-media.ts b/packages/webcodecs/src/convert-media.ts
index 522990ad71..1261555bdf 100644
--- a/packages/webcodecs/src/convert-media.ts
+++ b/packages/webcodecs/src/convert-media.ts
@@ -252,7 +252,7 @@ export const convertMedia = async function <
 		onVideoTrack: userVideoResolver ?? null,
 		logLevel,
 		outputContainer: container,
-		rotate: rotate ?? 0,
+		rotate: rotate ?? null,
 		resizeOperation: resize ?? null,
 	});
 
diff --git a/packages/webcodecs/src/default-on-video-track-handler.ts b/packages/webcodecs/src/default-on-video-track-handler.ts
index 636fff8c4f..fc186f661e 100644
--- a/packages/webcodecs/src/default-on-video-track-handler.ts
+++ b/packages/webcodecs/src/default-on-video-track-handler.ts
@@ -47,7 +47,7 @@ export const defaultOnVideoTrackHandler: ConvertMediaOnVideoTrackHandler =
 			return Promise.resolve({
 				type: 'reencode',
 				videoCodec: defaultVideoCodec,
-				rotate: rotate - track.rotation,
+				rotate,
 				resize: resizeOperation,
 			});
 		}
diff --git a/packages/webcodecs/src/on-video-track.ts b/packages/webcodecs/src/on-video-track.ts
index c1a626fa26..24cb21d119 100644
--- a/packages/webcodecs/src/on-video-track.ts
+++ b/packages/webcodecs/src/on-video-track.ts
@@ -41,7 +41,7 @@ export const makeVideoTrackHandler =
 		onVideoTrack: ConvertMediaOnVideoTrackHandler | null;
 		logLevel: MediaParserLogLevel;
 		outputContainer: ConvertMediaContainer;
-		rotate: number;
+		rotate: number | null;
 		resizeOperation: ResizeOperation | null;
 		progressTracker: ProgressTracker;
 	}): MediaParserOnVideoTrack =>
@@ -55,19 +55,21 @@ export const makeVideoTrackHandler =
 		const canCopyTrack = canCopyVideoTrack({
 			inputContainer,
 			outputContainer,
-			rotationToApply: rotate,
+			rotationToApply: rotate ?? track.rotation,
 			inputTrack: track,
 			resizeOperation,
 			outputVideoCodec: defaultVideoCodec,
 		});
 
+		const rotateToApply = rotate ?? track.rotation;
+
 		const videoOperation = await (onVideoTrack ?? defaultOnVideoTrackHandler)({
 			track,
 			defaultVideoCodec:
 				defaultVideoCodec ?? getDefaultVideoCodec({container: outputContainer}),
 			logLevel,
 			outputContainer,
-			rotate,
+			rotate: rotateToApply,
 			inputContainer,
 			canCopyTrack,
 			resizeOperation,
@@ -98,7 +100,7 @@ export const makeVideoTrackHandler =
 			abortConversion,
 			controller,
 			logLevel,
-			rotate,
+			rotate: rotateToApply,
 			track,
 			onVideoFrame,
 			state,
diff --git a/packages/webcodecs/src/reencode-video-track.ts b/packages/webcodecs/src/reencode-video-track.ts
index 16ff9fd0de..fb7a547934 100644
--- a/packages/webcodecs/src/reencode-video-track.ts
+++ b/packages/webcodecs/src/reencode-video-track.ts
@@ -34,7 +34,7 @@ export const reencodeVideoTrack = async ({
 	progressTracker,
 }: {
 	videoOperation: VideoOperation;
-	rotate: number;
+	rotate: number | null;
 	track: MediaParserVideoTrack;
 	logLevel: MediaParserLogLevel;
 	abortConversion: (errCause: Error) => void;
@@ -52,7 +52,8 @@ export const reencodeVideoTrack = async ({
 		);
 	}
 
-	const rotation = (videoOperation.rotate ?? rotate) - track.rotation;
+	const rotationToApply = videoOperation.rotate ?? rotate ?? track.rotation;
+	const rotation = rotationToApply - track.rotation;
 
 	const {height: newHeight, width: newWidth} =
 		calculateNewDimensionsFromRotateAndScale({
