diff --git a/Mage.Sets/src/mage/cards/a/AyaraWidowOfTheRealm.java b/Mage.Sets/src/mage/cards/a/AyaraWidowOfTheRealm.java
index 84ca4012..a363e7f6 100644
--- a/Mage.Sets/src/mage/cards/a/AyaraWidowOfTheRealm.java
+++ b/Mage.Sets/src/mage/cards/a/AyaraWidowOfTheRealm.java
@@ -112,10 +112,14 @@ class AyaraFurnaceQueenEffect extends OneShotEffect {
     @Override
     public boolean apply(Game game, Ability source) {
         Player player = game.getPlayer(source.getControllerId());
-        Card card = game.getCard(getTargetPointer().getFirst(game, source));
+        Card card = game.getCard(source.getFirstTarget());
         if (player == null || card == null) {
             return false;
         }
+        boolean result = player.moveCards(card, Zone.BATTLEFIELD, source, game);
+        if (!result) {
+            return false;
+        }
         Permanent permanent = CardUtil.getPermanentFromCardPutToBattlefield(card, game);
         if (permanent == null) {
             return false;
