diff --git a/packages/x-date-pickers/src/internals/hooks/useField/useField.utils.ts b/packages/x-date-pickers/src/internals/hooks/useField/useField.utils.ts
index 28b512b..5fa1bba 100644
--- a/packages/x-date-pickers/src/internals/hooks/useField/useField.utils.ts
+++ b/packages/x-date-pickers/src/internals/hooks/useField/useField.utils.ts
@@ -519,11 +519,11 @@ const transferDateSectionValue = (
       const mergedDateHours = adapter.getHours(dateToTransferTo);
 
       if (isAM && mergedDateHours >= 12) {
-        return adapter.addHours(dateToTransferTo, -12);
+        return adapter.setHours(dateToTransferTo, mergedDateHours - 12);
       }
 
       if (!isAM && mergedDateHours < 12) {
-        return adapter.addHours(dateToTransferTo, 12);
+        return adapter.setHours(dateToTransferTo, mergedDateHours + 12);
       }
 
       return dateToTransferTo;
