diff --git a/compat/src/Children.js b/compat/src/Children.js
index 0295d936..2df5b9d1 100644
--- a/compat/src/Children.js
+++ b/compat/src/Children.js
@@ -1,8 +1,10 @@
 import { toChildArray } from 'preact';
 
-const mapFn = (children, fn) => {
+const mapFn = (children, fn, context) => {
 	if (children == null) return null;
-	return toChildArray(toChildArray(children).map(fn));
+	return toChildArray(toChildArray(children).map(function(child, index) {
+		return context !== undefined ? fn.call(context, child, index) : fn(child, index);
+	}));
 };
 
 // This API is completely unnecessary for Preact, so it's basically passthrough.
