diff --git a/include/ylt/coro_rpc/impl/coro_rpc_client.hpp b/include/ylt/coro_rpc/impl/coro_rpc_client.hpp
index df53047..e51a84c 100644
--- a/include/ylt/coro_rpc/impl/coro_rpc_client.hpp
+++ b/include/ylt/coro_rpc/impl/coro_rpc_client.hpp
@@ -327,12 +327,8 @@ class coro_rpc_client {
       // do nothing, someone has reconnect the client
     }
 
-    if (config_.host.empty()) {
-      config_.host = std::move(host);
-    }
-    if (config_.port.empty()) {
-      config_.port = std::move(port);
-    }
+    config_.host = std::move(host);
+    config_.port = std::move(port);
     auto ret = co_await control_->socket_wrapper_.visit([&,
                                                          this](auto &socket) {
       return connect_impl(socket,
