diff --git a/Terminal.Gui/ViewBase/Layout/PosAlign.cs b/Terminal.Gui/ViewBase/Layout/PosAlign.cs
index 9e04f1b..f22b317 100644
--- a/Terminal.Gui/ViewBase/Layout/PosAlign.cs
+++ b/Terminal.Gui/ViewBase/Layout/PosAlign.cs
@@ -71,7 +71,9 @@ public record PosAlign : Pos
 
             if (posAlign is { })
             {
-                dimensionsSum += dimension == Dimension.Width ? view.Frame.Width : view.Frame.Height;
+                dimensionsSum += dimension == Dimension.Width
+                                     ? view.Width.Calculate (0, dimensionsSum, view, dimension)
+                                     : view.Height.Calculate (0, dimensionsSum, view, dimension);
             }
         }
 
