diff --git a/execution/rerun/rerun.go b/execution/rerun/rerun.go
index 9ac1725..3fe0942 100644
--- a/execution/rerun/rerun.go
+++ b/execution/rerun/rerun.go
@@ -97,10 +97,16 @@ func ListenFailedScenarios(wg *sync.WaitGroup, specDirs []string) {
 }
 
 func prepareScenarioFailedMetadata(res *result.ScenarioResult, sce *gauge.Scenario, executionInfo *gauge_messages.ExecutionInfo) {
+	specPath := executionInfo.GetCurrentSpec().GetFileName()
+	failedScenario := util.RelPathToProjectRoot(specPath)
+	failedKey := fmt.Sprintf("%s:%v", failedScenario, sce.Span.Start)
 	if res.GetFailed() {
-		specPath := executionInfo.GetCurrentSpec().GetFileName()
-		failedScenario := util.RelPathToProjectRoot(specPath)
-		failedMeta.addFailedItem(specPath, fmt.Sprintf("%s:%v", failedScenario, sce.Span.Start))
+		failedMeta.addFailedItem(specPath, failedKey)
+	} else {
+		delete(failedMeta.failedItemsMap[specPath], failedKey)
+		if len(failedMeta.failedItemsMap[specPath]) == 0 {
+			delete(failedMeta.failedItemsMap, specPath)
+		}
 	}
 }
