diff --git a/iwyu.cc b/iwyu.cc
index ab2eef7..d3083d1 100644
--- a/iwyu.cc
+++ b/iwyu.cc
@@ -1355,7 +1355,7 @@ class IwyuBaseAstVisitor : public BaseAstVisitor<Derived> {
     if (!fwd_decl) {
       if (const auto* func_decl = dyn_cast<FunctionDecl>(decl)) {
         if (const FunctionTemplateDecl* ft_decl =
-                func_decl->getPrimaryTemplate()) {
+                func_decl->getDescribedFunctionTemplate()) {
           VERRS(5) << "No fwd-decl found, looking for function template decl\n";
           for (const NamedDecl* redecl : ft_decl->redecls()) {
             if (GetFileEntry(redecl) == macro_def_file) {
@@ -1740,6 +1740,15 @@ class IwyuBaseAstVisitor : public BaseAstVisitor<Derived> {
     if (CanIgnoreCurrentASTNode())
       return true;
 
+    if (decl->getTemplateSpecializationKind() == clang::TSK_ExplicitSpecialization) {
+      // Explicit function template specializations require a declaration of
+      // the primary template. This is true even for deleted specializations,
+      // which Clang doesn't consider definitions.
+      if (FunctionTemplateDecl* tpl_decl =
+              decl->getTemplateSpecializationInfo()->getTemplate())
+        ReportDeclUse(CurrentLoc(), tpl_decl, nullptr, UF_RedeclUse);
+    }
+
     if (decl->isThisDeclarationADefinition()) {
       // For non-member function definitions, report use of all previously seen
       // decls.
