diff --git a/packages/webcodecs/src/default-on-video-track-handler.ts b/packages/webcodecs/src/default-on-video-track-handler.ts
index 636fff8c4f..0a2187e2e8 100644
--- a/packages/webcodecs/src/default-on-video-track-handler.ts
+++ b/packages/webcodecs/src/default-on-video-track-handler.ts
@@ -47,7 +47,7 @@ export const defaultOnVideoTrackHandler: ConvertMediaOnVideoTrackHandler =
 			return Promise.resolve({
 				type: 'reencode',
 				videoCodec: defaultVideoCodec,
-				rotate: rotate - track.rotation,
+				rotate: 2 * track.rotation + rotate,
 				resize: resizeOperation,
 			});
 		}
