diff --git a/src/main/java/com/github/tomakehurst/wiremock/standalone/CommandLineOptions.java b/src/main/java/com/github/tomakehurst/wiremock/standalone/CommandLineOptions.java
index 24467b787..07336370e 100644
--- a/src/main/java/com/github/tomakehurst/wiremock/standalone/CommandLineOptions.java
+++ b/src/main/java/com/github/tomakehurst/wiremock/standalone/CommandLineOptions.java
@@ -160,7 +160,7 @@ public class CommandLineOptions implements Options {
         .withRequiredArg();
     optionParser.accepts(BIND_ADDRESS, "The IP to listen connections").withRequiredArg();
     optionParser.accepts(CONTAINER_THREADS, "The number of container threads").withRequiredArg();
-    optionParser.accepts(TIMEOUT, "The default global timeout.");
+    optionParser.accepts(TIMEOUT, "The default global timeout.").withRequiredArg();
     optionParser.accepts(
         DISABLE_OPTIMIZE_XML_FACTORIES_LOADING,
         "Whether to disable optimize XML factories loading or not.");
diff --git a/src/test/java/com/github/tomakehurst/wiremock/standalone/CommandLineOptionsTest.java b/src/test/java/com/github/tomakehurst/wiremock/standalone/CommandLineOptionsTest.java
index 598c429f5..4bdf5c380 100644
--- a/src/test/java/com/github/tomakehurst/wiremock/standalone/CommandLineOptionsTest.java
+++ b/src/test/java/com/github/tomakehurst/wiremock/standalone/CommandLineOptionsTest.java
@@ -857,6 +857,7 @@ public class CommandLineOptionsTest {
     assertThat(proxyTimeout, is(Options.DEFAULT_TIMEOUT));
   }
 
+
   @Test
   void testProxyPassThroughOptionPassedAsFalse() {
     CommandLineOptions options = new CommandLineOptions("--proxy-pass-through", "false");
