diff --git a/include/chaiscript/dispatchkit/bootstrap_stl.hpp b/include/chaiscript/dispatchkit/bootstrap_stl.hpp
index f56304d..c36134e 100644
--- a/include/chaiscript/dispatchkit/bootstrap_stl.hpp
+++ b/include/chaiscript/dispatchkit/bootstrap_stl.hpp
@@ -18,6 +18,7 @@
 #include <functional>
 #include <memory>
 #include <stdexcept>
+#include <type_traits>
 #include <typeinfo>
 #include <vector>
 
@@ -443,14 +444,20 @@ namespace chaiscript::bootstrap::standard_library {
           }),
           "front");
 
-    back_insertion_sequence_type<VectorType>(type, m);
-    sequence_type<VectorType>(type, m);
+    if constexpr (std::is_copy_constructible_v<typename VectorType::value_type>) {
+      back_insertion_sequence_type<VectorType>(type, m);
+      sequence_type<VectorType>(type, m);
+    }
     random_access_container_type<VectorType>(type, m);
-    resizable_type<VectorType>(type, m);
+    if constexpr (std::is_copy_constructible_v<typename VectorType::value_type>) {
+      resizable_type<VectorType>(type, m);
+    }
     reservable_type<VectorType>(type, m);
     container_type<VectorType>(type, m);
     default_constructible_type<VectorType>(type, m);
-    assignable_type<VectorType>(type, m);
+    if constexpr (std::is_copy_constructible_v<typename VectorType::value_type>) {
+      assignable_type<VectorType>(type, m);
+    }
     input_range_type<VectorType>(type, m);
 
     if (typeid(VectorType) == typeid(std::vector<Boxed_Value>)) {
