diff --git a/packages/svelte/src/compiler/phases/2-analyze/visitors/Identifier.js b/packages/svelte/src/compiler/phases/2-analyze/visitors/Identifier.js
index cced326f9..0984280aa 100644
--- a/packages/svelte/src/compiler/phases/2-analyze/visitors/Identifier.js
+++ b/packages/svelte/src/compiler/phases/2-analyze/visitors/Identifier.js
@@ -93,7 +93,9 @@ export function Identifier(node, context) {
 			context.state.expression.references.add(binding);
 			context.state.expression.has_state ||=
 				binding.kind !== 'static' &&
-				!binding.is_function() &&
+				// Props should always be treated as having state, even if they have a function default value,
+				// because the parent component might update them
+				((binding.kind === 'prop' || binding.kind === 'bindable_prop') || !binding.is_function()) &&
 				!context.state.scope.evaluate(node).is_known;
 		}
