diff --git a/Zend/zend_compile.c b/Zend/zend_compile.c
index db611be49ab..652dcb7dca7 100644
--- a/Zend/zend_compile.c
+++ b/Zend/zend_compile.c
@@ -2726,7 +2726,8 @@ static inline bool zend_is_call(zend_ast *ast) /* {{{ */
 	return ast->kind == ZEND_AST_CALL
 		|| ast->kind == ZEND_AST_METHOD_CALL
 		|| ast->kind == ZEND_AST_NULLSAFE_METHOD_CALL
-		|| ast->kind == ZEND_AST_STATIC_CALL;
+		|| ast->kind == ZEND_AST_STATIC_CALL
+		|| ast->kind == ZEND_AST_PIPE;
 }
 /* }}} */
 
@@ -6468,15 +6469,14 @@ static void zend_compile_pipe(znode *result, zend_ast *ast)
 	znode operand_result;
 	zend_compile_expr(&operand_result, operand_ast);
 
-	/* Wrap simple values in a ZEND_QM_ASSIGN opcode to ensure references
-	 * always fail. They will already fail in complex cases like arrays,
-	 * so those don't need a wrapper. */
+	/* Wrap the operand in a ZEND_QM_ASSIGN opcode to prevent references to it.
+	 * This ensures that piping into a callable returning by-ref behaves like:
+	 *   $c = callable(...); $c($operand);
+	 * In particular, this must prevent the by-ref return from complaining if the
+	 * piped call is itself returned by reference.
+	 */
 	znode wrapped_operand_result;
-	if (operand_result.op_type & (IS_CV|IS_VAR)) {
-		zend_emit_op_tmp(&wrapped_operand_result, ZEND_QM_ASSIGN, &operand_result, NULL);
-	} else {
-		wrapped_operand_result = operand_result;
-	}
+	zend_emit_op_tmp(&wrapped_operand_result, ZEND_QM_ASSIGN, &operand_result, NULL);
 
 	/* Turn the operand into a function parameter list. */
 	zend_ast *arg_list_ast = zend_ast_create_list(1, ZEND_AST_ARG_LIST, zend_ast_create_znode(&wrapped_operand_result));
