diff --git a/internal/cmd/config.go b/internal/cmd/config.go
index 0e4b216..9436e1c 100644
--- a/internal/cmd/config.go
+++ b/internal/cmd/config.go
@@ -569,6 +569,7 @@ func newConfig(options ...configOption) (*Config, error) {
 		"splitList":                   c.splitListTemplateFunc,
 		"squote":                      c.squoteTemplateFunc,
 		"stat":                        c.statTemplateFunc,
+		"stdinIsATTY":                 c.stdinIsATTYInitTemplateFunc,
 		"toIni":                       c.toIniTemplateFunc,
 		"toPrettyJson":                c.toPrettyJsonTemplateFunc,
 		"toString":                    c.toStringTemplateFunc,
