diff --git a/smithy-model/src/main/java/software/amazon/smithy/model/shapes/SmithyIdlModelSerializer.java b/smithy-model/src/main/java/software/amazon/smithy/model/shapes/SmithyIdlModelSerializer.java
index 6773c71..7608894 100644
--- a/smithy-model/src/main/java/software/amazon/smithy/model/shapes/SmithyIdlModelSerializer.java
+++ b/smithy-model/src/main/java/software/amazon/smithy/model/shapes/SmithyIdlModelSerializer.java
@@ -987,7 +987,12 @@ public final class SmithyIdlModelSerializer {
         }
 
         private void serializeShapeId(StringNode node) {
-            codeWriter.writeInline("$I", node.getValue());
+            ShapeId shapeId = ShapeId.from(node.getValue());
+            if (model.getShape(shapeId.withoutMember()).isPresent()) {
+                codeWriter.writeInline("$I", shapeId);
+            } else {
+                serializeString(node);
+            }
         }
 
         private void serializeNumber(NumberNode node) {
