diff --git a/packages/solid-router/src/Matches.tsx b/packages/solid-router/src/Matches.tsx
index cf83c6b50..5df5cee55 100644
--- a/packages/solid-router/src/Matches.tsx
+++ b/packages/solid-router/src/Matches.tsx
@@ -7,6 +7,7 @@ import { Transitioner } from './Transitioner'
 import { matchContext } from './matchContext'
 import { Match } from './Match'
 import { SafeFragment } from './SafeFragment'
+import { HeadContent } from './HeadContent'
 import type {
   AnyRouter,
   DeepPartial,
@@ -50,6 +51,7 @@ export function Matches() {
 
   const inner = (
     <ResolvedSuspense fallback={pendingElement}>
+      <HeadContent />
       {!router.isServer && <Transitioner />}
       <MatchesInner />
     </ResolvedSuspense>
diff --git a/packages/solid-start-client/src/StartClient.tsx b/packages/solid-start-client/src/StartClient.tsx
index 195632021..244c3e540 100644
--- a/packages/solid-start-client/src/StartClient.tsx
+++ b/packages/solid-start-client/src/StartClient.tsx
@@ -21,18 +21,7 @@ export function StartClient(props: { router: AnyRouter }) {
       children={() => (
         <Dummy>
           <Dummy>
-            <RouterProvider
-              router={props.router}
-              InnerWrap={(props) => (
-                <Dummy>
-                  <Dummy>
-                    <HeadContent />
-                    {props.children}
-                  </Dummy>
-                  <Dummy />
-                </Dummy>
-              )}
-            />
+            <RouterProvider router={props.router} />
           </Dummy>
         </Dummy>
       )}
