diff --git a/internal/chezmoi/sourcestate.go b/internal/chezmoi/sourcestate.go
index 2087c7f..306e84d 100644
--- a/internal/chezmoi/sourcestate.go
+++ b/internal/chezmoi/sourcestate.go
@@ -334,7 +334,7 @@ func NewSourceState(options ...SourceStateOption) *SourceState {
 }
 
 // A PreAddFunc is called before a new source state entry is added.
-type PreAddFunc func(targetRelPath RelPath, fileInfo fs.FileInfo) error
+type PreAddFunc func(targetRelPath RelPath, fileInfo fs.FileInfo, options *AddOptions) error
 
 // A ReplaceFunc is called before a source state entry is replaced.
 type ReplaceFunc func(targetRelPath RelPath, newSourceStateEntry, oldSourceStateEntry SourceStateEntry) error
@@ -541,7 +541,7 @@ DEST_ABS_PATH:
 		}
 
 		if options.PreAddFunc != nil && destAbsPathInfo != nil {
-			switch err := options.PreAddFunc(targetRelPath, destAbsPathInfo); {
+			switch err := options.PreAddFunc(targetRelPath, destAbsPathInfo, options); {
 			case errors.Is(err, fs.SkipDir):
 				continue DEST_ABS_PATH
 			case err != nil:
diff --git a/internal/cmd/addcmd.go b/internal/cmd/addcmd.go
index 8ca1cbc..75db9c0 100644
--- a/internal/cmd/addcmd.go
+++ b/internal/cmd/addcmd.go
@@ -87,9 +87,9 @@ func (c *Config) defaultOnIgnoreFunc(targetRelPath chezmoi.RelPath) {
 	}
 }
 
-func (c *Config) defaultPreAddFunc(targetRelPath chezmoi.RelPath, fileInfo fs.FileInfo) error {
+func (c *Config) defaultPreAddFunc(targetRelPath chezmoi.RelPath, fileInfo fs.FileInfo, options *chezmoi.AddOptions) error {
 	// Scan unencrypted files for secrets, if configured.
-	if c.Add.Secrets.String() != severityIgnore && fileInfo.Mode().Type() == 0 && !c.Add.Encrypt {
+	if c.Add.Secrets.String() != severityIgnore && fileInfo.Mode().Type() == 0 && !options.Encrypt {
 		absPath := c.DestDirAbsPath.Join(targetRelPath)
 		content, err := c.destSystem.ReadFile(absPath)
 		if err != nil {
