diff --git a/p2p/transport/quicreuse/connmgr.go b/p2p/transport/quicreuse/connmgr.go
index dedca748..a61a59b9 100644
--- a/p2p/transport/quicreuse/connmgr.go
+++ b/p2p/transport/quicreuse/connmgr.go
@@ -247,7 +247,7 @@ func (c *ConnManager) ListenQUICAndAssociate(association any, addr ma.Multiaddr,
 			tr.associate(association)
 		}
 	}
-	l, err := entry.ln.Add(tlsConf, allowWindowIncrease, func() { c.onListenerClosed(key) })
+	l, err := entry.ln.Add(tlsConf, allowWindowIncrease, func() { c.onListenerClosed(key, association) })
 	if err != nil {
 		if entry.refCount <= 0 {
 			entry.ln.Close()
@@ -259,12 +259,17 @@ func (c *ConnManager) ListenQUICAndAssociate(association any, addr ma.Multiaddr,
 	return l, nil
 }
 
-func (c *ConnManager) onListenerClosed(key string) {
+func (c *ConnManager) onListenerClosed(key string, association any) {
 	c.quicListenersMu.Lock()
 	defer c.quicListenersMu.Unlock()
 
 	entry := c.quicListeners[key]
 	entry.refCount = entry.refCount - 1
+	if association != nil {
+		if tr, ok := entry.ln.transport.(*refcountedTransport); ok {
+			tr.removeAssociation(association)
+		}
+	}
 	if entry.refCount <= 0 {
 		delete(c.quicListeners, key)
 		entry.ln.Close()
diff --git a/p2p/transport/quicreuse/reuse.go b/p2p/transport/quicreuse/reuse.go
index 1593e226..0a0de281 100644
--- a/p2p/transport/quicreuse/reuse.go
+++ b/p2p/transport/quicreuse/reuse.go
@@ -109,6 +109,21 @@ func (c *refcountedTransport) associate(a any) {
 	c.assocations[a] = struct{}{}
 }
 
+func (c *refcountedTransport) removeAssociation(a any) {
+	if a == nil {
+		return
+	}
+	c.mutex.Lock()
+	defer c.mutex.Unlock()
+	if c.assocations == nil {
+		return
+	}
+	delete(c.assocations, a)
+	if len(c.assocations) == 0 {
+		c.assocations = nil
+	}
+}
+
 // hasAssociation returns true if the transport has the given association.
 // If it is a nil association, it will always return true.
 func (c *refcountedTransport) hasAssociation(a any) bool {
