diff --git a/shapely/geometry/collection.py b/shapely/geometry/collection.py
index cb7cc6e..0654680 100644
--- a/shapely/geometry/collection.py
+++ b/shapely/geometry/collection.py
@@ -33,7 +33,7 @@ class GeometryCollection(BaseMultipartGeometry):
 
     def __new__(self, geoms=None):
         """Create a new GeometryCollection."""
-        if not geoms:
+        if geoms is None or (hasattr(geoms, "__len__") and len(geoms) == 0):
             # TODO better empty constructor
             return shapely.from_wkt("GEOMETRYCOLLECTION EMPTY")
         if isinstance(geoms, BaseGeometry):
