diff --git a/src/cc65/coptstop.c b/src/cc65/coptstop.c
index 7a9416b51..924f097f0 100644
--- a/src/cc65/coptstop.c
+++ b/src/cc65/coptstop.c
@@ -1776,7 +1776,8 @@ static int RegAPreCondOk (StackOpData* D)
                     /* Cannot optimize */
                     break;
                 } else if ((Desc->Flags & OP_RHS_LOAD_DIRECT) != 0) {
-                    if ((RhsLo->Flags & LI_DIRECT) == 0) {
+                    if ((RhsLo->Flags & LI_DIRECT) == 0 ||
+                        ((Desc->Flags & OP_RHS_REMOVE_DIRECT) != 0 && (Rhs->X.Flags & LI_DIRECT) == 0)) {
                         /* Cannot optimize */
                         break;
                     }
