diff --git a/src/useTranslation.tsx b/src/useTranslation.tsx
index 33f98ac..89341c7 100644
--- a/src/useTranslation.tsx
+++ b/src/useTranslation.tsx
@@ -4,8 +4,8 @@ import wrapTWithDefaultNs from './wrapTWithDefaultNs'
 import I18nContext from './context'
 import createTranslation from './createTranslation'
 
-function useTranslationInPages(defaultNS?: string): I18n {
-  const ctx = useContext(I18nContext)
+function useTranslationInPages(defaultNS?: string, context?: I18n): I18n {
+  const ctx = context || useContext(I18nContext)
   return useMemo(
     () => ({
       ...ctx,
@@ -17,6 +17,12 @@ function useTranslationInPages(defaultNS?: string): I18n {
 
 export default function useTranslation(defaultNS?: string): I18n {
   const appDir = globalThis.__NEXT_TRANSLATE__
+
+  if (appDir?.config && I18nContext) {
+    const ctx = useContext(I18nContext)
+    if (ctx.lang) return useTranslationInPages(defaultNS, ctx)
+  }
+
   const useT = appDir?.config ? createTranslation : useTranslationInPages
   return useT(defaultNS)
 }
