diff --git a/src/MudBlazor/Components/Select/MudSelect.razor.cs b/src/MudBlazor/Components/Select/MudSelect.razor.cs
index f84998b..a48617c 100644
--- a/src/MudBlazor/Components/Select/MudSelect.razor.cs
+++ b/src/MudBlazor/Components/Select/MudSelect.razor.cs
@@ -703,17 +703,6 @@ namespace MudBlazor
             UpdateIcon();
             StateHasChanged();
 
-            //Scroll the active item on each opening
-            if (_activeItemId != null)
-            {
-                var index = Items.FindIndex(x => x.ItemId == _activeItemId);
-                if (index > 0)
-                {
-                    var item = Items[index];
-                    await ScrollToItemAsync(item);
-                }
-            }
-
             //disable escape propagation: if selectmenu is open, only the select popover should close and underlying components should not handle escape key
             await KeyInterceptorService.UpdateKeyAsync(ElementId, new("Escape", stopDown: "key+none"));
         }
@@ -915,6 +904,20 @@ namespace MudBlazor
             return InvokeAsync(StateHasChanged);
         }
 
+        private async ValueTask ScrollToActiveItemAsync()
+        {
+            if (_activeItemId is null)
+            {
+                return;
+            }
+
+            var index = Items.FindIndex(x => x.ItemId == _activeItemId);
+            if (index > 0)
+            {
+                await ScrollToItemAsync(Items[index]);
+            }
+        }
+
         private void UpdateSelectAllChecked()
         {
             if (MultiSelection && SelectAll)
@@ -1453,6 +1456,8 @@ namespace MudBlazor
                     {
                         await HighlightItemForValueAsync(ReadValue);
                     }
+
+                    await ScrollToActiveItemAsync();
                 });
             }
         }
