diff --git a/packages/core/src/utils/EntityComparator.ts b/packages/core/src/utils/EntityComparator.ts
index 7b81203..f3a8686 100644
--- a/packages/core/src/utils/EntityComparator.ts
+++ b/packages/core/src/utils/EntityComparator.ts
@@ -646,6 +646,19 @@ export class EntityComparator {
           .map(childProp => {
             const childDataKey =
               meta.embeddable || prop.object ? dataKey + this.wrap(childProp.embedded![1]) : this.wrap(childProp.name);
+
+            if (childProp.kind === ReferenceKind.EMBEDDED && !childProp.object && !childProp.array) {
+              return meta.props
+                .filter(
+                  p =>
+                    p.embedded?.[0] === childProp.name &&
+                    // object for JSON embeddable
+                    (p.object || p.persist !== false),
+                )
+                .map(nestedProp => `${padding}  ret${this.wrap(nestedProp.name)} = null;`)
+                .join('\n');
+            }
+
             return `${padding}  ret${childDataKey} = null;`;
           })
           .join('\n') + `\n`;
