diff --git a/packages/wxt/src/core/utils/env.ts b/packages/wxt/src/core/utils/env.ts
index 38871ca..f10c34d 100644
--- a/packages/wxt/src/core/utils/env.ts
+++ b/packages/wxt/src/core/utils/env.ts
@@ -32,14 +32,7 @@ export function loadEnv(mode: string, browser: TargetBrowser) {
     }),
   );
 
-  // Make a copy of `process.env` so that `dotenv-expand` doesn't re-assign the
-  // expanded values to the global `process.env`.
-  const processEnv = { ...process.env } as Record<string, string>;
-
-  expand({
-    parsed,
-    processEnv,
-  });
+  expand({ parsed });
 
   return parsed;
 }
