diff --git a/docs/checks.md b/docs/checks.md
index 6d77d5df..18855428 100644
--- a/docs/checks.md
+++ b/docs/checks.md
@@ -628,7 +628,7 @@ Signed releases attest to the provenance of the artifact.
 This check looks for the following filenames in the project's last five
 [release assets](https://docs.github.com/en/repositories/releasing-projects-on-github/about-releases):
 [*.minisig](https://github.com/jedisct1/minisign), *.asc (pgp),
-*.sig, *.sign, *.sigstore, [*.intoto.jsonl](https://slsa.dev).
+*.sig, *.sign, *.sigstore, *.sigstore.json, [*.intoto.jsonl](https://slsa.dev).
 
 If a signature is found in the assets for each release, a score of 8 is given.
 If a [SLSA provenance file](https://slsa.dev/spec/v0.1/index) is found in the assets for each release (*.intoto.jsonl), the maximum score of 10 is given.
diff --git a/docs/checks/internal/checks.yaml b/docs/checks/internal/checks.yaml
index e770895a..fb81ee37 100644
--- a/docs/checks/internal/checks.yaml
+++ b/docs/checks/internal/checks.yaml
@@ -667,7 +667,7 @@ checks:
       This check looks for the following filenames in the project's last five
       [release assets](https://docs.github.com/en/repositories/releasing-projects-on-github/about-releases):
       [*.minisig](https://github.com/jedisct1/minisign), *.asc (pgp),
-      *.sig, *.sign, *.sigstore, [*.intoto.jsonl](https://slsa.dev).
+      *.sig, *.sign, *.sigstore, *.sigstore.json, [*.intoto.jsonl](https://slsa.dev).
 
       If a signature is found in the assets for each release, a score of 8 is given.
       If a [SLSA provenance file](https://slsa.dev/spec/v0.1/index) is found in the assets for each release (*.intoto.jsonl), the maximum score of 10 is given.
diff --git a/probes/releasesAreSigned/impl.go b/probes/releasesAreSigned/impl.go
index 66efdbae..ddb8ce34 100644
--- a/probes/releasesAreSigned/impl.go
+++ b/probes/releasesAreSigned/impl.go
@@ -41,7 +41,7 @@ const (
 	releaseLookBack = 5
 )
 
-var signatureExtensions = []string{".asc", ".minisig", ".sig", ".sign", ".sigstore"}
+var signatureExtensions = []string{".asc", ".minisig", ".sig", ".sign", ".sigstore.json", ".sigstore"}
 
 func Run(raw *checker.RawResults) ([]finding.Finding, string, error) {
 	if raw == nil {
