diff --git a/IPython/core/inputtransformer2.py b/IPython/core/inputtransformer2.py
index e2157d636..c30959198 100644
--- a/IPython/core/inputtransformer2.py
+++ b/IPython/core/inputtransformer2.py
@@ -44,7 +44,7 @@ def leading_indent(lines):
     if not lines:
         return lines
     m = _indent_re.match(lines[0])
-    if not m or lines[0].strip().startswith("#"):
+    if not m:
         return lines
     space = m.group(0)
     n = len(space)
