diff --git a/mcp/event.go b/mcp/event.go
index 62dd2ad..c9ed9c0 100644
--- a/mcp/event.go
+++ b/mcp/event.go
@@ -374,12 +374,10 @@ func (s *MemoryEventStore) purge() {
 		changed := false
 		for _, sm := range s.store {
 			for _, dl := range sm {
-				if dl.size > 0 {
+				if len(dl.data) > 0 {
 					r := dl.removeFirst()
-					if r > 0 {
-						changed = true
-						s.nBytes -= r
-					}
+					changed = true
+					s.nBytes -= r
 				}
 			}
 		}
