diff --git a/ooxml/XSSF/UserModel/XSSFRow.cs b/ooxml/XSSF/UserModel/XSSFRow.cs
index c196e1b..20a9064 100644
--- a/ooxml/XSSF/UserModel/XSSFRow.cs
+++ b/ooxml/XSSF/UserModel/XSSFRow.cs
@@ -381,6 +381,14 @@ public ICell CreateCell(int columnIndex, CellType type)
             {
                 xcell.CellStyle = RowStyle;
             }
+            else
+            {
+                ICellStyle columnStyle = _sheet.GetColumnStyle(columnIndex);
+                if (columnStyle != null && columnStyle.Index != 0)
+                {
+                    xcell.CellStyle = columnStyle;
+                }
+            }
 
             _cells[columnIndex] = xcell;
             UpdateCacheOnAdd(columnIndex);
