diff --git a/src/cluster.c b/src/cluster.c
index dd83a3324..536a423b7 100644
--- a/src/cluster.c
+++ b/src/cluster.c
@@ -1593,6 +1593,8 @@ void readwriteCommand(client *c) {
 void resetClusterStats(void) {
     if (!server.cluster_enabled) return;
 
+    memset(server.cluster->stats_bus_messages_sent, 0, sizeof(server.cluster->stats_bus_messages_sent));
+    memset(server.cluster->stats_bus_messages_received, 0, sizeof(server.cluster->stats_bus_messages_received));
     clusterSlotStatResetAll();
 }
 
