diff --git a/Core/Relationships/RelationshipResolver.cs b/Core/Relationships/RelationshipResolver.cs
index 0c1314a..bd698c4 100644
--- a/Core/Relationships/RelationshipResolver.cs
+++ b/Core/Relationships/RelationshipResolver.cs
@@ -282,7 +282,8 @@ private void AddModulesToInstall(CkanModule[] modules)
                 try
                 {
                     candidates = resolved.Candidates(descriptor, modlist.Values,
-                                                     registry, game);
+                                                     registry, game,
+                                                     options.proceed_with_inconsistencies);
                     log.DebugFormat("Got {0} candidates for {1}",
                                     candidates.Count, descriptor);
                 }
diff --git a/Core/Relationships/ResolvedRelationshipsTree.cs b/Core/Relationships/ResolvedRelationshipsTree.cs
index 4ced90d..8a14ce3 100644
--- a/Core/Relationships/ResolvedRelationshipsTree.cs
+++ b/Core/Relationships/ResolvedRelationshipsTree.cs
@@ -61,7 +61,8 @@ public IEnumerable<ResolvedRelationship[]> Unsatisfied()
         public IReadOnlyList<CkanModule> Candidates(RelationshipDescriptor          rel,
                                                     IReadOnlyCollection<CkanModule> installing,
                                                     IRegistryQuerier                registry,
-                                                    IGame                           game)
+                                                    IGame                           game,
+                                                    bool                            includeInconsistent = false)
         {
             var candidates = new List<CkanModule>();
             var unresolved = new List<ResolvedRelationship[]>();
@@ -91,7 +92,14 @@ public IEnumerable<ResolvedRelationship[]> Unsatisfied()
                                   .ToArray()
                             is { Length: > 0 } badRels)
                         {
-                            unresolved.AddRange(badRels);
+                            if (includeInconsistent)
+                            {
+                                candidates.Add(module);
+                            }
+                            else
+                            {
+                                unresolved.AddRange(badRels);
+                            }
                         }
                         else if (rrs.SelectMany(subRR => subRR.BadRelationships(installing))
                                     .Select(tuple => tuple.Item2.Type == RelationshipType.Depends
@@ -102,7 +110,14 @@ public IEnumerable<ResolvedRelationship[]> Unsatisfied()
                                     .ToArray()
                                  is { Length: > 0 } badRRs)
                         {
-                            unresolved.AddRange(badRRs);
+                            if (includeInconsistent)
+                            {
+                                candidates.Add(module);
+                            }
+                            else
+                            {
+                                unresolved.AddRange(badRRs);
+                            }
                         }
                         else
                         {
