diff --git a/lib/codecept.js b/lib/codecept.js
index 59d77cd3..76f4e8f4 100644
--- a/lib/codecept.js
+++ b/lib/codecept.js
@@ -96,6 +96,7 @@ class Codecept {
 
       // mask sensitive data
       global.maskSensitiveData = this.config.maskSensitiveData || false
+      global.maskSensitiveDataPatterns = this.config.maskSensitiveDataPatterns || []
     }
   }
 
diff --git a/lib/config.js b/lib/config.js
index 3eaa49e5..eeee43bc 100644
--- a/lib/config.js
+++ b/lib/config.js
@@ -17,6 +17,7 @@ const defaultConfig = {
   teardown: null,
   hooks: [],
   gherkin: {},
+  maskSensitiveDataPatterns: [],
   plugins: {
     screenshotOnFail: {
       enabled: true, // will be disabled by default in 2.0
diff --git a/lib/output.js b/lib/output.js
index a551174a..0b211597 100644
--- a/lib/output.js
+++ b/lib/output.js
@@ -59,7 +59,7 @@ module.exports = {
    * @param {string} msg
    */
   debug(msg) {
-    const _msg = isMaskedData() ? maskSensitiveData(msg) : msg
+    const _msg = isMaskedData() ? maskSensitiveData(msg, getMaskPatterns()) : msg
     if (outputLevel >= 2) {
       print(' '.repeat(this.stepShift), styles.debug(`${figures.pointerSmall} ${_msg}`))
     }
@@ -70,7 +70,7 @@ module.exports = {
    * @param {string} msg
    */
   log(msg) {
-    const _msg = isMaskedData() ? maskSensitiveData(msg) : msg
+    const _msg = isMaskedData() ? maskSensitiveData(msg, getMaskPatterns()) : msg
     if (outputLevel >= 3) {
       print(' '.repeat(this.stepShift), styles.log(truncate(`   ${_msg}`, this.spaceShift)))
     }
@@ -124,7 +124,7 @@ module.exports = {
       stepLine += colors.grey(step.comment.split('\n').join('\n' + ' '.repeat(4)))
     }
 
-    const _stepLine = isMaskedData() ? maskSensitiveData(stepLine) : stepLine
+    const _stepLine = isMaskedData() ? maskSensitiveData(stepLine, getMaskPatterns()) : stepLine
     print(' '.repeat(this.stepShift), truncate(_stepLine, this.spaceShift))
   },
 
@@ -282,3 +282,7 @@ function truncate(msg, gap = 0) {
 function isMaskedData() {
   return global.maskSensitiveData === true || false
 }
+
+function getMaskPatterns() {
+  return global.maskSensitiveDataPatterns || []
+}
