diff --git a/src/OrchardCore.Modules/OrchardCore.Markdown/Razor/ContentRazorHelperExtensions.cs b/src/OrchardCore.Modules/OrchardCore.Markdown/Razor/ContentRazorHelperExtensions.cs
index a700bd1..79e2afa 100644
--- a/src/OrchardCore.Modules/OrchardCore.Markdown/Razor/ContentRazorHelperExtensions.cs
+++ b/src/OrchardCore.Modules/OrchardCore.Markdown/Razor/ContentRazorHelperExtensions.cs
@@ -40,7 +40,7 @@ public static class ContentRazorHelperExtensions
         var html = markdownService.ToHtml(markdown ?? string.Empty);
 
         // TODO: provide context argument (optional on this helper as with the liquid helper?).
-        html = await shortcodeService.ProcessAsync(markdown);
+        html = await shortcodeService.ProcessAsync(html);
 
         if (sanitize)
         {
