diff --git a/langchain4j-onnx-scoring/src/main/java/dev/langchain4j/model/scoring/onnx/OnnxScoringBertCrossEncoder.java b/langchain4j-onnx-scoring/src/main/java/dev/langchain4j/model/scoring/onnx/OnnxScoringBertCrossEncoder.java
index b1a74da..57ad1e5 100644
--- a/langchain4j-onnx-scoring/src/main/java/dev/langchain4j/model/scoring/onnx/OnnxScoringBertCrossEncoder.java
+++ b/langchain4j-onnx-scoring/src/main/java/dev/langchain4j/model/scoring/onnx/OnnxScoringBertCrossEncoder.java
@@ -14,7 +14,7 @@ import java.util.*;
 
 import static ai.onnxruntime.OnnxTensor.createTensor;
 
-class OnnxScoringBertCrossEncoder {
+class OnnxScoringBertCrossEncoder implements AutoCloseable {
 
     private final OrtEnvironment environment;
     private final OrtSession session;
@@ -113,4 +113,12 @@ class OnnxScoringBertCrossEncoder {
         return 1 / (1 + Math.exp(-x));
     }
 
+    @Override
+    public void close() {
+        try {
+            session.close();
+        } catch (OrtException e) {
+            throw new RuntimeException(e);
+        }
+    }
 }
diff --git a/langchain4j-onnx-scoring/src/main/java/dev/langchain4j/model/scoring/onnx/OnnxScoringModel.java b/langchain4j-onnx-scoring/src/main/java/dev/langchain4j/model/scoring/onnx/OnnxScoringModel.java
index 060e242..4ab6c66 100644
--- a/langchain4j-onnx-scoring/src/main/java/dev/langchain4j/model/scoring/onnx/OnnxScoringModel.java
+++ b/langchain4j-onnx-scoring/src/main/java/dev/langchain4j/model/scoring/onnx/OnnxScoringModel.java
@@ -2,7 +2,7 @@ package dev.langchain4j.model.scoring.onnx;
 
 import ai.onnxruntime.OrtSession;
 
-public class OnnxScoringModel extends AbstractInProcessScoringModel {
+public class OnnxScoringModel extends AbstractInProcessScoringModel implements AutoCloseable {
 
     private static final int DEFAULT_MODEL_MAX_LENGTH = 510; // 512 - 2 (special tokens [CLS] and [SEP])
 
@@ -29,4 +29,9 @@ public class OnnxScoringModel extends AbstractInProcessScoringModel {
     protected OnnxScoringBertCrossEncoder model() {
         return this.onnxBertBiEncoder;
     }
+
+    @Override
+    public void close() {
+        onnxBertBiEncoder.close();
+    }
 }
