diff --git a/java/opendataloader-pdf-core/src/main/java/org/opendataloader/pdf/processors/TextLineProcessor.java b/java/opendataloader-pdf-core/src/main/java/org/opendataloader/pdf/processors/TextLineProcessor.java
index 86cfc96..9d04369 100644
--- a/java/opendataloader-pdf-core/src/main/java/org/opendataloader/pdf/processors/TextLineProcessor.java
+++ b/java/opendataloader-pdf-core/src/main/java/org/opendataloader/pdf/processors/TextLineProcessor.java
@@ -43,7 +43,13 @@ public class TextLineProcessor {
         for (IObject content : contents) {
             if (content instanceof TextChunk) {
                 TextChunk textChunk = (TextChunk) content;
-                if (textChunk.isWhiteSpaceChunk() || textChunk.isEmpty()) {
+                if (textChunk.isEmpty()) {
+                    continue;
+                }
+                if (textChunk.isWhiteSpaceChunk()) {
+                    if (!previousLine.getTextChunks().isEmpty()) {
+                        previousLine.add(textChunk);
+                    }
                     continue;
                 }
                 TextLine currentLine = new TextLine(textChunk);
