diff --git a/tests/unit/s2n_connection_test.c b/tests/unit/s2n_connection_test.c
index 4830f71f0..e3e99ed74 100644
--- a/tests/unit/s2n_connection_test.c
+++ b/tests/unit/s2n_connection_test.c
@@ -265,6 +265,10 @@ int main(int argc, char **argv)
                 EXPECT_EQUAL(expected_output[i], output);
 
                 EXPECT_SUCCESS(s2n_connection_get_selected_digest_algorithm(conn, &output));
+
+                /* If no cipher suite has been negotiated yet, s2n_connection_get_selected_digest_algorithm
+                 * should return the digest corresponding to server_cert_sig_scheme.
+                 */
                 EXPECT_EQUAL(expected_output[i], output);
             } else {
                 EXPECT_SUCCESS(s2n_connection_get_selected_client_cert_digest_algorithm(conn, &output));
@@ -399,6 +403,7 @@ int main(int argc, char **argv)
             EXPECT_SUCCESS(s2n_io_pair_close(&io_pair));
             EXPECT_SUCCESS(s2n_config_free(config));
         };
+
     };
 
     /* s2n_connection_set_max_fragment_length */
diff --git a/tls/s2n_connection.c b/tls/s2n_connection.c
index 060bb3c7d..5de138514 100644
--- a/tls/s2n_connection.c
+++ b/tls/s2n_connection.c
@@ -1639,6 +1639,16 @@ int s2n_connection_get_selected_digest_algorithm(struct s2n_connection *conn,
     POSIX_ENSURE_REF(conn);
     POSIX_ENSURE_REF(converted_scheme);
 
+    /* When using non-ephemeral RSA key exchange ("RSA kex"), the server does not send
+     * a signature as part of the handshake.
+     */
+    if (conn->secure && conn->secure->cipher_suite && conn->secure->cipher_suite->key_exchange_alg
+            && !conn->secure->cipher_suite->key_exchange_alg->is_ephemeral
+            && conn->actual_protocol_version <= S2N_TLS12) {
+        *converted_scheme = S2N_TLS_HASH_NONE;
+        return S2N_SUCCESS;
+    }
+
     POSIX_GUARD_RESULT(s2n_signature_scheme_to_tls_iana(
             conn->handshake_params.server_cert_sig_scheme, converted_scheme));
 
@@ -1693,6 +1703,19 @@ int s2n_connection_get_selected_signature_algorithm(struct s2n_connection *conn,
     POSIX_ENSURE_REF(conn);
     POSIX_ENSURE_REF(converted_scheme);
 
+    /* When using non-ephemeral RSA key exchange ("RSA kex"), the server does not send
+     * a signature as part of the handshake. Therefore, no server signature scheme is
+     * actually negotiated.
+     *
+     * See RFC 5246 section 7.4.3 for when ServerKeyExchange is omitted.
+     */
+    if (conn->secure && conn->secure->cipher_suite && conn->secure->cipher_suite->key_exchange_alg
+            && !conn->secure->cipher_suite->key_exchange_alg->is_ephemeral
+            && conn->actual_protocol_version <= S2N_TLS12) {
+        *converted_scheme = S2N_TLS_SIGNATURE_ANONYMOUS;
+        return S2N_SUCCESS;
+    }
+
     POSIX_GUARD_RESULT(s2n_signature_scheme_to_signature_algorithm(
             conn->handshake_params.server_cert_sig_scheme, converted_scheme));
